// @canonical/react-components@4.2.0/dist/components/ConfirmationButton/ConfirmationButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.ConfirmationButton = void 0;
var _react = _interopRequireDefault(require("react"));
var _ActionButton = _interopRequireDefault(require("../ActionButton"));
var _ConfirmationModal = _interopRequireDefault(require("../ConfirmationModal"));
var _external = require("../../external");
const _excluded = ["confirmationModalProps", "onHoverText", "shiftClickEnabled", "showShiftClickHint", "preModalOpenHook"],
  _excluded2 = ["renderInPortal", "portalRenderer"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
const generateTitle = title => {
  if (typeof title === "string") {
    return title;
  }
  if (typeof title === "number") {
    return title.toString();
  }
  return null;
};
/**
 * `ConfirmationButton` is a specialised version of the [ActionButton](?path=/docs/actionbutton--default-story) component that uses a [ConfirmationModal](?path=/docs/confirmationmodal--default-story) to prompt a confirmation from the user before executing an action.
 */
const ConfirmationButton = _ref => {
  let {
      confirmationModalProps,
      onHoverText,
      shiftClickEnabled = false,
      showShiftClickHint = false,
      preModalOpenHook
    } = _ref,
    actionButtonProps = _objectWithoutProperties(_ref, _excluded);
  const {
    openPortal,
    closePortal,
    isOpen,
    Portal
  } = (0, _external.usePortal)();
  const _ref2 = confirmationModalProps,
    {
      renderInPortal: _ignoredRenderInPortal,
      portalRenderer: _ignoredPortalRenderer
    } = _ref2,
    safeConfirmationModalProps = _objectWithoutProperties(_ref2, _excluded2);
  const handleCancelModal = () => {
    closePortal();
    if (confirmationModalProps.close) {
      confirmationModalProps.close();
    }
  };
  const handleConfirmModal = e => {
    closePortal();
    confirmationModalProps.onConfirm(e);
  };
  const handleShiftClick = e => {
    if (e.shiftKey) {
      confirmationModalProps.onConfirm(e);
    } else {
      openPortal(e);
    }
  };
  const handleClick = e => {
    if (preModalOpenHook) {
      const result = preModalOpenHook(e);
      if (!result) return;
    }
    shiftClickEnabled ? handleShiftClick(e) : openPortal(e);
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, isOpen && /*#__PURE__*/_react.default.createElement(_ConfirmationModal.default, _extends({}, safeConfirmationModalProps, {
    close: handleCancelModal,
    confirmButtonLabel: confirmationModalProps.confirmButtonLabel,
    onConfirm: handleConfirmModal,
    portalRenderer: Portal
  }), confirmationModalProps.children, showShiftClickHint && /*#__PURE__*/_react.default.createElement("p", {
    className: "p-text--small u-text--muted u-hide--small"
  }, "Next time, you can skip this confirmation by holding", " ", /*#__PURE__*/_react.default.createElement("code", null, "SHIFT"), " and clicking the action.")), /*#__PURE__*/_react.default.createElement(_ActionButton.default, _extends({}, actionButtonProps, {
    onClick: handleClick,
    title: generateTitle(onHoverText !== null && onHoverText !== void 0 ? onHoverText : confirmationModalProps.confirmButtonLabel)
  }), actionButtonProps.children));
};
exports.ConfirmationButton = ConfirmationButton;
var _default = exports.default = ConfirmationButton;
