// @canonical/react-components@4.2.0/dist/components/ConfirmationButton/ConfirmationButton.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithPreModalOpenHook = exports.IconOnly = exports.IconAndLabel = exports.Default = exports.BaseAppearance = void 0;
var _react = _interopRequireWildcard(require("react"));
var _ConfirmationButton = _interopRequireDefault(require("./ConfirmationButton"));
var _Icon = _interopRequireWildcard(require("../Icon"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const doNothing = () => {};
const meta = {
  component: _ConfirmationButton.default,
  tags: ["autodocs"]
};
const Default = exports.Default = {
  render: () => /*#__PURE__*/_react.default.createElement(_ConfirmationButton.default, {
    confirmationModalProps: {
      title: "Confirm delete",
      confirmButtonLabel: "Delete",
      onConfirm: doNothing,
      children: /*#__PURE__*/_react.default.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/_react.default.createElement("br", null), "You cannot undo this action.")
    }
  }, "Delete"),
  name: "Default"
};
var _default = exports.default = meta;
const IconAndLabel = exports.IconAndLabel = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [isLoading, setLoading] = (0, _react.useState)(false);
    const brieflyLoad = () => {
      setLoading(true);
      setTimeout(() => setLoading(false), 2000);
    };
    return /*#__PURE__*/_react.default.createElement(_ConfirmationButton.default, {
      confirmationModalProps: {
        title: "Confirm delete",
        confirmButtonLabel: "Delete",
        onConfirm: brieflyLoad,
        children: /*#__PURE__*/_react.default.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/_react.default.createElement("br", null), "You cannot undo this action.")
      },
      className: "has-icon",
      disabled: isLoading,
      loading: isLoading,
      shiftClickEnabled: true,
      showShiftClickHint: true
    }, /*#__PURE__*/_react.default.createElement(_Icon.default, {
      name: _Icon.ICONS.delete
    }), /*#__PURE__*/_react.default.createElement("span", null, "Delete"));
  },
  name: "Icon and label"
};
const IconOnly = exports.IconOnly = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [isLoading, setLoading] = (0, _react.useState)(false);
    const brieflyLoad = () => {
      setLoading(true);
      setTimeout(() => setLoading(false), 2000);
    };
    return /*#__PURE__*/_react.default.createElement(_ConfirmationButton.default, {
      confirmationModalProps: {
        title: "Confirm delete",
        confirmButtonLabel: "Delete",
        onConfirm: brieflyLoad,
        children: /*#__PURE__*/_react.default.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/_react.default.createElement("br", null), "You cannot undo this action.")
      },
      className: "has-icon",
      disabled: isLoading,
      loading: isLoading,
      shiftClickEnabled: true,
      showShiftClickHint: true
    }, /*#__PURE__*/_react.default.createElement(_Icon.default, {
      name: _Icon.ICONS.delete
    }));
  },
  name: "Icon only"
};
const BaseAppearance = exports.BaseAppearance = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [isLoading, setLoading] = (0, _react.useState)(false);
    const brieflyLoad = () => {
      setLoading(true);
      setTimeout(() => setLoading(false), 2000);
    };
    return /*#__PURE__*/_react.default.createElement(_ConfirmationButton.default, {
      appearance: "base",
      confirmationModalProps: {
        title: "Confirm delete",
        confirmButtonLabel: "Delete",
        onConfirm: brieflyLoad,
        children: /*#__PURE__*/_react.default.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/_react.default.createElement("br", null), "You cannot undo this action.")
      },
      className: "has-icon",
      disabled: isLoading,
      loading: isLoading,
      shiftClickEnabled: true,
      showShiftClickHint: true
    }, /*#__PURE__*/_react.default.createElement(_Icon.default, {
      name: _Icon.ICONS.delete
    }));
  },
  name: "Base appearance"
};
const WithPreModalOpenHook = exports.WithPreModalOpenHook = {
  render: () => {
    const dontOpenModal = () => false;
    const openModal = () => true;
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_ConfirmationButton.default, {
      preModalOpenHook: dontOpenModal,
      confirmationModalProps: {
        title: "Confirm delete",
        confirmButtonLabel: "Delete",
        children: /*#__PURE__*/_react.default.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/_react.default.createElement("br", null), "You cannot undo this action.")
      }
    }, "Don't open modal"), /*#__PURE__*/_react.default.createElement(_ConfirmationButton.default, {
      preModalOpenHook: openModal,
      confirmationModalProps: {
        title: "Confirm delete",
        confirmButtonLabel: "Delete",
        children: /*#__PURE__*/_react.default.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/_react.default.createElement("br", null), "You cannot undo this action.")
      }
    }, "Open modal"));
  },
  name: "With preModalOpenHook handler"
};
