// @canonical/react-components@4.2.0/dist/components/PrefixedIpInput/PrefixedIpInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _PrefixedInput = _interopRequireDefault(require("../PrefixedInput"));
var _utils = require("./utils");
const _excluded = ["cidr", "help", "onIpChange", "ip", "name"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
const PrefixedIpInput = _ref => {
  let {
      cidr,
      help,
      onIpChange,
      ip,
      name
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  const [networkAddress] = cidr.split("/");
  const isIPV4 = (0, _utils.isIPv4)(networkAddress);
  const [immutable, editable] = (0, _utils.getImmutableAndEditable)(cidr);
  const inputValue = isIPV4 ? ip.split(".").slice(immutable.split(".").length).join(".") : ip.replace(immutable, "");
  const getIPv4MaxLength = () => {
    const immutableOctetsLength = immutable.split(".").length;
    const lengths = [15, 11, 7, 3]; // Corresponding to 0-3 immutable octets
    return lengths[immutableOctetsLength];
  };
  const maxLength = isIPV4 ? getIPv4MaxLength() : editable.length;
  const placeholder = props.disabled ? "" : editable;
  const setIp = editableValue => {
    const fullIp = editableValue ? isIPV4 ? "".concat(immutable, ".").concat(editableValue) : "".concat(immutable).concat(editableValue) : "";
    onIpChange(fullIp);
  };
  const handlePaste = e => {
    e.preventDefault();
    const pastedText = e.clipboardData.getData("text");
    if (isIPV4) {
      const octets = pastedText.split(".");
      const trimmed = octets.slice(0 - editable.split(".").length);
      const ip = trimmed.join(".");
      setIp(ip);
    } else {
      const ip = pastedText.replace(immutable, "");
      setIp(ip);
    }
  };
  return /*#__PURE__*/_react.default.createElement(_PrefixedInput.default, _extends({
    help: help ? help : /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, " ", isIPV4 ? /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, " ", "The available range in this subnet is", " ", /*#__PURE__*/_react.default.createElement("code", null, immutable, ".", editable, " ")) : /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, " ", "The available IPV6 address range is", " ", /*#__PURE__*/_react.default.createElement("code", null, immutable, editable, " ")), "."),
    immutableText: isIPV4 ? "".concat(immutable, ".") : immutable,
    maxLength: maxLength,
    name: name,
    onPaste: handlePaste,
    value: inputValue,
    onChange: e => {
      setIp(e.target.value);
    },
    placeholder: placeholder
  }, props));
};
var _default = exports.default = PrefixedIpInput;
