// @canonical/react-components@4.2.0/dist/components/PrefixedIpInput/PrefixedIpInput.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithCustomHelp = exports.Required = exports.IPv6WithValue = exports.IPv6Default = exports.IPv4WithValue = exports.IPv4WithError = exports.IPv4Disabled = exports.IPv4Default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _PrefixedIpInput = _interopRequireDefault(require("./PrefixedIpInput"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const PrefixedIpInputWrapper = args => {
  const [ip, setIp] = (0, _react.useState)(args.ip);
  return /*#__PURE__*/_react.default.createElement(_PrefixedIpInput.default, _extends({}, args, {
    ip: ip,
    onIpChange: newIp => {
      var _args$onIpChange;
      setIp(newIp);
      (_args$onIpChange = args.onIpChange) === null || _args$onIpChange === void 0 || _args$onIpChange.call(args, newIp);
    }
  }));
};
const meta = {
  component: _PrefixedIpInput.default,
  tags: ["autodocs"],
  argTypes: {
    ip: {
      control: "text"
    },
    cidr: {
      control: "text"
    },
    label: {
      control: "text"
    },
    name: {
      control: "text"
    },
    error: {
      control: "text"
    },
    help: {
      control: "text"
    },
    disabled: {
      control: "boolean"
    },
    required: {
      control: "boolean"
    }
  },
  render: args => /*#__PURE__*/_react.default.createElement(PrefixedIpInputWrapper, args)
};
var _default = exports.default = meta;
const IPv4Default = exports.IPv4Default = {
  name: "IPv4 default",
  args: {
    cidr: "192.168.1.0/24",
    ip: "",
    name: "ip-address",
    label: "IP Address",
    onIpChange: ip => console.log("IP changed:", ip)
  }
};
const IPv4WithValue = exports.IPv4WithValue = {
  name: "IPv4 with value",
  args: {
    cidr: "192.168.1.0/24",
    ip: "192.168.1.100",
    name: "ip-address",
    label: "IP Address",
    onIpChange: ip => console.log("IP changed:", ip)
  }
};
const IPv4WithError = exports.IPv4WithError = {
  name: "IPv4 with error",
  args: {
    cidr: "192.168.1.0/24",
    ip: "192.168.1.256",
    name: "ip-address",
    label: "IP Address",
    error: "Invalid IP address",
    onIpChange: ip => console.log("IP changed:", ip)
  }
};
const IPv4Disabled = exports.IPv4Disabled = {
  name: "IPv4 disabled",
  args: {
    cidr: "192.168.1.0/24",
    ip: "192.168.1.50",
    name: "ip-address",
    label: "IP Address",
    disabled: true,
    onIpChange: ip => console.log("IP changed:", ip)
  }
};
const IPv6Default = exports.IPv6Default = {
  name: "IPv6 default",
  args: {
    cidr: "2001:db8::/32",
    ip: "",
    name: "ipv6-address",
    label: "IPv6 Address",
    onIpChange: ip => console.log("IP changed:", ip)
  }
};
const IPv6WithValue = exports.IPv6WithValue = {
  name: "IPv6 with value",
  args: {
    cidr: "2001:db8::/32",
    ip: "2001:db8::1",
    name: "ipv6-address",
    label: "IPv6 Address",
    onIpChange: ip => console.log("IP changed:", ip)
  }
};
const WithCustomHelp = exports.WithCustomHelp = {
  args: {
    cidr: "10.0.0.0/16",
    ip: "",
    name: "ip-address",
    label: "IP Address",
    help: "Enter a custom IP address for this device",
    onIpChange: ip => console.log("IP changed:", ip)
  }
};
const Required = exports.Required = {
  args: {
    cidr: "192.168.0.0/24",
    ip: "",
    name: "ip-address",
    label: "IP Address",
    required: true,
    onIpChange: ip => console.log("IP changed:", ip)
  }
};
