// @canonical/react-components@4.2.0/dist/esm/components/ColumnSelector/ColumnSelector.stories.js
import React, { useState } from "react";
import ColumnSelector from "./ColumnSelector";
import MainTable from "../MainTable";
import { getRows, FILESYSTEM, SNAPSHOTS, USER_HIDEABLE_COLUMNS, DESCRIPTION, CLUSTER_MEMBER, COLUMN_HEADERS } from "./ColumnSelectorStoriesHelper";
import { visibleHeaderColumns, visibleRowColumns } from "./columnSelectorHelper";
var meta = {
  component: ColumnSelector,
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [userHidden, setUserHidden] = useState([FILESYSTEM, SNAPSHOTS, CLUSTER_MEMBER]);
    var sizeHidden = [DESCRIPTION];
    var hiddenColumns = userHidden.concat(sizeHidden);
    var ROWS = getRows();
    var setUserHiddenAndSaveInLocalStorage = columns => {
      setUserHidden(columns);
    };
    var displayedRows = visibleRowColumns(ROWS, hiddenColumns);
    var displayedHeaders = visibleHeaderColumns(COLUMN_HEADERS.map(t => {
      return {
        content: t,
        sortKey: t
      };
    }), hiddenColumns);
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(ColumnSelector, {
      columns: USER_HIDEABLE_COLUMNS,
      userHidden: userHidden,
      sizeHidden: sizeHidden,
      setUserHidden: setUserHiddenAndSaveInLocalStorage
    }), /*#__PURE__*/React.createElement(MainTable, {
      rows: displayedRows,
      headers: displayedHeaders
    }));
  },
  name: "Default"
};
