// @canonical/react-components@4.2.0/dist/esm/components/ColumnSelector/ColumnSelector.js
import React from "react";
import classnames from "classnames";
import Tooltip from "../Tooltip";
import ContextualMenu from "../ContextualMenu";
import Icon from "../Icon";
import CheckboxInput from "../CheckboxInput";
import "./ColumnSelector.scss";
/**
This is a [React](https://reactjs.org/) component that extends from the Vanilla [Select](https://vanillaframework.io/docs/base/forms#select) element.
The aim of this component is to provide a dropdown menu to control the visibility of columns within a table.
This component allows users to customize their view, hiding or showing columns as needed, while also handling columns that are automatically hidden on smaller screens.
*/
var ColumnSelector = _ref => {
  var {
    className,
    columns,
    userHidden,
    sizeHidden,
    setUserHidden
  } = _ref;
  var selectedCount = columns.length - userHidden.length;
  var toggleHiddenColumn = column => {
    if (userHidden.includes(column)) {
      setUserHidden(userHidden.filter(c => c !== column));
    } else {
      setUserHidden([...userHidden, column]);
    }
  };
  var wrapTooltip = (element, column) => {
    if (!sizeHidden.includes(column)) return element;
    return /*#__PURE__*/React.createElement(Tooltip, {
      message: /*#__PURE__*/React.createElement(React.Fragment, null, "Screen is too narrow to fit the column.", /*#__PURE__*/React.createElement("br", null), "Disable columns above or use a bigger screen."),
      position: "left"
    }, element);
  };
  return /*#__PURE__*/React.createElement(ContextualMenu, {
    className: classnames(className, "column-selector-toggle"),
    dropdownProps: {
      "aria-label": "columns menu"
    },
    position: "right",
    toggleClassName: "has-icon",
    toggleProps: {
      "aria-label": "Columns selection toggle"
    },
    toggleLabel: /*#__PURE__*/React.createElement(Icon, {
      name: "settings"
    }),
    toggleAppearance: "base",
    title: "Columns"
  }, /*#__PURE__*/React.createElement("div", {
    className: "column-selector-column-list"
  }, /*#__PURE__*/React.createElement(CheckboxInput, {
    checked: userHidden.length === 0,
    indeterminate: selectedCount > 0 && selectedCount < columns.length,
    label: "".concat(selectedCount, " out of ").concat(columns.length, " columns selected"),
    onChange: () => {
      if (userHidden.length > 0) {
        setUserHidden([]);
      } else {
        setUserHidden(columns);
      }
    }
  }), /*#__PURE__*/React.createElement("hr", null), columns.map(column => /*#__PURE__*/React.createElement("div", {
    key: column
  }, wrapTooltip(/*#__PURE__*/React.createElement(CheckboxInput, {
    "aria-label": column,
    labelClassName: classnames({
      "size-hidden": sizeHidden.includes(column)
    }),
    checked: !userHidden.includes(column),
    label: column,
    onChange: () => {
      toggleHiddenColumn(column);
    },
    disabled: sizeHidden.includes(column)
  }), column)))));
};
export default ColumnSelector;
