// @canonical/react-components@4.2.0/dist/esm/components/ApplicationLayout/AppNavigationBar/AppNavigationBar.stories.js
import React from "react";
import { useState } from "react";
import Application from "../Application";
import AppMain from "../AppMain";
import Panel from "../../Panel";
import Strip from "../../Strip";
import AppNavigationBar from "./AppNavigationBar";
var meta = {
  component: AppNavigationBar,
  tags: ["autodocs"]
};
export default meta;
/**
 * In most common cases an `AppNavigationBar` should contain a `<Panel>`.
 *
 * `AppNavigationBar` should be a direct child of an `<Application>` or when using `ApplicationLayout`
 * it will already be included.
 *
 * The toggle state should control the collapsed state of `AppNavigation`.
 */
export var Default = {
  args: {
    children: "AppNavigation"
  }
};
export var States = {
  render: args => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [collapsed, setCollapsed] = useState(true);
    return /*#__PURE__*/React.createElement(Application, null, /*#__PURE__*/React.createElement(AppNavigationBar, args, /*#__PURE__*/React.createElement(Panel, {
      dark: true,
      logo: {
        icon: "https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg",
        name: "https://assets.ubuntu.com/v1/2e04d794-logo-jaas.svg",
        nameAlt: "JAAS",
        href: "/"
      },
      toggle: {
        label: "Menu",
        onClick: () => setCollapsed(!collapsed)
      }
    })), /*#__PURE__*/React.createElement(AppMain, null, /*#__PURE__*/React.createElement(Strip, null, /*#__PURE__*/React.createElement("p", null, "Reduce your window size to mobile to see the navigation bar."), /*#__PURE__*/React.createElement("p", null, "Pass the collapsed state to the main menu:", " ", /*#__PURE__*/React.createElement("code", null, "<AppNavigation collapsed={".concat(collapsed, "} />"))))));
  }
};
