// @canonical/react-components@4.2.0/dist/esm/components/ApplicationLayout/AppStatus/AppStatus.stories.js
import React from "react";
import Application from "../Application";
import Button from "../../Button";
import Panel from "../../Panel";
import AppStatus from "./AppStatus";
import AppMain from "../AppMain";
import Strip from "../../Strip";
import Spinner from "../../Spinner";
var meta = {
  component: AppStatus,
  tags: ["autodocs"]
};
export default meta;
/**
 * `AppStatus` can contain a `<Panel>` to display the content as intended in the
 * application layout.
 *
 * `AppStatus` should be a direct child of an `<Application>` or when using `ApplicationLayout`
 * it will automatically wrap the content passed to the `status` prop.
 */
export var Default = {
  render: args => {
    return /*#__PURE__*/React.createElement(Application, null, /*#__PURE__*/React.createElement(AppMain, null, /*#__PURE__*/React.createElement(Strip, null, /*#__PURE__*/React.createElement("p", null, "Scroll to the bottom to see the status bar."))), /*#__PURE__*/React.createElement(AppStatus, args, /*#__PURE__*/React.createElement(Panel, {
      wrapContent: false
    }, /*#__PURE__*/React.createElement("p", {
      className: "u-no-margin--bottom"
    }, /*#__PURE__*/React.createElement(Spinner, null), " Update in progress...", " ", /*#__PURE__*/React.createElement(Button, {
      appearance: "base",
      onClick: () => {},
      dense: true,
      className: "u-no-margin"
    }, "Retry")))));
  }
};
