// @canonical/react-components@4.2.0/dist/esm/components/Card/Card.stories.js
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import React from "react";
import Card from "./";
import Col from "../Col";
import Row from "../Row";
var meta = {
  component: Card,
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  args: {
    title: "We'd love to have you join us as a partner.",
    children: "If you are an independent software vendor or bundle author, it's easy to apply. You can find out more below."
  }
};
export var Header = {
  args: {
    title: "Raspberry Pi2 and Pi3",
    thumbnail: "https://assets.ubuntu.com/v1/dca2e4c4-raspberry-logo.png",
    children: "For fun, for education and for profit, the RPi makes device development personal and entertaining. With support for both the Pi2 and the new Pi3, Ubuntu Core supports the world’s most beloved board."
  }
};
export var Highlighted = {
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    highlighted: true
  })
};
export var Overlay = {
  args: {
    title: "Web browsing",
    overlay: true,
    children: "Renowned for speed and security, Ubuntu and Firefox make browsing\nthe web a pleasure again. Ubuntu also includes Chrome, Opera and\nother browsers that can be installed from the Ubuntu Software\nCentre."
  },
  render: args => /*#__PURE__*/React.createElement("section", {
    className: "p-strip--image is-light",
    style: {
      backgroundImage: "url('https://assets.ubuntu.com/v1/0a98afcd-screenshot_desktop.jpg')"
    }
  }, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
    size: 6,
    emptyLarge: 7
  }, /*#__PURE__*/React.createElement(Card, args)))),
  parameters: {
    percy: {
      waitForTimeout: 1000 // Wait for 1 second before taking a snapshot to allow the image to load
    }
  }
};
