// @canonical/react-components@4.2.0/dist/esm/components/ThemeSwitcher/ThemeSwitcher.js
import React from "react";
import { useState } from "react";
var LOCAL_STORAGE_KEY = "theme";
var THEME_SYSTEM = "system";
var THEME_DARK = "dark";
var THEME_LIGHT = "light";
export var loadTheme = () => {
  var saved = localStorage.getItem(LOCAL_STORAGE_KEY);
  return saved || THEME_SYSTEM;
};
var saveTheme = theme => {
  localStorage.setItem(LOCAL_STORAGE_KEY, theme);
};
export var isDarkTheme = theme => {
  if (theme === THEME_SYSTEM) {
    return window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)").matches;
  }
  return theme === THEME_DARK;
};
export var applyTheme = theme => {
  if (isDarkTheme(theme)) {
    document.body.classList.add("is-dark");
  } else {
    document.body.classList.remove("is-dark");
  }
};

/**
 * This is a [React](https://reactjs.org/) component for the [Vanilla framework](https://docs.vanillaframework.io).
 *
 * The ThemeSwitcher component allows users to switch between different themes: dark, light, and system. It saves the selected theme in local storage and applies it to the document body. You can use it in user settings.
 *
 * In your root component, call the exported functions `loadTheme` and `applyTheme`, such as in the example below:
 * ```
 * useEffect(() => {
 *   const theme = loadTheme();
 *   applyTheme(theme);
 * }, []);
 * ```
 */
var ThemeSwitcher = () => {
  var [activeTheme, setActiveTheme] = useState(loadTheme());
  var themeButton = theme => {
    return /*#__PURE__*/React.createElement("button", {
      className: "p-segmented-control__button",
      type: "button",
      "aria-selected": activeTheme === theme ? "true" : "false",
      onClick: () => {
        saveTheme(theme);
        setActiveTheme(theme);
        applyTheme(theme);
      }
    }, theme);
  };
  return /*#__PURE__*/React.createElement("div", {
    className: "p-segmented-control"
  }, /*#__PURE__*/React.createElement("div", {
    className: "p-segmented-control__list",
    "aria-label": "Theme switcher"
  }, themeButton(THEME_DARK), themeButton(THEME_LIGHT), themeButton(THEME_SYSTEM)));
};
export default ThemeSwitcher;
