// @canonical/react-components@4.2.0/dist/components/Badge/Badge.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.UndefinedLargeNumber = exports.RoundedLargeNumber = exports.Default = exports.ColourCoding = void 0;
var _react = _interopRequireDefault(require("react"));
var _Badge = _interopRequireWildcard(require("./Badge"));
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _Badge.default,
  tags: ["autodocs"],
  argTypes: {
    badgeType: {
      control: {
        options: _Badge.BadgeType,
        type: "select"
      }
    }
  },
  args: {
    badgeType: _Badge.BadgeType.UNDEFINED_LARGE_NUMBER
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default",
  args: {
    value: 9
  }
};
const ColourCoding = exports.ColourCoding = {
  render: () => /*#__PURE__*/_react.default.createElement(_Badge.default, {
    value: 9,
    isNegative: true
  }),
  name: "Colour Coding"
};

/**
 * When the amount of items is only relevant for small values, this variant should be used. When the value grows further than 999, the badge should show "999+".
 */
const UndefinedLargeNumber = exports.UndefinedLargeNumber = {
  render: () => /*#__PURE__*/_react.default.createElement(_Badge.default, {
    value: 1000,
    badgeType: _Badge.BadgeType.UNDEFINED_LARGE_NUMBER
  }),
  name: "Undefined Large Number"
};

/**
 * When the value of the badge represents a type of information where differences between large amounts of items are relevant, it should include a decimal unit prefix and round the value.
 */
const RoundedLargeNumber = exports.RoundedLargeNumber = {
  render: () => /*#__PURE__*/_react.default.createElement(_Badge.default, {
    value: 1234,
    badgeType: _Badge.BadgeType.ROUNDED_LARGE_NUMBER
  }),
  name: "Rounded Large Number"
};
