// @canonical/react-components@4.2.0/dist/esm/components/Form/Form.stories.js
import React from "react";
import Form from "./Form";
import Input from "../Input";
import Select from "../Select";
var meta = {
  component: Form,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: {
        disable: true
      }
    }
  }
};
export default meta;
export var Default = {
  name: "Default",
  args: {
    children: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Input, {
      type: "text",
      id: "exampleInputEmail12",
      placeholder: "example@canonical.com",
      label: "Email address"
    }), /*#__PURE__*/React.createElement(Input, {
      type: "password",
      id: "exampleInputPassword12",
      placeholder: "******",
      label: "Password"
    }), /*#__PURE__*/React.createElement(Input, {
      type: "file",
      id: "exampleInputFile2",
      label: "File input"
    }), /*#__PURE__*/React.createElement(Input, {
      type: "checkbox",
      id: "CheckMe2",
      label: "I agree to receive information about Canonical’s products and services."
    }))
  }
};
export var Fieldset = {
  render: () => /*#__PURE__*/React.createElement(Form, null, /*#__PURE__*/React.createElement("fieldset", null, /*#__PURE__*/React.createElement(Input, {
    placeholder: "Joe",
    id: "list-input-12",
    type: "text",
    label: "First name"
  }), /*#__PURE__*/React.createElement(Input, {
    placeholder: "Bloggs",
    id: "list-input-22",
    type: "text",
    label: "Last name"
  }), /*#__PURE__*/React.createElement(Input, {
    placeholder: "example@canonical.com",
    id: "list-input-32",
    type: "text",
    label: "Email address"
  }))),
  name: "Fieldset"
};
export var Inline = {
  render: () => /*#__PURE__*/React.createElement(Form, {
    inline: true
  }, /*#__PURE__*/React.createElement(Input, {
    type: "text",
    id: "username-inline2",
    className: "p-form__control",
    label: "Username",
    help: "30 characters or fewer."
  }), /*#__PURE__*/React.createElement(Input, {
    type: "text",
    id: "address-inline22",
    "aria-invalid": "true",
    label: "Email address",
    error: "Please enter a valid email address."
  })),
  name: "Inline"
};
export var Stacked = {
  render: () => /*#__PURE__*/React.createElement(Form, {
    stacked: true
  }, /*#__PURE__*/React.createElement(Input, {
    type: "text",
    id: "full-name-stacked2",
    label: "Full name",
    stacked: true
  }), /*#__PURE__*/React.createElement(Input, {
    type: "text",
    id: "username-stacked2",
    label: "Username",
    stacked: true,
    help: "30 characters or fewer."
  }), /*#__PURE__*/React.createElement(Input, {
    type: "text",
    id: "username-stacked-error2",
    label: "Email address",
    stacked: true,
    error: "This field is required"
  }), /*#__PURE__*/React.createElement(Input, {
    type: "text",
    id: "address-optional-stacked2",
    label: "Address line 1",
    stacked: true
  }), /*#__PURE__*/React.createElement(Input, {
    type: "text",
    id: "address-optional-stacked3",
    label: "Address line 2",
    stacked: true
  })),
  name: "Stacked"
};
export var Disabled = {
  render: () => /*#__PURE__*/React.createElement(Form, null, /*#__PURE__*/React.createElement(Input, {
    label: "Email address",
    type: "text",
    id: "disabled-input2",
    placeholder: "example@canonical.com",
    disabled: true
  })),
  name: "Disabled"
};
export var Validation = {
  render: () => /*#__PURE__*/React.createElement(Form, null, /*#__PURE__*/React.createElement(Input, {
    type: "text",
    id: "exampleTextInputError2",
    placeholder: "example@canonical.com",
    label: "Email address",
    error: "This field is required."
  }), /*#__PURE__*/React.createElement(Input, {
    type: "text",
    id: "exampleTextInputCaution2",
    placeholder: "14",
    label: "Mail configuration ID",
    caution: "No validation is performed in preview mode."
  }), /*#__PURE__*/React.createElement(Input, {
    type: "text",
    id: "exampleTextInputSuccess2",
    placeholder: "**** **** **** ****",
    label: "Card number",
    success: "Verified."
  }), /*#__PURE__*/React.createElement(Select, {
    id: "exampleSelectInputError3",
    defaultValue: "",
    options: [{
      value: "",
      disabled: true,
      label: "--Select an option--"
    }, {
      value: "1",
      label: "Cosmic Cuttlefish"
    }, {
      value: "2",
      label: "Bionic Beaver"
    }, {
      value: "3",
      label: "Xenial Xerus"
    }],
    label: "Ubuntu releases",
    error: "You need to select an OS to complete your install."
  })),
  name: "Validation"
};
export var Required = {
  render: () => /*#__PURE__*/React.createElement(Form, null, /*#__PURE__*/React.createElement(Input, {
    type: "text",
    id: "exampleTextInputError3",
    placeholder: "e.g joe@bloggs.com",
    error: "This field is required.",
    label: "Email address",
    required: true
  })),
  name: "Required"
};
