// @canonical/react-components@4.2.0/dist/esm/components/SideNavigation/SideNavigationItem/SideNavigationItem.stories.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import React from "react";
import SideNavigation from "../SideNavigation";
import SideNavigationItem from "./SideNavigationItem";
import SideNavigationText from "../SideNavigationText";
var meta = {
  component: SideNavigationItem,
  render: args => /*#__PURE__*/React.createElement(SideNavigation, {
    dark: false,
    items: [{
      items: [/*#__PURE__*/React.createElement(SideNavigationItem, _extends({
        key: "item"
      }, args, {
        dark: false
      }))]
    }]
  }),
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  args: {
    icon: "drag",
    label: "Models",
    href: "/models"
  }
};

/**
 * The navigation item can appear as a link and accept the props from [`SideNavigationLink`](/docs/components-sidenavigation-sidenavigationlink--docs).
 */
export var Links = {
  args: {
    icon: "drag",
    label: "Models",
    href: "/models"
  }
};

/**
 * By providing the `nonInteractive` prop the navigation item will appear as a
 * text entry and accept the props from
 * [`SideNavigationText`](/docs/components-sidenavigation-sidenavigationtext--docs).
 */
export var Text = {
  args: {
    icon: "drag",
    label: "Models",
    nonInteractive: true
  }
};

/**
 * The navigation item can display custom content by providing the `children` prop.
 * In this case, any other attributes provided to the object will be given to
 * the list item.
 */
export var CustomContent = {
  args: {
    className: "custom-class",
    children: /*#__PURE__*/React.createElement(SideNavigationText, null, "This item has been given a custom class")
  }
};
