// @canonical/react-components@4.2.0/dist/esm/components/EmptyState/EmptyState.stories.js
import React from "react";
import EmptyState from "./EmptyState";
import Icon, { ICONS } from "../Icon";
import Button from "../Button";
var boxImage = /*#__PURE__*/React.createElement("img", {
  src: "https://assets.ubuntu.com/v1/c17e0d92-container.svg",
  style: {
    maxWidth: "10rem",
    maxHeight: "10rem"
  },
  alt: "empty state"
});
var doNothing = () => {};
var meta = {
  component: EmptyState,
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  render: () => /*#__PURE__*/React.createElement(EmptyState, {
    image: boxImage,
    title: "This application doesn't have any configuration parameters"
  }),
  name: "Default"
};
export var WithContent = {
  render: () => /*#__PURE__*/React.createElement(EmptyState, {
    image: boxImage,
    title: "No instances found"
  }, /*#__PURE__*/React.createElement("p", null, "There are no instances in this project. Spin up your first instance!"), /*#__PURE__*/React.createElement("p", null, /*#__PURE__*/React.createElement("a", {
    href: "#todo",
    target: "_blank"
  }, "How to create instances", /*#__PURE__*/React.createElement(Icon, {
    name: ICONS.externalLink
  }))), /*#__PURE__*/React.createElement(Button, {
    appearance: "positive",
    onClick: doNothing
  }, "Create instance")),
  name: "With content"
};
