// @canonical/react-components@4.2.0/dist/components/SearchBox/SearchBox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Label = void 0;
var _classnames = _interopRequireDefault(require("classnames"));
var _react = _interopRequireWildcard(require("react"));
var _Icon = _interopRequireDefault(require("../Icon"));
const _excluded = ["name", "id", "autocomplete", "className", "disabled", "externallyControlled", "onChange", "onSearch", "onClear", "placeholder", "shouldBlurOnSearch", "shouldRefocusAfterReset", "value"];
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
let Label = exports.Label = /*#__PURE__*/function (Label) {
  Label["Clear"] = "Clear search field";
  Label["Search"] = "Search";
  return Label;
}({});
/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [SearchBox](https://docs.vanillaframework.io/patterns/search-box/).
 *
 * Search boxes enable search functionality on a page and are typically used in a navigation bar. SearchBox elements should be wrapped within a form element that handles its own submit action.
 */
const SearchBox = /*#__PURE__*/_react.default.forwardRef((_ref, forwardedRef) => {
  let {
      name = "search",
      id = "search",
      autocomplete = "on",
      className,
      disabled,
      externallyControlled,
      onChange,
      onSearch,
      onClear,
      placeholder = "Search",
      shouldBlurOnSearch = true,
      shouldRefocusAfterReset,
      value
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  const internalRef = (0, _react.useRef)(null);
  const resetInput = () => {
    onChange === null || onChange === void 0 || onChange("");
    onClear === null || onClear === void 0 || onClear();
    if (internalRef.current) {
      internalRef.current.value = "";
      if (shouldRefocusAfterReset) internalRef.current.focus();
    }
  };
  const triggerSearch = () => {
    onSearch === null || onSearch === void 0 || onSearch(externallyControlled ? value : internalRef.current.value);
  };
  const onKeyDown = e => {
    if (e.key === "Enter" && internalRef.current.checkValidity()) {
      if (shouldBlurOnSearch) {
        internalRef.current.blur();
      }
      triggerSearch();
    }
  };
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)("p-search-box", className)
  }, /*#__PURE__*/_react.default.createElement("label", {
    className: "u-off-screen",
    htmlFor: id
  }, placeholder || "Search"), /*#__PURE__*/_react.default.createElement("input", _extends({
    autoComplete: autocomplete,
    className: "p-search-box__input",
    disabled: disabled,
    id: id,
    name: name,
    onChange: evt => onChange === null || onChange === void 0 ? void 0 : onChange(evt.target.value),
    onKeyDown: onKeyDown,
    placeholder: placeholder,
    ref: input => {
      internalRef.current = input;
      // Handle both function and object refs.
      if (typeof forwardedRef === "function") {
        forwardedRef(input);
      } else if (forwardedRef) {
        forwardedRef.current = input;
      }
    },
    type: "search",
    defaultValue: externallyControlled ? undefined : value,
    value: externallyControlled ? value : undefined
  }, props)), value && /*#__PURE__*/_react.default.createElement("button", {
    className: "p-search-box__reset",
    disabled: disabled,
    onClick: resetInput,
    type: "reset"
  }, /*#__PURE__*/_react.default.createElement(_Icon.default, {
    name: "close"
  }, Label.Clear)), /*#__PURE__*/_react.default.createElement("button", {
    className: "p-search-box__button",
    disabled: disabled,
    onClick: triggerSearch
  }, /*#__PURE__*/_react.default.createElement(_Icon.default, {
    name: "search"
  }, Label.Search)));
});
SearchBox.displayName = "SearchBox";
var _default = exports.default = SearchBox;
