// @canonical/react-components@4.2.0/dist/components/SearchBox/SearchBox.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Navigation = exports.ExternalState = exports.Disabled = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _SearchBox = _interopRequireDefault(require("./SearchBox"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _SearchBox.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default"
};
const Disabled = exports.Disabled = {
  render: () => /*#__PURE__*/_react.default.createElement(_SearchBox.default, {
    disabled: true
  }),
  name: "Disabled"
};

/**
 * If you wish to control the value of the input via external state you can set
the `externallyControlled` prop and provide an `onChange` method to update the
state and the `value` from state.
 */
const ExternalState = exports.ExternalState = {
  render: () => /*#__PURE__*/_react.default.createElement(_SearchBox.default, {
    externallyControlled: true,
    shouldRefocusAfterReset: true,
    onChange: () => {},
    value: "..."
  }),
  name: "External state"
};
const Navigation = exports.Navigation = {
  render: () => /*#__PURE__*/_react.default.createElement("header", {
    id: "navigation",
    className: "p-navigation"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "p-navigation__row--full-width"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "p-navigation__banner"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "p-navigation__logo"
  }, /*#__PURE__*/_react.default.createElement("a", {
    className: "p-navigation__item",
    href: "#test"
  }, /*#__PURE__*/_react.default.createElement("img", {
    className: "p-navigation__image",
    src: "https://assets.ubuntu.com/v1/5d6da5c4-logo-canonical-aubergine.svg",
    alt: "",
    width: "95"
  }))), /*#__PURE__*/_react.default.createElement("a", {
    href: "#navigation",
    className: "p-navigation__toggle--open",
    title: "menu"
  }, "Menu"), /*#__PURE__*/_react.default.createElement("a", {
    href: "#navigation-closed",
    className: "p-navigation__toggle--close",
    title: "close menu"
  }, "Close menu")), /*#__PURE__*/_react.default.createElement("nav", {
    className: "p-navigation__nav"
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "u-off-screen"
  }, /*#__PURE__*/_react.default.createElement("a", {
    href: "#main-content"
  }, "Jump to main content")), /*#__PURE__*/_react.default.createElement("ul", {
    className: "p-navigation__items",
    role: "menu"
  }, /*#__PURE__*/_react.default.createElement("li", {
    className: "p-navigation__item",
    role: "menuitem"
  }, /*#__PURE__*/_react.default.createElement("a", {
    href: "#test",
    className: "p-navigation__link"
  }, "Products")), /*#__PURE__*/_react.default.createElement("li", {
    className: "p-navigation__item",
    role: "menuitem"
  }, /*#__PURE__*/_react.default.createElement("a", {
    href: "#test",
    className: "p-navigation__link"
  }, "Services")), /*#__PURE__*/_react.default.createElement("li", {
    className: "p-navigation__item",
    role: "menuitem"
  }, /*#__PURE__*/_react.default.createElement("a", {
    href: "#test",
    className: "p-navigation__link"
  }, "Partners")), /*#__PURE__*/_react.default.createElement("li", {
    className: "p-navigation__item",
    role: "menuitem"
  }, /*#__PURE__*/_react.default.createElement("a", {
    href: "#test",
    className: "p-navigation__link"
  }, "About")), /*#__PURE__*/_react.default.createElement("li", {
    className: "p-navigation__item",
    role: "menuitem"
  }, /*#__PURE__*/_react.default.createElement("a", {
    href: "#test",
    className: "p-navigation__link"
  }, "Partners"))), /*#__PURE__*/_react.default.createElement(_SearchBox.default, null)))),
  name: "Navigation"
};
