// @canonical/react-components@4.2.0/dist/esm/components/Notifications/Notification/Notification.stories.js
import React from "react";
import { useState } from "react";
import Notification, { NotificationSeverity } from "./Notification";
var meta = {
  component: Notification,
  tags: ["autodocs"],
  argTypes: {
    borderless: {
      control: {
        type: "boolean"
      }
    },
    children: {
      control: {
        type: "text"
      }
    },
    inline: {
      control: {
        type: "boolean"
      }
    },
    severity: {
      control: {
        type: "radio"
      },
      options: Object.values(NotificationSeverity)
    },
    timestamp: {
      control: {
        type: "text"
      }
    },
    title: {
      control: {
        type: "text"
      }
    }
  },
  args: {
    borderless: false,
    inline: false,
    severity: NotificationSeverity.INFORMATION
  }
};
export default meta;
/**
 * The information severity should be used to convey an information message.
 */
export var Information = {
  name: "Information",
  args: {
    children: "Anyone with access can view your invited users.",
    severity: "information",
    title: "Permissions changed"
  }
};

/**
 * The caution severity should be used to convey information that is not critical but the user should be aware of.
 */
export var Caution = {
  name: "Caution",
  args: {
    children: "Custom storage configuration is only supported on Ubuntu, CentOS and RHEL.",
    severity: "caution",
    title: "Blocked"
  }
};

/**
 * The negative severity should be used to convey information that is critical and the user should take action.
 */
export var Negative = {
  name: "Negative",
  args: {
    children: "Node must be connected to a network.",
    severity: "negative",
    title: "Error"
  }
};

/**
 * The positive severity should be used to convey success or completion.
 */
export var Positive = {
  name: "Positive",
  args: {
    children: "Code successfully reformatted.",
    severity: "positive",
    title: "Success"
  }
};

/**
 * When vertical space is limited, you can use the inline variant.
 */
export var Inline = {
  name: "Inline",
  args: {
    children: "Not enough space.",
    inline: true,
    severity: "negative",
    title: "Error:"
  }
};

/**
 * In cases where a notification sits inside another component, such as a table cell or a card, it may be useful to remove the outer border and highlight bar.
 */
export var Borderless = {
  name: "Borderless",
  args: {
    borderless: true,
    children: "Only 8GB storage remaining.",
    severity: "caution",
    title: "Warning"
  }
};

/**
 * Actions will appear below the notification message.
 */
export var Actions = {
  name: "Actions",
  args: {
    actions: [{
      label: "Action 1",
      onClick: () => null
    }, {
      label: "Action 2",
      onClick: () => null
    }],
    children: "Body lorem ipsum dolor sit amet consequiteor. Lorem ipsum dolor sit amet consequiteor.",
    severity: "information",
    title: "Title"
  }
};

/**
 * Notifications that can be dismissed should be given an onDismiss function.
 */
export var Dismissible = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [show, setShow] = useState(true);
    if (!show) {
      return null;
    }
    return /*#__PURE__*/React.createElement(Notification, {
      onDismiss: () => setShow(false),
      title: "Dismissible"
    }, "Click the dismiss button to get rid of me.");
  },
  name: "Dismissible"
};

/**
 * Notifications can be automatically dismissed after a timeout interval.
 */
export var Timeout = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [show, setShow] = useState(true);
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [timer, setTimer] = useState(0);
    if (!show) {
      return null;
    }
    return /*#__PURE__*/React.createElement(Notification, {
      actions: [{
        label: "Start timer",
        onClick: () => setTimer(3000)
      }],
      onDismiss: () => setShow(false),
      timeout: timer,
      title: "Timeout"
    }, "Click the action to start a 3 second timer.");
  },
  name: "Timeout"
};

/**
 * For notifications in which recency is important, you can include a section for time.
 */
export var Timestamp = {
  name: "Timestamp",
  args: {
    children: "Body lorem ipsum dolor sit amet consequiteor. Lorem ipsum dolor sit amet consequiteor.",
    severity: "information",
    timestamp: "1h ago",
    title: "Title"
  }
};
