// @canonical/react-components@4.2.0/dist/esm/components/SearchAndFilter/FilterPanelSection/FilterPanelSection.js
import _pt from "prop-types";
import React, { useEffect, useEffectEvent, useRef, useState } from "react";
import Chip from "../../Chip";
import { overflowingChipsCount, isChipInArray } from "../utils";
import { highlightSubString } from "../../../utils";
var FilterPanelSection = _ref => {
  var {
    data,
    searchData,
    searchTerm = "",
    sectionHidden,
    toggleSelected
  } = _ref;
  var {
    chips,
    heading
  } = data;
  var [overflowCounter, setOverflowCounter] = useState(0);
  var [expanded, setExpanded] = useState(false);
  var chipWrapper = useRef(null);
  var handleChipClick = chip => {
    toggleSelected(chip);
  };

  // If the offsetTop is more than double height of a single chip, consider it
  // overflowing
  var updateFlowCount = useEffectEvent(() => {
    var _chipWrapper$current;
    var chips = chipWrapper === null || chipWrapper === void 0 || (_chipWrapper$current = chipWrapper.current) === null || _chipWrapper$current === void 0 ? void 0 : _chipWrapper$current.querySelectorAll(".p-chip");
    var overflowCount = overflowingChipsCount(chips, 2);
    setOverflowCounter(overflowCount);
  });

  // Check if search term characters matches any characters in panel heading
  var searchTermInHeading = highlightSubString(heading, searchTerm).match;

  // Serialise chip values into string so it can be interrogated with subString
  var chipValues = [];
  Object.entries(chips).forEach(chipValue => {
    chipValues.push(chipValue[1].value);
  });

  // Search chips for character match with search term
  var searchTermInChips = highlightSubString(chipValues.toString(), searchTerm).match;
  var panelSectionVisible = searchTermInHeading || searchTermInChips || searchTerm === "";

  // Update overflow count when component is resized
  useEffect(() => {
    var resizeObserverSupported = typeof ResizeObserver !== "undefined";
    var wrapper = chipWrapper === null || chipWrapper === void 0 ? void 0 : chipWrapper.current;
    var wrapperWidthObserver;
    if (resizeObserverSupported && panelSectionVisible) {
      wrapperWidthObserver = new ResizeObserver(() => {
        updateFlowCount();
      });
      wrapperWidthObserver.observe(wrapper);
    } else {
      updateFlowCount();
    }
    return () => {
      var _wrapperWidthObserver;
      resizeObserverSupported && ((_wrapperWidthObserver = wrapperWidthObserver) === null || _wrapperWidthObserver === void 0 ? void 0 : _wrapperWidthObserver.disconnect());
    };
  }, [panelSectionVisible]);

  // When overflow counter is clicked, all chips are shown
  var showAllChips = () => {
    setExpanded(true);
  };
  return /*#__PURE__*/React.createElement(React.Fragment, null, panelSectionVisible && /*#__PURE__*/React.createElement("div", {
    className: "p-filter-panel-section"
  }, heading && chips.length > 0 && /*#__PURE__*/React.createElement("h3", {
    className: "p-filter-panel-section__heading",
    dangerouslySetInnerHTML: {
      __html: highlightSubString(heading, searchTerm).text
    }
  }), /*#__PURE__*/React.createElement("div", {
    className: "p-filter-panel-section__chips",
    "aria-expanded": expanded,
    ref: chipWrapper
  }, chips === null || chips === void 0 ? void 0 : chips.map(chip => {
    // If search term has been added to input, only matching chips
    // should display
    var searchTermInChip = highlightSubString(chip.value, searchTerm).match;
    var chipVisible = searchTermInChip || searchTerm === "" || highlightSubString(heading, searchTerm).match;
    return chipVisible && !sectionHidden && /*#__PURE__*/React.createElement(Chip, {
      key: "".concat(chip.lead, "+").concat(chip.value),
      lead: chip.lead,
      value: chip.value,
      selected: isChipInArray(chip, searchData),
      subString: searchTerm,
      onClick: () => handleChipClick(chip)
    });
  }), overflowCounter > 0 && !expanded && /*#__PURE__*/React.createElement("span", {
    className: "p-filter-panel-section__counter",
    onClick: showAllChips,
    onKeyPress: showAllChips,
    tabIndex: 0
  }, "+", overflowCounter))));
};
FilterPanelSection.propTypes = {
  searchData: _pt.array.isRequired,
  searchTerm: _pt.string.isRequired,
  sectionHidden: _pt.bool,
  toggleSelected: _pt.func.isRequired
};
export default FilterPanelSection;
