// @canonical/react-components@4.2.0/dist/components/SummaryButton/SummaryButton.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.SummaryOnly = exports.Loading = exports.Default = exports.ActionOnly = void 0;
var _react = _interopRequireDefault(require("react"));
var _SummaryButton = _interopRequireDefault(require("./SummaryButton"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const doNothing = () => {};
const meta = {
  component: _SummaryButton.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: () => /*#__PURE__*/_react.default.createElement(_SummaryButton.default, {
    summary: "Showing 15 out of 100 items.",
    label: "Show more",
    onClick: doNothing
  }),
  name: "Default"
};

/**
 * Summary text is optional.
 */
const ActionOnly = exports.ActionOnly = {
  render: () => /*#__PURE__*/_react.default.createElement(_SummaryButton.default, {
    label: "Show more items",
    onClick: doNothing
  }),
  name: "Action only"
};

/**
 * Action is optional as well. When `onClick` prop is not provided component will render just the summary text.
 */
const SummaryOnly = exports.SummaryOnly = {
  render: () => /*#__PURE__*/_react.default.createElement(_SummaryButton.default, {
    summary: "Showing 100 out of 100 items.",
    label: "Show more items",
    onClick: doNothing
  }),
  name: "Summary only"
};

/**
 * When data is loaded asynchronusly `isLoading` prop should be set to `true` to shows a spinner animation in place of the action button.
 */
const Loading = exports.Loading = {
  render: () => /*#__PURE__*/_react.default.createElement(_SummaryButton.default, {
    summary: "Showing 15 out of 100 items.",
    label: "Show more",
    isLoading: true,
    onClick: doNothing
  }),
  name: "Loading"
};
