// @canonical/react-components@4.2.0/dist/components/ScrollableTable/ScrollableTable.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Responsive = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _ScrollableTable = _interopRequireDefault(require("./ScrollableTable"));
var _index = require("../../index");
var _blocks = require("@storybook/blocks");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
const meta = {
  component: _ScrollableTable.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const StoryExample = args => {
  const headers = [{
    content: "Fact"
  }, {
    content: "Relevancy score"
  }, {
    content: "Size"
  }, {
    content: "ID"
  }];
  const facts = ["Dragons are mythical creatures", "They can fly", "They breathe fire", "They hoard treasure", "They are often depicted as large reptiles", "They appear in various cultures' folklore", "They can be friendly or hostile", "They are often associated with wisdom", "They can shapeshift in some legends", "They are sometimes guardians of sacred places", "They can be found in literature and movies", "They are often depicted with wings", "They can be of various colors", "They are sometimes associated with knights"];
  const rows = facts.map(item => {
    return {
      columns: [{
        content: item,
        role: "rowheader"
      }, {
        content: 1
      }, {
        content: "1 GiB"
      }, {
        content: 2
      }]
    };
  });
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("h1", null, "Above contents"), /*#__PURE__*/_react.default.createElement(_ScrollableTable.default, {
    belowIds: args.belowIds,
    dependencies: [headers, rows],
    tableId: "facts-about-dragons"
  }, /*#__PURE__*/_react.default.createElement(_index.MainTable, {
    headers: headers,
    rows: rows,
    responsive: args["responsive"],
    id: "facts-about-dragons"
  })), /*#__PURE__*/_react.default.createElement("div", {
    id: "footer"
  }, /*#__PURE__*/_react.default.createElement("h2", {
    className: "u-no-margin"
  }, "Below contents"), /*#__PURE__*/_react.default.createElement("div", null, "here be dragons.")));
};
const Default = exports.Default = {
  args: {
    belowIds: ["footer"]
  },
  render: StoryExample,
  parameters: {
    docs: {
      page: () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_blocks.Title, null), /*#__PURE__*/_react.default.createElement(_blocks.Subtitle, null), /*#__PURE__*/_react.default.createElement(_blocks.Description, null))
    }
  }
};
const Responsive = exports.Responsive = {
  args: {
    belowIds: ["footer"]
  },
  render: args => StoryExample(_objectSpread(_objectSpread({}, args), {}, {
    responsive: true
  })),
  parameters: {
    docs: {
      page: () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_blocks.Title, null), /*#__PURE__*/_react.default.createElement(_blocks.Subtitle, null), /*#__PURE__*/_react.default.createElement(_blocks.Description, null))
    },
    // Percy settings to capture at multiple widths
    percy: {
      responsiveSnapshotCapture: true,
      widths: [768, 1280]
    }
  }
};
