// @canonical/react-components@4.2.0/dist/components/Notifications/Notification/Notification.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.NotificationSeverity = exports.Label = exports.DefaultTitles = void 0;
var _classnames = _interopRequireDefault(require("classnames"));
var _react = _interopRequireWildcard(require("react"));
var _Button = _interopRequireWildcard(require("../../Button"));
var _utils = require("../../../utils");
const _excluded = ["actions", "borderless", "children", "className", "close", "inline", "messageElement", "onDismiss", "severity", "status", "timeout", "timestamp", "title", "titleElement", "type"];
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
let Label = exports.Label = /*#__PURE__*/function (Label) {
  Label["Close"] = "Close notification";
  return Label;
}({});
const NotificationSeverity = exports.NotificationSeverity = {
  CAUTION: "caution",
  INFORMATION: "information",
  NEGATIVE: "negative",
  POSITIVE: "positive"
};
const DefaultTitles = exports.DefaultTitles = {
  [NotificationSeverity.CAUTION]: "Warning",
  [NotificationSeverity.INFORMATION]: "Info",
  [NotificationSeverity.NEGATIVE]: "Error",
  [NotificationSeverity.POSITIVE]: "Success"
};

/**
 * The props for the Notification component.
 */

/**
This is a [React](https://reactjs.org/) component for the Vanilla [Notification](https://docs.vanillaframework.io/patterns/notification/).

Notifications are used to display global information, our notification variants consist of default, caution, negative or positive.

The `NotificationProvider` component can be used to manage the notification state globally. Wrap your application components with it and then in any child component you can get the helper with `const notify = useNotify()` to easily trigger notifications:

* `notify.info("You should keep this in mind")`
* `notify.failure("Error when fooing a bar", errorObjectFromCatch)`
* `notify.success("The bar was foo'd", "Success")`

The `NotificationConsumer` component can be used to display notifications. It will automatically display any notifications that are added to the `NotificationProvider` state.

The Notification API has changed since v0.18 in order to support new variants
and align with our component prop naming conventions.

| Deprecated prop names | Replaced by |
| --------------------- | ----------- |
| `type`                | `severity`  |
| `status`              | `title`     |
| `close`               | `onDismiss` |

The `notificationTypes` const has also been replaced with `NotificationSeverity`
to reflect the new prop name.
 */
const Notification = _ref => {
  let {
      actions,
      borderless = false,
      children,
      className,
      close,
      inline = false,
      messageElement: MessageComponent = "p",
      onDismiss,
      severity = NotificationSeverity.INFORMATION,
      status,
      timeout,
      timestamp,
      title,
      titleElement: TitleComponent = "h5",
      type
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  const timeoutId = (0, _react.useRef)(null);
  const hasActions = (actions === null || actions === void 0 ? void 0 : actions.length) > 0;
  const showMeta = !!timestamp || hasActions;
  (0, _react.useEffect)(() => {
    if (timeout && onDismiss) {
      timeoutId.current = setTimeout(() => onDismiss(), timeout);
    }
    return () => clearTimeout(timeoutId.current);
  }, [onDismiss, timeout]);
  if (_utils.IS_DEV && (close || status || type)) {
    console.warn("The Notification component is using deprecated props. Refer to the deprecated list for details: https://canonical.github.io/react-components/?path=/docs/notification--information#deprecated");
  }
  return /*#__PURE__*/_react.default.createElement("div", _extends({
    className: (0, _classnames.default)(className, {
      ["p-notification--".concat(severity)]: !!severity,
      "p-notification": !severity,
      "is-borderless": borderless,
      "is-inline": inline
    })
  }, props), /*#__PURE__*/_react.default.createElement("div", {
    className: "p-notification__content"
  }, title && /*#__PURE__*/_react.default.createElement(TitleComponent, {
    className: "p-notification__title",
    "data-testid": "notification-title"
  }, title), inline && /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, "\u2002"), /*#__PURE__*/_react.default.createElement(MessageComponent, {
    className: "p-notification__message"
  }, children), onDismiss && /*#__PURE__*/_react.default.createElement("button", {
    type: "button",
    className: "p-notification__close",
    "data-testid": "notification-close-button",
    onClick: onDismiss
  }, Label.Close)), showMeta && /*#__PURE__*/_react.default.createElement("div", {
    className: "p-notification__meta",
    "data-testid": "notification-meta"
  }, timestamp && /*#__PURE__*/_react.default.createElement("span", {
    className: "p-notification__timestamp",
    "data-testid": "notification-timestamp"
  }, timestamp), hasActions ? /*#__PURE__*/_react.default.createElement("div", {
    className: "p-notification__actions"
  }, actions.map((action, i) => (0, _utils.isReactNode)(action) ? action : /*#__PURE__*/_react.default.createElement(_Button.default, {
    type: "button",
    appearance: _Button.ButtonAppearance.LINK,
    className: "p-notification__action",
    "data-testid": "notification-action",
    key: "".concat(action.label, "-").concat(i),
    onClick: action.onClick
  }, action.label))) : null));
};
var _default = exports.default = Notification;
