// @canonical/react-components@4.2.0/dist/esm/components/Code/Code.js
import _pt from "prop-types";
var _excluded = ["children", "className", "inline", "copyable", "numbered"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import React from "react";
import CodeSnippet, { CodeSnippetBlockAppearance } from "../CodeSnippet";
import { IS_DEV } from "../../utils";
/**
 * @deprecated Code component is deprecated. Use CodeSnippet component or inline `<code>` instead.
 */
var Code = _ref => {
  var {
      children,
      className,
      inline,
      copyable,
      numbered
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  if (IS_DEV) {
    console.warn("Code component is deprecated. Use CodeSnippet component or inline `<code>` instead.");
  }
  if (inline) {
    return /*#__PURE__*/React.createElement("code", _extends({
      className: className
    }, props), children);
  } else {
    var appearance = null;
    if (numbered) {
      appearance = CodeSnippetBlockAppearance.NUMBERED;
    } else if (copyable) {
      appearance = CodeSnippetBlockAppearance.LINUX_PROMPT;
    }
    return /*#__PURE__*/React.createElement(CodeSnippet, {
      blocks: [{
        appearance,
        code: children
      }]
    });
  }
};
Code.propTypes = {
  children: _pt.string.isRequired,
  inline: _pt.bool,
  copyable: _pt.bool,
  numbered: _pt.bool
};
export default Code;
