// @canonical/react-components@4.2.0/dist/esm/components/SearchBox/SearchBox.stories.js
import React from "react";
import SearchBox from "./SearchBox";
var meta = {
  component: SearchBox,
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  name: "Default"
};
export var Disabled = {
  render: () => /*#__PURE__*/React.createElement(SearchBox, {
    disabled: true
  }),
  name: "Disabled"
};

/**
 * If you wish to control the value of the input via external state you can set
the `externallyControlled` prop and provide an `onChange` method to update the
state and the `value` from state.
 */
export var ExternalState = {
  render: () => /*#__PURE__*/React.createElement(SearchBox, {
    externallyControlled: true,
    shouldRefocusAfterReset: true,
    onChange: () => {},
    value: "..."
  }),
  name: "External state"
};
export var Navigation = {
  render: () => /*#__PURE__*/React.createElement("header", {
    id: "navigation",
    className: "p-navigation"
  }, /*#__PURE__*/React.createElement("div", {
    className: "p-navigation__row--full-width"
  }, /*#__PURE__*/React.createElement("div", {
    className: "p-navigation__banner"
  }, /*#__PURE__*/React.createElement("div", {
    className: "p-navigation__logo"
  }, /*#__PURE__*/React.createElement("a", {
    className: "p-navigation__item",
    href: "#test"
  }, /*#__PURE__*/React.createElement("img", {
    className: "p-navigation__image",
    src: "https://assets.ubuntu.com/v1/5d6da5c4-logo-canonical-aubergine.svg",
    alt: "",
    width: "95"
  }))), /*#__PURE__*/React.createElement("a", {
    href: "#navigation",
    className: "p-navigation__toggle--open",
    title: "menu"
  }, "Menu"), /*#__PURE__*/React.createElement("a", {
    href: "#navigation-closed",
    className: "p-navigation__toggle--close",
    title: "close menu"
  }, "Close menu")), /*#__PURE__*/React.createElement("nav", {
    className: "p-navigation__nav"
  }, /*#__PURE__*/React.createElement("span", {
    className: "u-off-screen"
  }, /*#__PURE__*/React.createElement("a", {
    href: "#main-content"
  }, "Jump to main content")), /*#__PURE__*/React.createElement("ul", {
    className: "p-navigation__items",
    role: "menu"
  }, /*#__PURE__*/React.createElement("li", {
    className: "p-navigation__item",
    role: "menuitem"
  }, /*#__PURE__*/React.createElement("a", {
    href: "#test",
    className: "p-navigation__link"
  }, "Products")), /*#__PURE__*/React.createElement("li", {
    className: "p-navigation__item",
    role: "menuitem"
  }, /*#__PURE__*/React.createElement("a", {
    href: "#test",
    className: "p-navigation__link"
  }, "Services")), /*#__PURE__*/React.createElement("li", {
    className: "p-navigation__item",
    role: "menuitem"
  }, /*#__PURE__*/React.createElement("a", {
    href: "#test",
    className: "p-navigation__link"
  }, "Partners")), /*#__PURE__*/React.createElement("li", {
    className: "p-navigation__item",
    role: "menuitem"
  }, /*#__PURE__*/React.createElement("a", {
    href: "#test",
    className: "p-navigation__link"
  }, "About")), /*#__PURE__*/React.createElement("li", {
    className: "p-navigation__item",
    role: "menuitem"
  }, /*#__PURE__*/React.createElement("a", {
    href: "#test",
    className: "p-navigation__link"
  }, "Partners"))), /*#__PURE__*/React.createElement(SearchBox, null)))),
  name: "Navigation"
};
