// @canonical/react-components@4.2.0/dist/esm/components/SearchBox/SearchBox.js
var _excluded = ["name", "id", "autocomplete", "className", "disabled", "externallyControlled", "onChange", "onSearch", "onClear", "placeholder", "shouldBlurOnSearch", "shouldRefocusAfterReset", "value"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import classNames from "classnames";
import React, { useRef } from "react";
import Icon from "../Icon";
export var Label = /*#__PURE__*/function (Label) {
  Label["Clear"] = "Clear search field";
  Label["Search"] = "Search";
  return Label;
}({});
/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [SearchBox](https://docs.vanillaframework.io/patterns/search-box/).
 *
 * Search boxes enable search functionality on a page and are typically used in a navigation bar. SearchBox elements should be wrapped within a form element that handles its own submit action.
 */
var SearchBox = /*#__PURE__*/React.forwardRef((_ref, forwardedRef) => {
  var {
      name = "search",
      id = "search",
      autocomplete = "on",
      className,
      disabled,
      externallyControlled,
      onChange: _onChange,
      onSearch,
      onClear,
      placeholder = "Search",
      shouldBlurOnSearch = true,
      shouldRefocusAfterReset,
      value
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  var internalRef = useRef(null);
  var resetInput = () => {
    _onChange === null || _onChange === void 0 || _onChange("");
    onClear === null || onClear === void 0 || onClear();
    if (internalRef.current) {
      internalRef.current.value = "";
      if (shouldRefocusAfterReset) internalRef.current.focus();
    }
  };
  var triggerSearch = () => {
    onSearch === null || onSearch === void 0 || onSearch(externallyControlled ? value : internalRef.current.value);
  };
  var onKeyDown = e => {
    if (e.key === "Enter" && internalRef.current.checkValidity()) {
      if (shouldBlurOnSearch) {
        internalRef.current.blur();
      }
      triggerSearch();
    }
  };
  return /*#__PURE__*/React.createElement("div", {
    className: classNames("p-search-box", className)
  }, /*#__PURE__*/React.createElement("label", {
    className: "u-off-screen",
    htmlFor: id
  }, placeholder || "Search"), /*#__PURE__*/React.createElement("input", _extends({
    autoComplete: autocomplete,
    className: "p-search-box__input",
    disabled: disabled,
    id: id,
    name: name,
    onChange: evt => _onChange === null || _onChange === void 0 ? void 0 : _onChange(evt.target.value),
    onKeyDown: onKeyDown,
    placeholder: placeholder,
    ref: input => {
      internalRef.current = input;
      // Handle both function and object refs.
      if (typeof forwardedRef === "function") {
        forwardedRef(input);
      } else if (forwardedRef) {
        forwardedRef.current = input;
      }
    },
    type: "search",
    defaultValue: externallyControlled ? undefined : value,
    value: externallyControlled ? value : undefined
  }, props)), value && /*#__PURE__*/React.createElement("button", {
    className: "p-search-box__reset",
    disabled: disabled,
    onClick: resetInput,
    type: "reset"
  }, /*#__PURE__*/React.createElement(Icon, {
    name: "close"
  }, Label.Clear)), /*#__PURE__*/React.createElement("button", {
    className: "p-search-box__button",
    disabled: disabled,
    onClick: triggerSearch
  }, /*#__PURE__*/React.createElement(Icon, {
    name: "search"
  }, Label.Search)));
});
SearchBox.displayName = "SearchBox";
export default SearchBox;
