// @canonical/react-components@4.2.0/dist/components/Form/Form.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Validation = exports.Stacked = exports.Required = exports.Inline = exports.Fieldset = exports.Disabled = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _Form = _interopRequireDefault(require("./Form"));
var _Input = _interopRequireDefault(require("../Input"));
var _Select = _interopRequireDefault(require("../Select"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _Form.default,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: {
        disable: true
      }
    }
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default",
  args: {
    children: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_Input.default, {
      type: "text",
      id: "exampleInputEmail12",
      placeholder: "example@canonical.com",
      label: "Email address"
    }), /*#__PURE__*/_react.default.createElement(_Input.default, {
      type: "password",
      id: "exampleInputPassword12",
      placeholder: "******",
      label: "Password"
    }), /*#__PURE__*/_react.default.createElement(_Input.default, {
      type: "file",
      id: "exampleInputFile2",
      label: "File input"
    }), /*#__PURE__*/_react.default.createElement(_Input.default, {
      type: "checkbox",
      id: "CheckMe2",
      label: "I agree to receive information about Canonical’s products and services."
    }))
  }
};
const Fieldset = exports.Fieldset = {
  render: () => /*#__PURE__*/_react.default.createElement(_Form.default, null, /*#__PURE__*/_react.default.createElement("fieldset", null, /*#__PURE__*/_react.default.createElement(_Input.default, {
    placeholder: "Joe",
    id: "list-input-12",
    type: "text",
    label: "First name"
  }), /*#__PURE__*/_react.default.createElement(_Input.default, {
    placeholder: "Bloggs",
    id: "list-input-22",
    type: "text",
    label: "Last name"
  }), /*#__PURE__*/_react.default.createElement(_Input.default, {
    placeholder: "example@canonical.com",
    id: "list-input-32",
    type: "text",
    label: "Email address"
  }))),
  name: "Fieldset"
};
const Inline = exports.Inline = {
  render: () => /*#__PURE__*/_react.default.createElement(_Form.default, {
    inline: true
  }, /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "text",
    id: "username-inline2",
    className: "p-form__control",
    label: "Username",
    help: "30 characters or fewer."
  }), /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "text",
    id: "address-inline22",
    "aria-invalid": "true",
    label: "Email address",
    error: "Please enter a valid email address."
  })),
  name: "Inline"
};
const Stacked = exports.Stacked = {
  render: () => /*#__PURE__*/_react.default.createElement(_Form.default, {
    stacked: true
  }, /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "text",
    id: "full-name-stacked2",
    label: "Full name",
    stacked: true
  }), /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "text",
    id: "username-stacked2",
    label: "Username",
    stacked: true,
    help: "30 characters or fewer."
  }), /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "text",
    id: "username-stacked-error2",
    label: "Email address",
    stacked: true,
    error: "This field is required"
  }), /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "text",
    id: "address-optional-stacked2",
    label: "Address line 1",
    stacked: true
  }), /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "text",
    id: "address-optional-stacked3",
    label: "Address line 2",
    stacked: true
  })),
  name: "Stacked"
};
const Disabled = exports.Disabled = {
  render: () => /*#__PURE__*/_react.default.createElement(_Form.default, null, /*#__PURE__*/_react.default.createElement(_Input.default, {
    label: "Email address",
    type: "text",
    id: "disabled-input2",
    placeholder: "example@canonical.com",
    disabled: true
  })),
  name: "Disabled"
};
const Validation = exports.Validation = {
  render: () => /*#__PURE__*/_react.default.createElement(_Form.default, null, /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "text",
    id: "exampleTextInputError2",
    placeholder: "example@canonical.com",
    label: "Email address",
    error: "This field is required."
  }), /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "text",
    id: "exampleTextInputCaution2",
    placeholder: "14",
    label: "Mail configuration ID",
    caution: "No validation is performed in preview mode."
  }), /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "text",
    id: "exampleTextInputSuccess2",
    placeholder: "**** **** **** ****",
    label: "Card number",
    success: "Verified."
  }), /*#__PURE__*/_react.default.createElement(_Select.default, {
    id: "exampleSelectInputError3",
    defaultValue: "",
    options: [{
      value: "",
      disabled: true,
      label: "--Select an option--"
    }, {
      value: "1",
      label: "Cosmic Cuttlefish"
    }, {
      value: "2",
      label: "Bionic Beaver"
    }, {
      value: "3",
      label: "Xenial Xerus"
    }],
    label: "Ubuntu releases",
    error: "You need to select an OS to complete your install."
  })),
  name: "Validation"
};
const Required = exports.Required = {
  render: () => /*#__PURE__*/_react.default.createElement(_Form.default, null, /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "text",
    id: "exampleTextInputError3",
    placeholder: "e.g joe@bloggs.com",
    error: "This field is required.",
    label: "Email address",
    required: true
  })),
  name: "Required"
};
