// @canonical/react-components@4.2.0/dist/esm/components/ContextualMenu/ContextualMenu.stories.js
import React from "react";
import Button from "../Button";
import ContextualMenu from "./ContextualMenu";
import Modal from "../Modal";
var ScrollTemplate = args => /*#__PURE__*/React.createElement("div", {
  style: {
    maxWidth: "30rem",
    maxHeight: "10rem",
    overflow: "auto",
    padding: "1rem"
  }
}, /*#__PURE__*/React.createElement(ContextualMenu, args), Array(3).fill(/*#__PURE__*/React.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Consequuntur cum dicta beatae nostrum eligendi similique earum, dolorem fuga quis, sequi voluptates architecto ipsa dolorum eaque rem expedita inventore voluptas odit aspernatur alias molestias facere.")), /*#__PURE__*/React.createElement(ContextualMenu, args));
var Template = args => /*#__PURE__*/React.createElement("div", {
  className: "u-align--center"
}, /*#__PURE__*/React.createElement(ContextualMenu, args));
var meta = {
  component: ContextualMenu,
  render: Template,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: {
        disable: true
      }
    },
    links: {
      control: {
        disable: true
      }
    }
  }
};
export default meta;
/**
 * The contextual menu will provide a visual wrapper to any provided children. Visibility can be toggled via the `visible` prop.
 */
export var Default = {
  name: "Default",
  args: {
    children: /*#__PURE__*/React.createElement("span", {
      style: {
        padding: "1rem"
      }
    }, "This is a menu."),
    closeOnOutsideClick: false,
    constrainPanelWidth: false,
    position: "left",
    visible: true
  }
};
export var Toggle = {
  name: "Toggle",
  args: {
    links: [{
      children: "Link 1",
      onClick: () => {}
    }, {
      children: "Link 2",
      onClick: () => {}
    }],
    hasToggleIcon: true,
    position: "right",
    toggleLabel: "Click me!"
  }
};
export var OverflowingContainer = {
  render: ScrollTemplate.bind({}),
  name: "Overflowing container",
  args: {
    links: [{
      children: "Link 1",
      onClick: () => {}
    }, {
      children: "Long Link 2",
      onClick: () => {}
    }],
    hasToggleIcon: true,
    position: "right",
    toggleLabel: "Click me!"
  }
};
export var ChildFunction = {
  name: "Child function",
  args: {
    children: close => /*#__PURE__*/React.createElement(Button, {
      onClick: close
    }, "child function"),
    hasToggleIcon: true,
    position: "right",
    toggleLabel: "Click me!"
  }
};
export var ChildElement = {
  name: "Child element",
  args: {
    children: /*#__PURE__*/React.createElement(Button, null, "child element"),
    hasToggleIcon: true,
    position: "right",
    toggleLabel: "Click me!"
  }
};
export var InsideModal = {
  name: "Inside modal",
  args: {
    links: Array.from({
      length: 5
    }, (_, index) => ({
      children: "Link ".concat(index + 1),
      onClick: () => {}
    })),
    hasToggleIcon: true,
    position: "right",
    toggleLabel: "Click me!"
  },
  render: args => /*#__PURE__*/React.createElement(Modal, {
    title: "Contextual Menu inside Modal"
  }, /*#__PURE__*/React.createElement(Template, args))
};
