// @canonical/react-components@4.2.0/dist/esm/components/ScrollableTable/ScrollableTable.stories.js
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import React from "react";
import ScrollableTable from "./ScrollableTable";
import { MainTable } from "../../index";
import { Description, Subtitle, Title } from "@storybook/blocks";
var meta = {
  component: ScrollableTable,
  tags: ["autodocs"]
};
export default meta;
var StoryExample = args => {
  var headers = [{
    content: "Fact"
  }, {
    content: "Relevancy score"
  }, {
    content: "Size"
  }, {
    content: "ID"
  }];
  var facts = ["Dragons are mythical creatures", "They can fly", "They breathe fire", "They hoard treasure", "They are often depicted as large reptiles", "They appear in various cultures' folklore", "They can be friendly or hostile", "They are often associated with wisdom", "They can shapeshift in some legends", "They are sometimes guardians of sacred places", "They can be found in literature and movies", "They are often depicted with wings", "They can be of various colors", "They are sometimes associated with knights"];
  var rows = facts.map(item => {
    return {
      columns: [{
        content: item,
        role: "rowheader"
      }, {
        content: 1
      }, {
        content: "1 GiB"
      }, {
        content: 2
      }]
    };
  });
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("h1", null, "Above contents"), /*#__PURE__*/React.createElement(ScrollableTable, {
    belowIds: args.belowIds,
    dependencies: [headers, rows],
    tableId: "facts-about-dragons"
  }, /*#__PURE__*/React.createElement(MainTable, {
    headers: headers,
    rows: rows,
    responsive: args["responsive"],
    id: "facts-about-dragons"
  })), /*#__PURE__*/React.createElement("div", {
    id: "footer"
  }, /*#__PURE__*/React.createElement("h2", {
    className: "u-no-margin"
  }, "Below contents"), /*#__PURE__*/React.createElement("div", null, "here be dragons.")));
};
export var Default = {
  args: {
    belowIds: ["footer"]
  },
  render: StoryExample,
  parameters: {
    docs: {
      page: () => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Title, null), /*#__PURE__*/React.createElement(Subtitle, null), /*#__PURE__*/React.createElement(Description, null))
    }
  }
};
export var Responsive = {
  args: {
    belowIds: ["footer"]
  },
  render: args => StoryExample(_objectSpread(_objectSpread({}, args), {}, {
    responsive: true
  })),
  parameters: {
    docs: {
      page: () => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Title, null), /*#__PURE__*/React.createElement(Subtitle, null), /*#__PURE__*/React.createElement(Description, null))
    },
    // Percy settings to capture at multiple widths
    percy: {
      responsiveSnapshotCapture: true,
      widths: [768, 1280]
    }
  }
};
