// @canonical/react-components@4.2.0/dist/esm/components/LoginPageLayout/LoginPageLayout.js
import _pt from "prop-types";
import React from "react";
import Card from "../Card";
import Col from "../Col";
import Navigation from "../Navigation";
import Row from "../Row";
import { Theme } from "../../enums";
import "./LoginPageLayout.scss";
var defaultLogo = {
  src: "https://assets.ubuntu.com/v1/82818827-CoF_white.svg",
  title: "Canonical",
  url: "/"
};
/**
 * This is a layout component that is used to display a page with a title and children.
 * The LoginPageLayout recommended usages are in the login flow like registration, sign up and error pages.
 */
var LoginPageLayout = _ref => {
  var {
    children,
    title,
    logo = defaultLogo
  } = _ref;
  return /*#__PURE__*/React.createElement(Row, {
    className: "p-strip page-row"
  }, /*#__PURE__*/React.createElement(Col, {
    emptyLarge: 4,
    size: 6
  }, /*#__PURE__*/React.createElement(Card, {
    className: "u-no-padding page-card"
  }, /*#__PURE__*/React.createElement(Navigation, {
    logo: logo,
    theme: Theme.DARK
  }), /*#__PURE__*/React.createElement("div", {
    className: "p-card__inner page-inner"
  }, /*#__PURE__*/React.createElement("h1", {
    className: "p-heading--4"
  }, title), /*#__PURE__*/React.createElement("div", null, children)))));
};
LoginPageLayout.propTypes = {
  title: _pt.string.isRequired,
  children: _pt.node,
  logo: _pt.shape({
    src: _pt.string.isRequired,
    title: _pt.string.isRequired,
    url: _pt.string.isRequired
  })
};
export default LoginPageLayout;
