// @canonical/react-components@4.2.0/dist/esm/components/LoginPageLayout/LoginPageLayout.stories.js
import React from "react";
import CodeSnippet from "../CodeSnippet/CodeSnippet";
import Spinner from "../Spinner/Spinner";
import LoginPageLayout from "./LoginPageLayout";
var meta = {
  component: LoginPageLayout,
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  args: {
    title: "This is the title"
  }
};
export var LoginPage = {
  args: {
    title: "Sign in",
    children: /*#__PURE__*/React.createElement(Spinner, null)
  }
};
export var ErrorPage = {
  args: {
    title: "Sign in failed",
    children: /*#__PURE__*/React.createElement(CodeSnippet, {
      blocks: [{
        wrapLines: true,
        code: /*#__PURE__*/React.createElement(React.Fragment, null, "An error occurred. Try signing in again.")
      }]
    })
  }
};
export var RegistrationPage = {
  args: {
    title: "Sign up",
    children: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("p", null, "Fill in the form below to create an account"), /*#__PURE__*/React.createElement("form", null, /*#__PURE__*/React.createElement("input", {
      type: "text",
      placeholder: "Username"
    }), /*#__PURE__*/React.createElement("input", {
      type: "email",
      placeholder: "Email"
    }), /*#__PURE__*/React.createElement("input", {
      type: "password",
      placeholder: "Password"
    }), /*#__PURE__*/React.createElement("button", null, "Sign up"))),
    logo: {
      src: "https://assets.ubuntu.com/v1/04242fd4-anbox-cloud-logo.svg",
      title: "Anbox Cloud",
      url: "/"
    }
  }
};
