// @canonical/react-components@4.2.0/dist/components/Stepper/Step/Step.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _Step = _interopRequireDefault(require("./Step"));
var _Stepper = _interopRequireDefault(require("../Stepper"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const meta = {
  component: _Step.default,
  render: args => /*#__PURE__*/_react.default.createElement(_Stepper.default, {
    variant: "horizontal",
    steps: [/*#__PURE__*/_react.default.createElement(_Step.default, _extends({
      key: "step"
    }, args))]
  }),
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default",
  args: {
    title: "Step 1",
    index: 1,
    enabled: false,
    hasProgressLine: false,
    iconName: "number",
    handleClick: () => {}
  }
};
