// @canonical/react-components@4.2.0/dist/components/Stepper/Step/Step.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _classnames = _interopRequireDefault(require("classnames"));
var _react = _interopRequireDefault(require("react"));
var _Icon = _interopRequireDefault(require("../../Icon"));
var _Link = _interopRequireDefault(require("../../Link"));
require("./Step.scss");
const _excluded = ["hasProgressLine", "index", "title", "label", "linkProps", "enabled", "selected", "iconName", "iconClassName", "handleClick"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
const Step = _ref => {
  let {
      hasProgressLine,
      index,
      title,
      label,
      linkProps,
      enabled,
      selected = false,
      iconName,
      iconClassName,
      handleClick
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  const stepStatusClass = enabled ? "step-enabled" : "step-disabled";
  return /*#__PURE__*/_react.default.createElement("div", _extends({
    className: (0, _classnames.default)("step", {
      "progress-line": hasProgressLine,
      "step-selected": selected
    })
  }, props), iconName === "number" ? /*#__PURE__*/_react.default.createElement("span", {
    className: (0, _classnames.default)("step-number", {
      "step-number-disabled": !enabled
    })
  }, index) : /*#__PURE__*/_react.default.createElement(_Icon.default, {
    name: iconName,
    className: (0, _classnames.default)("step-status-icon", iconClassName)
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "step-content"
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: (0, _classnames.default)(stepStatusClass),
    onClick: handleClick
  }, title), label && /*#__PURE__*/_react.default.createElement("span", {
    className: (0, _classnames.default)("step-optional-content", "u-no-margin--bottom", {
      "step-disabled": !enabled
    })
  }, label), linkProps && /*#__PURE__*/_react.default.createElement(_Link.default, _extends({
    className: "p-text--small u-no-margin--bottom step-optional-content"
  }, linkProps), linkProps.children)));
};
Step.propTypes = {
  hasProgressLine: _propTypes.default.bool.isRequired,
  index: _propTypes.default.number.isRequired,
  title: _propTypes.default.string.isRequired,
  label: _propTypes.default.string,
  enabled: _propTypes.default.bool.isRequired,
  selected: _propTypes.default.bool,
  iconName: _propTypes.default.string.isRequired,
  handleClick: _propTypes.default.func.isRequired
};
var _default = exports.default = Step;
