// @canonical/react-components@4.2.0/dist/components/FormikField/FormikField.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Fields = exports.Errors = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _FormikField = _interopRequireDefault(require("./FormikField"));
var _Select = _interopRequireDefault(require("../Select"));
var _formik = require("formik");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _FormikField.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  args: {
    name: "username",
    label: "Username",
    type: "text"
  },
  render: args => /*#__PURE__*/_react.default.createElement(_formik.Formik, {
    initialValues: {
      username: ""
    },
    onSubmit: () => {}
  }, /*#__PURE__*/_react.default.createElement(_FormikField.default, args))
};
const Fields = exports.Fields = {
  args: {
    component: _Select.default,
    name: "release",
    label: "Release",
    options: [{
      value: "",
      disabled: true,
      label: "Select an option"
    }, {
      value: "1",
      label: "Cosmic Cuttlefish"
    }, {
      value: "2",
      label: "Bionic Beaver"
    }, {
      value: "3",
      label: "Xenial Xerus"
    }]
  },
  parameters: {
    docs: {
      description: {
        story: "\nAny React Components input can be provided to FormikField (e.g. Input, Textarea or Select) or you may provide a custom component.\n\nAny additional props that need to be passed can be given to FormikField.\n        "
      }
    }
  },
  render: args => /*#__PURE__*/_react.default.createElement(_formik.Formik, {
    initialValues: {
      release: ""
    },
    onSubmit: () => {}
  }, /*#__PURE__*/_react.default.createElement(_FormikField.default, args))
};
const Errors = exports.Errors = {
  args: Default.args,
  parameters: {
    docs: {
      description: {
        story: "\nFormik parameters are passed to the field using Formik's `useField`. This means that validation and errors, state handlers etc. should all just work.\n        "
      }
    }
  },
  render: args => /*#__PURE__*/_react.default.createElement(_formik.Formik, {
    initialErrors: {
      username: "This username has already been taken."
    },
    initialTouched: {
      username: true
    },
    initialValues: {
      username: ""
    },
    onSubmit: () => {}
  }, /*#__PURE__*/_react.default.createElement(_FormikField.default, args))
};
