// @canonical/react-components@4.2.0/dist/components/CheckboxInput/CheckboxInput.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Required = exports.Inline = exports.Indeterminate = exports.Disabled = exports.Default = exports.Children = void 0;
var _react = _interopRequireDefault(require("react"));
var _CheckboxInput = _interopRequireDefault(require("./CheckboxInput"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _CheckboxInput.default,
  tags: ["autodocs"],
  argTypes: {
    label: {
      control: {
        type: "text"
      }
    },
    disabled: {
      control: {
        type: "boolean"
      }
    },
    required: {
      control: {
        type: "boolean"
      }
    },
    inline: {
      control: {
        type: "boolean"
      }
    },
    indeterminate: {
      control: {
        type: "boolean"
      }
    }
  },
  args: {
    label: "CheckboxInput"
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default"
};
const Children = exports.Children = {
  render: () => /*#__PURE__*/_react.default.createElement(_CheckboxInput.default, {
    label: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, "I agree to", /*#__PURE__*/_react.default.createElement("a", {
      href: "http://ubuntu.com/legal"
    }, "Terms and Conditions")),
    name: "CheckboxInput"
  }),
  name: "Label with child elements"
};
const Disabled = exports.Disabled = {
  render: () => /*#__PURE__*/_react.default.createElement(_CheckboxInput.default, {
    label: "Disabled",
    disabled: true
  }),
  name: "Disabled"
};
const Required = exports.Required = {
  render: () => /*#__PURE__*/_react.default.createElement(_CheckboxInput.default, {
    label: "Required",
    required: true
  }),
  name: "Required"
};
const Inline = exports.Inline = {
  render: () => /*#__PURE__*/_react.default.createElement(_CheckboxInput.default, {
    label: "Inline",
    inline: true
  }),
  name: "Inline"
};
const Indeterminate = exports.Indeterminate = {
  render: () => /*#__PURE__*/_react.default.createElement(_CheckboxInput.default, {
    label: "Indeterminate",
    indeterminate: true
  }),
  name: "Indeterminate"
};
