// @canonical/react-components@4.2.0/dist/components/ApplicationLayout/AppMain/AppMain.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = exports.Content = void 0;
var _Application = _interopRequireDefault(require("../Application"));
var _Button = _interopRequireDefault(require("../../Button"));
var _react = _interopRequireDefault(require("react"));
var _Panel = _interopRequireDefault(require("../../Panel"));
var _AppMain = _interopRequireDefault(require("./AppMain"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _AppMain.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
/**
 * In most common cases an `AppMain` should contain a `<Panel>` to display the
 * content as intended in the application layout.
 *
 * `AppMain` should be a direct child of an `<Application>` or when using `ApplicationLayout`
 * it will automatically wrap the component's children.
 */
const Default = exports.Default = {
  args: {
    children: "AppMain"
  }
};
const Content = exports.Content = {
  render: args => {
    return /*#__PURE__*/_react.default.createElement(_Application.default, null, /*#__PURE__*/_react.default.createElement(_AppMain.default, args, /*#__PURE__*/_react.default.createElement(_Panel.default, {
      controls: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_Button.default, {
        appearance: "positive",
        onClick: () => {},
        className: "u-no-margin--bottom"
      }, "Add"), /*#__PURE__*/_react.default.createElement(_Button.default, {
        appearance: "negative",
        onClick: () => {},
        className: "u-no-margin--bottom"
      }, "Delete")),
      title: "App main"
    }, /*#__PURE__*/_react.default.createElement("p", null, "App main content."), /*#__PURE__*/_react.default.createElement("p", null, "Scroll to the right to see the controls."))));
  }
};
