// @canonical/react-components@4.2.0/dist/components/Button/Button.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Small = exports.PositiveDisabled = exports.Positive = exports.NegativeDisabled = exports.Negative = exports.LinkDisabled = exports.Link = exports.Inline = exports.IconText = exports.Icon = exports.DisabledWithTooltip = exports.Dense = exports.DefaultDisabled = exports.Default = exports.ButtonAsLinkDisabled = exports.ButtonAsLink = exports.BrandDisabled = exports.Brand = exports.BaseDisabled = exports.Base = void 0;
var _react = _interopRequireDefault(require("react"));
var _Tooltip = _interopRequireDefault(require("../Tooltip"));
var _Button = _interopRequireWildcard(require("./Button"));
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _Button.default,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: {
        type: "text"
      }
    },
    appearance: {
      control: {
        options: _Button.ButtonAppearance,
        type: "select"
      }
    }
  },
  args: {
    children: "Click me!",
    appearance: _Button.ButtonAppearance.DEFAULT
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default",
  args: {
    children: "Button"
  },
  parameters: {
    docs: {
      description: {
        story: "Button with default appearance"
      }
    }
  }
};
const DefaultDisabled = exports.DefaultDisabled = {
  name: "Default disabled",
  args: {
    children: "Button",
    disabled: true
  },
  parameters: {
    docs: {
      description: {
        story: "Disabled button with default appearance"
      }
    }
  }
};
const Base = exports.Base = {
  name: "Base",
  args: {
    children: "Base button",
    appearance: "base"
  },
  parameters: {
    docs: {
      description: {
        story: "Button with base appearance"
      }
    }
  }
};
const BaseDisabled = exports.BaseDisabled = {
  name: "Base disabled",
  args: {
    children: "Base button disabled",
    appearance: "base",
    disabled: true
  },
  parameters: {
    docs: {
      description: {
        story: "Disabled button with base appearance"
      }
    }
  }
};
const Link = exports.Link = {
  name: "Link",
  args: {
    children: "Link button",
    appearance: "link"
  },
  parameters: {
    docs: {
      description: {
        story: "Button with link appearance"
      }
    }
  }
};
const LinkDisabled = exports.LinkDisabled = {
  name: "Link disabled",
  args: {
    children: "Link button disabled",
    appearance: "link",
    disabled: true
  },
  parameters: {
    docs: {
      description: {
        story: "Disabled button with link appearance"
      }
    }
  }
};
const Positive = exports.Positive = {
  name: "Positive",
  args: {
    children: "Positive button",
    appearance: "positive"
  },
  parameters: {
    docs: {
      description: {
        story: "Button with positive appearance"
      }
    }
  }
};
const PositiveDisabled = exports.PositiveDisabled = {
  name: "Positive disabled",
  args: {
    children: "Positive button disabled",
    appearance: "positive",
    disabled: true
  },
  parameters: {
    docs: {
      description: {
        story: "Disabled button with positive appearance"
      }
    }
  }
};
const Negative = exports.Negative = {
  name: "Negative",
  args: {
    children: "Negative button",
    appearance: "negative"
  },
  parameters: {
    docs: {
      description: {
        story: "Button with negative appearance"
      }
    }
  }
};
const NegativeDisabled = exports.NegativeDisabled = {
  name: "Negative disabled",
  args: {
    children: "Negative button disabled",
    appearance: "negative",
    disabled: true
  },
  parameters: {
    docs: {
      description: {
        story: "Disabled button with negative appearance"
      }
    }
  }
};
const Brand = exports.Brand = {
  name: "Brand",
  args: {
    children: "Brand button",
    appearance: "brand"
  },
  parameters: {
    docs: {
      description: {
        story: "Button with brand appearance"
      }
    }
  }
};
const BrandDisabled = exports.BrandDisabled = {
  name: "Brand disabled",
  args: {
    children: "Brand button disabled",
    appearance: "brand",
    disabled: true
  },
  parameters: {
    docs: {
      description: {
        story: "Disabled button with brand appearance"
      }
    }
  }
};
const Inline = exports.Inline = {
  render: () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("span", null, "Everything you need to get started with Vanilla."), /*#__PURE__*/_react.default.createElement(_Button.default, {
    appearance: "neutral",
    inline: true
  }, "Inline button")),
  name: "Inline",
  parameters: {
    docs: {
      description: {
        story: "Button that displays on the same line as the sibling elements"
      }
    }
  }
};
const Dense = exports.Dense = {
  render: () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("span", null, "Everything you need to get started with Vanilla."), /*#__PURE__*/_react.default.createElement(_Button.default, {
    dense: true
  }, "Dense button")),
  name: "Dense",
  parameters: {
    docs: {
      description: {
        story: "Button with reduced vertical padding"
      }
    }
  }
};
const Small = exports.Small = {
  render: () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_Button.default, {
    small: true
  }, "Small button"), /*#__PURE__*/_react.default.createElement(_Button.default, {
    small: true,
    dense: true
  }, "Small dense button")),
  name: "Small",
  parameters: {
    docs: {
      description: {
        story: "Button that can fit small text inside"
      }
    }
  }
};
const Icon = exports.Icon = {
  name: "Icon",
  args: {
    children: /*#__PURE__*/_react.default.createElement("i", {
      className: "p-icon--plus"
    }),
    hasIcon: true
  },
  parameters: {
    docs: {
      description: {
        story: "Button with an icon inside"
      }
    }
  }
};
const IconText = exports.IconText = {
  name: "Icon & text",
  args: {
    children: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("i", {
      className: "p-icon--plus"
    }), " ", /*#__PURE__*/_react.default.createElement("span", null, "Button with icon & text")),
    hasIcon: true
  },
  parameters: {
    docs: {
      description: {
        story: "Button with both an icon and text inside"
      }
    }
  }
};
const DisabledWithTooltip = exports.DisabledWithTooltip = {
  name: "Disabled with tooltip",
  args: {
    children: "Disabled button with a tooltip",
    disabled: true
  },
  render: args => /*#__PURE__*/_react.default.createElement("div", {
    style: {
      paddingTop: "3rem"
    }
  }, /*#__PURE__*/_react.default.createElement(_Tooltip.default, {
    message: "This button is disabled"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, args))),
  parameters: {
    docs: {
      description: {
        story: "Disabled button that displays a tooltip when hovered"
      }
    }
  }
};
const ButtonAsLink = exports.ButtonAsLink = {
  name: "Button as link",
  args: {
    children: "Button as link",
    appearance: "base",
    element: "a",
    href: "#test"
  },
  parameters: {
    docs: {
      description: {
        story: "Button that is rendered as an `a` element"
      }
    }
  }
};
const ButtonAsLinkDisabled = exports.ButtonAsLinkDisabled = {
  name: "Button as link disabled",
  args: {
    children: "Button as link disabled",
    appearance: "base",
    element: "a",
    href: "#test",
    disabled: true
  },
  parameters: {
    docs: {
      description: {
        story: "Disabled button that is rendered as an `a` element"
      }
    }
  }
};
