// @canonical/react-components@4.2.0/dist/esm/components/Input/Input.stories.js
import React from "react";
import Input from "./Input";
var meta = {
  component: Input,
  tags: ["autodocs"],
  argTypes: {
    caution: {
      control: {
        type: "text"
      }
    },
    error: {
      control: {
        type: "text"
      }
    },
    help: {
      control: {
        type: "text"
      }
    },
    helpClassName: {
      control: {
        type: "text"
      }
    },
    label: {
      control: {
        type: "text"
      }
    },
    success: {
      control: {
        type: "text"
      }
    },
    id: {
      control: {
        disable: true
      }
    },
    placeholder: {
      control: {
        type: "text"
      }
    }
  }
};
export default meta;
export var TextInput = {
  name: "Text input",
  args: {
    type: "text",
    id: "exampleTextInput2",
    placeholder: "example@canonical.com",
    label: "Email address",
    help: "Additional description for the field",
    helpClassName: "u-no-margin--bottom"
  }
};
export var Stacked = {
  render: () => /*#__PURE__*/React.createElement(Input, {
    type: "text",
    id: "exampleTextInput3",
    placeholder: "example@canonical.com",
    stacked: true,
    label: "Email address"
  }),
  name: "Stacked"
};
export var Disabled = {
  render: () => /*#__PURE__*/React.createElement(Input, {
    type: "text",
    id: "exampleTextInput4",
    placeholder: "example@canonical.com",
    disabled: true,
    label: "Email address"
  }),
  name: "Disabled"
};
export var Error = {
  render: () => /*#__PURE__*/React.createElement(Input, {
    type: "text",
    id: "exampleTextInput5",
    placeholder: "example@canonical.com",
    error: "This field is required.",
    label: "Email address"
  }),
  name: "Error"
};
export var Success = {
  render: () => /*#__PURE__*/React.createElement(Input, {
    type: "text",
    id: "exampleTextInput6",
    placeholder: "example@canonical.com",
    success: "Verified.",
    label: "Email address"
  }),
  name: "Success"
};
export var Caution = {
  render: () => /*#__PURE__*/React.createElement(Input, {
    type: "text",
    id: "exampleTextInput7",
    placeholder: "example@canonical.com",
    caution: "No validation is performed in preview mode.",
    label: "Email address"
  }),
  name: "Caution"
};
export var Required = {
  render: () => /*#__PURE__*/React.createElement(Input, {
    type: "text",
    id: "exampleTextInput8",
    placeholder: "example@canonical.com",
    required: true,
    label: "Email address"
  }),
  name: "Required"
};
export var Checkbox = {
  render: () => /*#__PURE__*/React.createElement(Input, {
    type: "checkbox",
    id: "checkExample12",
    defaultChecked: true,
    label: "HTML"
  }),
  name: "Checkbox"
};
export var RadioButton = {
  render: () => /*#__PURE__*/React.createElement(Input, {
    label: "Linux",
    type: "radio",
    name: "RadioOptions",
    id: "Radio12",
    defaultValue: "option1",
    defaultChecked: true,
    help: "Ubuntu"
  }),
  name: "Radio button"
};
