// @canonical/react-components@4.2.0/dist/esm/components/Input/Input.js
var _excluded = ["caution", "className", "error", "help", "helpClassName", "id", "label", "labelClassName", "required", "stacked", "success", "takeFocus", "takeFocusDelay", "type", "wrapperClassName"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import classNames from "classnames";
import React, { useEffect, useId, useRef } from "react";
import Field from "../Field";
import CheckboxInput from "../CheckboxInput";
import RadioInput from "../RadioInput";

/**
 * The props for the Input component.
 */

/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Input](https://docs.vanillaframework.io/base/forms/#input).
 *
 * An input field where the user can enter data, which can vary in many ways, depending on the type attribute.
 */
var Input = _ref => {
  var {
      caution,
      className,
      error,
      help,
      helpClassName,
      id,
      label,
      labelClassName,
      required,
      stacked,
      success,
      takeFocus,
      takeFocusDelay,
      type,
      wrapperClassName
    } = _ref,
    inputProps = _objectWithoutProperties(_ref, _excluded);
  var inputRef = useRef(null);
  var fieldLabel = !["checkbox", "radio"].includes(type) ? label : "";
  var defaultInputId = useId();
  var inputId = id || defaultInputId;
  var validationId = useId();
  var helpId = useId();
  var hasError = !!error;
  var description = [help ? helpId : null, success ? validationId : null].filter(Boolean).join(" ");
  var commonProps = _objectSpread({
    "aria-describedby": !description ? undefined : description,
    "aria-errormessage": hasError ? validationId : null,
    "aria-invalid": hasError,
    id: inputId,
    required: required
  }, inputProps);
  useEffect(() => {
    if (takeFocus) {
      if (takeFocusDelay) {
        setTimeout(() => inputRef.current.focus(), takeFocusDelay);
      } else {
        inputRef.current.focus();
      }
    }
  }, [takeFocus, takeFocusDelay]);
  var input;
  if (type === "checkbox") {
    input = /*#__PURE__*/React.createElement(CheckboxInput, _extends({
      label: label,
      labelClassName: labelClassName
    }, commonProps));
  } else if (type === "radio") {
    input = /*#__PURE__*/React.createElement(RadioInput, _extends({
      label: label,
      labelClassName: labelClassName
    }, commonProps));
  } else {
    input = /*#__PURE__*/React.createElement("input", _extends({
      className: classNames("p-form-validation__input", className),
      ref: inputRef,
      type: type
    }, commonProps));
  }
  return /*#__PURE__*/React.createElement(Field, {
    caution: caution,
    className: wrapperClassName,
    error: error,
    forId: inputId,
    help: help,
    helpClassName: helpClassName,
    helpId: helpId,
    isTickElement: type === "checkbox" || type === "radio",
    label: fieldLabel,
    labelClassName: labelClassName,
    required: required,
    stacked: stacked,
    success: success,
    validationId: validationId
  }, input);
};
export default Input;
