// @canonical/react-components@4.2.0/dist/esm/components/List/List.stories.js
import React from "react";
import List from "./List";
import StatusLabel from "../StatusLabel";
var meta = {
  component: List,
  tags: ["autodocs"],
  argTypes: {
    items: {
      control: {
        disable: true
      }
    },
    stepped: {
      control: {
        disable: true
      }
    }
  }
};
export default meta;
export var Default = {
  name: "Default",
  args: {
    items: ["Digital signage", "Robotics", "Gateways"]
  }
};
export var Ticked = {
  render: () => /*#__PURE__*/React.createElement(List, {
    items: ["Hardware guidance and sizing", "Fixed-price deployment", "Reference architecture"],
    ticked: true
  }),
  name: "Ticked"
};
export var HorizontalDivider = {
  render: () => /*#__PURE__*/React.createElement(List, {
    items: ["Single-command install and upgrade", "Postgres database for reliable state", "High availability for all services"],
    divided: true
  }),
  name: "Horizontal divider"
};
export var TickedWithHorizontalDivider = {
  render: () => /*#__PURE__*/React.createElement(List, {
    items: ["Balance compute load in the cloud", "Optimise for specific workloads", "Assess overcommit ratios"],
    divided: true,
    ticked: true
  }),
  name: "Ticked with horizontal divider"
};
export var Inline = {
  render: () => /*#__PURE__*/React.createElement(List, {
    items: ["Community", "Careers", "Press centre"],
    inline: true
  }),
  name: "Inline"
};
export var Middot = {
  render: () => /*#__PURE__*/React.createElement(List, {
    items: ["Legal information", "Data privacy", "Report a bug on this site"],
    middot: true
  }),
  name: "Middot"
};
export var Stretch = {
  render: () => /*#__PURE__*/React.createElement(List, {
    items: [{
      content: /*#__PURE__*/React.createElement("h3", null, "Documentation")
    }, {
      content: /*#__PURE__*/React.createElement(StatusLabel, {
        appearance: "positive"
      }, "New"),
      className: "u-vertically-center u-align--right"
    }],
    stretch: true
  }),
  name: "Stretch"
};
export var VerticalStepped = {
  render: () => /*#__PURE__*/React.createElement(List, {
    items: [{
      title: "Log in to JAAS",
      content: "Ensure you have an Ubuntu SSO account before contacting JAAS. Log in to JAAS now."
    }, {
      title: "Configure a model",
      content: 'Applications are contained within models and are installed via charms. Configure your model by pressing the "Start a new model" button.'
    }, {
      title: "Credentials and SSH keys",
      content: "After having selected a cloud, a form will appear for submitting your credentials to JAAS. The below resources are available if you need help with gathering credentials."
    }],
    stepped: true
  }),
  name: "Vertical stepped"
};
export var HorizontalStepped = {
  render: () => /*#__PURE__*/React.createElement(List, {
    items: [{
      title: "Log in to JAAS",
      content: "Ensure you have an Ubuntu SSO account before contacting JAAS. Log in to JAAS now."
    }, {
      title: "Configure a model",
      content: 'Applications are contained within models and are installed via charms. Configure your model by pressing the "Start a new model" button.'
    }],
    stepped: true,
    detailed: true
  }),
  name: "Horizontal stepped"
};
export var Split = {
  render: () => /*#__PURE__*/React.createElement(List, {
    items: ["Jointly shape the OpenStack architecture", "We help you plan your cloud hardware requirements", "We build OpenStack in your data center", "We operate the cloud to an SLA", "Transparent audit, logging, monitoring and management", "When your team is ready, we hand over the keys"],
    divided: true,
    split: true,
    ticked: true
  }),
  name: "Split"
};
