// @canonical/react-components@4.2.0/dist/components/CustomSelect/CustomSelect.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.StandardOptions = exports.ManualSearchable = exports.DisabledOptions = exports.CustomOptionsAndSelectedLabel = exports.CustomOptions = exports.AutoSearchable = exports.AsyncSearchable = void 0;
var _CustomSelect = _interopRequireDefault(require("./CustomSelect"));
var _react = _interopRequireWildcard(require("react"));
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectDestructuringEmpty(t) { if (null == t) throw new TypeError("Cannot destructure " + t); }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const generateStandardOptions = num => Array(num).fill(null).map((_, i) => ({
  value: "option-".concat(i + 1),
  label: "Option ".concat(i + 1),
  text: "Option ".concat(i + 1),
  disabled: false
}));
const generateCustomOptions = () => {
  return [{
    value: "smile",
    label: /*#__PURE__*/_react.default.createElement("div", null, "\uD83D\uDE00"),
    text: "Smile",
    disabled: false
  }, {
    value: "grin",
    label: /*#__PURE__*/_react.default.createElement("div", null, "\uD83D\uDE01"),
    text: "Grin",
    disabled: false
  }, {
    value: "cry",
    label: /*#__PURE__*/_react.default.createElement("div", null, "\uD83D\uDE2D"),
    text: "Cry",
    disabled: false
  }, {
    value: "angry",
    label: /*#__PURE__*/_react.default.createElement("div", null, "\uD83D\uDE21"),
    text: "Angry",
    disabled: false
  }, {
    value: "sad",
    label: /*#__PURE__*/_react.default.createElement("div", null, "\uD83D\uDE22"),
    text: "Sad",
    disabled: false
  }];
};
const generateCustomOptionsWithSelectedLabel = () => {
  const options = [{
    type: "ovn",
    name: "ovntest",
    config: {
      "security.acls": "foo,bar"
    }
  }, {
    type: "bridge",
    name: "lxdbr0",
    config: {}
  }, {
    type: "bridge",
    name: "microbr0",
    config: {}
  }, {
    type: "macvlan",
    name: "macvlantest",
    config: {}
  }].map(network => {
    var _network$config$secur;
    return {
      label: /*#__PURE__*/_react.default.createElement("div", {
        className: "label",
        style: {
          display: "flex",
          gap: "5px"
        }
      }, /*#__PURE__*/_react.default.createElement("span", {
        title: network.name,
        className: "network-option u-truncate",
        style: {
          width: "12rem"
        }
      }, network.name), /*#__PURE__*/_react.default.createElement("span", {
        title: network.type,
        className: "network-option u-truncate",
        style: {
          width: "8rem"
        }
      }, network.type), /*#__PURE__*/_react.default.createElement("span", {
        title: "network ACLs",
        className: "network-option u-truncate u-align--right",
        style: {
          paddingRight: "8px",
          width: "4rem"
        }
      }, ((_network$config$secur = network.config["security.acls"]) === null || _network$config$secur === void 0 ? void 0 : _network$config$secur.length) || "-")),
      value: network.name,
      text: "".concat(network.name, " - ").concat(network.type),
      disabled: false,
      selectedLabel: /*#__PURE__*/_react.default.createElement("span", null, network.name, "\xA0", /*#__PURE__*/_react.default.createElement("span", {
        className: "u-text--muted"
      }, "(", network.type, ")"))
    };
  });
  return options;
};
const getHeader = () => {
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "header",
    style: {
      backgroundColor: "$colors--theme--background-alt",
      display: "flex",
      gap: "$sph--small",
      padding: "$sph--x-small $sph--small",
      position: "sticky",
      top: 0
    }
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "network-option u-no-margin--bottom",
    style: {
      color: "$colors--theme--text-default",
      width: "12rem"
    }
  }, "Name"), /*#__PURE__*/_react.default.createElement("span", {
    className: "network-option u-no-margin--bottom",
    style: {
      color: "$colors--theme--text-default",
      width: "8rem"
    }
  }, "Type"), /*#__PURE__*/_react.default.createElement("span", {
    className: "network-option u-no-margin--bottom",
    style: {
      color: "$colors--theme--text-default",
      width: "4rem"
    }
  }, "ACLs"));
};
const Template = _ref => {
  let props = _extends({}, (_objectDestructuringEmpty(_ref), _ref));
  const [selected, setSelected] = (0, _react.useState)(props.value || "");
  return /*#__PURE__*/_react.default.createElement(_CustomSelect.default, _extends({}, props, {
    value: selected,
    onChange: value => setSelected(value)
  }));
};
const meta = {
  component: _CustomSelect.default,
  render: Template,
  tags: ["autodocs"],
  args: {
    name: "customSelect",
    label: "Custom Select",
    defaultToggleLabel: "Select an option",
    searchable: "auto",
    searchPlaceholder: "Search",
    initialPosition: "left"
  },
  argTypes: {
    searchable: {
      options: ["auto", "always", "async", "never"],
      control: {
        type: "select"
      }
    },
    initialPosition: {
      options: ["left", "right"],
      control: {
        type: "select"
      }
    }
  }
};
var _default = exports.default = meta;
/**
 * If `label` is of `string` type. You do not have to do anything extra to render it.
 */
const StandardOptions = exports.StandardOptions = {
  args: {
    options: generateStandardOptions(10)
  }
};

/**
 * If `label` is of `ReactNode` type. You can render custom content.
 * In this case, the `text` property for each option is required and is used for display in the toggle, search and sort functionalities.
 */
const CustomOptions = exports.CustomOptions = {
  args: {
    options: generateCustomOptions()
  }
};

/**
 * If `label` is of `ReactNode` type. You can render custom content.
 * In this case, the `selectedLabel` for each option is provided and will be displayed in the toggle instead of `text`
 * The `text` property for each option is still required and is used for search and sort functionalities.
 */
const CustomOptionsAndSelectedLabel = exports.CustomOptionsAndSelectedLabel = {
  args: {
    options: generateCustomOptionsWithSelectedLabel(),
    header: getHeader(),
    dropdownClassName: "network-select-dropdown"
  }
};

/**
 * For each option, if `disabled` is set to `true`, the option will be disabled.
 */
const DisabledOptions = exports.DisabledOptions = {
  args: {
    options: generateStandardOptions(5).map((option, i) => _objectSpread(_objectSpread({}, option), {}, {
      disabled: i % 2 === 0
    }))
  }
};

/**
 * Search is enabled by default when there are 5 or more options.
 */
const AutoSearchable = exports.AutoSearchable = {
  args: {
    options: generateStandardOptions(5),
    searchable: "auto"
  }
};

/**
 * Search can be enabled manually by setting `searchable` to `always`.
 */
const ManualSearchable = exports.ManualSearchable = {
  args: {
    options: generateStandardOptions(4),
    searchable: "always"
  }
};

/**
 * Search can be enabled manually by setting `searchable` to `async`.
 * This will always show the search input regardless of the number of options.
 */
const AsyncSearchable = exports.AsyncSearchable = {
  args: {
    options: generateStandardOptions(0),
    searchable: "async"
  }
};
