// @canonical/react-components@4.2.0/dist/esm/components/FormikField/FormikField.stories.js
import React from "react";
import FormikField from "./FormikField";
import Select from "../Select";
import { Formik } from "formik";
var meta = {
  component: FormikField,
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  args: {
    name: "username",
    label: "Username",
    type: "text"
  },
  render: args => /*#__PURE__*/React.createElement(Formik, {
    initialValues: {
      username: ""
    },
    onSubmit: () => {}
  }, /*#__PURE__*/React.createElement(FormikField, args))
};
export var Fields = {
  args: {
    component: Select,
    name: "release",
    label: "Release",
    options: [{
      value: "",
      disabled: true,
      label: "Select an option"
    }, {
      value: "1",
      label: "Cosmic Cuttlefish"
    }, {
      value: "2",
      label: "Bionic Beaver"
    }, {
      value: "3",
      label: "Xenial Xerus"
    }]
  },
  parameters: {
    docs: {
      description: {
        story: "\nAny React Components input can be provided to FormikField (e.g. Input, Textarea or Select) or you may provide a custom component.\n\nAny additional props that need to be passed can be given to FormikField.\n        "
      }
    }
  },
  render: args => /*#__PURE__*/React.createElement(Formik, {
    initialValues: {
      release: ""
    },
    onSubmit: () => {}
  }, /*#__PURE__*/React.createElement(FormikField, args))
};
export var Errors = {
  args: Default.args,
  parameters: {
    docs: {
      description: {
        story: "\nFormik parameters are passed to the field using Formik's `useField`. This means that validation and errors, state handlers etc. should all just work.\n        "
      }
    }
  },
  render: args => /*#__PURE__*/React.createElement(Formik, {
    initialErrors: {
      username: "This username has already been taken."
    },
    initialTouched: {
      username: true
    },
    initialValues: {
      username: ""
    },
    onSubmit: () => {}
  }, /*#__PURE__*/React.createElement(FormikField, args))
};
