// @canonical/react-components@4.2.0/dist/esm/components/Field/Field.js
var _excluded = ["caution", "children", "className", "error", "forId", "help", "helpClassName", "helpId", "isSelect", "isTickElement", "label", "labelClassName", "labelFirst", "required", "stacked", "success", "validationId"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import classNames from "classnames";
import React from "react";
import Label from "../Label";
import Col from "../Col";

/**
 * The props for the Field component.
 */

var generateHelpText = _ref => {
  var {
    help,
    helpId,
    helpClassName,
    isTickElement
  } = _ref;
  return help ? /*#__PURE__*/React.createElement("p", {
    className: classNames("p-form-help-text", helpClassName, {
      "is-tick-element": isTickElement
    }),
    id: helpId
  }, help) : null;
};
var generateError = (error, caution, success, validationId) => {
  if (!error && !caution && !success) {
    return null;
  }
  return /*#__PURE__*/React.createElement("p", {
    className: "p-form-validation__message",
    id: validationId
  }, error || caution || success);
};
var generateLabel = (forId, required, label, labelClassName, stacked) => {
  if (!label) {
    return null;
  }
  var labelNode = /*#__PURE__*/React.createElement(Label, {
    className: labelClassName,
    forId: forId,
    required: required
  }, label);
  if (stacked) {
    return /*#__PURE__*/React.createElement(Col, {
      size: 4
    }, labelNode);
  }
  return labelNode;
};
var generateContent = _ref2 => {
  var {
    isSelect,
    children,
    labelFirst,
    labelNode,
    help,
    helpClassName,
    error,
    caution,
    success,
    validationId,
    helpId,
    isTickElement
  } = _ref2;
  return /*#__PURE__*/React.createElement("div", {
    className: "p-form__control u-clearfix"
  }, isSelect ? /*#__PURE__*/React.createElement("div", {
    className: "p-form-validation__select-wrapper"
  }, children) : children, !labelFirst && labelNode, generateHelpText({
    helpId,
    help,
    helpClassName,
    isTickElement
  }), generateError(error, caution, success, validationId));
};
var Field = _ref3 => {
  var {
      caution,
      children,
      className,
      error,
      forId,
      help,
      helpClassName,
      helpId,
      isSelect,
      isTickElement,
      label,
      labelClassName,
      labelFirst = true,
      required,
      stacked,
      success,
      validationId
    } = _ref3,
    props = _objectWithoutProperties(_ref3, _excluded);
  var labelNode = generateLabel(forId, required, label, labelClassName, stacked);
  var content = generateContent({
    isSelect,
    isTickElement,
    children,
    labelFirst,
    labelNode,
    help,
    helpClassName,
    error,
    caution,
    success,
    validationId,
    helpId
  });
  return /*#__PURE__*/React.createElement("div", _extends({
    className: classNames("p-form__group", "p-form-validation", className, {
      "is-error": error,
      "is-caution": caution,
      "is-success": success,
      row: stacked
    })
  }, props), labelFirst && labelNode, stacked ? /*#__PURE__*/React.createElement(Col, {
    size: 8
  }, content) : content);
};
export default Field;
