// @canonical/react-components@4.2.0/dist/components/Panel/Panel.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Logo = exports.Header = exports.Default = exports.CustomHeader = void 0;
var _react = _interopRequireDefault(require("react"));
var _Panel = _interopRequireDefault(require("./Panel"));
var _Button = _interopRequireDefault(require("../Button"));
var _Icon = _interopRequireDefault(require("../Icon"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _Panel.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  args: {
    children: "Panel content",
    title: "Panel"
  }
};
const Header = exports.Header = {
  args: {
    controls: /*#__PURE__*/_react.default.createElement(_Button.default, {
      appearance: "positive"
    }, /*#__PURE__*/_react.default.createElement(_Icon.default, {
      name: "plus",
      light: true
    }), " Create"),
    title: "Panel title",
    titleComponent: "h1"
  }
};

/**
 * The logo may be provided as attributes to use the standard logo. If this is
 * not sufficient the a `ReactNode` can be passed to the `logo` prop instead.
 */
const Logo = exports.Logo = {
  args: {
    logo: {
      icon: "https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg",
      name: "https://assets.ubuntu.com/v1/a85f7947-juju_black-text-only.svg",
      nameAlt: "Juju"
    }
  }
};

/**
 * If the default header does not meet your needs then a `ReactNode` can be
 * passed to the `header` prop to replace the header.
 */
const CustomHeader = exports.CustomHeader = {
  args: {
    header: /*#__PURE__*/_react.default.createElement("div", {
      className: "p-panel__header"
    }, "This header replaces the entire header area")
  }
};
