// @canonical/react-components@4.2.0/dist/hooks/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "useClickOutside", {
  enumerable: true,
  get: function () {
    return _useOnClickOutside.useClickOutside;
  }
});
Object.defineProperty(exports, "useId", {
  enumerable: true,
  get: function () {
    return _useId.useId;
  }
});
Object.defineProperty(exports, "useListener", {
  enumerable: true,
  get: function () {
    return _useListener.useListener;
  }
});
Object.defineProperty(exports, "useOnClickOutside", {
  enumerable: true,
  get: function () {
    return _useOnClickOutside.useOnClickOutside;
  }
});
Object.defineProperty(exports, "useOnEscapePressed", {
  enumerable: true,
  get: function () {
    return _useOnEscapePressed.useOnEscapePressed;
  }
});
Object.defineProperty(exports, "usePagination", {
  enumerable: true,
  get: function () {
    return _usePagination.usePagination;
  }
});
Object.defineProperty(exports, "usePrefersReducedMotion", {
  enumerable: true,
  get: function () {
    return _usePrefersReducedMotion.usePrefersReducedMotion;
  }
});
Object.defineProperty(exports, "usePrevious", {
  enumerable: true,
  get: function () {
    return _usePrevious.usePrevious;
  }
});
Object.defineProperty(exports, "useThrottle", {
  enumerable: true,
  get: function () {
    return _useThrottle.useThrottle;
  }
});
Object.defineProperty(exports, "useWindowFitment", {
  enumerable: true,
  get: function () {
    return _useWindowFitment.useWindowFitment;
  }
});
var _useOnClickOutside = require("./useOnClickOutside");
var _useId = require("./useId");
var _useListener = require("./useListener");
var _useOnEscapePressed = require("./useOnEscapePressed");
var _usePrevious = require("./usePrevious");
var _useThrottle = require("./useThrottle");
var _usePagination = require("./usePagination");
var _usePrefersReducedMotion = require("./usePrefersReducedMotion");
var _useWindowFitment = require("./useWindowFitment");
