// @canonical/react-components@4.2.0/dist/esm/components/Stepper/Stepper.stories.js
import React from "react";
import Stepper from "./Stepper";
import Step from "./Step";
var meta = {
  component: Stepper,
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  name: "Default",
  args: {
    variant: "vertical",
    steps: [/*#__PURE__*/React.createElement(Step, {
      key: "Default Step 1",
      title: "Step 1",
      index: 1,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      handleClick: () => {}
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Default Step 2",
      title: "Step 2",
      index: 2,
      enabled: true,
      hasProgressLine: true,
      iconName: "spinner",
      iconClassName: "u-animation--spin",
      handleClick: () => {}
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Default Step 3",
      title: "Step 3",
      index: 3,
      enabled: false,
      hasProgressLine: false,
      iconName: "number",
      handleClick: () => {}
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Default Step 4",
      title: "Step 4",
      index: 4,
      enabled: false,
      hasProgressLine: false,
      iconName: "number",
      handleClick: () => {}
    })]
  }
};
export var HorizontalStepper = {
  name: "Horizontal Stepper",
  render: () => /*#__PURE__*/React.createElement(Stepper, {
    variant: "horizontal",
    steps: [/*#__PURE__*/React.createElement(Step, {
      key: "Horizontal Step 1",
      title: "Step 1",
      index: 1,
      enabled: false,
      hasProgressLine: true,
      iconName: "success",
      handleClick: () => {}
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Horizontal Step 2",
      title: "Step 2",
      index: 2,
      enabled: false,
      hasProgressLine: true,
      iconName: "error",
      handleClick: () => {}
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Horizontal Step 3",
      title: "Step 3",
      index: 3,
      enabled: true,
      hasProgressLine: true,
      iconName: "spinner",
      iconClassName: "u-animation--spin",
      handleClick: () => {}
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Horizontal Step 4",
      title: "Step 4",
      index: 4,
      enabled: false,
      hasProgressLine: false,
      iconName: "number",
      handleClick: () => {}
    })]
  })
};
export var VerticalOptional = {
  name: "Vertical variant: Optional labels and links",
  render: () => /*#__PURE__*/React.createElement(Stepper, {
    steps: [/*#__PURE__*/React.createElement(Step, {
      key: "Vertical optional Step 1",
      title: "Step 1",
      index: 1,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      label: "Optional label",
      handleClick: () => {}
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Vertical optional Step 2",
      title: "Step 2",
      index: 2,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      handleClick: () => {},
      label: "Optional label"
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Vertical optional Step 3",
      title: "Step 3",
      index: 3,
      enabled: true,
      hasProgressLine: true,
      iconName: "number",
      label: "Optional label",
      linkProps: {
        children: "Optional link"
      },
      handleClick: () => {}
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Vertical optional Step 4",
      title: "Step 4",
      index: 4,
      enabled: false,
      hasProgressLine: false,
      linkProps: {
        children: "Optional link"
      },
      iconName: "number",
      handleClick: () => {}
    })]
  })
};
export var HorizontalOptional = {
  name: "Horizontal variant: Optional labels and links",
  render: () => /*#__PURE__*/React.createElement(Stepper, {
    variant: "horizontal",
    steps: [/*#__PURE__*/React.createElement(Step, {
      key: "Horizontal optional Step 1",
      title: "Step 1",
      index: 1,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      label: "Optional label",
      handleClick: () => {}
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Horizontal optional Step 2",
      title: "Step 2",
      index: 2,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      handleClick: () => {},
      label: "Optional label"
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Horizontal optional Step 3",
      title: "Step 3",
      index: 3,
      enabled: true,
      hasProgressLine: true,
      iconName: "number",
      label: "Optional label",
      linkProps: {
        children: "Optional link"
      },
      handleClick: () => {}
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Horizontal optional Step 4",
      title: "Step 4",
      index: 4,
      enabled: false,
      hasProgressLine: false,
      iconName: "number",
      linkProps: {
        children: "Optional link"
      },
      handleClick: () => {}
    })]
  })
};
export var VerticalSelected = {
  name: "Vertical variant: Selected step",
  render: () => /*#__PURE__*/React.createElement("div", {
    style: {
      width: "50%"
    }
  }, /*#__PURE__*/React.createElement(Stepper, {
    steps: [/*#__PURE__*/React.createElement(Step, {
      key: "Vertical Selected Step 1",
      title: "Step 1",
      index: 1,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      label: "Optional label",
      handleClick: () => {}
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Vertical Selected Step 2",
      title: "Step 2",
      index: 2,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      handleClick: () => {},
      label: "Optional label",
      selected: true
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Vertical Selected Step 3",
      title: "Step 3",
      index: 3,
      enabled: true,
      hasProgressLine: true,
      iconName: "number",
      label: "Optional label",
      linkProps: {
        children: "Optional link"
      },
      handleClick: () => {}
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Vertical Selected Step 4",
      title: "Step 4",
      index: 4,
      enabled: false,
      hasProgressLine: false,
      linkProps: {
        children: "Optional link"
      },
      iconName: "number",
      label: "Optional label",
      handleClick: () => {}
    })]
  }))
};
export var HorizontalSelected = {
  name: "Horizontal variant: Selected step",
  render: () => /*#__PURE__*/React.createElement(Stepper, {
    variant: "horizontal",
    steps: [/*#__PURE__*/React.createElement(Step, {
      key: "Horizontal Selected Step 1",
      title: "Step 1",
      index: 1,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      label: "Optional label",
      handleClick: () => {}
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Horizontal Selected Step 2",
      title: "Step 2",
      index: 2,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      selected: true,
      handleClick: () => {},
      label: "Optional label"
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Horizontal Selected Step 3",
      title: "Step 3",
      index: 3,
      enabled: true,
      hasProgressLine: true,
      iconName: "number",
      label: "Optional label",
      linkProps: {
        children: "Optional link"
      },
      handleClick: () => {}
    }), /*#__PURE__*/React.createElement(Step, {
      key: "Horizontal Selected Step 4",
      title: "Step 4",
      index: 4,
      enabled: false,
      hasProgressLine: false,
      iconName: "number",
      label: "Optional label",
      linkProps: {
        children: "Optional link"
      },
      handleClick: () => {}
    })]
  })
};
