// @canonical/react-components@4.2.0/dist/esm/components/ModularTable/ModularTable.stories.js
import { useState } from "react";
import React from "react";
import SummaryButton from "../SummaryButton";
import ModularTable from "./ModularTable";
import { ICONS } from "../Icon";
var getSampleData = function getSampleData() {
  var i = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 0;
  return [{
    data: ++i * 100 + i
  }, {
    data: ++i * 100 + i
  }, {
    data: ++i * 100 + i
  }, {
    data: ++i * 100 + i
  }, {
    data: ++i * 100 + i
  }];
};
var meta = {
  component: ModularTable,
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  render: () => /*#__PURE__*/React.createElement(ModularTable, {
    getCellProps: _ref => {
      var {
        value
      } = _ref;
      return {
        className: value === "1 minute" ? "p-heading--5" : ""
      };
    }
    // eslint-disable-next-line react-hooks/rules-of-hooks
    ,
    columns: React.useMemo(() => [{
      Header: "ID",
      accessor: "buildId",
      Cell: function (_Cell) {
        function Cell(_x) {
          return _Cell.apply(this, arguments);
        }
        Cell.toString = function () {
          return _Cell.toString();
        };
        return Cell;
      }(_ref2 => {
        var {
          value
        } = _ref2;
        return /*#__PURE__*/React.createElement("a", {
          href: "#test"
        }, "#", value);
      })
    }, {
      Header: "Architecture",
      accessor: "arch"
    }, {
      Header: "Build Duration",
      accessor: "duration",
      className: "u-hide--small"
    }, {
      Header: "Result",
      accessor: "result",
      Cell: function (_Cell2) {
        function Cell(_x2) {
          return _Cell2.apply(this, arguments);
        }
        Cell.toString = function () {
          return _Cell2.toString();
        };
        return Cell;
      }(_ref3 => {
        var {
          value
        } = _ref3;
        switch (value) {
          case "released":
            return "Released";
          case "failed":
            return "Failed";
          default:
            return "Unknown";
        }
      }),
      getCellIcon: _ref4 => {
        var {
          value
        } = _ref4;
        switch (value) {
          case "released":
            return ICONS.success;
          case "failed":
            return ICONS.error;
          default:
            return false;
        }
      }
    }, {
      Header: "Build Finished",
      accessor: "finished",
      className: "u-align-text--right"
    }], [])
    // eslint-disable-next-line react-hooks/rules-of-hooks
    ,
    data: React.useMemo(() => [{
      buildId: "5432",
      arch: "arm64",
      duration: "5 minutes",
      result: "released",
      finished: "10 minutes ago"
    }, {
      buildId: "1234",
      arch: "armhf",
      duration: "5 minutes",
      result: "failed",
      finished: "over 1 year ago"
    }, {
      buildId: "1111",
      arch: "test64",
      duration: "1 minute",
      result: "other",
      finished: "ages ago"
    }], [])
  }),
  name: "Default"
};
export var Empty = {
  render: () => /*#__PURE__*/React.createElement(ModularTable, {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    columns: React.useMemo(() => [{
      Header: "ID",
      accessor: "buildId"
    }, {
      Header: "Architecture",
      accessor: "arch"
    }, {
      Header: "Build Duration",
      accessor: "duration",
      className: "u-hide--small"
    }, {
      Header: "Result",
      accessor: "result"
    }, {
      Header: "Build Finished",
      accessor: "finished",
      className: "u-align-text--right"
    }], []),
    data: [],
    emptyMsg: "Waiting for data..."
  }),
  name: "Empty"
};
export var Sortable = {
  render: () => /*#__PURE__*/React.createElement(ModularTable, {
    sortable: true,
    initialSortColumn: "duration",
    initialSortDirection: "ascending",
    getCellProps: _ref5 => {
      var {
        value
      } = _ref5;
      return {
        className: value === "1 minute" ? "p-heading--5" : ""
      };
    }
    // eslint-disable-next-line react-hooks/rules-of-hooks
    ,
    columns: React.useMemo(() => [{
      Header: "ID",
      accessor: "buildId",
      sortType: "basic",
      Cell: function (_Cell3) {
        function Cell(_x3) {
          return _Cell3.apply(this, arguments);
        }
        Cell.toString = function () {
          return _Cell3.toString();
        };
        return Cell;
      }(_ref6 => {
        var {
          value
        } = _ref6;
        return /*#__PURE__*/React.createElement("a", {
          href: "#test"
        }, "#", value);
      })
    }, {
      Header: "Architecture",
      accessor: "arch",
      sortType: "basic"
    }, {
      Header: "Build Duration",
      accessor: "duration",
      className: "u-hide--small",
      sortType: "basic"
    }, {
      Header: "Result",
      accessor: "result",
      sortType: "basic",
      Cell: function (_Cell4) {
        function Cell(_x4) {
          return _Cell4.apply(this, arguments);
        }
        Cell.toString = function () {
          return _Cell4.toString();
        };
        return Cell;
      }(_ref7 => {
        var {
          value
        } = _ref7;
        switch (value) {
          case "released":
            return "Released";
          case "failed":
            return "Failed";
          default:
            return "Unknown";
        }
      }),
      getCellIcon: _ref8 => {
        var {
          value
        } = _ref8;
        switch (value) {
          case "released":
            return ICONS.success;
          case "failed":
            return ICONS.error;
          default:
            return false;
        }
      }
    }, {
      Header: "Build Finished",
      accessor: "finished",
      className: "u-align-text--right",
      sortType: "basic"
    }], [])
    // eslint-disable-next-line react-hooks/rules-of-hooks
    ,
    data: React.useMemo(() => [{
      buildId: "5432",
      arch: "arm64",
      duration: "5 minutes",
      result: "released",
      finished: "10 minutes ago"
    }, {
      buildId: "1234",
      arch: "armhf",
      duration: "5 minutes",
      result: "failed",
      finished: "over 1 year ago"
    }, {
      buildId: "1111",
      arch: "test64",
      duration: "1 minute",
      result: "other",
      finished: "ages ago"
    }], [])
  }),
  name: "Sortable"
};

/**
 * Subrows can be used to group table rows under a heading that needs to appear
at the start of the group. In the following example sorting the columns will
sort the group rows, then sort the sub-rows, but the group header always comes first.
 */
export var Subrows = {
  render: () => /*#__PURE__*/React.createElement(ModularTable, {
    sortable: true
    // eslint-disable-next-line react-hooks/rules-of-hooks
    ,
    columns: React.useMemo(() => [{
      Header: "Flavour",
      accessor: "flavour",
      sortType: "basic",
      Cell: function (_Cell5) {
        function Cell(_x5) {
          return _Cell5.apply(this, arguments);
        }
        Cell.toString = function () {
          return _Cell5.toString();
        };
        return Cell;
      }(props => "depth" in props.row && props.row.depth === 0 ? /*#__PURE__*/React.createElement("strong", null, props.value) : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("input", {
        type: "checkbox"
      }), " ", props.value))
    }, {
      Header: "Scoops",
      accessor: "scoops",
      sortType: "basic",
      Cell: function (_Cell6) {
        function Cell(_x6) {
          return _Cell6.apply(this, arguments);
        }
        Cell.toString = function () {
          return _Cell6.toString();
        };
        return Cell;
      }(props => "depth" in props.row && props.row.depth === 0 ? /*#__PURE__*/React.createElement("span", {
        className: "u-text--muted"
      }, props.value) : props.value)
    }], [])
    // eslint-disable-next-line react-hooks/rules-of-hooks
    ,
    data: React.useMemo(() => [{
      flavour: "Sorbet",
      scoops: "1-2",
      subRows: [{
        flavour: "Lemon",
        scoops: 2
      }, {
        flavour: "Mango",
        scoops: 1
      }, {
        flavour: "Raspberry",
        scoops: 2
      }]
    }, {
      flavour: "Ice-cream",
      scoops: "1-3",
      subRows: [{
        flavour: "Chocolate",
        scoops: 1
      }, {
        flavour: "Vanilla",
        scoops: 3
      }, {
        flavour: "Caramel",
        scoops: 2
      }]
    }], [])
  }),
  name: "Subrows"
};
export var Subheading = {
  render: () => /*#__PURE__*/React.createElement(ModularTable, {
    getCellProps: _ref9 => {
      var {
        value
      } = _ref9;
      return {
        className: value === "1 minute" ? "p-heading--5" : ""
      };
    }
    // eslint-disable-next-line react-hooks/rules-of-hooks
    ,
    columns: React.useMemo(() => [{
      Header: "ID",
      accessor: "buildId",
      Cell: function (_Cell7) {
        function Cell(_x7) {
          return _Cell7.apply(this, arguments);
        }
        Cell.toString = function () {
          return _Cell7.toString();
        };
        return Cell;
      }(_ref0 => {
        var {
          value
        } = _ref0;
        return /*#__PURE__*/React.createElement("a", {
          href: "#test"
        }, "#", value);
      })
    }, {
      Header: "Architecture",
      accessor: "arch"
    }, {
      Header: "Build Duration",
      accessor: "duration",
      className: "u-hide--small"
    }, {
      Header: "Result",
      accessor: "result",
      Cell: function (_Cell8) {
        function Cell(_x8) {
          return _Cell8.apply(this, arguments);
        }
        Cell.toString = function () {
          return _Cell8.toString();
        };
        return Cell;
      }(_ref1 => {
        var {
          value
        } = _ref1;
        switch (value) {
          case "released":
            return "Released";
          case "failed":
            return "Failed";
          default:
            return "Unknown";
        }
      }),
      getCellIcon: _ref10 => {
        var {
          value
        } = _ref10;
        switch (value) {
          case "released":
            return ICONS.success;
          case "failed":
            return ICONS.error;
          default:
            return false;
        }
      }
    }, {
      Header: "Build Finished",
      accessor: "finished",
      className: "u-align-text--right"
    }], [])
    // eslint-disable-next-line react-hooks/rules-of-hooks
    ,
    data: React.useMemo(() => [{
      buildId: "5432",
      arch: "arm64",
      duration: "5 minutes",
      result: "released",
      finished: "10 minutes ago"
    }, {
      buildId: "1234",
      arch: "armhf",
      duration: "5 minutes",
      result: "failed",
      finished: "over 1 year ago"
    }, {
      buildId: "1111",
      arch: "test64",
      duration: "1 minute",
      result: "other",
      finished: "ages ago"
    }], []),
    subhead: /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("th", {
      colSpan: 5,
      className: "p-text--default",
      style: {
        textTransform: "none"
      }
    }, "Showing 3 of 3 results"))
  }),
  name: "Subheading"
};

/**
Example below shows a basic `ModularTable` with `SummaryButton` component in the footer.
```
export const getSampleData = (i = 0) => [
  { data: ++i * 100 + i },
  { data: ++i * 100 + i },
  { data: ++i * 100 + i },
  { data: ++i * 100 + i },
  { data: ++i * 100 + i },
];
```
 */
export var LoadMore = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [data, setData] = useState(getSampleData());
    var hasMore = data.length < 20;
    var loadMore = () => {
      setData(data.concat(getSampleData(data.length)));
    };
    var columns = [{
      Header: "Id",
      accessor: (_d, i) => i
    }, {
      Header: "Data",
      accessor: "data"
    }];
    var footer = hasMore ? /*#__PURE__*/React.createElement("div", {
      className: "u-align--right"
    }, /*#__PURE__*/React.createElement(SummaryButton, {
      summary: "Showing ".concat(data.length, " of 20 items."),
      label: "Show 5 more",
      onClick: loadMore
    })) : null;
    return /*#__PURE__*/React.createElement(ModularTable, {
      data: data,
      columns: columns,
      footer: footer
    });
  },
  name: "Load more"
};
