// @canonical/react-components@4.2.0/dist/esm/components/ModularTable/ModularTable.js
var _excluded = ["data", "columns", "emptyMsg", "footer", "sortable", "getHeaderProps", "getRowProps", "getCellProps", "getRowId", "initialSortColumn", "initialSortDirection", "autoResetSortBy", "subhead"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import React, { useMemo, isValidElement } from "react";
import { useTable, useSortBy } from "react-table";
import Table from "../Table";
import TableRow from "../TableRow";
import TableHeader from "../TableHeader";
import TableCell from "../TableCell";
import Icon from "../Icon";
var generateCell = (cell, getCellProps) => {
  var _cell$column$getCellI, _cell$column;
  var hasColumnIcon = cell.column.getCellIcon;
  var iconName = hasColumnIcon && ((_cell$column$getCellI = (_cell$column = cell.column).getCellIcon) === null || _cell$column$getCellI === void 0 ? void 0 : _cell$column$getCellI.call(_cell$column, cell));
  return /*#__PURE__*/React.createElement(TableCell, cell.getCellProps([{
    className: cell.column.className
  }, {
    className: hasColumnIcon ? "p-table__cell--icon-placeholder" : ""
  }, _objectSpread({}, getCellProps === null || getCellProps === void 0 ? void 0 : getCellProps(cell))]), iconName && /*#__PURE__*/React.createElement(Icon, {
    name: iconName
  }), cell.render("Cell"));
};
var generateRows = (rows, prepareRow, getRowProps, getCellProps) => {
  var tableRows = [];
  rows.forEach(row => {
    var _row$subRows;
    // This function is responsible for lazily preparing a row for rendering.
    // Any row that you intend to render in your table needs to be passed to this function before every render.
    // see: https://react-table.tanstack.com/docs/api/useTable#instance-properties
    prepareRow(row);
    tableRows.push(/*#__PURE__*/React.createElement(TableRow, row.getRowProps(getRowProps === null || getRowProps === void 0 ? void 0 : getRowProps(row)), row.cells.map(cell => generateCell(cell, getCellProps))));
    if ((_row$subRows = row.subRows) !== null && _row$subRows !== void 0 && _row$subRows.length) {
      tableRows = tableRows.concat(generateRows(row.subRows, prepareRow, getRowProps, getCellProps));
    }
  });
  return tableRows;
};

/**
This is a [React](https://reactjs.org/) component to support many table use cases.
 
ModularTable components accepts `columns` and `data` arguments in the same format as [`useTable`](https://react-table.tanstack.com/docs/api/useTable) hook of the ReactTable library.

`columns` - The core columns configuration object for the entire table. https://react-table.tanstack.com/docs/api/useTable#column-options
`data` - The data array that you want to display on the table.
### Important note!
Values passed to both of these params have to be memoized (for example via{" "}
  <code>React.useMemo</code>). Memoization ensures that our data isn't recreated
  on every render. If we didn't use <code>React.useMemo</code>, the table would
  think it was receiving new data on every render and attempt to recalulate a
  lot of logic every single time.

#### Custom column options

In addition to standard column propeties from [`useTable`](https://react-table.tanstack.com/docs/api/useTable) `ModularTable` accepts some custom properties.

##### Class names

Custom `className` can be used to provide a specific CSS class name that will be added to all cells in given column.
You can also provide `getHeaderProps`, `getRowProps` and `getCellProps` to resolve additional custom props for a specific element. More on this in [`useTable - cell properties`](https://react-table.tanstack.com/docs/api/useTable#cell-properties).

```js
getCellProps={({ value, column }) => ({
  className: `table__cell--${column.id} ${value ===  "1 minute" ? "p-heading--5" : ""}`,
})}
columns = {
  Header: "Hidden on mobile",
  accessor: "example",
  className: "u-hide--small"
}
```

##### Icons

To show an icon in the cells of a column `getCellIcon` function should be defined. The function takes a cell data as an argument and should return one of built in icons (from the `ICONS` const), a string with a custom icon name, or `false` if no icon should be shown.

The `ICONS` const contains all [the Vanilla built in icons](https://vanillaframework.io/docs/patterns/icons) such as "plus", "minus", "success", "error", etc.

Product specific icons can be used as well, assuming that the product provides the necessary CSS styling and the icon follows the Vanilla naming convention `p-icon--{name}`.

```js
columns = {
  Header: "With icons",
  accessor: "status",
  getCellIcon: ({ value }) => {
    return value === "released" ? ICONS.success : false;
  },
};
```
 */
function ModularTable(_ref) {
  var {
      data,
      columns,
      emptyMsg,
      footer,
      sortable,
      getHeaderProps,
      getRowProps,
      getCellProps,
      getRowId,
      initialSortColumn,
      initialSortDirection,
      autoResetSortBy = false,
      subhead
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  var sortBy = useMemo(() => initialSortColumn ? [{
    id: initialSortColumn,
    desc: initialSortDirection === "descending"
  }] : [], [initialSortColumn, initialSortDirection]);
  var {
    getTableProps,
    getTableBodyProps,
    headerGroups,
    rows,
    prepareRow
  } = useTable({
    columns,
    data,
    getRowId: getRowId || undefined,
    initialState: {
      sortBy
    },
    autoResetSortBy
  }, sortable ? useSortBy : undefined);
  var showEmpty = !!emptyMsg && (!rows || rows.length === 0);

  // Function returns whether table can be sorted by a specific column.
  // Returns true if sorting is enabled for the column and there is text
  // or JSX provided for the header, otherwise returns false.
  var isColumnSortable = column => column.canSort && (/*#__PURE__*/isValidElement(column.Header) || (typeof column.Header === "string" || typeof column.Header === "number") && !!String(column.Header).trim());
  var getColumnSortDirection = column => {
    if (!isColumnSortable(column)) {
      return undefined;
    }
    if (!column.isSorted) {
      return "none";
    }
    return column.isSortedDesc ? "descending" : "ascending";
  };
  return /*#__PURE__*/React.createElement(Table, _extends({}, getTableProps(), props), /*#__PURE__*/React.createElement("thead", null, headerGroups.map((headerGroup, i) => /*#__PURE__*/React.createElement(TableRow, _extends({}, headerGroup.getHeaderGroupProps(), {
    key: i
  }), headerGroup.headers.map((column, j) => /*#__PURE__*/React.createElement(TableHeader, _extends({
    key: j,
    sort: getColumnSortDirection(column)
  }, column.getHeaderProps([{
    className: column.className
  }, {
    className: column.getCellIcon ? "p-table__cell--icon-placeholder" : ""
  }, _objectSpread({}, getHeaderProps === null || getHeaderProps === void 0 ? void 0 : getHeaderProps(column)),
  // Only call this if we want it to be sortable too.
  sortable && isColumnSortable(column) ? column.getSortByToggleProps({
    title: undefined
  }) : {}])), column.render("Header"))))), subhead), /*#__PURE__*/React.createElement("tbody", getTableBodyProps(), generateRows(rows, prepareRow, getRowProps, getCellProps), showEmpty && /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableCell, {
    colSpan: columns.length
  }, emptyMsg)), footer && /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableCell, {
    colSpan: columns.length
  }, footer))));
}
export default ModularTable;
