// @canonical/react-components@4.2.0/dist/components/Modal/Modal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Modal = void 0;
var _classnames = _interopRequireDefault(require("classnames"));
var _react = _interopRequireWildcard(require("react"));
const _excluded = ["buttonRow", "children", "className", "close", "focusRef", "title", "shouldPropagateClickEvent", "closeOnOutsideClick"];
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Modal](https://docs.vanillaframework.io/patterns/modal/).
 *
 * The modal component can be used to overlay an area of the screen which can contain a prompt, dialog or interaction.
 */
const Modal = _ref => {
  let {
      buttonRow,
      children,
      className,
      close,
      focusRef,
      title,
      shouldPropagateClickEvent = false,
      closeOnOutsideClick = true
    } = _ref,
    wrapperProps = _objectWithoutProperties(_ref, _excluded);
  // list of focusable selectors is based on this Stack Overflow answer:
  // https://stackoverflow.com/a/30753870/3732840
  const focusableElementSelectors = 'a[href]:not([tabindex="-1"]), button:not([disabled]), textarea:not([disabled]):not([tabindex="-1"]), input:not([disabled]):not([tabindex="-1"]), select:not([disabled]):not([tabindex="-1"]), area[href]:not([tabindex="-1"]), iframe:not([tabindex="-1"]), [tabindex]:not([tabindex="-1"]), [contentEditable=true]:not([tabindex="-1"])';
  const descriptionId = (0, _react.useId)();
  const titleId = (0, _react.useId)();
  const shouldClose = (0, _react.useRef)(false);
  const modalRef = (0, _react.useRef)(null);
  const closeButtonRef = (0, _react.useRef)(null);
  const handleTabKey = event => {
    const focusableModalElements = modalRef.current.querySelectorAll(focusableElementSelectors);
    if (focusableModalElements.length > 0) {
      const firstElement = focusableModalElements[0];
      const lastElement = focusableModalElements[focusableModalElements.length - 1];
      if (!event.shiftKey && document.activeElement === lastElement) {
        firstElement.focus();
        event.preventDefault();
      }
      if (event.shiftKey && document.activeElement === firstElement) {
        lastElement.focus();
        return event.preventDefault();
      }
    }
  };
  const handleEscKey = event => {
    if ("nativeEvent" in event && event.nativeEvent.stopImmediatePropagation) {
      event.nativeEvent.stopImmediatePropagation();
    } else if ("stopImmediatePropagation" in event) {
      event.stopImmediatePropagation();
    } else if (event.stopPropagation) {
      event.stopPropagation();
    }
    if (close) {
      close();
    }
  };
  (0, _react.useEffect)(() => {
    if (focusRef !== null && focusRef !== void 0 && focusRef.current) {
      focusRef.current.focus();
    } else if (closeButtonRef.current) {
      closeButtonRef.current.focus();
    } else {
      modalRef.current.focus();
    }
  }, [focusRef]);
  (0, _react.useEffect)(() => {
    const keyListenersMap = new Map([["Escape", handleEscKey], ["Tab", handleTabKey]]);
    const keyDown = event => {
      const listener = keyListenersMap.get(event.code);
      return listener && listener(event);
    };
    document.addEventListener("keydown", keyDown);
    return () => {
      document.removeEventListener("keydown", keyDown);
    };
  });
  const handleContentOnMouseDown = () => {
    shouldClose.current = false;
  };
  const handleContentOnMouseUp = () => {
    shouldClose.current = false;
  };
  const handleOverlayOnMouseDown = event => {
    if (event.target === modalRef.current && closeOnOutsideClick) {
      shouldClose.current = true;
    }
  };
  const handleClose = event => {
    if (!shouldPropagateClickEvent) {
      event.stopPropagation();
    }
    if (close) {
      close();
    }
  };
  const handleOverlayOnClick = event => {
    if (shouldClose.current) {
      handleClose(event);
    }
  };
  return /*#__PURE__*/_react.default.createElement("div", _extends({
    className: (0, _classnames.default)("p-modal", className),
    onClick: handleOverlayOnClick,
    onMouseDown: handleOverlayOnMouseDown
  }, wrapperProps, {
    ref: modalRef
  }), /*#__PURE__*/_react.default.createElement("section", {
    className: "p-modal__dialog",
    role: "dialog",
    "aria-modal": "true",
    "aria-labelledby": titleId,
    "aria-describedby": descriptionId,
    onMouseDown: handleContentOnMouseDown,
    onMouseUp: handleContentOnMouseUp
  }, !!title && /*#__PURE__*/_react.default.createElement("header", {
    className: "p-modal__header"
  }, /*#__PURE__*/_react.default.createElement("h2", {
    className: "p-modal__title",
    id: titleId
  }, title), close && /*#__PURE__*/_react.default.createElement("button", {
    type: "button",
    className: "p-modal__close",
    "aria-label": "Close active modal",
    onClick: handleClose,
    ref: closeButtonRef
  }, "Close")), /*#__PURE__*/_react.default.createElement("div", {
    id: descriptionId
  }, children), !!buttonRow && /*#__PURE__*/_react.default.createElement("footer", {
    className: "p-modal__footer"
  }, buttonRow)));
};
exports.Modal = Modal;
var _default = exports.default = Modal;
