// @canonical/react-components@4.2.0/dist/components/Modal/Modal.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Focus = exports.Default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _Button = _interopRequireDefault(require("../Button"));
var _Modal = _interopRequireDefault(require("./Modal"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const Template = args => {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      height: "100vh",
      position: "relative"
    }
  }, /*#__PURE__*/_react.default.createElement(_Modal.default, args));
};
const meta = {
  component: _Modal.default,
  render: Template,
  tags: ["autodocs"],
  argTypes: {
    buttonRow: {
      control: {
        disable: true
      }
    },
    children: {
      control: {
        disable: true
      }
    },
    closeOnOutsideClick: {
      control: "boolean"
    }
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: _ref => {
    let {
      closeOnOutsideClick
    } = _ref;
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [modalOpen, setModalOpen] = (0, _react.useState)(true);
    const closeHandler = () => setModalOpen(false);
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
      onClick: () => setModalOpen(true)
    }, "Open modal"), modalOpen ? /*#__PURE__*/_react.default.createElement(_Modal.default, {
      close: closeHandler,
      title: "Confirm delete",
      closeOnOutsideClick: closeOnOutsideClick,
      buttonRow: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
        className: "u-no-margin--bottom",
        onClick: closeHandler
      }, "Cancel"), /*#__PURE__*/_react.default.createElement("button", {
        className: "p-button--negative u-no-margin--bottom"
      }, "Delete"))
    }, /*#__PURE__*/_react.default.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/_react.default.createElement("br", null), "You cannot undo this action.")) : null);
  },
  name: "Default"
};
const Focus = exports.Focus = {
  render: _ref2 => {
    let {
      closeOnOutsideClick
    } = _ref2;
    /* eslint-disable react-hooks/rules-of-hooks */
    const [modalOpen, setModalOpen] = (0, _react.useState)(true);
    const buttonRef = (0, _react.useRef)(null);
    /* eslint-enable react-hooks/rules-of-hooks */

    const closeHandler = () => setModalOpen(false);
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
      onClick: () => setModalOpen(true)
    }, "Open modal"), modalOpen ? /*#__PURE__*/_react.default.createElement(_Modal.default, {
      close: closeHandler,
      title: "Confirm delete",
      closeOnOutsideClick: closeOnOutsideClick,
      buttonRow: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
        className: "u-no-margin--bottom",
        onClick: closeHandler
      }, "Cancel"), /*#__PURE__*/_react.default.createElement("button", {
        className: "p-button--negative u-no-margin--bottom"
      }, "Delete")),
      focusRef: buttonRef
    }, /*#__PURE__*/_react.default.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/_react.default.createElement("br", null), "You cannot undo this action."), /*#__PURE__*/_react.default.createElement("p", null, /*#__PURE__*/_react.default.createElement(_Button.default, {
      appearance: "link",
      ref: buttonRef
    }, "More information"))) : null);
  },
  name: "Focus"
};
