// @canonical/react-components@4.2.0/dist/esm/utils.js
import { isValidElement } from "react";
export var IS_DEV = process.env.NODE_ENV === "development";

/**
 * Find substring and wrap in <strong /> tag
 * @param {string} str - The string to search
 * @param {string} subString - The substring to find
 * @return {Obj} newStr - Object with text and match bool
 */
export var highlightSubString = (str, subString) => {
  if (typeof str !== "string" || typeof subString !== "string") {
    return {
      text: str || "",
      match: false
    };
  }
  var escapedSubstring = subString.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
  var caseInsensitiveRegex = new RegExp(escapedSubstring, "gi");
  var newStr = str.replace(caseInsensitiveRegex, match => "<strong>".concat(match, "</strong>"));
  return {
    text: subString === "" ? str : newStr,
    match: newStr !== str
  };
};

/**
 * Whether a navigation item is an anchor.
 * @param link - The navigation item.
 */
export var isNavigationAnchor = link => !!link.url;

/**
 * Whether a navigation item is a button.
 * @param link - The navigation item.
 */
export var isNavigationButton = link => !link.url;

/**
 * A typeguard for whether an element is a ReactNode.
 */
export var isReactNode = element => /*#__PURE__*/isValidElement(element);
export var getElementAbsoluteHeight = element => {
  if (!element) {
    return 0;
  }
  var style = window.getComputedStyle(element);
  var margin = toFloat(style.marginTop) + toFloat(style.marginBottom);
  return element.offsetHeight + margin + 1;
};
export var getAbsoluteHeightBelowById = belowId => {
  var element = belowId ? document.getElementById(belowId) : undefined;
  if (!element) {
    return 0;
  }
  return getElementAbsoluteHeight(element);
};
export var getParentsBottomSpacing = element => {
  var sum = 0;
  while (element.parentElement) {
    element = element.parentElement;
    var style = window.getComputedStyle(element);
    var margin = toFloat(style.marginBottom);
    var padding = toFloat(style.paddingBottom);
    sum += margin + padding;
  }
  return sum;
};
export var toFloat = value => {
  var result = parseFloat(value);
  return Number.isNaN(result) ? 0 : result;
};
