// @canonical/react-components@4.2.0/dist/esm/components/CodeSnippet/CodeSnippetBlock.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import classNames from "classnames";
import React from "react";
import CodeSnippetDropdown from "./CodeSnippetDropdown";
export var CodeSnippetBlockAppearance = {
  LINUX_PROMPT: "linuxPrompt",
  NUMBERED: "numbered",
  URL: "url",
  WINDOWS_PROMPT: "windowsPrompt"
};
export default function CodeSnippetBlock(_ref) {
  var {
    appearance,
    code,
    content,
    dropdowns,
    stacked = false,
    title,
    wrapLines = false
  } = _ref;
  var className = "p-code-snippet__block";
  var isNumbered = appearance === CodeSnippetBlockAppearance.NUMBERED;
  var hasIcon = appearance === CodeSnippetBlockAppearance.LINUX_PROMPT || appearance === CodeSnippetBlockAppearance.WINDOWS_PROMPT || appearance === CodeSnippetBlockAppearance.URL;
  var numberedCode;
  if (isNumbered) {
    className += "--numbered";

    // wrap code lines in spans (and preserve the whitespace)
    var lines;
    if (Array.isArray(code)) {
      lines = code;
    } else if (typeof code === "string") {
      lines = code.split(/\r?\n/);
    } else {
      // If the code is not able to be split over multiple lines, then display
      // a single line number.
      lines = [code];
    }
    numberedCode = lines.map((line, i) => /*#__PURE__*/React.createElement(React.Fragment, {
      key: "p-code-snippet__line-".concat(i)
    }, /*#__PURE__*/React.createElement("span", {
      className: "p-code-snippet__line"
    }, line), "\n"));
  } else if (hasIcon) {
    className += "--icon";
  }
  className = classNames(className, {
    "is-windows-prompt": appearance === CodeSnippetBlockAppearance.WINDOWS_PROMPT,
    "is-url": appearance === CodeSnippetBlockAppearance.URL,
    "is-wrapped": wrapLines
  });
  var hasDropdowns = dropdowns && dropdowns.length;
  return /*#__PURE__*/React.createElement(React.Fragment, null, (title || hasDropdowns) && /*#__PURE__*/React.createElement("div", {
    className: "p-code-snippet__header ".concat(stacked ? "is-stacked" : "")
  }, /*#__PURE__*/React.createElement("h5", {
    className: "p-code-snippet__title"
  }, title), hasDropdowns && /*#__PURE__*/React.createElement("div", {
    className: "p-code-snippet__dropdowns"
  }, dropdowns.map((dropdown, i) => /*#__PURE__*/React.createElement(CodeSnippetDropdown, _extends({}, dropdown, {
    key: "code-snippet-dropdown-".concat(i)
  }))))), /*#__PURE__*/React.createElement("pre", {
    className: className
  }, /*#__PURE__*/React.createElement("code", null, isNumbered ? numberedCode : code)), content);
}
