// @canonical/react-components@4.2.0/dist/esm/components/CodeSnippet/CodeSnippet.stories.js
import React from "react";
import { useState } from "react";
import CodeSnippet from "./CodeSnippet";
import { CodeSnippetBlockAppearance } from "./CodeSnippetBlock";
import Button from "../Button";
var meta = {
  component: CodeSnippet,
  tags: ["autodocs"],
  argTypes: {
    className: {
      control: {
        type: "text"
      }
    }
  }
};
export default meta;
export var Default = {
  render: () => /*#__PURE__*/React.createElement(CodeSnippet, {
    blocks: [{
      code: "Mode: all Settings: maas_url=http://192.168.122.1:5240/MAAS"
    }]
  }),
  name: "Default"
};

/**
 * Use the `title` prop to add a title to a code block.
 */
export var Title = {
  render: () => /*#__PURE__*/React.createElement(CodeSnippet, {
    blocks: [{
      title: "Output",
      code: "Mode: all Settings: maas_url=http://192.168.122.1:5240/MAAS"
    }]
  }),
  name: "Title"
};

/**
 * Pass an array of `blocks` to render multiple code blocks (for example input and output) within a single code snippet.
 */
export var MultipleBlocks = {
  render: () => /*#__PURE__*/React.createElement(CodeSnippet, {
    blocks: [{
      title: "JavaScript",
      code: "console.log('Vanilla');"
    }, {
      title: "Output",
      code: "Vanilla"
    }]
  }),
  name: "Multiple blocks"
};

/**
 * Use `appearance` to control the visual aspects of the code block. `"numbered"` appearance will add line numbers in the code blocks.
 * Values of `"linuxPrompt"`, `"windowsPrompt"`, `"url"` will add a relevant icon in the code block.
 */
export var Appearance = {
  render: () => /*#__PURE__*/React.createElement(CodeSnippet, {
    blocks: [{
      title: "Install on Linux",
      appearance: CodeSnippetBlockAppearance.LINUX_PROMPT,
      code: "snap install toto"
    }, {
      title: "Install on Windows",
      appearance: CodeSnippetBlockAppearance.WINDOWS_PROMPT,
      code: "snap install toto"
    }, {
      title: "Get from the Store",
      appearance: CodeSnippetBlockAppearance.URL,
      code: "http://snapcraft.io/toto"
    }]
  }),
  name: "Appearance"
};

/**
 * Set `wrapLines` prop to `true` to enable line wrapping inside the code block.
 */
export var WrapLines = {
  render: () => /*#__PURE__*/React.createElement(CodeSnippet, {
    blocks: [{
      appearance: CodeSnippetBlockAppearance.NUMBERED,
      wrapLines: true,
      code: "#!/bin/bash\nset -eu . $CONJURE_UP_SPELLSDIR/sdk/common.sh\nif [[ \"$JUJU_PROVIDERTYPE\" == \"lxd\" ]]; then\n  debug \"Running pre-deploy for $CONJURE_UP_SPELL\"\n  sed \"s/##MODEL##/$JUJU_MODEL/\" $(scriptPath)/lxd-profile.yaml | lxc profile edit \"juju-$JUJU_MODEL\" || exposeResult \"Failed to set profile\" $? \"false\"\nfi\nexposeResult \"Successful pre-deploy.\" 0 \"true\""
    }]
  }),
  name: "Wrap lines"
};

/**
 * Dropdown select menus can be added to the headers of code blocks to allow users to choose one of the options. The dropdowns options are passed via `dropdowns` property in the block options object.
 *
 * With the `options` being an array of option properties compatible with the `Select` options (`[{ label: string, value: string | number }]`).
 */
export var Dropdown = {
  render: () => {
    var code = {
      js: "console.log(\"Example 1\");",
      css: ".p-heading--2 { color: red; }",
      html: "<h1 class=\"p-heading--2\">How to use code snippets</h1>"
    };

    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [lang, setLang] = useState("html");
    return /*#__PURE__*/React.createElement(CodeSnippet, {
      blocks: [{
        code: code[lang],
        dropdowns: [{
          options: [{
            value: "js",
            label: "JS"
          }, {
            value: "css",
            label: "CSS"
          }, {
            value: "html",
            label: "HTML"
          }],
          value: lang,
          onChange: event => {
            setLang(event.target.value);
          }
        }]
      }]
    });
  },
  name: "Dropdown"
};

/**
 * Multiple dropdowns can be passed in if needed.
 */
export var Dropdowns = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [channel, setChannel] = useState("stable");
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [snap, setSnap] = useState("firefox");
    var code = "sudo snap install ".concat(snap, " ").concat(channel === "stable" ? "" : "--" + channel);
    return /*#__PURE__*/React.createElement(CodeSnippet, {
      blocks: [{
        title: "Install snap",
        code: code,
        dropdowns: [{
          options: [{
            value: "stable",
            label: "stable"
          }, {
            value: "candidate",
            label: "candidate"
          }, {
            value: "beta",
            label: "beta"
          }, {
            value: "edge",
            label: "edge"
          }],
          value: channel,
          onChange: event => {
            setChannel(event.target.value);
          }
        }, {
          options: [{
            value: "firefox",
            label: "Firefox"
          }, {
            value: "gimp",
            label: "Gimp"
          }, {
            value: "vlc",
            label: "VLC"
          }],
          value: snap,
          onChange: event => {
            setSnap(event.target.value);
          }
        }]
      }]
    });
  },
  name: "Dropdowns"
};

/**
 * If multiple dropdowns may overlap with long title you can use `stacked` variant, by setting the relevant property on code block options.
 */
export var DropdownsStacked = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [channel, setChannel] = useState("stable");
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [snap, setSnap] = useState("firefox");
    var code = "sudo snap install ".concat(snap, " ").concat(channel === "stable" ? "" : "--" + channel);
    return /*#__PURE__*/React.createElement(CodeSnippet, {
      blocks: [{
        title: "Install Firefox, Gimp or VLC as a snap from different channels using command line",
        code: code,
        stacked: true,
        dropdowns: [{
          options: [{
            value: "stable",
            label: "stable"
          }, {
            value: "candidate",
            label: "candidate"
          }, {
            value: "beta",
            label: "beta"
          }, {
            value: "edge",
            label: "edge"
          }],
          value: channel,
          onChange: event => {
            setChannel(event.target.value);
          }
        }, {
          options: [{
            value: "firefox",
            label: "Firefox"
          }, {
            value: "gimp",
            label: "Gimp"
          }, {
            value: "vlc",
            label: "VLC"
          }],
          value: snap,
          onChange: event => {
            setSnap(event.target.value);
          }
        }]
      }]
    });
  },
  name: "DropdownsStacked"
};

/**
 * Custom elements can be passed to a CodeBlock via the `content` prop. In these cases, a border will be added around the entire code snippet to visually associate the content with the code block.
 */
export var Content = {
  render: () => /*#__PURE__*/React.createElement(CodeSnippet, {
    blocks: [{
      title: "A button example",
      code: "<Button>Button</Button>",
      content: /*#__PURE__*/React.createElement("div", {
        style: {
          margin: "0.5rem"
        }
      }, /*#__PURE__*/React.createElement(Button, null, "Button"))
    }]
  }),
  name: "Content"
};

/**
 * It's possible to pass JSX instead of strings to the `code` parameter, either as a single element or an array (e.g. if you want to display line numbers).
 */
export var JsxCodeElements = {
  render: () => /*#__PURE__*/React.createElement(CodeSnippet, {
    blocks: [{
      code: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("strong", null, "snap"), " install ", /*#__PURE__*/React.createElement("a", {
        href: "#test"
      }, "toto"), /*#__PURE__*/React.createElement("br", null), /*#__PURE__*/React.createElement("strong", null, "apt"), " install ", /*#__PURE__*/React.createElement("a", {
        href: "#test"
      }, "toto"))
    }]
  }),
  name: "JSX code elements"
};
