// @canonical/react-components@4.2.0/dist/esm/components/Button/Button.stories.js
import React from "react";
import Tooltip from "../Tooltip";
import Button, { ButtonAppearance } from "./Button";
var meta = {
  component: Button,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: {
        type: "text"
      }
    },
    appearance: {
      control: {
        options: ButtonAppearance,
        type: "select"
      }
    }
  },
  args: {
    children: "Click me!",
    appearance: ButtonAppearance.DEFAULT
  }
};
export default meta;
export var Default = {
  name: "Default",
  args: {
    children: "Button"
  },
  parameters: {
    docs: {
      description: {
        story: "Button with default appearance"
      }
    }
  }
};
export var DefaultDisabled = {
  name: "Default disabled",
  args: {
    children: "Button",
    disabled: true
  },
  parameters: {
    docs: {
      description: {
        story: "Disabled button with default appearance"
      }
    }
  }
};
export var Base = {
  name: "Base",
  args: {
    children: "Base button",
    appearance: "base"
  },
  parameters: {
    docs: {
      description: {
        story: "Button with base appearance"
      }
    }
  }
};
export var BaseDisabled = {
  name: "Base disabled",
  args: {
    children: "Base button disabled",
    appearance: "base",
    disabled: true
  },
  parameters: {
    docs: {
      description: {
        story: "Disabled button with base appearance"
      }
    }
  }
};
export var Link = {
  name: "Link",
  args: {
    children: "Link button",
    appearance: "link"
  },
  parameters: {
    docs: {
      description: {
        story: "Button with link appearance"
      }
    }
  }
};
export var LinkDisabled = {
  name: "Link disabled",
  args: {
    children: "Link button disabled",
    appearance: "link",
    disabled: true
  },
  parameters: {
    docs: {
      description: {
        story: "Disabled button with link appearance"
      }
    }
  }
};
export var Positive = {
  name: "Positive",
  args: {
    children: "Positive button",
    appearance: "positive"
  },
  parameters: {
    docs: {
      description: {
        story: "Button with positive appearance"
      }
    }
  }
};
export var PositiveDisabled = {
  name: "Positive disabled",
  args: {
    children: "Positive button disabled",
    appearance: "positive",
    disabled: true
  },
  parameters: {
    docs: {
      description: {
        story: "Disabled button with positive appearance"
      }
    }
  }
};
export var Negative = {
  name: "Negative",
  args: {
    children: "Negative button",
    appearance: "negative"
  },
  parameters: {
    docs: {
      description: {
        story: "Button with negative appearance"
      }
    }
  }
};
export var NegativeDisabled = {
  name: "Negative disabled",
  args: {
    children: "Negative button disabled",
    appearance: "negative",
    disabled: true
  },
  parameters: {
    docs: {
      description: {
        story: "Disabled button with negative appearance"
      }
    }
  }
};
export var Brand = {
  name: "Brand",
  args: {
    children: "Brand button",
    appearance: "brand"
  },
  parameters: {
    docs: {
      description: {
        story: "Button with brand appearance"
      }
    }
  }
};
export var BrandDisabled = {
  name: "Brand disabled",
  args: {
    children: "Brand button disabled",
    appearance: "brand",
    disabled: true
  },
  parameters: {
    docs: {
      description: {
        story: "Disabled button with brand appearance"
      }
    }
  }
};
export var Inline = {
  render: () => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("span", null, "Everything you need to get started with Vanilla."), /*#__PURE__*/React.createElement(Button, {
    appearance: "neutral",
    inline: true
  }, "Inline button")),
  name: "Inline",
  parameters: {
    docs: {
      description: {
        story: "Button that displays on the same line as the sibling elements"
      }
    }
  }
};
export var Dense = {
  render: () => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("span", null, "Everything you need to get started with Vanilla."), /*#__PURE__*/React.createElement(Button, {
    dense: true
  }, "Dense button")),
  name: "Dense",
  parameters: {
    docs: {
      description: {
        story: "Button with reduced vertical padding"
      }
    }
  }
};
export var Small = {
  render: () => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Button, {
    small: true
  }, "Small button"), /*#__PURE__*/React.createElement(Button, {
    small: true,
    dense: true
  }, "Small dense button")),
  name: "Small",
  parameters: {
    docs: {
      description: {
        story: "Button that can fit small text inside"
      }
    }
  }
};
export var Icon = {
  name: "Icon",
  args: {
    children: /*#__PURE__*/React.createElement("i", {
      className: "p-icon--plus"
    }),
    hasIcon: true
  },
  parameters: {
    docs: {
      description: {
        story: "Button with an icon inside"
      }
    }
  }
};
export var IconText = {
  name: "Icon & text",
  args: {
    children: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("i", {
      className: "p-icon--plus"
    }), " ", /*#__PURE__*/React.createElement("span", null, "Button with icon & text")),
    hasIcon: true
  },
  parameters: {
    docs: {
      description: {
        story: "Button with both an icon and text inside"
      }
    }
  }
};
export var DisabledWithTooltip = {
  name: "Disabled with tooltip",
  args: {
    children: "Disabled button with a tooltip",
    disabled: true
  },
  render: args => /*#__PURE__*/React.createElement("div", {
    style: {
      paddingTop: "3rem"
    }
  }, /*#__PURE__*/React.createElement(Tooltip, {
    message: "This button is disabled"
  }, /*#__PURE__*/React.createElement(Button, args))),
  parameters: {
    docs: {
      description: {
        story: "Disabled button that displays a tooltip when hovered"
      }
    }
  }
};
export var ButtonAsLink = {
  name: "Button as link",
  args: {
    children: "Button as link",
    appearance: "base",
    element: "a",
    href: "#test"
  },
  parameters: {
    docs: {
      description: {
        story: "Button that is rendered as an `a` element"
      }
    }
  }
};
export var ButtonAsLinkDisabled = {
  name: "Button as link disabled",
  args: {
    children: "Button as link disabled",
    appearance: "base",
    element: "a",
    href: "#test",
    disabled: true
  },
  parameters: {
    docs: {
      description: {
        story: "Disabled button that is rendered as an `a` element"
      }
    }
  }
};
