// @canonical/react-components@4.2.0/dist/components/Tooltip/Tooltip.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.TargetingElements = exports.FollowMouse = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _Button = _interopRequireDefault(require("../Button"));
var _Tooltip = _interopRequireWildcard(require("./Tooltip"));
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const Template = args => {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      justifyContent: "center",
      padding: "3rem"
    }
  }, /*#__PURE__*/_react.default.createElement(_Tooltip.default, args, /*#__PURE__*/_react.default.createElement(_Button.default, {
    className: "u-no-margin--bottom"
  }, "Hover over me!")));
};
const meta = {
  component: _Tooltip.default,
  render: Template,
  tags: ["autodocs"],
  argTypes: {
    position: {
      control: {
        type: "radio"
      },
      options: Object.values(_Tooltip.position)
    }
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: Template.bind({}),
  name: "Default",
  args: {
    message: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, "Tooltip", " ", /*#__PURE__*/_react.default.createElement("a", {
      href: "#todo",
      target: "_blank"
    }, "Link1"))
  }
};
const FollowMouse = exports.FollowMouse = {
  render: Template.bind({}),
  name: "Follow mouse",
  args: {
    followMouse: true,
    message: "Tooltip message to display.",
    position: "top-center"
  }
};

/**
 * The tooltip elements are displayed inside a [Portal](https://reactjs.org/docs/portals.html) so that the tooltip is not constrained by its current position in the DOM.
 * Sometimes however, you need to target the tooltip elements. This can be achieved by passing class names to the tooltip elements. You can also pass classes to the elements that trigger and position the tooltip.
 */
const TargetingElements = exports.TargetingElements = {
  render: Template.bind({}),
  name: "Targeting elements",
  args: {
    className: "custom-wrapping-class",
    positionElementClassName: "custom-position-class",
    tooltipClassName: "custom-tooltip-class",
    message: "Tooltip message to display."
  }
};
