// @canonical/react-components@4.2.0/dist/components/Chip/Chip.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithIcon = exports.WithBadge = exports.ReadOnly = exports.LeadValue = exports.Inline = exports.Dismissible = exports.Dense = exports.Default = exports.Appearance = void 0;
var _react = _interopRequireDefault(require("react"));
var _Chip = _interopRequireDefault(require("./Chip"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _Chip.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default",
  args: {
    value: "Default"
  }
};
const LeadValue = exports.LeadValue = {
  render: () => /*#__PURE__*/_react.default.createElement(_Chip.default, {
    lead: "Owner",
    value: "Bob"
  }),
  name: "Lead-value"
};
const Appearance = exports.Appearance = {
  render: () => /*#__PURE__*/_react.default.createElement(_Chip.default, {
    lead: "Type",
    value: "Positive",
    appearance: "positive"
  }),
  name: "Appearance"
};
const Dismissible = exports.Dismissible = {
  render: () => /*#__PURE__*/_react.default.createElement(_Chip.default, {
    lead: "Owner",
    value: "Bob",
    onDismiss: () => {
      console.log("onDismiss called");
    }
  }),
  name: "Dismissible"
};
const Dense = exports.Dense = {
  render: () => /*#__PURE__*/_react.default.createElement(_Chip.default, {
    lead: "Owner",
    value: "Bob",
    isDense: true
  }),
  name: "Dense"
};
const Inline = exports.Inline = {
  render: () => /*#__PURE__*/_react.default.createElement("p", null, "This is text with an inline", " ", /*#__PURE__*/_react.default.createElement(_Chip.default, {
    value: "chip",
    isDense: true,
    isInline: true
  })),
  name: "Inline"
};
const ReadOnly = exports.ReadOnly = {
  render: () => /*#__PURE__*/_react.default.createElement(_Chip.default, {
    lead: "Owner",
    value: "Bob",
    isReadOnly: true
  }),
  name: "Read-only"
};
const WithIcon = exports.WithIcon = {
  render: () => /*#__PURE__*/_react.default.createElement(_Chip.default, {
    value: "Users",
    iconName: "user"
  }),
  name: "With Icon"
};
const WithBadge = exports.WithBadge = {
  render: () => /*#__PURE__*/_react.default.createElement(_Chip.default, {
    lead: "Owner",
    value: "Bob",
    badge: /*#__PURE__*/_react.default.createElement("span", {
      className: "p-badge",
      "aria-label": "More than 2.5 billion items"
    }, "2.5B")
  }),
  name: "With Badge"
};
