// @canonical/react-components@4.2.0/dist/components/Chip/Chip.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Label = exports.ChipType = void 0;
var _react = _interopRequireDefault(require("react"));
var _utils = require("../../utils");
var _classnames = _interopRequireDefault(require("classnames"));
const _excluded = ["appearance", "lead", "onClick", "onDismiss", "quoteValue", "selected", "subString", "isReadOnly", "isDense", "isInline", "iconName", "badge", "value"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
let Label = exports.Label = /*#__PURE__*/function (Label) {
  Label["Dismiss"] = "Dismiss";
  return Label;
}({});
const ChipType = exports.ChipType = {
  CAUTION: "caution",
  INFORMATION: "information",
  NEGATIVE: "negative",
  POSITIVE: "positive"
};
/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Chip](https://vanillaframework.io/docs/patterns/chip).
 *
 * It can be used to display a small value attached to a component.
 */
const Chip = _ref => {
  let {
      appearance,
      lead = "",
      onClick,
      onDismiss,
      quoteValue,
      selected,
      subString = "",
      isReadOnly = false,
      isDense = false,
      isInline = false,
      iconName,
      badge,
      value
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  // When user tabs over chip and presses Enter or Space key, chip will trigger
  // click functionality
  const onKeyDown = e => {
    // The " " value is what is returned for the spacebar
    if (e.key === " " || e.key === "Enter") {
      if (typeof onClick === "function") {
        onClick({
          lead: lead,
          value: value
        });
      }
    }
  };
  const chipValue = (0, _utils.highlightSubString)(value, subString).text;
  const chipContent = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, iconName && /*#__PURE__*/_react.default.createElement("i", {
    className: "p-icon--".concat(iconName)
  }), lead && /*#__PURE__*/_react.default.createElement("span", {
    className: "p-chip__lead"
  }, lead.toUpperCase()), /*#__PURE__*/_react.default.createElement("span", {
    className: "p-chip__value",
    dangerouslySetInnerHTML: {
      __html: quoteValue ? "'".concat(chipValue, "'") : chipValue
    }
  }), badge && badge);
  const chipClassName = (0, _classnames.default)({
    ["p-chip--".concat(appearance)]: !!appearance,
    "p-chip": !appearance,
    "is-dense": isDense,
    "is-readonly": isReadOnly,
    "is-inline": isInline
  }, props.className);
  if (isReadOnly) {
    return /*#__PURE__*/_react.default.createElement("span", _extends({}, props, {
      className: chipClassName
    }), chipContent);
  } else if (onDismiss) {
    return /*#__PURE__*/_react.default.createElement("span", _extends({}, props, {
      className: chipClassName
    }), chipContent, /*#__PURE__*/_react.default.createElement("button", {
      className: "p-chip__dismiss",
      onClick: onDismiss,
      type: "button"
    }, /*#__PURE__*/_react.default.createElement("i", {
      className: "p-icon--close"
    }, Label.Dismiss)));
  } else {
    return /*#__PURE__*/_react.default.createElement("button", _extends({}, props, {
      "aria-pressed": selected,
      className: chipClassName,
      onClick: onClick,
      onKeyDown: e => onKeyDown(e),
      type: "button"
    }), chipContent);
  }
};
var _default = exports.default = Chip;
