// @canonical/react-components@4.2.0/dist/components/PrefixedInput/PrefixedInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));
var _Input = _interopRequireDefault(require("../Input"));
var _Field = _interopRequireDefault(require("../Field"));
require("./PrefixedInput.scss");
const _excluded = ["immutableText", "disabled", "className", "wrapperClassName", "label", "error", "help", "caution", "success", "id", "required"];
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
const PrefixedInput = _ref => {
  let {
      immutableText,
      disabled,
      className,
      wrapperClassName,
      label,
      error,
      help,
      caution,
      success,
      id,
      required
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  const defaultId = (0, _react.useId)();
  const inputId = id || defaultId;
  return /*#__PURE__*/_react.default.createElement(_Field.default, {
    label: label,
    error: error,
    help: help,
    caution: caution,
    success: success,
    required: required,
    forId: inputId,
    className: (0, _classnames.default)("p-prefixed-input", wrapperClassName)
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)("p-prefixed-input__flex-container", {
      "is-disabled": disabled,
      "is-error": !!error,
      "is-caution": !!caution,
      "is-success": !!success
    })
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "p-prefixed-input__prefix"
  }, immutableText), /*#__PURE__*/_react.default.createElement(_Input.default, _extends({}, props, {
    id: inputId,
    disabled: disabled,
    required: required
    // IMPORTANT: pass nulls so Input doesn't trigger its own Field wrapper
    ,
    label: null,
    error: null,
    help: null,
    caution: null,
    success: null,
    type: "text",
    className: (0, _classnames.default)("p-prefixed-input__input-field", className)
  }))));
};
PrefixedInput.propTypes = {
  immutableText: _propTypes.default.string.isRequired
};
var _default = exports.default = PrefixedInput;
