// @canonical/react-components@4.2.0/dist/components/PrefixedInput/PrefixedInput.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithSuccess = exports.WithLabel = exports.WithHelpText = exports.WithError = exports.WithCaution = exports.Required = exports.Disabled = exports.Default = void 0;
var _PrefixedInput = _interopRequireDefault(require("./PrefixedInput"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _PrefixedInput.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  args: {
    immutableText: "https://",
    placeholder: "example.com"
  }
};
const WithLabel = exports.WithLabel = {
  args: {
    immutableText: "https://",
    label: "Website URL",
    placeholder: "example.com"
  }
};
const Disabled = exports.Disabled = {
  args: {
    immutableText: "@",
    label: "Username",
    placeholder: "username",
    disabled: true
  }
};
const WithError = exports.WithError = {
  args: {
    immutableText: "https://",
    label: "Website URL",
    placeholder: "example.com",
    error: "Invalid URL format"
  }
};
const WithSuccess = exports.WithSuccess = {
  args: {
    immutableText: "https://",
    label: "Website URL",
    placeholder: "example.com",
    success: "Valid URL format"
  }
};
const WithCaution = exports.WithCaution = {
  args: {
    immutableText: "https://",
    label: "Website URL",
    placeholder: "example.com",
    caution: "This URL might be problematic"
  }
};
const WithHelpText = exports.WithHelpText = {
  args: {
    immutableText: "User ID:",
    label: "User Identifier",
    placeholder: " Enter user ID",
    help: "This will be used to identify your account"
  }
};
const Required = exports.Required = {
  args: {
    immutableText: "https://",
    label: "Website URL",
    placeholder: "example.com",
    required: true
  }
};
