// @canonical/react-components@4.2.0/dist/esm/components/Stepper/Step/Step.js
import _pt from "prop-types";
var _excluded = ["hasProgressLine", "index", "title", "label", "linkProps", "enabled", "selected", "iconName", "iconClassName", "handleClick"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import classNames from "classnames";
import React from "react";
import Icon from "../../Icon";
import Link from "../../Link";
import "./Step.scss";
var Step = _ref => {
  var {
      hasProgressLine,
      index,
      title,
      label,
      linkProps,
      enabled,
      selected = false,
      iconName,
      iconClassName,
      handleClick
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  var stepStatusClass = enabled ? "step-enabled" : "step-disabled";
  return /*#__PURE__*/React.createElement("div", _extends({
    className: classNames("step", {
      "progress-line": hasProgressLine,
      "step-selected": selected
    })
  }, props), iconName === "number" ? /*#__PURE__*/React.createElement("span", {
    className: classNames("step-number", {
      "step-number-disabled": !enabled
    })
  }, index) : /*#__PURE__*/React.createElement(Icon, {
    name: iconName,
    className: classNames("step-status-icon", iconClassName)
  }), /*#__PURE__*/React.createElement("div", {
    className: "step-content"
  }, /*#__PURE__*/React.createElement("span", {
    className: classNames(stepStatusClass),
    onClick: handleClick
  }, title), label && /*#__PURE__*/React.createElement("span", {
    className: classNames("step-optional-content", "u-no-margin--bottom", {
      "step-disabled": !enabled
    })
  }, label), linkProps && /*#__PURE__*/React.createElement(Link, _extends({
    className: "p-text--small u-no-margin--bottom step-optional-content"
  }, linkProps), linkProps.children)));
};
Step.propTypes = {
  hasProgressLine: _pt.bool.isRequired,
  index: _pt.number.isRequired,
  title: _pt.string.isRequired,
  label: _pt.string,
  enabled: _pt.bool.isRequired,
  selected: _pt.bool,
  iconName: _pt.string.isRequired,
  handleClick: _pt.func.isRequired
};
export default Step;
