// @canonical/react-components@4.2.0/dist/components/SideNavigation/SideNavigation.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = exports.CustomContent = void 0;
var _react = _interopRequireDefault(require("react"));
var _Badge = _interopRequireDefault(require("../Badge"));
var _SideNavigation = _interopRequireDefault(require("./SideNavigation"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _SideNavigation.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
/**
 * Menu items can be provided as links, text or custom components. To provide attributes to individual menus then each menu can be provided as a object containing an items array: `{ className: "menu-one", items: [...] }`.
 */
const Default = exports.Default = {
  args: {
    items: [{
      className: "menu-one",
      items: [{
        icon: "drag",
        label: "Models",
        href: "/models",
        status: /*#__PURE__*/_react.default.createElement(_Badge.default, {
          value: 9,
          isNegative: true
        })
      }, {
        icon: "menu",
        label: "Controllers",
        nonInteractive: true
      }, {
        icon: "user",
        label: "Permissions",
        href: "/users"
      }]
    }, {
      className: "menu-two",
      items: [{
        icon: "information",
        label: "Docs",
        href: "/docs"
      }, {
        label: "Logout",
        href: "/logout"
      }]
    }]
  }
};

/**
 * `children` can be provided instead of `items` in cases where custom content
 * is required.
 */
const CustomContent = exports.CustomContent = {
  args: {
    children: /*#__PURE__*/_react.default.createElement("div", null, "Custom menu content.")
  }
};
