// @canonical/react-components@4.2.0/dist/esm/components/Switch/Switch.js
var _excluded = ["label", "disabled", "help", "helpClassName"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import React from "react";
import classNames from "classnames";
import { useId } from "react";
import "./Switch.scss";
/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Switch](https://vanillaframework.io/docs/patterns/switch) component.
 */
export var Switch = _ref => {
  var {
      label,
      disabled = false,
      help,
      helpClassName
    } = _ref,
    inputProps = _objectWithoutProperties(_ref, _excluded);
  var helpId = useId();
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("label", {
    className: "p-switch"
  }, /*#__PURE__*/React.createElement("input", _extends({
    type: "checkbox",
    className: "p-switch__input",
    role: "switch",
    disabled: disabled
  }, inputProps)), /*#__PURE__*/React.createElement("span", {
    className: "p-switch__slider"
  }), /*#__PURE__*/React.createElement("span", {
    className: "p-switch__label"
  }, label)), help && /*#__PURE__*/React.createElement("p", {
    className: classNames("p-form-help-text", "switch-help-text", helpClassName),
    id: helpId
  }, help));
};
export default Switch;
