// @canonical/react-components@4.2.0/dist/components/DoughnutChart/DoughnutChart.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;
var _DoughnutChart = _interopRequireDefault(require("./DoughnutChart"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _DoughnutChart.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
/**
 * The Doughnut Chart component visually represents data segments in a circular format, with tooltips that appear on hover, and segments that can be customized via props.
 */
const Default = exports.Default = {
  name: "Default",
  args: {
    chartID: "default",
    segmentHoverWidth: 45,
    segmentThickness: 40,
    segments: [{
      color: "#0E8420",
      tooltip: "Running",
      value: 10
    }, {
      color: "#CC7900",
      tooltip: "Stopped",
      value: 15
    }, {
      color: "#C7162B",
      tooltip: "Frozen",
      value: 5
    }, {
      color: "#000",
      tooltip: "Error",
      value: 5
    }],
    size: 150
  }
};
