// @canonical/react-components@4.2.0/dist/esm/components/Badge/Badge.stories.js
import React from "react";
import Badge, { BadgeType } from "./Badge";
var meta = {
  component: Badge,
  tags: ["autodocs"],
  argTypes: {
    badgeType: {
      control: {
        options: BadgeType,
        type: "select"
      }
    }
  },
  args: {
    badgeType: BadgeType.UNDEFINED_LARGE_NUMBER
  }
};
export default meta;
export var Default = {
  name: "Default",
  args: {
    value: 9
  }
};
export var ColourCoding = {
  render: () => /*#__PURE__*/React.createElement(Badge, {
    value: 9,
    isNegative: true
  }),
  name: "Colour Coding"
};

/**
 * When the amount of items is only relevant for small values, this variant should be used. When the value grows further than 999, the badge should show "999+".
 */
export var UndefinedLargeNumber = {
  render: () => /*#__PURE__*/React.createElement(Badge, {
    value: 1000,
    badgeType: BadgeType.UNDEFINED_LARGE_NUMBER
  }),
  name: "Undefined Large Number"
};

/**
 * When the value of the badge represents a type of information where differences between large amounts of items are relevant, it should include a decimal unit prefix and round the value.
 */
export var RoundedLargeNumber = {
  render: () => /*#__PURE__*/React.createElement(Badge, {
    value: 1234,
    badgeType: BadgeType.ROUNDED_LARGE_NUMBER
  }),
  name: "Rounded Large Number"
};
