// @canonical/react-components@4.2.0/dist/esm/components/TablePagination/TablePaginationControls/TablePaginationControls.js
import _pt from "prop-types";
var _excluded = ["className", "currentPage", "description", "displayDescription", "onInputPageChange", "itemName", "nextButtonProps", "onNextPage", "onPageChange", "onPageSizeChange", "onPreviousPage", "pageLimits", "pageSize", "previousButtonProps", "showPageInput", "totalItems", "visibleCount"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import Button from "../../Button";
import Icon from "../../Icon";
import Input from "../../Input";
import Select from "../../Select";
import React from "react";
import classnames from "classnames";
import { generatePagingOptions, getDescription, useFigureSmallScreen } from "../utils";
export var Label = /*#__PURE__*/function (Label) {
  Label["NEXT_PAGE"] = "Next page";
  Label["PREVIOUS_PAGE"] = "Previous page";
  Label["PAGE_NUMBER"] = "Page number";
  return Label;
}({});
var TablePaginationControls = _ref => {
  var {
      className,
      currentPage,
      description,
      displayDescription = true,
      onInputPageChange,
      itemName = "row",
      nextButtonProps,
      onNextPage,
      onPageChange,
      onPageSizeChange,
      onPreviousPage,
      pageLimits,
      pageSize,
      previousButtonProps,
      showPageInput = true,
      totalItems,
      visibleCount
    } = _ref,
    divProps = _objectWithoutProperties(_ref, _excluded);
  var isSmallScreen = useFigureSmallScreen();
  var totalPages = totalItems ? Math.ceil(totalItems / pageSize) : null;
  var descriptionDisplay = getDescription({
    description,
    visibleCount,
    isSmallScreen,
    totalItems,
    itemName,
    currentPage
  });
  var handleDecrementPage = currentPage => {
    if (currentPage > 1) {
      onPageChange === null || onPageChange === void 0 || onPageChange(currentPage - 1);
    }
    onPreviousPage === null || onPreviousPage === void 0 || onPreviousPage(typeof currentPage === "number" ? currentPage - 1 : null);
  };
  var handleIncrementPage = (currentPage, totalPages) => {
    if (currentPage < totalPages) {
      onPageChange === null || onPageChange === void 0 || onPageChange(currentPage + 1);
    }
    onNextPage === null || onNextPage === void 0 || onNextPage(typeof currentPage === "number" ? currentPage + 1 : null);
  };
  var handleInputPageChange = e => {
    var newPage = Math.min(totalPages, Math.max(1, parseInt(e.target.value)));
    onPageChange === null || onPageChange === void 0 || onPageChange(newPage);
    onInputPageChange === null || onInputPageChange === void 0 || onInputPageChange(Number(e.target.value));
  };
  var handlePageSizeChange = e => {
    onPageSizeChange(parseInt(e.target.value));
  };
  var isInputDisabled = typeof totalItems === "number" && (!totalPages || totalPages == 1);
  var maxPageValue = typeof totalPages === "number" ? totalPages : 1;
  return /*#__PURE__*/React.createElement("div", _extends({
    className: classnames("pagination", className)
  }, divProps, {
    role: "navigation"
  }), /*#__PURE__*/React.createElement("div", {
    className: "description",
    id: "pagination-description"
  }, displayDescription ? descriptionDisplay : null), /*#__PURE__*/React.createElement(Button, _extends({
    "aria-label": Label.PREVIOUS_PAGE,
    className: "back",
    appearance: "base",
    hasIcon: true,
    disabled: isInputDisabled || currentPage === 1,
    onClick: () => handleDecrementPage(currentPage)
  }, previousButtonProps), previousButtonProps !== null && previousButtonProps !== void 0 && previousButtonProps.children ? previousButtonProps.children : /*#__PURE__*/React.createElement(Icon, {
    name: "chevron-down"
  })), showPageInput ? /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Input, {
    id: "paginationPageInput",
    label: Label.PAGE_NUMBER,
    labelClassName: "u-off-screen",
    className: "u-no-margin--bottom pagination-input",
    onChange: handleInputPageChange,
    value: currentPage,
    type: "number",
    disabled: isInputDisabled,
    min: 1,
    max: maxPageValue
  }), " ", typeof totalPages === "number" ? /*#__PURE__*/React.createElement("div", {
    className: "pagination-item-count"
  }, "of\xA0", totalPages) : null) : null, /*#__PURE__*/React.createElement(Button, _extends({
    "aria-label": Label.NEXT_PAGE,
    className: "next",
    appearance: "base",
    hasIcon: true,
    disabled: isInputDisabled || currentPage === totalPages,
    onClick: () => handleIncrementPage(currentPage, totalPages)
  }, nextButtonProps), /*#__PURE__*/React.createElement(Icon, {
    name: "chevron-down"
  })), /*#__PURE__*/React.createElement(Select, {
    className: "u-no-margin--bottom",
    label: "Items per page",
    labelClassName: "u-off-screen",
    id: "itemsPerPage",
    options: generatePagingOptions(pageLimits),
    onChange: handlePageSizeChange,
    value: pageSize
  }));
};
TablePaginationControls.propTypes = {
  displayDescription: _pt.bool,
  onInputPageChange: _pt.func,
  onNextPage: _pt.func,
  onPreviousPage: _pt.func,
  visibleCount: _pt.number,
  showPageInput: _pt.bool
};
export default TablePaginationControls;
