// @canonical/react-components@4.2.0/dist/components/Field/Field.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _classnames = _interopRequireDefault(require("classnames"));
var _react = _interopRequireDefault(require("react"));
var _Label = _interopRequireDefault(require("../Label"));
var _Col = _interopRequireDefault(require("../Col"));
const _excluded = ["caution", "children", "className", "error", "forId", "help", "helpClassName", "helpId", "isSelect", "isTickElement", "label", "labelClassName", "labelFirst", "required", "stacked", "success", "validationId"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
/**
 * The props for the Field component.
 */

const generateHelpText = _ref => {
  let {
    help,
    helpId,
    helpClassName,
    isTickElement
  } = _ref;
  return help ? /*#__PURE__*/_react.default.createElement("p", {
    className: (0, _classnames.default)("p-form-help-text", helpClassName, {
      "is-tick-element": isTickElement
    }),
    id: helpId
  }, help) : null;
};
const generateError = (error, caution, success, validationId) => {
  if (!error && !caution && !success) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement("p", {
    className: "p-form-validation__message",
    id: validationId
  }, error || caution || success);
};
const generateLabel = (forId, required, label, labelClassName, stacked) => {
  if (!label) {
    return null;
  }
  const labelNode = /*#__PURE__*/_react.default.createElement(_Label.default, {
    className: labelClassName,
    forId: forId,
    required: required
  }, label);
  if (stacked) {
    return /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 4
    }, labelNode);
  }
  return labelNode;
};
const generateContent = _ref2 => {
  let {
    isSelect,
    children,
    labelFirst,
    labelNode,
    help,
    helpClassName,
    error,
    caution,
    success,
    validationId,
    helpId,
    isTickElement
  } = _ref2;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "p-form__control u-clearfix"
  }, isSelect ? /*#__PURE__*/_react.default.createElement("div", {
    className: "p-form-validation__select-wrapper"
  }, children) : children, !labelFirst && labelNode, generateHelpText({
    helpId,
    help,
    helpClassName,
    isTickElement
  }), generateError(error, caution, success, validationId));
};
const Field = _ref3 => {
  let {
      caution,
      children,
      className,
      error,
      forId,
      help,
      helpClassName,
      helpId,
      isSelect,
      isTickElement,
      label,
      labelClassName,
      labelFirst = true,
      required,
      stacked,
      success,
      validationId
    } = _ref3,
    props = _objectWithoutProperties(_ref3, _excluded);
  const labelNode = generateLabel(forId, required, label, labelClassName, stacked);
  const content = generateContent({
    isSelect,
    isTickElement,
    children,
    labelFirst,
    labelNode,
    help,
    helpClassName,
    error,
    caution,
    success,
    validationId,
    helpId
  });
  return /*#__PURE__*/_react.default.createElement("div", _extends({
    className: (0, _classnames.default)("p-form__group", "p-form-validation", className, {
      "is-error": error,
      "is-caution": caution,
      "is-success": success,
      row: stacked
    })
  }, props), labelFirst && labelNode, stacked ? /*#__PURE__*/_react.default.createElement(_Col.default, {
    size: 8
  }, content) : content);
};
var _default = exports.default = Field;
