// @canonical/react-components@4.2.0/dist/components/LoginPageLayout/LoginPageLayout.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.RegistrationPage = exports.LoginPage = exports.ErrorPage = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _CodeSnippet = _interopRequireDefault(require("../CodeSnippet/CodeSnippet"));
var _Spinner = _interopRequireDefault(require("../Spinner/Spinner"));
var _LoginPageLayout = _interopRequireDefault(require("./LoginPageLayout"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _LoginPageLayout.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  args: {
    title: "This is the title"
  }
};
const LoginPage = exports.LoginPage = {
  args: {
    title: "Sign in",
    children: /*#__PURE__*/_react.default.createElement(_Spinner.default, null)
  }
};
const ErrorPage = exports.ErrorPage = {
  args: {
    title: "Sign in failed",
    children: /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, {
      blocks: [{
        wrapLines: true,
        code: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, "An error occurred. Try signing in again.")
      }]
    })
  }
};
const RegistrationPage = exports.RegistrationPage = {
  args: {
    title: "Sign up",
    children: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("p", null, "Fill in the form below to create an account"), /*#__PURE__*/_react.default.createElement("form", null, /*#__PURE__*/_react.default.createElement("input", {
      type: "text",
      placeholder: "Username"
    }), /*#__PURE__*/_react.default.createElement("input", {
      type: "email",
      placeholder: "Email"
    }), /*#__PURE__*/_react.default.createElement("input", {
      type: "password",
      placeholder: "Password"
    }), /*#__PURE__*/_react.default.createElement("button", null, "Sign up"))),
    logo: {
      src: "https://assets.ubuntu.com/v1/04242fd4-anbox-cloud-logo.svg",
      title: "Anbox Cloud",
      url: "/"
    }
  }
};
