// @canonical/react-components@4.2.0/dist/components/ConfirmationModal/ConfirmationModal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.ConfirmationModal = void 0;
var _react = _interopRequireDefault(require("react"));
var _Button = _interopRequireDefault(require("../Button"));
var _Modal = _interopRequireDefault(require("../Modal"));
var _ActionButton = _interopRequireDefault(require("../ActionButton"));
var _external = require("../../external");
const _excluded = ["cancelButtonLabel", "cancelButtonProps", "children", "confirmButtonAppearance", "confirmButtonLabel", "confirmExtra", "onConfirm", "confirmButtonLoading", "confirmButtonDisabled", "confirmButtonProps", "renderInPortal", "portalRenderer"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
const InternalPortalRenderer = _ref => {
  let {
    children
  } = _ref;
  const {
    Portal
  } = (0, _external.usePortal)();
  return /*#__PURE__*/_react.default.createElement(Portal, null, children);
};

/**
 * `ConfirmationModal` is a specialised version of the [Modal](?path=/docs/modal--default-story) component to prompt a confirmation from the user before executing an action.
 */
const ConfirmationModal = _ref2 => {
  var _cancelButtonProps$ty;
  let {
      cancelButtonLabel = "Cancel",
      cancelButtonProps,
      children,
      confirmButtonAppearance = "negative",
      confirmButtonLabel,
      confirmExtra,
      onConfirm,
      confirmButtonLoading,
      confirmButtonDisabled,
      confirmButtonProps,
      renderInPortal = false,
      portalRenderer: PortalRenderer
    } = _ref2,
    props = _objectWithoutProperties(_ref2, _excluded);
  const handleClick = action => event => {
    if (!props.shouldPropagateClickEvent) {
      event.stopPropagation();
    }
    if (action) {
      action(event);
    }
  };
  const ModalElement = /*#__PURE__*/_react.default.createElement(_Modal.default, _extends({
    buttonRow: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, confirmExtra, /*#__PURE__*/_react.default.createElement(_Button.default, _extends({}, cancelButtonProps, {
      type: (_cancelButtonProps$ty = cancelButtonProps === null || cancelButtonProps === void 0 ? void 0 : cancelButtonProps.type) !== null && _cancelButtonProps$ty !== void 0 ? _cancelButtonProps$ty : "button",
      className: "u-no-margin--bottom",
      onClick: handleClick(props.close)
    }), cancelButtonLabel), /*#__PURE__*/_react.default.createElement(_ActionButton.default, _extends({}, confirmButtonProps, {
      appearance: confirmButtonAppearance,
      className: "u-no-margin--bottom",
      onClick: handleClick(onConfirm),
      loading: confirmButtonLoading,
      disabled: confirmButtonDisabled
    }), confirmButtonLabel))
  }, props), children);
  if (PortalRenderer) {
    return /*#__PURE__*/_react.default.createElement(PortalRenderer, null, ModalElement);
  } else if (renderInPortal) {
    return /*#__PURE__*/_react.default.createElement(InternalPortalRenderer, null, ModalElement);
  } else {
    return ModalElement;
  }
};
exports.ConfirmationModal = ConfirmationModal;
var _default = exports.default = ConfirmationModal;
