// @canonical/react-components@4.2.0/dist/esm/components/DoughnutChart/DoughnutChart.js
import _pt from "prop-types";
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import React, { useRef, useState } from "react";
import classNames from "classnames";
import Tooltip from "../Tooltip";
import "./DoughnutChart.scss";
export var TestIds = /*#__PURE__*/function (TestIds) {
  TestIds["Label"] = "label";
  TestIds["Segment"] = "segment";
  TestIds["Chart"] = "chart";
  TestIds["Section"] = "Section";
  return TestIds;
}({});
var DoughnutChart = _ref => {
  var {
    className,
    label,
    labelClassname,
    segmentHoverWidth,
    segmentThickness,
    segments,
    size,
    chartID
  } = _ref;
  var [tooltipMessage, setTooltipMessage] = useState(null);
  var id = useRef("doughnut-chart-".concat(chartID));
  var hoverIncrease = segmentHoverWidth - segmentThickness;
  var adjustedHoverWidth = segmentHoverWidth + hoverIncrease;
  // The canvas needs enough space so that the hover state does not get cut off.
  var canvasSize = size + adjustedHoverWidth - segmentThickness;
  var diameter = size - segmentThickness;
  var radius = diameter / 2;
  var circumference = Math.round(diameter * Math.PI);
  // Calculate the total value of all segments.
  var total = segments.reduce((totalValue, segment) => totalValue += segment.value, 0);

  // Pre-calculate the accumulated lengths.
  var segmentsWithOffsets = segments.reduce((acc, segment, i) => {
    var previousSegment = i > 0 ? acc[i - 1] : null;
    // The start position is the value of all previous segments.
    var startPosition = i === 0 ? 0 : previousSegment.startPosition + previousSegment.segmentLength;
    // The length of the segment (as a portion of the doughnut circumference)
    var segmentLength = segment.value / total * circumference;
    // The space left until the end of the circle.
    var remainingSpace = circumference - (segmentLength + startPosition);
    acc.push(_objectSpread(_objectSpread({}, segment), {}, {
      startPosition,
      segmentLength,
      remainingSpace
    }));
    return acc;
  }, []);

  // Map over the enriched data.
  var segmentNodes = segmentsWithOffsets.map((_ref2, i) => {
    var {
      color,
      tooltip,
      value,
      startPosition,
      segmentLength,
      remainingSpace
    } = _ref2;
    return /*#__PURE__*/React.createElement("circle", {
      className: "doughnut-chart__segment",
      cx: radius - segmentThickness / 2 - hoverIncrease,
      cy: radius + segmentThickness / 2 + hoverIncrease,
      "data-testid": TestIds.Segment,
      key: i,
      tabIndex: 0,
      "aria-label": tooltip ? "".concat(tooltip, ": ").concat(value) : "".concat(value),
      onMouseOut: tooltip ? () => {
        // Hide the tooltip.
        setTooltipMessage(null);
      } : undefined,
      onMouseOver: tooltip ? () => {
        setTooltipMessage(tooltip);
      } : undefined,
      r: radius,
      style: {
        stroke: color,
        strokeWidth: segmentThickness,
        // The dash array used is:
        // 1 - We want there to be a space before the first visible dash so
        //     by setting this to 0 we can use the next dash for the space.
        // 2 - This gap is the distance of all previous segments
        //     so that the segment starts in the correct spot.
        // 3 - A dash that is the length of the segment.
        // 4 - A gap from the end of the segment to the start of the circle
        //     so that the dash array doesn't repeat and be visible.
        strokeDasharray: "0 ".concat(startPosition.toFixed(2), " ").concat(segmentLength.toFixed(2), " ").concat(remainingSpace.toFixed(2))
      }
      // Rotate the segment so that the segments start at the top of
      // the chart.
      ,
      transform: "rotate(-90 ".concat(radius, ",").concat(radius, ")")
    });
  });
  return /*#__PURE__*/React.createElement("div", {
    className: classNames("doughnut-chart", className),
    style: {
      maxWidth: "".concat(canvasSize, "px")
    },
    "data-testid": TestIds.Chart
  }, /*#__PURE__*/React.createElement(Tooltip, {
    className: "doughnut-chart__tooltip",
    followMouse: true,
    message: tooltipMessage,
    position: "right"
  }, /*#__PURE__*/React.createElement("style", null, "#".concat(id.current, " .doughnut-chart__segment:hover {\n          stroke-width: ").concat(adjustedHoverWidth, " !important;\n        }")), /*#__PURE__*/React.createElement("svg", {
    className: "doughnut-chart__chart",
    id: id.current,
    viewBox: "0 0 ".concat(canvasSize, " ").concat(canvasSize),
    "data-testid": TestIds.Section,
    "aria-labelledby": "".concat(id.current, "-chart-title ").concat(id.current, "-chart-desc")
  }, label && /*#__PURE__*/React.createElement("title", {
    id: "".concat(id.current, "-chart-title")
  }, label), /*#__PURE__*/React.createElement("desc", {
    id: "".concat(id.current, "-chart-desc")
  }, segments.map(segment => {
    var description = "";
    if (segment.tooltip) description += "".concat(segment.tooltip, ": ");
    description += segment.value;
    return description;
  }).join(",")), /*#__PURE__*/React.createElement("mask", {
    id: "canvasMask"
  }, /*#__PURE__*/React.createElement("rect", {
    fill: "white",
    height: canvasSize,
    width: canvasSize,
    x: "0",
    y: "0"
  }), /*#__PURE__*/React.createElement("circle", {
    cx: canvasSize / 2,
    cy: canvasSize / 2,
    fill: "black",
    r: radius - segmentThickness / 2
  })), /*#__PURE__*/React.createElement("g", {
    mask: "url(#canvasMask)"
  }, /*#__PURE__*/React.createElement("rect", {
    fill: "transparent",
    height: canvasSize,
    width: canvasSize,
    x: "0",
    y: "0"
  }), /*#__PURE__*/React.createElement("g", null, segmentNodes)), label ? /*#__PURE__*/React.createElement("text", {
    x: radius + adjustedHoverWidth / 2,
    y: radius + adjustedHoverWidth / 2
  }, /*#__PURE__*/React.createElement("tspan", {
    className: classNames("doughnut-chart__label", labelClassname),
    "data-testid": TestIds.Label
  }, label)) : null)));
};
DoughnutChart.propTypes = {
  label: _pt.string,
  labelClassname: _pt.string,
  className: _pt.string,
  segmentHoverWidth: _pt.number.isRequired,
  segmentThickness: _pt.number.isRequired,
  segments: _pt.arrayOf(_pt.shape({
    color: _pt.string.isRequired,
    tooltip: _pt.string,
    value: _pt.number.isRequired
  })).isRequired,
  size: _pt.number.isRequired,
  chartID: _pt.string.isRequired
};
export default DoughnutChart;
