// @canonical/react-components@4.2.0/dist/components/Pagination/Pagination.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.TruncatedNumberedPagination = exports.DisabledPaginationItemsNumberedPagination = exports.DisabledControlsNumberedPagination = exports.DefaultButtonsOnlyPagination = exports.DefaultButtonLabelsButtonsOnlyPagination = exports.Default = exports.CustomButtonLabelsButtonsOnlyPagination = exports.CenteredNumberedPagination = exports.CenteredButtonsOnlyPagination = void 0;
var _react = _interopRequireDefault(require("react"));
var _Pagination = _interopRequireDefault(require("./Pagination"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _Pagination.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default",
  args: {
    itemsPerPage: 20,
    totalItems: 100,
    paginate: () => {},
    currentPage: 3
  }
};
const TruncatedNumberedPagination = exports.TruncatedNumberedPagination = {
  render: () => /*#__PURE__*/_react.default.createElement(_Pagination.default, {
    itemsPerPage: 10,
    totalItems: 1000,
    paginate: () => {},
    currentPage: 34
  }),
  name: "Truncated (Numbered Pagination)"
};
const DisabledControlsNumberedPagination = exports.DisabledControlsNumberedPagination = {
  render: () => /*#__PURE__*/_react.default.createElement(_Pagination.default, {
    itemsPerPage: 20,
    totalItems: 100,
    paginate: () => {},
    currentPage: 1
  }),
  name: "Disabled controls (Numbered Pagination)"
};
const CenteredNumberedPagination = exports.CenteredNumberedPagination = {
  render: () => /*#__PURE__*/_react.default.createElement(_Pagination.default, {
    itemsPerPage: 10,
    totalItems: 50,
    paginate: () => {},
    currentPage: 1,
    centered: true
  }),
  name: "Centered (Numbered Pagination)"
};
const DisabledPaginationItemsNumberedPagination = exports.DisabledPaginationItemsNumberedPagination = {
  render: () => /*#__PURE__*/_react.default.createElement(_Pagination.default, {
    itemsPerPage: 10,
    totalItems: 50,
    paginate: () => {},
    currentPage: 1,
    hideNumbers: true
  }),
  name: "Disabled pagination items (Numbered Pagination)"
};
const DefaultButtonsOnlyPagination = exports.DefaultButtonsOnlyPagination = {
  render: () => /*#__PURE__*/_react.default.createElement(_Pagination.default, {
    onForward: () => {},
    onBack: () => {}
  }),
  name: "Default (Buttons-only Pagination)"
};
const DefaultButtonLabelsButtonsOnlyPagination = exports.DefaultButtonLabelsButtonsOnlyPagination = {
  render: () => /*#__PURE__*/_react.default.createElement(_Pagination.default, {
    onForward: () => {},
    onBack: () => {},
    showLabels: true
  }),
  name: "Default button labels (Buttons-only Pagination)"
};
const CustomButtonLabelsButtonsOnlyPagination = exports.CustomButtonLabelsButtonsOnlyPagination = {
  render: () => /*#__PURE__*/_react.default.createElement(_Pagination.default, {
    onForward: () => {},
    onBack: () => {},
    showLabels: true,
    forwardLabel: "Custom forward label",
    backLabel: "Custom back label"
  }),
  name: "Custom button labels (Buttons-only Pagination)"
};
const CenteredButtonsOnlyPagination = exports.CenteredButtonsOnlyPagination = {
  render: () => /*#__PURE__*/_react.default.createElement(_Pagination.default, {
    onForward: () => {},
    onBack: () => {},
    centered: true
  }),
  name: "Centered (Buttons-only Pagination)"
};
