// @canonical/react-components@4.2.0/dist/esm/components/TablePagination/utils.js
import { Children, cloneElement, useEffect, useState } from "react";

/**
 * Determine if we are working with a small screen.
 * 'small screen' in this case is relative to the width of the description div
 */
export var figureSmallScreen = () => {
  var descriptionElement = document.getElementById("pagination-description");
  if (!descriptionElement) {
    return true;
  }
  return descriptionElement.getBoundingClientRect().width < 230;
};

/**
 * Iterate direct react child components and override the value of the prop specified by @param dataForwardProp
 * for those child components.
 * @param children - react node children to iterate
 * @param dataForwardProp - the name of the prop from the children components to override
 * @param data - actual data to be passed to the prop specified by @param dataForwardProp
 */
export var renderChildren = (children, dataForwardProp, data) => {
  return Children.map(children, child => {
    return /*#__PURE__*/cloneElement(child, {
      [dataForwardProp]: data
    });
  });
};
export var DEFAULT_PAGE_LIMITS = [50, 100, 200];
export var generatePagingOptions = pageLimits => {
  return pageLimits.map(limit => ({
    value: limit,
    label: "".concat(limit, "/page")
  }));
};
export var getDescription = _ref => {
  var {
    description,
    isSmallScreen,
    totalItems,
    itemName,
    visibleCount,
    currentPage
  } = _ref;
  if (description) {
    return description;
  }
  var closing = "";
  if (typeof totalItems === "number") {
    closing = " out of ".concat(totalItems);
  } else if (currentPage !== 1) {
    closing = " of more than ".concat(visibleCount);
  }
  if (isSmallScreen) {
    return "".concat(visibleCount).concat(closing);
  }
  if (visibleCount === totalItems && visibleCount > 1) {
    return "Showing all ".concat(totalItems, " ").concat(itemName, "s");
  }
  return "Showing ".concat(visibleCount).concat(closing, " ").concat(itemName).concat(totalItems !== 1 ? "s" : "");
};
export var useFigureSmallScreen = () => {
  var [isSmallScreen, setSmallScreen] = useState(false);
  useEffect(() => {
    var handleResize = () => {
      setSmallScreen(figureSmallScreen());
    };
    window.addEventListener("resize", handleResize);
    return () => {
      window.removeEventListener("resize", handleResize);
    };
  }, []);
  return isSmallScreen;
};
