// @canonical/react-components@4.2.0/dist/esm/components/SideNavigation/SideNavigation.stories.js
import React from "react";
import Badge from "../Badge";
import SideNavigation from "./SideNavigation";
var meta = {
  component: SideNavigation,
  tags: ["autodocs"]
};
export default meta;
/**
 * Menu items can be provided as links, text or custom components. To provide attributes to individual menus then each menu can be provided as a object containing an items array: `{ className: "menu-one", items: [...] }`.
 */
export var Default = {
  args: {
    items: [{
      className: "menu-one",
      items: [{
        icon: "drag",
        label: "Models",
        href: "/models",
        status: /*#__PURE__*/React.createElement(Badge, {
          value: 9,
          isNegative: true
        })
      }, {
        icon: "menu",
        label: "Controllers",
        nonInteractive: true
      }, {
        icon: "user",
        label: "Permissions",
        href: "/users"
      }]
    }, {
      className: "menu-two",
      items: [{
        icon: "information",
        label: "Docs",
        href: "/docs"
      }, {
        label: "Logout",
        href: "/logout"
      }]
    }]
  }
};

/**
 * `children` can be provided instead of `items` in cases where custom content
 * is required.
 */
export var CustomContent = {
  args: {
    children: /*#__PURE__*/React.createElement("div", null, "Custom menu content.")
  }
};
