// @canonical/react-components@4.2.0/dist/esm/components/ConfirmationButton/ConfirmationButton.js
var _excluded = ["confirmationModalProps", "onHoverText", "shiftClickEnabled", "showShiftClickHint", "preModalOpenHook"],
  _excluded2 = ["renderInPortal", "portalRenderer"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import React from "react";
import ActionButton from "../ActionButton";
import ConfirmationModal from "../ConfirmationModal";
import { usePortal } from "../../external";
var generateTitle = title => {
  if (typeof title === "string") {
    return title;
  }
  if (typeof title === "number") {
    return title.toString();
  }
  return null;
};
/**
 * `ConfirmationButton` is a specialised version of the [ActionButton](?path=/docs/actionbutton--default-story) component that uses a [ConfirmationModal](?path=/docs/confirmationmodal--default-story) to prompt a confirmation from the user before executing an action.
 */
export var ConfirmationButton = _ref => {
  var {
      confirmationModalProps,
      onHoverText,
      shiftClickEnabled = false,
      showShiftClickHint = false,
      preModalOpenHook
    } = _ref,
    actionButtonProps = _objectWithoutProperties(_ref, _excluded);
  var {
    openPortal,
    closePortal,
    isOpen,
    Portal
  } = usePortal();
  var _ref2 = confirmationModalProps,
    {
      renderInPortal: _ignoredRenderInPortal,
      portalRenderer: _ignoredPortalRenderer
    } = _ref2,
    safeConfirmationModalProps = _objectWithoutProperties(_ref2, _excluded2);
  var handleCancelModal = () => {
    closePortal();
    if (confirmationModalProps.close) {
      confirmationModalProps.close();
    }
  };
  var handleConfirmModal = e => {
    closePortal();
    confirmationModalProps.onConfirm(e);
  };
  var handleShiftClick = e => {
    if (e.shiftKey) {
      confirmationModalProps.onConfirm(e);
    } else {
      openPortal(e);
    }
  };
  var handleClick = e => {
    if (preModalOpenHook) {
      var result = preModalOpenHook(e);
      if (!result) return;
    }
    shiftClickEnabled ? handleShiftClick(e) : openPortal(e);
  };
  return /*#__PURE__*/React.createElement(React.Fragment, null, isOpen && /*#__PURE__*/React.createElement(ConfirmationModal, _extends({}, safeConfirmationModalProps, {
    close: handleCancelModal,
    confirmButtonLabel: confirmationModalProps.confirmButtonLabel,
    onConfirm: handleConfirmModal,
    portalRenderer: Portal
  }), confirmationModalProps.children, showShiftClickHint && /*#__PURE__*/React.createElement("p", {
    className: "p-text--small u-text--muted u-hide--small"
  }, "Next time, you can skip this confirmation by holding", " ", /*#__PURE__*/React.createElement("code", null, "SHIFT"), " and clicking the action.")), /*#__PURE__*/React.createElement(ActionButton, _extends({}, actionButtonProps, {
    onClick: handleClick,
    title: generateTitle(onHoverText !== null && onHoverText !== void 0 ? onHoverText : confirmationModalProps.confirmButtonLabel)
  }), actionButtonProps.children));
};
export default ConfirmationButton;
