// @canonical/react-components@4.2.0/dist/esm/components/ConfirmationButton/ConfirmationButton.stories.js
import React from "react";
import { useState } from "react";
import ConfirmationButton from "./ConfirmationButton";
import Icon, { ICONS } from "../Icon";
var doNothing = () => {};
var meta = {
  component: ConfirmationButton,
  tags: ["autodocs"]
};
export var Default = {
  render: () => /*#__PURE__*/React.createElement(ConfirmationButton, {
    confirmationModalProps: {
      title: "Confirm delete",
      confirmButtonLabel: "Delete",
      onConfirm: doNothing,
      children: /*#__PURE__*/React.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/React.createElement("br", null), "You cannot undo this action.")
    }
  }, "Delete"),
  name: "Default"
};
export default meta;
export var IconAndLabel = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [isLoading, setLoading] = useState(false);
    var brieflyLoad = () => {
      setLoading(true);
      setTimeout(() => setLoading(false), 2000);
    };
    return /*#__PURE__*/React.createElement(ConfirmationButton, {
      confirmationModalProps: {
        title: "Confirm delete",
        confirmButtonLabel: "Delete",
        onConfirm: brieflyLoad,
        children: /*#__PURE__*/React.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/React.createElement("br", null), "You cannot undo this action.")
      },
      className: "has-icon",
      disabled: isLoading,
      loading: isLoading,
      shiftClickEnabled: true,
      showShiftClickHint: true
    }, /*#__PURE__*/React.createElement(Icon, {
      name: ICONS.delete
    }), /*#__PURE__*/React.createElement("span", null, "Delete"));
  },
  name: "Icon and label"
};
export var IconOnly = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [isLoading, setLoading] = useState(false);
    var brieflyLoad = () => {
      setLoading(true);
      setTimeout(() => setLoading(false), 2000);
    };
    return /*#__PURE__*/React.createElement(ConfirmationButton, {
      confirmationModalProps: {
        title: "Confirm delete",
        confirmButtonLabel: "Delete",
        onConfirm: brieflyLoad,
        children: /*#__PURE__*/React.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/React.createElement("br", null), "You cannot undo this action.")
      },
      className: "has-icon",
      disabled: isLoading,
      loading: isLoading,
      shiftClickEnabled: true,
      showShiftClickHint: true
    }, /*#__PURE__*/React.createElement(Icon, {
      name: ICONS.delete
    }));
  },
  name: "Icon only"
};
export var BaseAppearance = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [isLoading, setLoading] = useState(false);
    var brieflyLoad = () => {
      setLoading(true);
      setTimeout(() => setLoading(false), 2000);
    };
    return /*#__PURE__*/React.createElement(ConfirmationButton, {
      appearance: "base",
      confirmationModalProps: {
        title: "Confirm delete",
        confirmButtonLabel: "Delete",
        onConfirm: brieflyLoad,
        children: /*#__PURE__*/React.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/React.createElement("br", null), "You cannot undo this action.")
      },
      className: "has-icon",
      disabled: isLoading,
      loading: isLoading,
      shiftClickEnabled: true,
      showShiftClickHint: true
    }, /*#__PURE__*/React.createElement(Icon, {
      name: ICONS.delete
    }));
  },
  name: "Base appearance"
};
export var WithPreModalOpenHook = {
  render: () => {
    var dontOpenModal = () => false;
    var openModal = () => true;
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(ConfirmationButton, {
      preModalOpenHook: dontOpenModal,
      confirmationModalProps: {
        title: "Confirm delete",
        confirmButtonLabel: "Delete",
        children: /*#__PURE__*/React.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/React.createElement("br", null), "You cannot undo this action.")
      }
    }, "Don't open modal"), /*#__PURE__*/React.createElement(ConfirmationButton, {
      preModalOpenHook: openModal,
      confirmationModalProps: {
        title: "Confirm delete",
        confirmButtonLabel: "Delete",
        children: /*#__PURE__*/React.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/React.createElement("br", null), "You cannot undo this action.")
      }
    }, "Open modal"));
  },
  name: "With preModalOpenHook handler"
};
