// @canonical/react-components@4.2.0/dist/esm/components/CheckboxInput/CheckboxInput.stories.js
import React from "react";
import CheckboxInput from "./CheckboxInput";
var meta = {
  component: CheckboxInput,
  tags: ["autodocs"],
  argTypes: {
    label: {
      control: {
        type: "text"
      }
    },
    disabled: {
      control: {
        type: "boolean"
      }
    },
    required: {
      control: {
        type: "boolean"
      }
    },
    inline: {
      control: {
        type: "boolean"
      }
    },
    indeterminate: {
      control: {
        type: "boolean"
      }
    }
  },
  args: {
    label: "CheckboxInput"
  }
};
export default meta;
export var Default = {
  name: "Default"
};
export var Children = {
  render: () => /*#__PURE__*/React.createElement(CheckboxInput, {
    label: /*#__PURE__*/React.createElement(React.Fragment, null, "I agree to", /*#__PURE__*/React.createElement("a", {
      href: "http://ubuntu.com/legal"
    }, "Terms and Conditions")),
    name: "CheckboxInput"
  }),
  name: "Label with child elements"
};
export var Disabled = {
  render: () => /*#__PURE__*/React.createElement(CheckboxInput, {
    label: "Disabled",
    disabled: true
  }),
  name: "Disabled"
};
export var Required = {
  render: () => /*#__PURE__*/React.createElement(CheckboxInput, {
    label: "Required",
    required: true
  }),
  name: "Required"
};
export var Inline = {
  render: () => /*#__PURE__*/React.createElement(CheckboxInput, {
    label: "Inline",
    inline: true
  }),
  name: "Inline"
};
export var Indeterminate = {
  render: () => /*#__PURE__*/React.createElement(CheckboxInput, {
    label: "Indeterminate",
    indeterminate: true
  }),
  name: "Indeterminate"
};
