// @canonical/react-components@4.2.0/dist/esm/components/PrefixedInput/PrefixedInput.stories.js
import PrefixedInput from "./PrefixedInput";
var meta = {
  component: PrefixedInput,
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  args: {
    immutableText: "https://",
    placeholder: "example.com"
  }
};
export var WithLabel = {
  args: {
    immutableText: "https://",
    label: "Website URL",
    placeholder: "example.com"
  }
};
export var Disabled = {
  args: {
    immutableText: "@",
    label: "Username",
    placeholder: "username",
    disabled: true
  }
};
export var WithError = {
  args: {
    immutableText: "https://",
    label: "Website URL",
    placeholder: "example.com",
    error: "Invalid URL format"
  }
};
export var WithSuccess = {
  args: {
    immutableText: "https://",
    label: "Website URL",
    placeholder: "example.com",
    success: "Valid URL format"
  }
};
export var WithCaution = {
  args: {
    immutableText: "https://",
    label: "Website URL",
    placeholder: "example.com",
    caution: "This URL might be problematic"
  }
};
export var WithHelpText = {
  args: {
    immutableText: "User ID:",
    label: "User Identifier",
    placeholder: " Enter user ID",
    help: "This will be used to identify your account"
  }
};
export var Required = {
  args: {
    immutableText: "https://",
    label: "Website URL",
    placeholder: "example.com",
    required: true
  }
};
