// @canonical/react-components@4.2.0/dist/esm/components/PrefixedInput/PrefixedInput.js
import _pt from "prop-types";
var _excluded = ["immutableText", "disabled", "className", "wrapperClassName", "label", "error", "help", "caution", "success", "id", "required"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import React, { useId } from "react";
import classNames from "classnames";
import Input from "../Input";
import Field from "../Field";
import "./PrefixedInput.scss";
var PrefixedInput = _ref => {
  var {
      immutableText,
      disabled,
      className,
      wrapperClassName,
      label,
      error,
      help,
      caution,
      success,
      id,
      required
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  var defaultId = useId();
  var inputId = id || defaultId;
  return /*#__PURE__*/React.createElement(Field, {
    label: label,
    error: error,
    help: help,
    caution: caution,
    success: success,
    required: required,
    forId: inputId,
    className: classNames("p-prefixed-input", wrapperClassName)
  }, /*#__PURE__*/React.createElement("div", {
    className: classNames("p-prefixed-input__flex-container", {
      "is-disabled": disabled,
      "is-error": !!error,
      "is-caution": !!caution,
      "is-success": !!success
    })
  }, /*#__PURE__*/React.createElement("span", {
    className: "p-prefixed-input__prefix"
  }, immutableText), /*#__PURE__*/React.createElement(Input, _extends({}, props, {
    id: inputId,
    disabled: disabled,
    required: required
    // IMPORTANT: pass nulls so Input doesn't trigger its own Field wrapper
    ,
    label: null,
    error: null,
    help: null,
    caution: null,
    success: null,
    type: "text",
    className: classNames("p-prefixed-input__input-field", className)
  }))));
};
PrefixedInput.propTypes = {
  immutableText: _pt.string.isRequired
};
export default PrefixedInput;
