// @canonical/react-components@4.2.0/dist/esm/components/Select/Select.js
var _excluded = ["label", "value"],
  _excluded2 = ["caution", "className", "error", "help", "helpClassName", "id", "label", "labelClassName", "onChange", "options", "required", "stacked", "success", "takeFocus", "wrapperClassName"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import classNames from "classnames";
import React, { useEffect, useId, useRef } from "react";
import Field from "../Field";

/**
 * The props for the Select component.
 */

var generateOptions = options => options === null || options === void 0 ? void 0 : options.map(_ref => {
  var {
      label,
      value
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/React.createElement("option", _extends({
    value: value,
    key: "".concat(value) || label
  }, props), label);
});

/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Select](https://docs.vanillaframework.io/base/forms/#select).
 *
 * Use the Select component to create a drop-down list.
 */
var Select = _ref2 => {
  var {
      caution,
      className,
      error,
      help,
      helpClassName,
      id,
      label,
      labelClassName,
      onChange: _onChange,
      options,
      required,
      stacked,
      success,
      takeFocus,
      wrapperClassName
    } = _ref2,
    selectProps = _objectWithoutProperties(_ref2, _excluded2);
  var selectRef = useRef(null);
  var validationId = useId();
  var defaultSelectId = useId();
  var selectId = id || defaultSelectId;
  var helpId = useId();
  var hasError = !!error;
  useEffect(() => {
    if (takeFocus) {
      selectRef.current.focus();
    }
  }, [takeFocus]);
  return /*#__PURE__*/React.createElement(Field, {
    caution: caution,
    className: wrapperClassName,
    error: error,
    forId: selectId,
    help: help,
    helpClassName: helpClassName,
    helpId: helpId,
    isSelect: true,
    label: label,
    labelClassName: labelClassName,
    required: required,
    stacked: stacked,
    success: success,
    validationId: validationId
  }, /*#__PURE__*/React.createElement("select", _extends({
    "aria-describedby": [help ? helpId : null, success ? validationId : null].filter(Boolean).join(" "),
    "aria-errormessage": hasError ? validationId : null,
    "aria-invalid": hasError,
    className: classNames("p-form-validation__input", className),
    id: selectId,
    onChange: evt => _onChange && _onChange(evt),
    ref: selectRef,
    required: required
  }, selectProps), generateOptions(options)));
};
export default Select;
