// @canonical/react-components@4.2.0/dist/esm/components/Notifications/ToastNotification/ToastNotification.stories.js
import React, { useEffect } from "react";
import { ToastNotificationProvider, useToastNotification } from "./index";
import Button from "../../Button";
var meta = {
  component: ToastNotificationProvider,
  tags: ["autodocs"],
  argTypes: {
    autoDismissDelay: {
      control: {
        type: "select"
      },
      options: [1000, 2000, 3000, 5000, 10000, 0],
      labels: {
        1000: "1 second",
        2000: "2 seconds",
        3000: "3 seconds",
        5000: "5 seconds (default)",
        10000: "10 seconds",
        0: "Persistent (no auto-hide)"
      },
      description: "Delay in milliseconds before hiding notification, or 0 for persistent notifications"
    }
  },
  args: {
    autoDismissDelay: 5000
  }
};
export default meta;
export var Default = {
  name: "Default",
  render: args => /*#__PURE__*/React.createElement(ToastNotificationStoryWrapper, args)
};
var ToastNotificationStoryWrapper = _ref => {
  var {
    autoDismissDelay
  } = _ref;
  useEffect(() => {
    var root = document.getElementById("storybook-root");
    if (root) {
      root.style.height = "90vh";
    }
  }, []);
  return /*#__PURE__*/React.createElement(ToastNotificationProvider, {
    autoDismissDelay: autoDismissDelay
  }, /*#__PURE__*/React.createElement(PreloadedList, null));
};
var PreloadedList = () => {
  var toastNotify = useToastNotification();
  useEffect(() => {
    toastNotify.success("Settings saved successfully");
    toastNotify.info("Your changes are syncing in the background");
    toastNotify.failure("Save failed", new Error("500 Internal Server Error"), "Please try again.");
    toastNotify.caution("You have unsaved changes.");
    toastNotify.toggleListView();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return /*#__PURE__*/React.createElement("div", {
    style: {
      margin: "2rem auto",
      display: "flex",
      flexDirection: "column",
      flexWrap: "wrap",
      alignContent: "flex-start",
      gap: "0.75rem"
    }
  }, /*#__PURE__*/React.createElement(Button, {
    onClick: () => toastNotify.success("Your changes have been saved.", [{
      label: "Undo",
      onClick: () => console.log("Undo clicked")
    }])
  }, "Add Success"), /*#__PURE__*/React.createElement(Button, {
    onClick: () => toastNotify.info("Your changes are syncing in the background.", "Syncing")
  }, "Add Info"), /*#__PURE__*/React.createElement(Button, {
    onClick: () => toastNotify.failure("Save failed", new Error("500 Internal Server Error"), "Please try again.", [{
      label: "Retry",
      onClick: () => console.log("Retry clicked")
    }])
  }, "Add Error"), /*#__PURE__*/React.createElement(Button, {
    onClick: () => toastNotify.caution("You have unsaved changes")
  }, "Add Warning"), /*#__PURE__*/React.createElement(Button, {
    onClick: toastNotify.toggleListView
  }, "Toggle List View"));
};
