// @canonical/react-components@4.2.0/dist/esm/components/Notifications/ToastNotification/ToastNotification.js
import _pt from "prop-types";
import Notification from "./..";
import { DefaultTitles } from "../Notification/Notification";
import { createPortal } from "react-dom";
import Animate from "./Animate";
import React from "react";
import "./Toast.scss";
var ToastNotification = _ref => {
  var _notification$title;
  var {
    notification,
    onDismiss: _onDismiss,
    show,
    onHoverStart,
    onHoverEnd
  } = _ref;
  if (!notification) {
    return null;
  }
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/createPortal(/*#__PURE__*/React.createElement(Animate, {
    show: show,
    from: {
      opacity: 0
    },
    to: {
      opacity: 1
    },
    exitAnimation: [{
      opacity: 1,
      transform: "translateY(0)"
    }, {
      opacity: 0,
      transform: "translateY(50px)"
    }],
    options: {
      duration: 200
    },
    className: "toast-animate"
  }, /*#__PURE__*/React.createElement("div", {
    className: "toast-notification",
    onMouseEnter: onHoverStart,
    onMouseLeave: onHoverEnd
  }, /*#__PURE__*/React.createElement(Notification, {
    title: (_notification$title = notification.title) !== null && _notification$title !== void 0 ? _notification$title : DefaultTitles[notification.type],
    actions: notification.actions,
    severity: notification.type,
    onDismiss: () => {
      _onDismiss([notification]);
    },
    className: "u-no-margin--bottom",
    timestamp: notification.timestamp,
    titleElement: "div",
    role: "alert"
  }, notification.message))), document.body));
};
ToastNotification.propTypes = {
  onDismiss: _pt.func.isRequired,
  show: _pt.bool.isRequired,
  onHoverStart: _pt.func,
  onHoverEnd: _pt.func
};
export default ToastNotification;
