// @canonical/react-components@4.2.0/dist/components/Tabs/Tabs.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.HorizontallyAligned = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _Row = _interopRequireDefault(require("../Row"));
var _Tabs = _interopRequireDefault(require("./Tabs"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _Tabs.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default",
  args: {
    links: [{
      active: true,
      label: "Summary"
    }, {
      active: false,
      label: "Network"
    }, {
      active: false,
      label: "Storage"
    }, {
      active: false,
      label: "Settings"
    }]
  }
};

/**
 * To horizontally align the Tabs with other content, they can be contained within a Row component to
provide correct gutters.
 */
const HorizontallyAligned = exports.HorizontallyAligned = {
  render: () => /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Tabs.default, {
    links: [{
      active: true,
      label: "Summary"
    }, {
      active: false,
      label: "Network"
    }, {
      active: false,
      label: "Storage"
    }, {
      active: false,
      label: "Settings"
    }]
  }), /*#__PURE__*/_react.default.createElement("p", null, "There should be gutters and the text should be horizontally aligned")),
  name: "Horizontally aligned"
};
