// @canonical/react-components@4.2.0/dist/esm/components/SummaryButton/SummaryButton.stories.js
import React from "react";
import SummaryButton from "./SummaryButton";
var doNothing = () => {};
var meta = {
  component: SummaryButton,
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  render: () => /*#__PURE__*/React.createElement(SummaryButton, {
    summary: "Showing 15 out of 100 items.",
    label: "Show more",
    onClick: doNothing
  }),
  name: "Default"
};

/**
 * Summary text is optional.
 */
export var ActionOnly = {
  render: () => /*#__PURE__*/React.createElement(SummaryButton, {
    label: "Show more items",
    onClick: doNothing
  }),
  name: "Action only"
};

/**
 * Action is optional as well. When `onClick` prop is not provided component will render just the summary text.
 */
export var SummaryOnly = {
  render: () => /*#__PURE__*/React.createElement(SummaryButton, {
    summary: "Showing 100 out of 100 items.",
    label: "Show more items",
    onClick: doNothing
  }),
  name: "Summary only"
};

/**
 * When data is loaded asynchronusly `isLoading` prop should be set to `true` to shows a spinner animation in place of the action button.
 */
export var Loading = {
  render: () => /*#__PURE__*/React.createElement(SummaryButton, {
    summary: "Showing 15 out of 100 items.",
    label: "Show more",
    isLoading: true,
    onClick: doNothing
  }),
  name: "Loading"
};
