// @canonical/react-components@4.2.0/dist/components/CodeSnippet/CodeSnippet.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WrapLines = exports.Title = exports.MultipleBlocks = exports.JsxCodeElements = exports.DropdownsStacked = exports.Dropdowns = exports.Dropdown = exports.Default = exports.Content = exports.Appearance = void 0;
var _react = _interopRequireWildcard(require("react"));
var _CodeSnippet = _interopRequireDefault(require("./CodeSnippet"));
var _CodeSnippetBlock = require("./CodeSnippetBlock");
var _Button = _interopRequireDefault(require("../Button"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const meta = {
  component: _CodeSnippet.default,
  tags: ["autodocs"],
  argTypes: {
    className: {
      control: {
        type: "text"
      }
    }
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: () => /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, {
    blocks: [{
      code: "Mode: all Settings: maas_url=http://192.168.122.1:5240/MAAS"
    }]
  }),
  name: "Default"
};

/**
 * Use the `title` prop to add a title to a code block.
 */
const Title = exports.Title = {
  render: () => /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, {
    blocks: [{
      title: "Output",
      code: "Mode: all Settings: maas_url=http://192.168.122.1:5240/MAAS"
    }]
  }),
  name: "Title"
};

/**
 * Pass an array of `blocks` to render multiple code blocks (for example input and output) within a single code snippet.
 */
const MultipleBlocks = exports.MultipleBlocks = {
  render: () => /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, {
    blocks: [{
      title: "JavaScript",
      code: "console.log('Vanilla');"
    }, {
      title: "Output",
      code: "Vanilla"
    }]
  }),
  name: "Multiple blocks"
};

/**
 * Use `appearance` to control the visual aspects of the code block. `"numbered"` appearance will add line numbers in the code blocks.
 * Values of `"linuxPrompt"`, `"windowsPrompt"`, `"url"` will add a relevant icon in the code block.
 */
const Appearance = exports.Appearance = {
  render: () => /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, {
    blocks: [{
      title: "Install on Linux",
      appearance: _CodeSnippetBlock.CodeSnippetBlockAppearance.LINUX_PROMPT,
      code: "snap install toto"
    }, {
      title: "Install on Windows",
      appearance: _CodeSnippetBlock.CodeSnippetBlockAppearance.WINDOWS_PROMPT,
      code: "snap install toto"
    }, {
      title: "Get from the Store",
      appearance: _CodeSnippetBlock.CodeSnippetBlockAppearance.URL,
      code: "http://snapcraft.io/toto"
    }]
  }),
  name: "Appearance"
};

/**
 * Set `wrapLines` prop to `true` to enable line wrapping inside the code block.
 */
const WrapLines = exports.WrapLines = {
  render: () => /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, {
    blocks: [{
      appearance: _CodeSnippetBlock.CodeSnippetBlockAppearance.NUMBERED,
      wrapLines: true,
      code: "#!/bin/bash\nset -eu . $CONJURE_UP_SPELLSDIR/sdk/common.sh\nif [[ \"$JUJU_PROVIDERTYPE\" == \"lxd\" ]]; then\n  debug \"Running pre-deploy for $CONJURE_UP_SPELL\"\n  sed \"s/##MODEL##/$JUJU_MODEL/\" $(scriptPath)/lxd-profile.yaml | lxc profile edit \"juju-$JUJU_MODEL\" || exposeResult \"Failed to set profile\" $? \"false\"\nfi\nexposeResult \"Successful pre-deploy.\" 0 \"true\""
    }]
  }),
  name: "Wrap lines"
};

/**
 * Dropdown select menus can be added to the headers of code blocks to allow users to choose one of the options. The dropdowns options are passed via `dropdowns` property in the block options object.
 *
 * With the `options` being an array of option properties compatible with the `Select` options (`[{ label: string, value: string | number }]`).
 */
const Dropdown = exports.Dropdown = {
  render: () => {
    const code = {
      js: "console.log(\"Example 1\");",
      css: ".p-heading--2 { color: red; }",
      html: "<h1 class=\"p-heading--2\">How to use code snippets</h1>"
    };

    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [lang, setLang] = (0, _react.useState)("html");
    return /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, {
      blocks: [{
        code: code[lang],
        dropdowns: [{
          options: [{
            value: "js",
            label: "JS"
          }, {
            value: "css",
            label: "CSS"
          }, {
            value: "html",
            label: "HTML"
          }],
          value: lang,
          onChange: event => {
            setLang(event.target.value);
          }
        }]
      }]
    });
  },
  name: "Dropdown"
};

/**
 * Multiple dropdowns can be passed in if needed.
 */
const Dropdowns = exports.Dropdowns = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [channel, setChannel] = (0, _react.useState)("stable");
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [snap, setSnap] = (0, _react.useState)("firefox");
    const code = "sudo snap install ".concat(snap, " ").concat(channel === "stable" ? "" : "--" + channel);
    return /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, {
      blocks: [{
        title: "Install snap",
        code: code,
        dropdowns: [{
          options: [{
            value: "stable",
            label: "stable"
          }, {
            value: "candidate",
            label: "candidate"
          }, {
            value: "beta",
            label: "beta"
          }, {
            value: "edge",
            label: "edge"
          }],
          value: channel,
          onChange: event => {
            setChannel(event.target.value);
          }
        }, {
          options: [{
            value: "firefox",
            label: "Firefox"
          }, {
            value: "gimp",
            label: "Gimp"
          }, {
            value: "vlc",
            label: "VLC"
          }],
          value: snap,
          onChange: event => {
            setSnap(event.target.value);
          }
        }]
      }]
    });
  },
  name: "Dropdowns"
};

/**
 * If multiple dropdowns may overlap with long title you can use `stacked` variant, by setting the relevant property on code block options.
 */
const DropdownsStacked = exports.DropdownsStacked = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [channel, setChannel] = (0, _react.useState)("stable");
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [snap, setSnap] = (0, _react.useState)("firefox");
    const code = "sudo snap install ".concat(snap, " ").concat(channel === "stable" ? "" : "--" + channel);
    return /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, {
      blocks: [{
        title: "Install Firefox, Gimp or VLC as a snap from different channels using command line",
        code: code,
        stacked: true,
        dropdowns: [{
          options: [{
            value: "stable",
            label: "stable"
          }, {
            value: "candidate",
            label: "candidate"
          }, {
            value: "beta",
            label: "beta"
          }, {
            value: "edge",
            label: "edge"
          }],
          value: channel,
          onChange: event => {
            setChannel(event.target.value);
          }
        }, {
          options: [{
            value: "firefox",
            label: "Firefox"
          }, {
            value: "gimp",
            label: "Gimp"
          }, {
            value: "vlc",
            label: "VLC"
          }],
          value: snap,
          onChange: event => {
            setSnap(event.target.value);
          }
        }]
      }]
    });
  },
  name: "DropdownsStacked"
};

/**
 * Custom elements can be passed to a CodeBlock via the `content` prop. In these cases, a border will be added around the entire code snippet to visually associate the content with the code block.
 */
const Content = exports.Content = {
  render: () => /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, {
    blocks: [{
      title: "A button example",
      code: "<Button>Button</Button>",
      content: /*#__PURE__*/_react.default.createElement("div", {
        style: {
          margin: "0.5rem"
        }
      }, /*#__PURE__*/_react.default.createElement(_Button.default, null, "Button"))
    }]
  }),
  name: "Content"
};

/**
 * It's possible to pass JSX instead of strings to the `code` parameter, either as a single element or an array (e.g. if you want to display line numbers).
 */
const JsxCodeElements = exports.JsxCodeElements = {
  render: () => /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, {
    blocks: [{
      code: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("strong", null, "snap"), " install ", /*#__PURE__*/_react.default.createElement("a", {
        href: "#test"
      }, "toto"), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("strong", null, "apt"), " install ", /*#__PURE__*/_react.default.createElement("a", {
        href: "#test"
      }, "toto"))
    }]
  }),
  name: "JSX code elements"
};
