// @canonical/react-components@4.2.0/dist/components/CodeSnippet/CodeSnippetBlock.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CodeSnippetBlockAppearance = void 0;
exports.default = CodeSnippetBlock;
var _classnames = _interopRequireDefault(require("classnames"));
var _react = _interopRequireDefault(require("react"));
var _CodeSnippetDropdown = _interopRequireDefault(require("./CodeSnippetDropdown"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const CodeSnippetBlockAppearance = exports.CodeSnippetBlockAppearance = {
  LINUX_PROMPT: "linuxPrompt",
  NUMBERED: "numbered",
  URL: "url",
  WINDOWS_PROMPT: "windowsPrompt"
};
function CodeSnippetBlock(_ref) {
  let {
    appearance,
    code,
    content,
    dropdowns,
    stacked = false,
    title,
    wrapLines = false
  } = _ref;
  let className = "p-code-snippet__block";
  const isNumbered = appearance === CodeSnippetBlockAppearance.NUMBERED;
  const hasIcon = appearance === CodeSnippetBlockAppearance.LINUX_PROMPT || appearance === CodeSnippetBlockAppearance.WINDOWS_PROMPT || appearance === CodeSnippetBlockAppearance.URL;
  let numberedCode;
  if (isNumbered) {
    className += "--numbered";

    // wrap code lines in spans (and preserve the whitespace)
    let lines;
    if (Array.isArray(code)) {
      lines = code;
    } else if (typeof code === "string") {
      lines = code.split(/\r?\n/);
    } else {
      // If the code is not able to be split over multiple lines, then display
      // a single line number.
      lines = [code];
    }
    numberedCode = lines.map((line, i) => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, {
      key: "p-code-snippet__line-".concat(i)
    }, /*#__PURE__*/_react.default.createElement("span", {
      className: "p-code-snippet__line"
    }, line), "\n"));
  } else if (hasIcon) {
    className += "--icon";
  }
  className = (0, _classnames.default)(className, {
    "is-windows-prompt": appearance === CodeSnippetBlockAppearance.WINDOWS_PROMPT,
    "is-url": appearance === CodeSnippetBlockAppearance.URL,
    "is-wrapped": wrapLines
  });
  const hasDropdowns = dropdowns && dropdowns.length;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, (title || hasDropdowns) && /*#__PURE__*/_react.default.createElement("div", {
    className: "p-code-snippet__header ".concat(stacked ? "is-stacked" : "")
  }, /*#__PURE__*/_react.default.createElement("h5", {
    className: "p-code-snippet__title"
  }, title), hasDropdowns && /*#__PURE__*/_react.default.createElement("div", {
    className: "p-code-snippet__dropdowns"
  }, dropdowns.map((dropdown, i) => /*#__PURE__*/_react.default.createElement(_CodeSnippetDropdown.default, _extends({}, dropdown, {
    key: "code-snippet-dropdown-".concat(i)
  }))))), /*#__PURE__*/_react.default.createElement("pre", {
    className: className
  }, /*#__PURE__*/_react.default.createElement("code", null, isNumbered ? numberedCode : code)), content);
}
