// @canonical/react-components@4.2.0/dist/esm/components/Tooltip/Tooltip.stories.js
import React from "react";
import Button from "../Button";
import Tooltip, { position } from "./Tooltip";
var Template = args => {
  return /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      justifyContent: "center",
      padding: "3rem"
    }
  }, /*#__PURE__*/React.createElement(Tooltip, args, /*#__PURE__*/React.createElement(Button, {
    className: "u-no-margin--bottom"
  }, "Hover over me!")));
};
var meta = {
  component: Tooltip,
  render: Template,
  tags: ["autodocs"],
  argTypes: {
    position: {
      control: {
        type: "radio"
      },
      options: Object.values(position)
    }
  }
};
export default meta;
export var Default = {
  render: Template.bind({}),
  name: "Default",
  args: {
    message: /*#__PURE__*/React.createElement(React.Fragment, null, "Tooltip", " ", /*#__PURE__*/React.createElement("a", {
      href: "#todo",
      target: "_blank"
    }, "Link1"))
  }
};
export var FollowMouse = {
  render: Template.bind({}),
  name: "Follow mouse",
  args: {
    followMouse: true,
    message: "Tooltip message to display.",
    position: "top-center"
  }
};

/**
 * The tooltip elements are displayed inside a [Portal](https://reactjs.org/docs/portals.html) so that the tooltip is not constrained by its current position in the DOM.
 * Sometimes however, you need to target the tooltip elements. This can be achieved by passing class names to the tooltip elements. You can also pass classes to the elements that trigger and position the tooltip.
 */
export var TargetingElements = {
  render: Template.bind({}),
  name: "Targeting elements",
  args: {
    className: "custom-wrapping-class",
    positionElementClassName: "custom-position-class",
    tooltipClassName: "custom-tooltip-class",
    message: "Tooltip message to display."
  }
};
