// @canonical/react-components@4.2.0/dist/esm/components/Tooltip/Tooltip.js
import classNames from "classnames";
import React, { useCallback, useEffect, useId, useRef, useState } from "react";
import { useWindowFitment, useListener } from "../../hooks";
import { usePortal } from "../../external";
export var position = {
  btmCenter: "btm-center",
  btmLeft: "btm-left",
  btmRight: "btm-right",
  left: "left",
  right: "right",
  topCenter: "top-center",
  topLeft: "top-left",
  topRight: "top-right"
};
var getPositionStyle = (pos, wrapperNode) => {
  if (!wrapperNode) {
    return null;
  }
  var dimensions = wrapperNode.getBoundingClientRect();
  var {
    x,
    y,
    height,
    width
  } = dimensions;
  var left = x + window.scrollX || 0;
  var top = y + window.scrollY || 0;
  switch (pos) {
    case "btm-center":
      left += width / 2;
      top += height;
      break;
    case "btm-left":
      top += height;
      break;
    case "btm-right":
      left += width;
      top += height;
      break;
    case "left":
      top += height / 2;
      break;
    case "right":
      left += width;
      top += height / 2;
      break;
    case "top-center":
      left += width / 2;
      break;
    case "top-left":
      break;
    case "top-right":
      left += width;
      break;
    default:
      break;
  }
  return {
    position: "absolute",
    left,
    top
  };
};
export var adjustForWindow = function adjustForWindow(position, fitsWindow) {
  var messageHeight = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 0;
  var newPosition = position;
  if (!fitsWindow.fromLeft.fitsLeft && newPosition === "left") {
    newPosition = "top-right";
  }
  if (fitsWindow.fromTop.spaceAbove < messageHeight / 2 && newPosition === "left") {
    newPosition = "btm-left";
  }
  if (fitsWindow.fromBottom.spaceBelow < messageHeight / 2 && newPosition === "left") {
    newPosition = "top-left";
  }
  if (!fitsWindow.fromRight.fitsRight && newPosition === "right") {
    newPosition = "top-left";
  }
  if (fitsWindow.fromTop.spaceAbove < messageHeight / 2 && newPosition === "right") {
    newPosition = "btm-right";
  }
  if (fitsWindow.fromBottom.spaceBelow < messageHeight / 2 && newPosition === "right") {
    newPosition = "top-right";
  }
  if (!fitsWindow.fromRight.fitsLeft && newPosition.endsWith("-right")) {
    newPosition = newPosition.replace("right", "left");
  }
  if (!fitsWindow.fromLeft.fitsRight && newPosition.endsWith("-left")) {
    newPosition = newPosition.replace("left", "right");
  }
  if (!fitsWindow.fromTop.fitsAbove && newPosition.startsWith("top")) {
    newPosition = newPosition.replace("top", "btm");
  }
  if (!fitsWindow.fromBottom.fitsBelow && newPosition.startsWith("btm")) {
    newPosition = newPosition.replace("btm", "top");
  }
  if (!fitsWindow.fromLeft.fitsRight && !fitsWindow.fromRight.fitsLeft && (newPosition.endsWith("-left") || newPosition.endsWith("-right"))) {
    newPosition = newPosition.replace("left", "center").replace("right", "center");
  }
  if (newPosition.endsWith("center") && (fitsWindow.fromCenter.fitsCentered.fitsRight || fitsWindow.fromCenter.fitsCentered.fitsLeft)) {
    if (!fitsWindow.fromCenter.fitsCentered.fitsRight) {
      newPosition = newPosition.replace("center", "right");
    }
    if (!fitsWindow.fromCenter.fitsCentered.fitsLeft) {
      newPosition = newPosition.replace("center", "left");
    }
  }
  return newPosition;
};

/**
 * This is the [React](https://reactjs.org/) component for Vanilla [Tooltips](https://vanillaframework.io/docs/patterns/tooltips).
 *
 * Tooltips are text labels that appear when the user hovers over, focuses on, or touches an element on the screen.
 *
 * They can be used to provide information about concepts/terms/actions that are not self-explanatory or well known.
 *
 * An alternative use of tooltips is to provide information on a disabled actionable element, e.g. for disabled buttons, providing information about why they are disabled.
 */
var Tooltip = _ref => {
  var {
    autoAdjust = true,
    children,
    className,
    followMouse = false,
    message,
    position = "top-left",
    positionElementClassName,
    tooltipClassName,
    zIndex,
    delay = 350
  } = _ref;
  var wrapperRef = useRef(null);
  var messageRef = useRef(null);
  var [adjustedPosition, setAdjustedPosition] = useState(position);
  var [positionStyle, setPositionStyle] = useState({
    position: "absolute",
    // Initially position the tooltip of the screen in case it gets shown
    // before setting the position.
    left: -9999999,
    top: -9999999
  });
  var {
    openPortal,
    closePortal,
    isOpen,
    Portal
  } = usePortal({
    programmaticallyOpen: true
  });
  var tooltipId = useId();
  var [timer, setTimer] = useState(null);
  var cancelableClosePortal = useCallback(() => {
    clearTimeout(timer);
    closePortal();
  }, [timer, closePortal]);
  useEffect(() => {
    if (isOpen && !followMouse && wrapperRef.current) {
      // Position the tooltip when it becomes visible.
      setPositionStyle(getPositionStyle(adjustedPosition, wrapperRef.current));
    }
  }, [adjustedPosition, isOpen, followMouse]);
  var mouseHandler = useCallback(evt => {
    // Set the position of the tooltip next to the mouse.
    setPositionStyle({
      // Don't allow the tooltip to block the mouse events.
      pointerEvents: "none",
      position: "absolute",
      left: evt.pageX,
      top: evt.pageY
    });
  }, []);
  var onUpdateWindowFitment = useCallback(fitsWindow => {
    var _messageRef$current$g, _messageRef$current;
    var messageHeight = (_messageRef$current$g = (_messageRef$current = messageRef.current) === null || _messageRef$current === void 0 ? void 0 : _messageRef$current.getBoundingClientRect().height) !== null && _messageRef$current$g !== void 0 ? _messageRef$current$g : 0;
    setAdjustedPosition(adjustForWindow(position, fitsWindow, messageHeight));
  }, [setAdjustedPosition, position]);

  // Handle mouse events.
  useListener(wrapperRef.current, mouseHandler, "mousemove", true, followMouse);

  // Handle adjusting the position of the tooltip so that it remains on screen.
  useWindowFitment(messageRef.current, wrapperRef.current, onUpdateWindowFitment, 20, isOpen, autoAdjust && followMouse);
  var handleKeyPress = useCallback(event => {
    if (event.key === "Escape") {
      cancelableClosePortal();
    }
  }, [cancelableClosePortal]);
  useEffect(() => {
    window.addEventListener("keypress", handleKeyPress);
    return () => {
      window.removeEventListener("keypress", handleKeyPress);
    };
  }, [handleKeyPress]);
  var handleBlur = e => {
    var _wrapperRef$current, _messageRef$current2, _wrapperRef$current2;
    // do not close if the focus is within the tooltip wrapper
    if (wrapperRef !== null && wrapperRef !== void 0 && (_wrapperRef$current = wrapperRef.current) !== null && _wrapperRef$current !== void 0 && _wrapperRef$current.contains(document.activeElement)) {
      return;
    }
    if (e.relatedTarget ? !((_messageRef$current2 = messageRef.current) !== null && _messageRef$current2 !== void 0 && _messageRef$current2.contains(e.relatedTarget)) && !((_wrapperRef$current2 = wrapperRef.current) !== null && _wrapperRef$current2 !== void 0 && _wrapperRef$current2.contains(e.relatedTarget)) : e.target !== messageRef.current) {
      cancelableClosePortal();
    }
  };
  var handleClick = e => {
    var _messageRef$current3;
    // ignore clicks within the tooltip message
    if ((_messageRef$current3 = messageRef.current) !== null && _messageRef$current3 !== void 0 && _messageRef$current3.contains(e.target)) {
      return;
    }
    e.target.focus();
    openPortal();
  };
  var handleFocus = () => {
    if (followMouse && wrapperRef.current) {
      // set initial position for the tooltip
      setPositionStyle(getPositionStyle(adjustedPosition, wrapperRef.current));
    }
    openPortal();
  };
  var delayedOpenPortal = useCallback(() => {
    if (isOpen) {
      return;
    }
    if (timer) {
      clearTimeout(timer);
    }
    var timeout = setTimeout(() => openPortal(), delay);
    setTimer(timeout);
  }, [delay, openPortal, timer, isOpen]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, message ? /*#__PURE__*/React.createElement("span", {
    className: className,
    onBlur: handleBlur,
    onClick: handleClick,
    onFocus: handleFocus,
    onMouseOut: handleBlur,
    onMouseOver: delayedOpenPortal
  }, /*#__PURE__*/React.createElement("span", {
    className: positionElementClassName,
    ref: wrapperRef,
    style: {
      display: "inline-block"
    }
  }, React.Children.map(children, child => child && /*#__PURE__*/React.isValidElement(child) ? /*#__PURE__*/React.cloneElement(child, {
    "aria-describedby": isOpen ? tooltipId : undefined
  }) : child)), isOpen ? /*#__PURE__*/React.createElement(Portal, null, /*#__PURE__*/React.createElement("span", {
    className: classNames("p-tooltip--".concat(adjustedPosition), "is-detached", tooltipClassName),
    "data-testid": "tooltip-portal",
    style: positionStyle
  }, /*#__PURE__*/React.createElement("span", {
    role: "tooltip",
    className: "p-tooltip__message",
    onClick: event => {
      // Prevent clicks inside the message from bubbling to parent
      // click handlers.
      event.stopPropagation();
    },
    ref: messageRef,
    id: tooltipId,
    style: {
      zIndex: zIndex
    }
  }, message))) : null) : /*#__PURE__*/React.createElement("span", {
    className: className
  }, children));
};
export default Tooltip;
