// @canonical/react-components@4.2.0/dist/esm/components/Modal/Modal.js
var _excluded = ["buttonRow", "children", "className", "close", "focusRef", "title", "shouldPropagateClickEvent", "closeOnOutsideClick"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import classNames from "classnames";
import React, { useId, useRef, useEffect } from "react";
/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Modal](https://docs.vanillaframework.io/patterns/modal/).
 *
 * The modal component can be used to overlay an area of the screen which can contain a prompt, dialog or interaction.
 */
export var Modal = _ref => {
  var {
      buttonRow,
      children,
      className,
      close,
      focusRef,
      title,
      shouldPropagateClickEvent = false,
      closeOnOutsideClick = true
    } = _ref,
    wrapperProps = _objectWithoutProperties(_ref, _excluded);
  // list of focusable selectors is based on this Stack Overflow answer:
  // https://stackoverflow.com/a/30753870/3732840
  var focusableElementSelectors = 'a[href]:not([tabindex="-1"]), button:not([disabled]), textarea:not([disabled]):not([tabindex="-1"]), input:not([disabled]):not([tabindex="-1"]), select:not([disabled]):not([tabindex="-1"]), area[href]:not([tabindex="-1"]), iframe:not([tabindex="-1"]), [tabindex]:not([tabindex="-1"]), [contentEditable=true]:not([tabindex="-1"])';
  var descriptionId = useId();
  var titleId = useId();
  var shouldClose = useRef(false);
  var modalRef = useRef(null);
  var closeButtonRef = useRef(null);
  var handleTabKey = event => {
    var focusableModalElements = modalRef.current.querySelectorAll(focusableElementSelectors);
    if (focusableModalElements.length > 0) {
      var firstElement = focusableModalElements[0];
      var lastElement = focusableModalElements[focusableModalElements.length - 1];
      if (!event.shiftKey && document.activeElement === lastElement) {
        firstElement.focus();
        event.preventDefault();
      }
      if (event.shiftKey && document.activeElement === firstElement) {
        lastElement.focus();
        return event.preventDefault();
      }
    }
  };
  var handleEscKey = event => {
    if ("nativeEvent" in event && event.nativeEvent.stopImmediatePropagation) {
      event.nativeEvent.stopImmediatePropagation();
    } else if ("stopImmediatePropagation" in event) {
      event.stopImmediatePropagation();
    } else if (event.stopPropagation) {
      event.stopPropagation();
    }
    if (close) {
      close();
    }
  };
  useEffect(() => {
    if (focusRef !== null && focusRef !== void 0 && focusRef.current) {
      focusRef.current.focus();
    } else if (closeButtonRef.current) {
      closeButtonRef.current.focus();
    } else {
      modalRef.current.focus();
    }
  }, [focusRef]);
  useEffect(() => {
    var keyListenersMap = new Map([["Escape", handleEscKey], ["Tab", handleTabKey]]);
    var keyDown = event => {
      var listener = keyListenersMap.get(event.code);
      return listener && listener(event);
    };
    document.addEventListener("keydown", keyDown);
    return () => {
      document.removeEventListener("keydown", keyDown);
    };
  });
  var handleContentOnMouseDown = () => {
    shouldClose.current = false;
  };
  var handleContentOnMouseUp = () => {
    shouldClose.current = false;
  };
  var handleOverlayOnMouseDown = event => {
    if (event.target === modalRef.current && closeOnOutsideClick) {
      shouldClose.current = true;
    }
  };
  var handleClose = event => {
    if (!shouldPropagateClickEvent) {
      event.stopPropagation();
    }
    if (close) {
      close();
    }
  };
  var handleOverlayOnClick = event => {
    if (shouldClose.current) {
      handleClose(event);
    }
  };
  return /*#__PURE__*/React.createElement("div", _extends({
    className: classNames("p-modal", className),
    onClick: handleOverlayOnClick,
    onMouseDown: handleOverlayOnMouseDown
  }, wrapperProps, {
    ref: modalRef
  }), /*#__PURE__*/React.createElement("section", {
    className: "p-modal__dialog",
    role: "dialog",
    "aria-modal": "true",
    "aria-labelledby": titleId,
    "aria-describedby": descriptionId,
    onMouseDown: handleContentOnMouseDown,
    onMouseUp: handleContentOnMouseUp
  }, !!title && /*#__PURE__*/React.createElement("header", {
    className: "p-modal__header"
  }, /*#__PURE__*/React.createElement("h2", {
    className: "p-modal__title",
    id: titleId
  }, title), close && /*#__PURE__*/React.createElement("button", {
    type: "button",
    className: "p-modal__close",
    "aria-label": "Close active modal",
    onClick: handleClose,
    ref: closeButtonRef
  }, "Close")), /*#__PURE__*/React.createElement("div", {
    id: descriptionId
  }, children), !!buttonRow && /*#__PURE__*/React.createElement("footer", {
    className: "p-modal__footer"
  }, buttonRow)));
};
export default Modal;
