// @canonical/react-components@4.2.0/dist/esm/components/Modal/Modal.stories.js
import { useRef, useState } from "react";
import React from "react";
import Button from "../Button";
import Modal from "./Modal";
var Template = args => {
  return /*#__PURE__*/React.createElement("div", {
    style: {
      height: "100vh",
      position: "relative"
    }
  }, /*#__PURE__*/React.createElement(Modal, args));
};
var meta = {
  component: Modal,
  render: Template,
  tags: ["autodocs"],
  argTypes: {
    buttonRow: {
      control: {
        disable: true
      }
    },
    children: {
      control: {
        disable: true
      }
    },
    closeOnOutsideClick: {
      control: "boolean"
    }
  }
};
export default meta;
export var Default = {
  render: _ref => {
    var {
      closeOnOutsideClick
    } = _ref;
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [modalOpen, setModalOpen] = useState(true);
    var closeHandler = () => setModalOpen(false);
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("button", {
      onClick: () => setModalOpen(true)
    }, "Open modal"), modalOpen ? /*#__PURE__*/React.createElement(Modal, {
      close: closeHandler,
      title: "Confirm delete",
      closeOnOutsideClick: closeOnOutsideClick,
      buttonRow: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("button", {
        className: "u-no-margin--bottom",
        onClick: closeHandler
      }, "Cancel"), /*#__PURE__*/React.createElement("button", {
        className: "p-button--negative u-no-margin--bottom"
      }, "Delete"))
    }, /*#__PURE__*/React.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/React.createElement("br", null), "You cannot undo this action.")) : null);
  },
  name: "Default"
};
export var Focus = {
  render: _ref2 => {
    var {
      closeOnOutsideClick
    } = _ref2;
    /* eslint-disable react-hooks/rules-of-hooks */
    var [modalOpen, setModalOpen] = useState(true);
    var buttonRef = useRef(null);
    /* eslint-enable react-hooks/rules-of-hooks */

    var closeHandler = () => setModalOpen(false);
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("button", {
      onClick: () => setModalOpen(true)
    }, "Open modal"), modalOpen ? /*#__PURE__*/React.createElement(Modal, {
      close: closeHandler,
      title: "Confirm delete",
      closeOnOutsideClick: closeOnOutsideClick,
      buttonRow: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("button", {
        className: "u-no-margin--bottom",
        onClick: closeHandler
      }, "Cancel"), /*#__PURE__*/React.createElement("button", {
        className: "p-button--negative u-no-margin--bottom"
      }, "Delete")),
      focusRef: buttonRef
    }, /*#__PURE__*/React.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/React.createElement("br", null), "You cannot undo this action."), /*#__PURE__*/React.createElement("p", null, /*#__PURE__*/React.createElement(Button, {
      appearance: "link",
      ref: buttonRef
    }, "More information"))) : null);
  },
  name: "Focus"
};
