// @canonical/react-components@4.2.0/dist/components/Notifications/ToastNotification/ToastNotification.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _index = require("./index");
var _Button = _interopRequireDefault(require("../../Button"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const meta = {
  component: _index.ToastNotificationProvider,
  tags: ["autodocs"],
  argTypes: {
    autoDismissDelay: {
      control: {
        type: "select"
      },
      options: [1000, 2000, 3000, 5000, 10000, 0],
      labels: {
        1000: "1 second",
        2000: "2 seconds",
        3000: "3 seconds",
        5000: "5 seconds (default)",
        10000: "10 seconds",
        0: "Persistent (no auto-hide)"
      },
      description: "Delay in milliseconds before hiding notification, or 0 for persistent notifications"
    }
  },
  args: {
    autoDismissDelay: 5000
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default",
  render: args => /*#__PURE__*/_react.default.createElement(ToastNotificationStoryWrapper, args)
};
const ToastNotificationStoryWrapper = _ref => {
  let {
    autoDismissDelay
  } = _ref;
  (0, _react.useEffect)(() => {
    const root = document.getElementById("storybook-root");
    if (root) {
      root.style.height = "90vh";
    }
  }, []);
  return /*#__PURE__*/_react.default.createElement(_index.ToastNotificationProvider, {
    autoDismissDelay: autoDismissDelay
  }, /*#__PURE__*/_react.default.createElement(PreloadedList, null));
};
const PreloadedList = () => {
  const toastNotify = (0, _index.useToastNotification)();
  (0, _react.useEffect)(() => {
    toastNotify.success("Settings saved successfully");
    toastNotify.info("Your changes are syncing in the background");
    toastNotify.failure("Save failed", new Error("500 Internal Server Error"), "Please try again.");
    toastNotify.caution("You have unsaved changes.");
    toastNotify.toggleListView();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      margin: "2rem auto",
      display: "flex",
      flexDirection: "column",
      flexWrap: "wrap",
      alignContent: "flex-start",
      gap: "0.75rem"
    }
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: () => toastNotify.success("Your changes have been saved.", [{
      label: "Undo",
      onClick: () => console.log("Undo clicked")
    }])
  }, "Add Success"), /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: () => toastNotify.info("Your changes are syncing in the background.", "Syncing")
  }, "Add Info"), /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: () => toastNotify.failure("Save failed", new Error("500 Internal Server Error"), "Please try again.", [{
      label: "Retry",
      onClick: () => console.log("Retry clicked")
    }])
  }, "Add Error"), /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: () => toastNotify.caution("You have unsaved changes")
  }, "Add Warning"), /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: toastNotify.toggleListView
  }, "Toggle List View"));
};
