// @canonical/react-components@4.2.0/dist/components/ApplicationLayout/AppAside/AppAside.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _Application = _interopRequireDefault(require("../Application"));
var _AppMain = _interopRequireDefault(require("../AppMain"));
var _Button = _interopRequireDefault(require("../../Button"));
var _Col = _interopRequireDefault(require("../../Col"));
var _Form = _interopRequireDefault(require("../../Form"));
var _Icon = _interopRequireDefault(require("../../Icon"));
var _Input = _interopRequireDefault(require("../../Input"));
var _Panel = _interopRequireDefault(require("../../Panel"));
var _Row = _interopRequireDefault(require("../../Row"));
var _AppAside = _interopRequireDefault(require("./AppAside"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); } /* eslint-disable react-hooks/rules-of-hooks */
const meta = {
  component: _AppAside.default,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: false
    }
  }
};
var _default = exports.default = meta;
/**
 * In most common cases an `AppAside` should contain a `<Panel>` to display the
 * content as intended in the application layout.
 *
 * `AppAside` should be a direct child of an `<Application>` or passed to the
 * application layout `<ApplicationLayout aside={<AppAside .../>}>`.
 */
const Default = exports.Default = {
  render: args => {
    const [pinned, setPinned] = (0, _react.useState)(false);
    const [width, setWidth] = (0, _react.useState)(null);
    const [collapsed, setCollapsed] = (0, _react.useState)(false);
    return /*#__PURE__*/_react.default.createElement(_Application.default, null, /*#__PURE__*/_react.default.createElement(_AppMain.default, null, /*#__PURE__*/_react.default.createElement("p", null, "Scroll to the right to see the panel."), /*#__PURE__*/_react.default.createElement(_Button.default, {
      onClick: () => setCollapsed(false)
    }, "Open"), /*#__PURE__*/_react.default.createElement(_Button.default, {
      onClick: () => setWidth("narrow")
    }, "Narrow"), /*#__PURE__*/_react.default.createElement(_Button.default, {
      onClick: () => setWidth(null)
    }, "Default"), /*#__PURE__*/_react.default.createElement(_Button.default, {
      onClick: () => setWidth("wide")
    }, "Wide")), /*#__PURE__*/_react.default.createElement(_AppAside.default, _extends({}, args, {
      pinned: pinned,
      wide: width === "wide",
      narrow: width === "narrow",
      collapsed: collapsed
    }), /*#__PURE__*/_react.default.createElement(_Panel.default, {
      controls: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_Button.default, {
        onClick: () => setPinned(!pinned),
        dense: true,
        className: "u-no-margin--bottom"
      }, "Pin"), /*#__PURE__*/_react.default.createElement(_Button.default, {
        appearance: "base",
        className: "u-no-margin--bottom",
        hasIcon: true,
        onClick: () => setCollapsed(!collapsed)
      }, /*#__PURE__*/_react.default.createElement(_Icon.default, {
        name: "close"
      }, "Close"))),
      title: "App aside"
    }, /*#__PURE__*/_react.default.createElement(_Form.default, {
      stacked: true
    }, /*#__PURE__*/_react.default.createElement(_Input.default, {
      label: "Full name",
      type: "text",
      name: "fullName",
      autoComplete: "name",
      stacked: true
    }), /*#__PURE__*/_react.default.createElement(_Input.default, {
      label: "Username",
      type: "text",
      name: "username-stacked",
      autoComplete: "username",
      "aria-describedby": "exampleHelpTextMessage",
      stacked: true,
      help: "30 characters or fewer."
    }), /*#__PURE__*/_react.default.createElement(_Input.default, {
      type: "text",
      label: "Email address",
      "aria-invalid": "true",
      name: "username-stackederror",
      autoComplete: "email",
      required: true,
      error: "This field is required.",
      stacked: true
    }), /*#__PURE__*/_react.default.createElement(_Input.default, {
      label: "Address line 1",
      type: "text",
      name: "address-optional-stacked",
      autoComplete: "address-line1",
      stacked: true
    }), /*#__PURE__*/_react.default.createElement(_Input.default, {
      label: "Address line 2",
      type: "text",
      name: "address-optional-stacked",
      autoComplete: "address-line3",
      stacked: true
    }), /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 12
    }, /*#__PURE__*/_react.default.createElement(_Button.default, {
      appearance: "positive",
      className: "u-float-right",
      name: "add-details"
    }, "Add details")))))));
  }
};
