// @canonical/react-components@4.2.0/dist/components/ApplicationLayout/AppNavigation/AppNavigation.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));
var _Application = _interopRequireDefault(require("../Application"));
var _AppMain = _interopRequireDefault(require("../AppMain"));
var _Button = _interopRequireDefault(require("../../Button"));
var _Icon = _interopRequireDefault(require("../../Icon"));
var _Panel = _interopRequireDefault(require("../../Panel"));
var _SideNavigation = _interopRequireDefault(require("../../SideNavigation"));
var _Strip = _interopRequireDefault(require("../../Strip"));
var _AppNavigation = _interopRequireDefault(require("./AppNavigation"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); } /* eslint-disable react-hooks/rules-of-hooks */
const meta = {
  component: _AppNavigation.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
/**
 * In most common cases an `AppNavigation` should contain a `<Panel>` and `<SideNavigation>`.
 *
 * `AppNavigation` should be a direct child of an `<Application>` or when using `ApplicationLayout`
 * it will wrap `navItems` and `sideNavigation`.
 */
const Default = exports.Default = {
  render: args => {
    const [pinned, setPinned] = (0, _react.useState)(false);
    const [collapsed, setCollapsed] = (0, _react.useState)(true);
    return /*#__PURE__*/_react.default.createElement(_Application.default, null, /*#__PURE__*/_react.default.createElement(_AppNavigation.default, _extends({}, args, {
      pinned: pinned,
      collapsed: collapsed
    }), /*#__PURE__*/_react.default.createElement(_Panel.default, {
      dark: true,
      controls: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_Button.default, {
        hasIcon: true,
        appearance: "base",
        className: (0, _classnames.default)("u-no-margin u-hide--medium", {
          "is-dark": true
        }),
        onClick: evt => {
          setCollapsed(true);
          // The menu stays open while its content has focus, so the
          // close button must blur to actually close the menu.
          evt.currentTarget.blur();
        }
      }, /*#__PURE__*/_react.default.createElement(_Icon.default, {
        name: "close",
        className: (0, _classnames.default)({
          "is-light": true
        })
      }, "Close menu")), /*#__PURE__*/_react.default.createElement(_Button.default, {
        hasIcon: true,
        appearance: "base",
        className: (0, _classnames.default)("u-no-margin u-hide--small", {
          "is-dark": true
        }),
        onClick: () => {
          setPinned(!pinned);
        }
      }, /*#__PURE__*/_react.default.createElement(_Icon.default, {
        name: pinned ? "close" : "pin",
        className: (0, _classnames.default)({
          "is-light": true
        })
      }, pinned ? "Unpin menu" : "Pin menu"))),
      controlsClassName: "u-hide--large",
      stickyHeader: true,
      logo: {
        icon: "https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg",
        name: "https://assets.ubuntu.com/v1/2e04d794-logo-jaas.svg",
        nameAlt: "JAAS",
        href: "/"
      }
    }, /*#__PURE__*/_react.default.createElement(_SideNavigation.default, {
      dark: true,
      items: [{
        items: [{
          icon: "drag",
          label: "Models",
          href: "/models"
        }, {
          icon: "menu",
          label: "Controllers",
          href: "/controllers"
        }, {
          icon: "user",
          label: "Permissions",
          href: "/permissions"
        }]
      }]
    }))), /*#__PURE__*/_react.default.createElement(_AppMain.default, null, /*#__PURE__*/_react.default.createElement(_Strip.default, null, /*#__PURE__*/_react.default.createElement("p", null, "These behaviours are only available at smaller screen sizes:"), /*#__PURE__*/_react.default.createElement(_Button.default, {
      onClick: () => setPinned(!pinned)
    }, "Pin"), /*#__PURE__*/_react.default.createElement(_Button.default, {
      onClick: () => setCollapsed(!collapsed)
    }, "Toggle"))));
  }
};
