// @canonical/react-components@4.2.0/dist/components/EmptyState/EmptyState.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithContent = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _EmptyState = _interopRequireDefault(require("./EmptyState"));
var _Icon = _interopRequireWildcard(require("../Icon"));
var _Button = _interopRequireDefault(require("../Button"));
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const boxImage = /*#__PURE__*/_react.default.createElement("img", {
  src: "https://assets.ubuntu.com/v1/c17e0d92-container.svg",
  style: {
    maxWidth: "10rem",
    maxHeight: "10rem"
  },
  alt: "empty state"
});
const doNothing = () => {};
const meta = {
  component: _EmptyState.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: () => /*#__PURE__*/_react.default.createElement(_EmptyState.default, {
    image: boxImage,
    title: "This application doesn't have any configuration parameters"
  }),
  name: "Default"
};
const WithContent = exports.WithContent = {
  render: () => /*#__PURE__*/_react.default.createElement(_EmptyState.default, {
    image: boxImage,
    title: "No instances found"
  }, /*#__PURE__*/_react.default.createElement("p", null, "There are no instances in this project. Spin up your first instance!"), /*#__PURE__*/_react.default.createElement("p", null, /*#__PURE__*/_react.default.createElement("a", {
    href: "#todo",
    target: "_blank"
  }, "How to create instances", /*#__PURE__*/_react.default.createElement(_Icon.default, {
    name: _Icon.ICONS.externalLink
  }))), /*#__PURE__*/_react.default.createElement(_Button.default, {
    appearance: "positive",
    onClick: doNothing
  }, "Create instance")),
  name: "With content"
};
