// @canonical/react-components@4.2.0/dist/utils.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.toFloat = exports.isReactNode = exports.isNavigationButton = exports.isNavigationAnchor = exports.highlightSubString = exports.getParentsBottomSpacing = exports.getElementAbsoluteHeight = exports.getAbsoluteHeightBelowById = exports.IS_DEV = void 0;
var _react = require("react");
const IS_DEV = exports.IS_DEV = process.env.NODE_ENV === "development";

/**
 * Find substring and wrap in <strong /> tag
 * @param {string} str - The string to search
 * @param {string} subString - The substring to find
 * @return {Obj} newStr - Object with text and match bool
 */
const highlightSubString = (str, subString) => {
  if (typeof str !== "string" || typeof subString !== "string") {
    return {
      text: str || "",
      match: false
    };
  }
  const escapedSubstring = subString.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
  const caseInsensitiveRegex = new RegExp(escapedSubstring, "gi");
  const newStr = str.replace(caseInsensitiveRegex, match => "<strong>".concat(match, "</strong>"));
  return {
    text: subString === "" ? str : newStr,
    match: newStr !== str
  };
};

/**
 * Whether a navigation item is an anchor.
 * @param link - The navigation item.
 */
exports.highlightSubString = highlightSubString;
const isNavigationAnchor = link => !!link.url;

/**
 * Whether a navigation item is a button.
 * @param link - The navigation item.
 */
exports.isNavigationAnchor = isNavigationAnchor;
const isNavigationButton = link => !link.url;

/**
 * A typeguard for whether an element is a ReactNode.
 */
exports.isNavigationButton = isNavigationButton;
const isReactNode = element => /*#__PURE__*/(0, _react.isValidElement)(element);
exports.isReactNode = isReactNode;
const getElementAbsoluteHeight = element => {
  if (!element) {
    return 0;
  }
  const style = window.getComputedStyle(element);
  const margin = toFloat(style.marginTop) + toFloat(style.marginBottom);
  return element.offsetHeight + margin + 1;
};
exports.getElementAbsoluteHeight = getElementAbsoluteHeight;
const getAbsoluteHeightBelowById = belowId => {
  const element = belowId ? document.getElementById(belowId) : undefined;
  if (!element) {
    return 0;
  }
  return getElementAbsoluteHeight(element);
};
exports.getAbsoluteHeightBelowById = getAbsoluteHeightBelowById;
const getParentsBottomSpacing = element => {
  let sum = 0;
  while (element.parentElement) {
    element = element.parentElement;
    const style = window.getComputedStyle(element);
    const margin = toFloat(style.marginBottom);
    const padding = toFloat(style.paddingBottom);
    sum += margin + padding;
  }
  return sum;
};
exports.getParentsBottomSpacing = getParentsBottomSpacing;
const toFloat = value => {
  const result = parseFloat(value);
  return Number.isNaN(result) ? 0 : result;
};
exports.toFloat = toFloat;
