// @canonical/react-components@4.2.0/dist/esm/components/PrefixedIpInput/PrefixedIpInput.js
var _excluded = ["cidr", "help", "onIpChange", "ip", "name"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import React from "react";
import PrefixedInput from "../PrefixedInput";
import { getImmutableAndEditable, isIPv4 } from "./utils";
var PrefixedIpInput = _ref => {
  var {
      cidr,
      help,
      onIpChange,
      ip,
      name
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  var [networkAddress] = cidr.split("/");
  var isIPV4 = isIPv4(networkAddress);
  var [immutable, editable] = getImmutableAndEditable(cidr);
  var inputValue = isIPV4 ? ip.split(".").slice(immutable.split(".").length).join(".") : ip.replace(immutable, "");
  var getIPv4MaxLength = () => {
    var immutableOctetsLength = immutable.split(".").length;
    var lengths = [15, 11, 7, 3]; // Corresponding to 0-3 immutable octets
    return lengths[immutableOctetsLength];
  };
  var maxLength = isIPV4 ? getIPv4MaxLength() : editable.length;
  var placeholder = props.disabled ? "" : editable;
  var setIp = editableValue => {
    var fullIp = editableValue ? isIPV4 ? "".concat(immutable, ".").concat(editableValue) : "".concat(immutable).concat(editableValue) : "";
    onIpChange(fullIp);
  };
  var handlePaste = e => {
    e.preventDefault();
    var pastedText = e.clipboardData.getData("text");
    if (isIPV4) {
      var octets = pastedText.split(".");
      var trimmed = octets.slice(0 - editable.split(".").length);
      var _ip = trimmed.join(".");
      setIp(_ip);
    } else {
      var _ip2 = pastedText.replace(immutable, "");
      setIp(_ip2);
    }
  };
  return /*#__PURE__*/React.createElement(PrefixedInput, _extends({
    help: help ? help : /*#__PURE__*/React.createElement(React.Fragment, null, " ", isIPV4 ? /*#__PURE__*/React.createElement(React.Fragment, null, " ", "The available range in this subnet is", " ", /*#__PURE__*/React.createElement("code", null, immutable, ".", editable, " ")) : /*#__PURE__*/React.createElement(React.Fragment, null, " ", "The available IPV6 address range is", " ", /*#__PURE__*/React.createElement("code", null, immutable, editable, " ")), "."),
    immutableText: isIPV4 ? "".concat(immutable, ".") : immutable,
    maxLength: maxLength,
    name: name,
    onPaste: handlePaste,
    value: inputValue,
    onChange: e => {
      setIp(e.target.value);
    },
    placeholder: placeholder
  }, props));
};
export default PrefixedIpInput;
