// @canonical/react-components@4.2.0/dist/components/PasswordToggle/PasswordToggle.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Label = void 0;
var _classnames = _interopRequireDefault(require("classnames"));
var _react = _interopRequireWildcard(require("react"));
var _Button = _interopRequireDefault(require("../Button"));
var _Field = _interopRequireDefault(require("../Field"));
var _Label = _interopRequireDefault(require("../Label"));
require("./PasswordToggle.scss");
const _excluded = ["caution", "className", "error", "help", "id", "label", "readOnly", "required", "success", "type", "wrapperClassName"];
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
let Label = exports.Label = /*#__PURE__*/function (Label) {
  Label["Hide"] = "Hide";
  Label["Show"] = "Show";
  return Label;
}({});
/**
 * The props for the Password Toggle component.
 */
/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Password Toggle](https://docs.vanillaframework.io/base/forms/#password-toggle).
 *
 * It can be used when an input needs to obscure its value, whilst giving the user a way to reveal it if needed.
 */
const PasswordToggle = /*#__PURE__*/_react.default.forwardRef((_ref, ref) => {
  let {
      caution,
      className,
      error,
      help,
      id,
      label,
      readOnly,
      required,
      success,
      type,
      wrapperClassName
    } = _ref,
    inputProps = _objectWithoutProperties(_ref, _excluded);
  const [isPassword, setIsPassword] = (0, _react.useState)(true);
  const validationId = (0, _react.useId)();
  const helpId = (0, _react.useId)();
  const hasError = !!error;
  const defaultPasswordToggleId = (0, _react.useId)();
  const passwordToggleId = id || defaultPasswordToggleId;
  const togglePassword = () => {
    if (isPassword) {
      setIsPassword(false);
    } else {
      setIsPassword(true);
    }
  };
  return /*#__PURE__*/_react.default.createElement(_Field.default, {
    caution: caution,
    className: wrapperClassName,
    error: error,
    help: help,
    helpId: helpId,
    required: required,
    success: success,
    validationId: validationId
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "password-toggle"
  }, /*#__PURE__*/_react.default.createElement("input", _extends({
    "aria-describedby": [help ? helpId : null, success ? validationId : null].filter(Boolean).join(" "),
    "aria-errormessage": hasError ? validationId : null,
    "aria-invalid": hasError,
    className: (0, _classnames.default)("p-form-validation__input", className),
    id: passwordToggleId,
    readOnly: readOnly,
    ref: ref,
    type: isPassword ? "password" : "text"
  }, inputProps)), /*#__PURE__*/_react.default.createElement("div", {
    className: "p-form-password-toggle"
  }, /*#__PURE__*/_react.default.createElement(_Label.default, {
    forId: passwordToggleId,
    required: required
  }, label), /*#__PURE__*/_react.default.createElement(_Button.default, {
    appearance: "base",
    type: type ? type : "button",
    className: "u-no-margin--bottom",
    hasIcon: true,
    "aria-controls": id,
    "aria-live": "polite",
    onClick: () => togglePassword()
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "p-form-password-toggle__label"
  }, isPassword ? Label.Show : Label.Hide), /*#__PURE__*/_react.default.createElement("i", {
    className: isPassword ? "p-icon--show" : "p-icon--hide"
  })))));
});
var _default = exports.default = PasswordToggle;
