// @canonical/react-components@4.2.0/dist/components/PasswordToggle/PasswordToggle.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Success = exports.ReadOnly = exports.Help = exports.Error = exports.Disabled = exports.Default = exports.Caution = void 0;
var _react = _interopRequireDefault(require("react"));
var _PasswordToggle = _interopRequireDefault(require("./PasswordToggle"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _PasswordToggle.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: () => /*#__PURE__*/_react.default.createElement(_PasswordToggle.default, {
    id: "examplePassword1",
    label: "Password"
  }),
  name: "Default"
};
const ReadOnly = exports.ReadOnly = {
  render: () => /*#__PURE__*/_react.default.createElement(_PasswordToggle.default, {
    id: "examplePassword2",
    label: "Password",
    readOnly: true,
    defaultValue: "password"
  }),
  name: "Read only"
};
const Error = exports.Error = {
  render: () => /*#__PURE__*/_react.default.createElement(_PasswordToggle.default, {
    error: "This field is required.",
    id: "examplePassword3",
    label: "Password",
    defaultValue: "password"
  }),
  name: "Error"
};
const Success = exports.Success = {
  render: () => /*#__PURE__*/_react.default.createElement(_PasswordToggle.default, {
    success: "Strong password",
    id: "examplePassword4",
    label: "Password",
    defaultValue: "password"
  }),
  name: "Success"
};
const Caution = exports.Caution = {
  render: () => /*#__PURE__*/_react.default.createElement(_PasswordToggle.default, {
    caution: "Weak password",
    id: "examplePassword5",
    label: "Password",
    defaultValue: "password"
  }),
  name: "Caution"
};
const Help = exports.Help = {
  render: () => /*#__PURE__*/_react.default.createElement(_PasswordToggle.default, {
    help: "Enter a secure password",
    id: "examplePassword6",
    label: "Password"
  }),
  name: "Help"
};
const Disabled = exports.Disabled = {
  render: () => /*#__PURE__*/_react.default.createElement(_PasswordToggle.default, {
    disabled: true,
    id: "examplePassword7",
    label: "Password",
    defaultValue: "password"
  }),
  name: "Disabled"
};
