// @canonical/react-components@4.2.0/dist/components/ModularTable/ModularTable.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Subrows = exports.Subheading = exports.Sortable = exports.LoadMore = exports.Empty = exports.Default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _SummaryButton = _interopRequireDefault(require("../SummaryButton"));
var _ModularTable = _interopRequireDefault(require("./ModularTable"));
var _Icon = require("../Icon");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const getSampleData = function () {
  let i = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 0;
  return [{
    data: ++i * 100 + i
  }, {
    data: ++i * 100 + i
  }, {
    data: ++i * 100 + i
  }, {
    data: ++i * 100 + i
  }, {
    data: ++i * 100 + i
  }];
};
const meta = {
  component: _ModularTable.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: () => /*#__PURE__*/_react.default.createElement(_ModularTable.default, {
    getCellProps: _ref => {
      let {
        value
      } = _ref;
      return {
        className: value === "1 minute" ? "p-heading--5" : ""
      };
    }
    // eslint-disable-next-line react-hooks/rules-of-hooks
    ,
    columns: _react.default.useMemo(() => [{
      Header: "ID",
      accessor: "buildId",
      Cell: _ref2 => {
        let {
          value
        } = _ref2;
        return /*#__PURE__*/_react.default.createElement("a", {
          href: "#test"
        }, "#", value);
      }
    }, {
      Header: "Architecture",
      accessor: "arch"
    }, {
      Header: "Build Duration",
      accessor: "duration",
      className: "u-hide--small"
    }, {
      Header: "Result",
      accessor: "result",
      Cell: _ref3 => {
        let {
          value
        } = _ref3;
        switch (value) {
          case "released":
            return "Released";
          case "failed":
            return "Failed";
          default:
            return "Unknown";
        }
      },
      getCellIcon: _ref4 => {
        let {
          value
        } = _ref4;
        switch (value) {
          case "released":
            return _Icon.ICONS.success;
          case "failed":
            return _Icon.ICONS.error;
          default:
            return false;
        }
      }
    }, {
      Header: "Build Finished",
      accessor: "finished",
      className: "u-align-text--right"
    }], [])
    // eslint-disable-next-line react-hooks/rules-of-hooks
    ,
    data: _react.default.useMemo(() => [{
      buildId: "5432",
      arch: "arm64",
      duration: "5 minutes",
      result: "released",
      finished: "10 minutes ago"
    }, {
      buildId: "1234",
      arch: "armhf",
      duration: "5 minutes",
      result: "failed",
      finished: "over 1 year ago"
    }, {
      buildId: "1111",
      arch: "test64",
      duration: "1 minute",
      result: "other",
      finished: "ages ago"
    }], [])
  }),
  name: "Default"
};
const Empty = exports.Empty = {
  render: () => /*#__PURE__*/_react.default.createElement(_ModularTable.default, {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    columns: _react.default.useMemo(() => [{
      Header: "ID",
      accessor: "buildId"
    }, {
      Header: "Architecture",
      accessor: "arch"
    }, {
      Header: "Build Duration",
      accessor: "duration",
      className: "u-hide--small"
    }, {
      Header: "Result",
      accessor: "result"
    }, {
      Header: "Build Finished",
      accessor: "finished",
      className: "u-align-text--right"
    }], []),
    data: [],
    emptyMsg: "Waiting for data..."
  }),
  name: "Empty"
};
const Sortable = exports.Sortable = {
  render: () => /*#__PURE__*/_react.default.createElement(_ModularTable.default, {
    sortable: true,
    initialSortColumn: "duration",
    initialSortDirection: "ascending",
    getCellProps: _ref5 => {
      let {
        value
      } = _ref5;
      return {
        className: value === "1 minute" ? "p-heading--5" : ""
      };
    }
    // eslint-disable-next-line react-hooks/rules-of-hooks
    ,
    columns: _react.default.useMemo(() => [{
      Header: "ID",
      accessor: "buildId",
      sortType: "basic",
      Cell: _ref6 => {
        let {
          value
        } = _ref6;
        return /*#__PURE__*/_react.default.createElement("a", {
          href: "#test"
        }, "#", value);
      }
    }, {
      Header: "Architecture",
      accessor: "arch",
      sortType: "basic"
    }, {
      Header: "Build Duration",
      accessor: "duration",
      className: "u-hide--small",
      sortType: "basic"
    }, {
      Header: "Result",
      accessor: "result",
      sortType: "basic",
      Cell: _ref7 => {
        let {
          value
        } = _ref7;
        switch (value) {
          case "released":
            return "Released";
          case "failed":
            return "Failed";
          default:
            return "Unknown";
        }
      },
      getCellIcon: _ref8 => {
        let {
          value
        } = _ref8;
        switch (value) {
          case "released":
            return _Icon.ICONS.success;
          case "failed":
            return _Icon.ICONS.error;
          default:
            return false;
        }
      }
    }, {
      Header: "Build Finished",
      accessor: "finished",
      className: "u-align-text--right",
      sortType: "basic"
    }], [])
    // eslint-disable-next-line react-hooks/rules-of-hooks
    ,
    data: _react.default.useMemo(() => [{
      buildId: "5432",
      arch: "arm64",
      duration: "5 minutes",
      result: "released",
      finished: "10 minutes ago"
    }, {
      buildId: "1234",
      arch: "armhf",
      duration: "5 minutes",
      result: "failed",
      finished: "over 1 year ago"
    }, {
      buildId: "1111",
      arch: "test64",
      duration: "1 minute",
      result: "other",
      finished: "ages ago"
    }], [])
  }),
  name: "Sortable"
};

/**
 * Subrows can be used to group table rows under a heading that needs to appear
at the start of the group. In the following example sorting the columns will
sort the group rows, then sort the sub-rows, but the group header always comes first.
 */
const Subrows = exports.Subrows = {
  render: () => /*#__PURE__*/_react.default.createElement(_ModularTable.default, {
    sortable: true
    // eslint-disable-next-line react-hooks/rules-of-hooks
    ,
    columns: _react.default.useMemo(() => [{
      Header: "Flavour",
      accessor: "flavour",
      sortType: "basic",
      Cell: props => "depth" in props.row && props.row.depth === 0 ? /*#__PURE__*/_react.default.createElement("strong", null, props.value) : /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("input", {
        type: "checkbox"
      }), " ", props.value)
    }, {
      Header: "Scoops",
      accessor: "scoops",
      sortType: "basic",
      Cell: props => "depth" in props.row && props.row.depth === 0 ? /*#__PURE__*/_react.default.createElement("span", {
        className: "u-text--muted"
      }, props.value) : props.value
    }], [])
    // eslint-disable-next-line react-hooks/rules-of-hooks
    ,
    data: _react.default.useMemo(() => [{
      flavour: "Sorbet",
      scoops: "1-2",
      subRows: [{
        flavour: "Lemon",
        scoops: 2
      }, {
        flavour: "Mango",
        scoops: 1
      }, {
        flavour: "Raspberry",
        scoops: 2
      }]
    }, {
      flavour: "Ice-cream",
      scoops: "1-3",
      subRows: [{
        flavour: "Chocolate",
        scoops: 1
      }, {
        flavour: "Vanilla",
        scoops: 3
      }, {
        flavour: "Caramel",
        scoops: 2
      }]
    }], [])
  }),
  name: "Subrows"
};
const Subheading = exports.Subheading = {
  render: () => /*#__PURE__*/_react.default.createElement(_ModularTable.default, {
    getCellProps: _ref9 => {
      let {
        value
      } = _ref9;
      return {
        className: value === "1 minute" ? "p-heading--5" : ""
      };
    }
    // eslint-disable-next-line react-hooks/rules-of-hooks
    ,
    columns: _react.default.useMemo(() => [{
      Header: "ID",
      accessor: "buildId",
      Cell: _ref0 => {
        let {
          value
        } = _ref0;
        return /*#__PURE__*/_react.default.createElement("a", {
          href: "#test"
        }, "#", value);
      }
    }, {
      Header: "Architecture",
      accessor: "arch"
    }, {
      Header: "Build Duration",
      accessor: "duration",
      className: "u-hide--small"
    }, {
      Header: "Result",
      accessor: "result",
      Cell: _ref1 => {
        let {
          value
        } = _ref1;
        switch (value) {
          case "released":
            return "Released";
          case "failed":
            return "Failed";
          default:
            return "Unknown";
        }
      },
      getCellIcon: _ref10 => {
        let {
          value
        } = _ref10;
        switch (value) {
          case "released":
            return _Icon.ICONS.success;
          case "failed":
            return _Icon.ICONS.error;
          default:
            return false;
        }
      }
    }, {
      Header: "Build Finished",
      accessor: "finished",
      className: "u-align-text--right"
    }], [])
    // eslint-disable-next-line react-hooks/rules-of-hooks
    ,
    data: _react.default.useMemo(() => [{
      buildId: "5432",
      arch: "arm64",
      duration: "5 minutes",
      result: "released",
      finished: "10 minutes ago"
    }, {
      buildId: "1234",
      arch: "armhf",
      duration: "5 minutes",
      result: "failed",
      finished: "over 1 year ago"
    }, {
      buildId: "1111",
      arch: "test64",
      duration: "1 minute",
      result: "other",
      finished: "ages ago"
    }], []),
    subhead: /*#__PURE__*/_react.default.createElement("tr", null, /*#__PURE__*/_react.default.createElement("th", {
      colSpan: 5,
      className: "p-text--default",
      style: {
        textTransform: "none"
      }
    }, "Showing 3 of 3 results"))
  }),
  name: "Subheading"
};

/**
Example below shows a basic `ModularTable` with `SummaryButton` component in the footer.
```
export const getSampleData = (i = 0) => [
  { data: ++i * 100 + i },
  { data: ++i * 100 + i },
  { data: ++i * 100 + i },
  { data: ++i * 100 + i },
  { data: ++i * 100 + i },
];
```
 */
const LoadMore = exports.LoadMore = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [data, setData] = (0, _react.useState)(getSampleData());
    const hasMore = data.length < 20;
    const loadMore = () => {
      setData(data.concat(getSampleData(data.length)));
    };
    const columns = [{
      Header: "Id",
      accessor: (_d, i) => i
    }, {
      Header: "Data",
      accessor: "data"
    }];
    const footer = hasMore ? /*#__PURE__*/_react.default.createElement("div", {
      className: "u-align--right"
    }, /*#__PURE__*/_react.default.createElement(_SummaryButton.default, {
      summary: "Showing ".concat(data.length, " of 20 items."),
      label: "Show 5 more",
      onClick: loadMore
    })) : null;
    return /*#__PURE__*/_react.default.createElement(_ModularTable.default, {
      data: data,
      columns: columns,
      footer: footer
    });
  },
  name: "Load more"
};
