// @canonical/react-components@4.2.0/dist/components/Slider/Slider.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Slider = exports.FILLED_COLOR = exports.EMPTY_COLOR = void 0;
var _react = _interopRequireWildcard(require("react"));
var _Field = _interopRequireDefault(require("../Field"));
const _excluded = ["caution", "disabled", "error", "help", "id", "inputDisabled", "label", "max", "min", "onChange", "required", "showInput"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
const FILLED_COLOR = exports.FILLED_COLOR = "#0066CC";
const EMPTY_COLOR = exports.EMPTY_COLOR = "#D9D9D9";
/**
 * This is the [React](https://reactjs.org/) component for the Vanilla [Slider](https://vanillaframework.io/docs/patterns/slider).
 *
 * Sliders should be used to let a user specify a numeric value between a defined
minimum and maximum value, however the precise value is not especially important.
 */
const Slider = _ref => {
  var _navigator;
  let {
      caution,
      disabled = false,
      error,
      help,
      id,
      inputDisabled = false,
      label,
      max,
      min,
      onChange,
      required = false,
      showInput = false
    } = _ref,
    inputProps = _objectWithoutProperties(_ref, _excluded);
  const validationId = (0, _react.useId)();
  const helpId = (0, _react.useId)();
  const defaultSliderId = (0, _react.useId)();
  const sliderId = id || defaultSliderId;
  const hasError = !!error;
  let style = {};
  if ((_navigator = navigator) !== null && _navigator !== void 0 && (_navigator = _navigator.userAgent) !== null && _navigator !== void 0 && _navigator.includes("AppleWebKit")) {
    var _inputProps$value;
    // Range inputs on Webkit browsers don't have a built-in "filled" portion,
    // so instead it is handled here as a background.
    const val = (_inputProps$value = inputProps.value) !== null && _inputProps$value !== void 0 ? _inputProps$value : inputProps.defaultValue;
    const filledPercentage = "".concat((Number(val) - min) / (max - min) * 100, "%");
    style = {
      background: "linear-gradient(\n        to right,\n        ".concat(FILLED_COLOR, " 0%,\n        ").concat(FILLED_COLOR, " ").concat(filledPercentage, ",\n        ").concat(EMPTY_COLOR, " ").concat(filledPercentage, ",\n        ").concat(EMPTY_COLOR, " 100%\n      )")
    };
  }
  return /*#__PURE__*/_react.default.createElement(_Field.default, {
    caution: caution,
    error: error,
    help: help,
    forId: sliderId,
    helpId: helpId,
    label: label,
    required: required,
    validationId: validationId
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "p-slider__wrapper"
  }, /*#__PURE__*/_react.default.createElement("input", _extends({
    "aria-describedby": help ? helpId : null,
    "aria-errormessage": hasError ? validationId : null,
    "aria-invalid": hasError,
    disabled: disabled,
    id: sliderId,
    max: max,
    min: min,
    onChange: onChange,
    required: required,
    style: style,
    type: "range"
  }, inputProps)), showInput && /*#__PURE__*/_react.default.createElement("input", _extends({
    "aria-label": typeof label === "string" ? label : undefined,
    "aria-describedby": help ? helpId : null,
    "aria-errormessage": hasError ? validationId : null,
    "aria-invalid": hasError,
    className: "p-slider__input",
    disabled: disabled || inputDisabled,
    max: max,
    min: min,
    onChange: onChange,
    type: "number"
  }, inputProps))));
};
exports.Slider = Slider;
var _default = exports.default = Slider;
