// @canonical/react-components@4.2.0/dist/components/Slider/Slider.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithInput = exports.Default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _Slider = _interopRequireDefault(require("./Slider"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const Template = args => {
  const [selectedValue, setSelectedValue] = (0, _react.useState)(50);
  return /*#__PURE__*/_react.default.createElement(_Slider.default, _extends({}, args, {
    onChange: e => setSelectedValue(Number(e.target.value)),
    value: selectedValue
  }));
};
const meta = {
  component: _Slider.default,
  render: Template,
  tags: ["autodocs"],
  argTypes: {
    caution: {
      control: {
        type: "text"
      }
    },
    error: {
      control: {
        type: "text"
      }
    },
    help: {
      control: {
        type: "text"
      }
    },
    label: {
      control: {
        type: "text"
      }
    }
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: Template.bind({}),
  name: "Default",
  args: {
    disabled: false,
    label: "Volume",
    max: 100,
    min: 0,
    showInput: false
  }
};

/**
 * You can also optionally provide a numeric representation of the slider value.
 */
const WithInput = exports.WithInput = {
  render: Template.bind({}),
  name: "With input",
  args: {
    disabled: false,
    inputDisabled: false,
    label: "Volume",
    max: 100,
    min: 0,
    showInput: true
  }
};
