// @canonical/react-components@4.2.0/dist/components/Textarea/Textarea.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.DynamicHeight = exports.Default = void 0;
var _Textarea = _interopRequireDefault(require("./Textarea"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _Textarea.default,
  tags: ["autodocs"],
  argTypes: {
    caution: {
      control: {
        type: "text"
      }
    },
    error: {
      control: {
        type: "text"
      }
    },
    help: {
      control: {
        type: "text"
      }
    },
    label: {
      control: {
        type: "text"
      }
    },
    success: {
      control: {
        type: "text"
      }
    },
    id: {
      control: {
        disable: true
      }
    }
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default",
  args: {
    label: "Tell us about your project or interest",
    id: "textarea2",
    rows: 3,
    defaultValue: "Ubuntu"
  }
};
const DynamicHeight = exports.DynamicHeight = {
  name: "Dynamic Height",
  args: {
    label: "Text area that will dynamically adjust its height based on content",
    defaultValue: "Ubuntu",
    grow: true
  }
};
