// @canonical/react-components@4.2.0/dist/components/Link/Link.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Soft = exports.Inverted = exports.Default = exports.BackToTop = void 0;
var _react = _interopRequireDefault(require("react"));
var _Link = _interopRequireDefault(require("./Link"));
var _Strip = _interopRequireDefault(require("../Strip"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _Link.default,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: {
        type: "text"
      }
    }
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default",
  args: {
    children: "Fast, secure and simple, Ubuntu powers millions of PCs worldwide"
  }
};
const Soft = exports.Soft = {
  render: () => /*#__PURE__*/_react.default.createElement(_Link.default, {
    soft: true
  }, "Learn about MAAS"),
  name: "Soft"
};
const Inverted = exports.Inverted = {
  render: () => /*#__PURE__*/_react.default.createElement(_Strip.default, {
    type: "dark"
  }, /*#__PURE__*/_react.default.createElement(_Link.default, {
    inverted: true
  }, "Package & publish your app")),
  name: "Inverted"
};
const BackToTop = exports.BackToTop = {
  render: () => /*#__PURE__*/_react.default.createElement(_Link.default, {
    top: true
  }, "Back to top"),
  name: "Back to top"
};
