// @canonical/react-components@4.2.0/dist/esm/components/ApplicationLayout/ApplicationLayout.stories.js
/* eslint-disable react-hooks/rules-of-hooks */
import React, { useState } from "react";
import Form from "../Form";
import Input from "../Input";
import Button from "../Button";
import Row from "../Row";
import Col from "../Col";
import ApplicationLayout from "./ApplicationLayout";
import AppAside from "./AppAside";
import Panel from "../Panel";
import Icon from "../Icon";
import SideNavigation from "../SideNavigation";
var meta = {
  component: ApplicationLayout,
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  render: () => {
    var [showAside, setShowAside] = useState(true);
    var [asidePinned, setAsidePinned] = useState(false);
    return /*#__PURE__*/React.createElement(ApplicationLayout, {
      logo: {
        icon: "https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg",
        name: "https://assets.ubuntu.com/v1/2e04d794-logo-jaas.svg",
        nameAlt: "JAAS",
        href: "/"
      },
      navItems: [{
        items: [{
          icon: "drag",
          label: "Models",
          href: "/models"
        }, {
          icon: "menu",
          label: "Controllers",
          href: "/controllers"
        }, {
          icon: "user",
          label: "Permissions",
          href: "/users"
        }]
      }],
      aside: showAside ? /*#__PURE__*/React.createElement(AppAside, {
        pinned: asidePinned
      }, /*#__PURE__*/React.createElement(Panel, {
        title: "Aside panel",
        controls: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Button, {
          onClick: () => setAsidePinned(!asidePinned),
          dense: true,
          className: "u-no-margin"
        }, "Pin aside"), /*#__PURE__*/React.createElement(Button, {
          appearance: "base",
          className: "u-no-margin--bottom",
          hasIcon: true,
          onClick: () => {
            setShowAside(false);
            setAsidePinned(false);
          }
        }, /*#__PURE__*/React.createElement(Icon, {
          name: "close"
        }, "Close")))
      }, /*#__PURE__*/React.createElement(Form, {
        stacked: true
      }, /*#__PURE__*/React.createElement(Input, {
        label: "Full name",
        type: "text",
        name: "fullName",
        autoComplete: "name",
        stacked: true
      }), /*#__PURE__*/React.createElement(Input, {
        label: "Username",
        type: "text",
        name: "username-stacked",
        autoComplete: "username",
        "aria-describedby": "exampleHelpTextMessage",
        stacked: true,
        help: "30 characters or fewer."
      }), /*#__PURE__*/React.createElement(Input, {
        type: "text",
        label: "Email address",
        "aria-invalid": "true",
        name: "username-stackederror",
        autoComplete: "email",
        required: true,
        error: "This field is required.",
        stacked: true
      }), /*#__PURE__*/React.createElement(Input, {
        label: "Address line 1",
        type: "text",
        name: "address-optional-stacked",
        autoComplete: "address-line1",
        stacked: true
      }), /*#__PURE__*/React.createElement(Input, {
        label: "Address line 2",
        type: "text",
        name: "address-optional-stacked",
        autoComplete: "address-line3",
        stacked: true
      }), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
        size: 12
      }, /*#__PURE__*/React.createElement(Button, {
        appearance: "positive",
        className: "u-float-right",
        name: "add-details"
      }, "Add details")))))) : null,
      status: /*#__PURE__*/React.createElement(Button, {
        onClick: () => setShowAside(!showAside),
        dense: true,
        appearance: "base",
        className: "u-no-margin"
      }, "Toggle aside")
    }, /*#__PURE__*/React.createElement(Panel, {
      title: "Application Layout"
    }, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 12
    }, "Content"))));
  }
};

/**
 * The pinned and collapsed states can be handled externally if you need access
 * to these outside the application layout.
 */
export var ExternalStates = {
  render: () => {
    var [menuPinned, setMenuPinned] = useState(false);
    var [menuCollapsed, setMenuCollapsed] = useState(true);
    return /*#__PURE__*/React.createElement(ApplicationLayout, {
      menuPinned: menuPinned,
      onPinMenu: setMenuPinned,
      menuCollapsed: menuCollapsed,
      onCollapseMenu: setMenuCollapsed,
      logo: {
        icon: "https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg",
        name: "https://assets.ubuntu.com/v1/2e04d794-logo-jaas.svg",
        nameAlt: "JAAS",
        href: "/"
      },
      navItems: [{
        items: [{
          icon: "drag",
          label: "Models",
          href: "/models"
        }, {
          icon: "menu",
          label: "Controllers",
          href: "/controllers"
        }, {
          icon: "user",
          label: "Permissions",
          href: "/users"
        }]
      }]
    }, /*#__PURE__*/React.createElement(Panel, {
      title: "Application Layout"
    }, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 12
    }, "Content"))));
  }
};

/**
 * Menu items can either be provided as an array given to `navItems` or when
 * more control is needed then by proving `sideNavigation`.
 *
 * For more details about how to provide navigation items see the
 * [`SideNavigation`](/docs/components-sidenavigation--docs) docs.
 */
export var Navigation = {
  render: () => {
    var [menuPinned, setMenuPinned] = useState(false);
    var [menuCollapsed, setMenuCollapsed] = useState(true);
    return /*#__PURE__*/React.createElement(ApplicationLayout, {
      menuPinned: menuPinned,
      onPinMenu: setMenuPinned,
      menuCollapsed: menuCollapsed,
      onCollapseMenu: setMenuCollapsed,
      logo: {
        icon: "https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg",
        name: "https://assets.ubuntu.com/v1/2e04d794-logo-jaas.svg",
        nameAlt: "JAAS",
        href: "/"
      },
      sideNavigation: /*#__PURE__*/React.createElement(SideNavigation, {
        items: [{
          items: [{
            icon: "drag",
            label: "Models",
            href: "/models"
          }, {
            icon: "menu",
            label: "Controllers",
            href: "/controllers"
          }, {
            icon: "user",
            label: "Permissions",
            href: "/users"
          }]
        }]
      })
    }, /*#__PURE__*/React.createElement(Panel, {
      title: "Application Layout"
    }, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 12
    }, "Content"))));
  }
};

/**
 * App navigation can also be completely opted out of, for some
 * particular scenarios (e.g. login page, error pages).
 */
export var NoNavigation = {
  render: () => {
    return /*#__PURE__*/React.createElement(ApplicationLayout, null, /*#__PURE__*/React.createElement(Panel, {
      title: "Application Layout with no navigation"
    }, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 12
    }, "Content"))));
  }
};
