// @canonical/react-components@4.2.0/dist/components/Input/Input.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _classnames = _interopRequireDefault(require("classnames"));
var _react = _interopRequireWildcard(require("react"));
var _Field = _interopRequireDefault(require("../Field"));
var _CheckboxInput = _interopRequireDefault(require("../CheckboxInput"));
var _RadioInput = _interopRequireDefault(require("../RadioInput"));
const _excluded = ["caution", "className", "error", "help", "helpClassName", "id", "label", "labelClassName", "required", "stacked", "success", "takeFocus", "takeFocusDelay", "type", "wrapperClassName"];
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
/**
 * The props for the Input component.
 */

/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Input](https://docs.vanillaframework.io/base/forms/#input).
 *
 * An input field where the user can enter data, which can vary in many ways, depending on the type attribute.
 */
const Input = _ref => {
  let {
      caution,
      className,
      error,
      help,
      helpClassName,
      id,
      label,
      labelClassName,
      required,
      stacked,
      success,
      takeFocus,
      takeFocusDelay,
      type,
      wrapperClassName
    } = _ref,
    inputProps = _objectWithoutProperties(_ref, _excluded);
  const inputRef = (0, _react.useRef)(null);
  const fieldLabel = !["checkbox", "radio"].includes(type) ? label : "";
  const defaultInputId = (0, _react.useId)();
  const inputId = id || defaultInputId;
  const validationId = (0, _react.useId)();
  const helpId = (0, _react.useId)();
  const hasError = !!error;
  const description = [help ? helpId : null, success ? validationId : null].filter(Boolean).join(" ");
  const commonProps = _objectSpread({
    "aria-describedby": !description ? undefined : description,
    "aria-errormessage": hasError ? validationId : null,
    "aria-invalid": hasError,
    id: inputId,
    required: required
  }, inputProps);
  (0, _react.useEffect)(() => {
    if (takeFocus) {
      if (takeFocusDelay) {
        setTimeout(() => inputRef.current.focus(), takeFocusDelay);
      } else {
        inputRef.current.focus();
      }
    }
  }, [takeFocus, takeFocusDelay]);
  let input;
  if (type === "checkbox") {
    input = /*#__PURE__*/_react.default.createElement(_CheckboxInput.default, _extends({
      label: label,
      labelClassName: labelClassName
    }, commonProps));
  } else if (type === "radio") {
    input = /*#__PURE__*/_react.default.createElement(_RadioInput.default, _extends({
      label: label,
      labelClassName: labelClassName
    }, commonProps));
  } else {
    input = /*#__PURE__*/_react.default.createElement("input", _extends({
      className: (0, _classnames.default)("p-form-validation__input", className),
      ref: inputRef,
      type: type
    }, commonProps));
  }
  return /*#__PURE__*/_react.default.createElement(_Field.default, {
    caution: caution,
    className: wrapperClassName,
    error: error,
    forId: inputId,
    help: help,
    helpClassName: helpClassName,
    helpId: helpId,
    isTickElement: type === "checkbox" || type === "radio",
    label: fieldLabel,
    labelClassName: labelClassName,
    required: required,
    stacked: stacked,
    success: success,
    validationId: validationId
  }, input);
};
var _default = exports.default = Input;
