// @canonical/react-components@4.2.0/dist/components/Input/Input.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.TextInput = exports.Success = exports.Stacked = exports.Required = exports.RadioButton = exports.Error = exports.Disabled = exports.Checkbox = exports.Caution = void 0;
var _react = _interopRequireDefault(require("react"));
var _Input = _interopRequireDefault(require("./Input"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _Input.default,
  tags: ["autodocs"],
  argTypes: {
    caution: {
      control: {
        type: "text"
      }
    },
    error: {
      control: {
        type: "text"
      }
    },
    help: {
      control: {
        type: "text"
      }
    },
    helpClassName: {
      control: {
        type: "text"
      }
    },
    label: {
      control: {
        type: "text"
      }
    },
    success: {
      control: {
        type: "text"
      }
    },
    id: {
      control: {
        disable: true
      }
    },
    placeholder: {
      control: {
        type: "text"
      }
    }
  }
};
var _default = exports.default = meta;
const TextInput = exports.TextInput = {
  name: "Text input",
  args: {
    type: "text",
    id: "exampleTextInput2",
    placeholder: "example@canonical.com",
    label: "Email address",
    help: "Additional description for the field",
    helpClassName: "u-no-margin--bottom"
  }
};
const Stacked = exports.Stacked = {
  render: () => /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "text",
    id: "exampleTextInput3",
    placeholder: "example@canonical.com",
    stacked: true,
    label: "Email address"
  }),
  name: "Stacked"
};
const Disabled = exports.Disabled = {
  render: () => /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "text",
    id: "exampleTextInput4",
    placeholder: "example@canonical.com",
    disabled: true,
    label: "Email address"
  }),
  name: "Disabled"
};
const Error = exports.Error = {
  render: () => /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "text",
    id: "exampleTextInput5",
    placeholder: "example@canonical.com",
    error: "This field is required.",
    label: "Email address"
  }),
  name: "Error"
};
const Success = exports.Success = {
  render: () => /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "text",
    id: "exampleTextInput6",
    placeholder: "example@canonical.com",
    success: "Verified.",
    label: "Email address"
  }),
  name: "Success"
};
const Caution = exports.Caution = {
  render: () => /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "text",
    id: "exampleTextInput7",
    placeholder: "example@canonical.com",
    caution: "No validation is performed in preview mode.",
    label: "Email address"
  }),
  name: "Caution"
};
const Required = exports.Required = {
  render: () => /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "text",
    id: "exampleTextInput8",
    placeholder: "example@canonical.com",
    required: true,
    label: "Email address"
  }),
  name: "Required"
};
const Checkbox = exports.Checkbox = {
  render: () => /*#__PURE__*/_react.default.createElement(_Input.default, {
    type: "checkbox",
    id: "checkExample12",
    defaultChecked: true,
    label: "HTML"
  }),
  name: "Checkbox"
};
const RadioButton = exports.RadioButton = {
  render: () => /*#__PURE__*/_react.default.createElement(_Input.default, {
    label: "Linux",
    type: "radio",
    name: "RadioOptions",
    id: "Radio12",
    defaultValue: "option1",
    defaultChecked: true,
    help: "Ubuntu"
  }),
  name: "Radio button"
};
