// @canonical/react-components@4.2.0/dist/esm/components/Pagination/PaginationButton/PaginationButton.js
import classNames from "classnames";
import React from "react";
export var Label = /*#__PURE__*/function (Label) {
  Label["Next"] = "Next page";
  Label["Previous"] = "Previous page";
  return Label;
}({});
var PaginationButton = _ref => {
  var {
    direction,
    onClick,
    disabled = false,
    showLabel,
    label
  } = _ref;
  var buttonLabel = label || (direction === "back" ? Label.Previous : Label.Next);
  return /*#__PURE__*/React.createElement("li", {
    className: "p-pagination__item"
  }, /*#__PURE__*/React.createElement("button", {
    className: classNames({
      "p-pagination__link--previous": direction === "back",
      "p-pagination__link--next": direction === "forward"
    }),
    disabled: disabled,
    onClick: onClick,
    type: "button"
  }, direction === "forward" && showLabel && /*#__PURE__*/React.createElement("span", null, buttonLabel), /*#__PURE__*/React.createElement("i", {
    className: "p-icon--chevron-down"
  }, buttonLabel), direction === "back" && showLabel && /*#__PURE__*/React.createElement("span", null, buttonLabel)));
};
export default PaginationButton;
