// @canonical/react-components@4.2.0/dist/components/Select/Select.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.SelectMultiple = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _Select = _interopRequireDefault(require("./Select"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _Select.default,
  tags: ["autodocs"],
  argTypes: {
    caution: {
      control: {
        type: "text"
      }
    },
    error: {
      control: {
        type: "text"
      }
    },
    help: {
      control: {
        type: "text"
      }
    },
    label: {
      control: {
        type: "text"
      }
    },
    success: {
      control: {
        type: "text"
      }
    },
    id: {
      control: {
        disable: true
      }
    },
    options: {
      control: {
        disable: true
      }
    }
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Select",
  args: {
    name: "exampleSelect",
    id: "exampleSelect2",
    defaultValue: "",
    options: [{
      value: "",
      disabled: true,
      label: "Select an option"
    }, {
      value: "1",
      label: "Cosmic Cuttlefish"
    }, {
      value: "2",
      label: "Bionic Beaver"
    }, {
      value: "3",
      label: "Xenial Xerus"
    }],
    label: "Ubuntu releases"
  }
};
const SelectMultiple = exports.SelectMultiple = {
  render: () => /*#__PURE__*/_react.default.createElement(_Select.default, {
    name: "exampleSelectMulti",
    id: "exampleSelectMulti2",
    options: [{
      value: "",
      disabled: true,
      label: "Select..."
    }, {
      value: "1",
      label: "Cosmic Cuttlefish"
    }, {
      value: "2",
      label: "Bionic Beaver"
    }, {
      value: "3",
      label: "Xenial Xerus"
    }],
    label: "Ubuntu releases",
    multiple: true
  }),
  name: "Select multiple"
};
