// @canonical/react-components@4.2.0/dist/components/Select/Select.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _classnames = _interopRequireDefault(require("classnames"));
var _react = _interopRequireWildcard(require("react"));
var _Field = _interopRequireDefault(require("../Field"));
const _excluded = ["label", "value"],
  _excluded2 = ["caution", "className", "error", "help", "helpClassName", "id", "label", "labelClassName", "onChange", "options", "required", "stacked", "success", "takeFocus", "wrapperClassName"];
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
/**
 * The props for the Select component.
 */

const generateOptions = options => options === null || options === void 0 ? void 0 : options.map(_ref => {
  let {
      label,
      value
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/_react.default.createElement("option", _extends({
    value: value,
    key: "".concat(value) || label
  }, props), label);
});

/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Select](https://docs.vanillaframework.io/base/forms/#select).
 *
 * Use the Select component to create a drop-down list.
 */
const Select = _ref2 => {
  let {
      caution,
      className,
      error,
      help,
      helpClassName,
      id,
      label,
      labelClassName,
      onChange,
      options,
      required,
      stacked,
      success,
      takeFocus,
      wrapperClassName
    } = _ref2,
    selectProps = _objectWithoutProperties(_ref2, _excluded2);
  const selectRef = (0, _react.useRef)(null);
  const validationId = (0, _react.useId)();
  const defaultSelectId = (0, _react.useId)();
  const selectId = id || defaultSelectId;
  const helpId = (0, _react.useId)();
  const hasError = !!error;
  (0, _react.useEffect)(() => {
    if (takeFocus) {
      selectRef.current.focus();
    }
  }, [takeFocus]);
  return /*#__PURE__*/_react.default.createElement(_Field.default, {
    caution: caution,
    className: wrapperClassName,
    error: error,
    forId: selectId,
    help: help,
    helpClassName: helpClassName,
    helpId: helpId,
    isSelect: true,
    label: label,
    labelClassName: labelClassName,
    required: required,
    stacked: stacked,
    success: success,
    validationId: validationId
  }, /*#__PURE__*/_react.default.createElement("select", _extends({
    "aria-describedby": [help ? helpId : null, success ? validationId : null].filter(Boolean).join(" "),
    "aria-errormessage": hasError ? validationId : null,
    "aria-invalid": hasError,
    className: (0, _classnames.default)("p-form-validation__input", className),
    id: selectId,
    onChange: evt => onChange && onChange(evt),
    ref: selectRef,
    required: required
  }, selectProps), generateOptions(options)));
};
var _default = exports.default = Select;
