// @canonical/react-components@4.2.0/dist/esm/components/ContextualMenu/ContextualMenuDropdown/ContextualMenuDropdown.js
var _excluded = ["children", "className", "onClick"],
  _excluded2 = ["adjustedPosition", "autoAdjust", "handleClose", "constrainPanelWidth", "dropdownClassName", "dropdownContent", "id", "isOpen", "links", "position", "positionCoords", "positionNode", "scrollOverflow", "setAdjustedPosition", "contextualMenuClassName"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import classNames from "classnames";
import React, { useCallback, useEffect, useRef, useState } from "react";
import { useWindowFitment } from "../../../hooks";
import Button from "../../Button";
export var Label = /*#__PURE__*/function (Label) {
  Label["Dropdown"] = "sub";
  return Label;
}({});

/**
 * The type of the menu links.
 * @template L - The type of the link props.
 */

/**
 * The props for the ContextualMenuDropdown component.
 * @template L - The type of the link props.
 */

/**
 * Calculate the styles for the menu.
 * @param position - The menu position.
 * @param positionCoords - The coordinates of the position node.
 * @param constrainPanelWidth - Whether the menu width should be constrained to the position width.
 */
var getPositionStyle = (position, verticalPosition, positionCoords, constrainPanelWidth) => {
  if (!positionCoords) {
    return null;
  }
  var {
    height,
    left,
    top,
    width
  } = positionCoords;
  var topPos = verticalPosition === "bottom" ? top + height + (window.scrollY || 0) : top + (window.scrollY || 0);
  var leftPos = left;
  switch (position) {
    case "left":
      leftPos = left;
      break;
    case "center":
      leftPos = left + width / 2;
      break;
    case "right":
      leftPos = left + width;
      break;
    default:
      break;
  }
  return _objectSpread({
    position: "absolute",
    left: leftPos,
    top: topPos
  }, constrainPanelWidth ? {
    width
  } : null);
};

/**
 * Calculate the adjusted position in relation to the window.
 * @param position - The requested position.
 * @param fitsWindow - The window fitment info.
 * @return The new position.
 */
export var adjustForWindow = (position, fitsWindow) => {
  var newPosition = position;
  if (!fitsWindow.fromRight.fitsLeft && newPosition === "right") {
    newPosition = "left";
  }
  if (!fitsWindow.fromLeft.fitsRight && newPosition === "left") {
    newPosition = "right";
  }
  // If the menu doesn't fit to the left or the right then center it.
  if (!fitsWindow.fromLeft.fitsRight && !fitsWindow.fromRight.fitsLeft && (newPosition === "left" || newPosition === "right")) {
    newPosition = "center";
  }
  // If the menu doesn't fit when centered then find a new position.
  if (newPosition === "center" && (!fitsWindow.fromCenter.fitsCentered.fitsRight || !fitsWindow.fromCenter.fitsCentered.fitsLeft)) {
    if (fitsWindow.fromLeft.fitsRight) {
      newPosition = "left";
    }
    if (fitsWindow.fromRight.fitsLeft) {
      newPosition = "right";
    }
  }
  return newPosition;
};

/**
 * Generate a menu link
 * @template L - The type of the link props.
 * @param link - A button
 * @param key - A key for the DOM.
 * @param handleClose - The function to close the menu.
 */
var generateLink = (link, key, handleClose) => {
  var {
      children,
      className,
      onClick
    } = link,
    props = _objectWithoutProperties(link, _excluded);
  return /*#__PURE__*/React.createElement(Button, _extends({
    className: classNames("p-contextual-menu__link", className),
    key: key,
    role: "menuitem",
    onClick: onClick ? evt => {
      handleClose(evt.nativeEvent);
      onClick(evt);
    } : null
  }, props), children);
};
var getClosestScrollableParent = node => {
  var currentNode = node;
  while (currentNode && currentNode !== document.body) {
    var {
      overflowY,
      overflowX
    } = window.getComputedStyle(currentNode);
    if (["auto", "scroll", "overlay"].includes(overflowY) && ["auto", "scroll", "overlay"].includes(overflowX)) {
      return currentNode;
    }
    currentNode = currentNode.parentElement;
  }
  return document.body;
};

// nearest parents z-index that is not 0 or auto
export var getNearestParentsZIndex = element => {
  if (!window || !element) {
    return "0";
  }
  var zIndex = window.getComputedStyle(element, null).getPropertyValue("z-index");
  if (!element.parentElement) {
    return zIndex;
  }
  if (zIndex === "auto" || zIndex === "0" || zIndex === "") {
    return getNearestParentsZIndex(element.parentElement);
  }
  return zIndex;
};
var ContextualMenuDropdown = _ref => {
  var {
      adjustedPosition,
      autoAdjust,
      handleClose,
      constrainPanelWidth,
      dropdownClassName,
      dropdownContent,
      id,
      isOpen,
      links,
      position,
      positionCoords,
      positionNode,
      scrollOverflow,
      setAdjustedPosition,
      contextualMenuClassName
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded2);
  var dropdown = useRef(null);
  var [verticalPosition, setVerticalPosition] = useState("bottom");
  var [positionStyle, setPositionStyle] = useState(getPositionStyle(adjustedPosition, verticalPosition, positionCoords, constrainPanelWidth));
  var [maxHeight, setMaxHeight] = useState();
  // Update the styles to position the menu.
  var updatePositionStyle = useCallback(() => {
    setPositionStyle(getPositionStyle(adjustedPosition, verticalPosition, positionCoords, constrainPanelWidth));
  }, [adjustedPosition, positionCoords, verticalPosition, constrainPanelWidth]);
  var updateVerticalPosition = useCallback(() => {
    var _dropdown$current$get;
    if (!positionNode) {
      return;
    }
    var scrollableParent = getClosestScrollableParent(positionNode);
    if (!scrollableParent) {
      return;
    }
    var scrollableParentRect = scrollableParent.getBoundingClientRect();
    var toggleRect = positionNode.getBoundingClientRect();

    // Calculate the rect in relation to the scrollableParent
    var relativeToScrollParentRect = {
      top: toggleRect.top - scrollableParentRect.top,
      bottom: toggleRect.bottom - scrollableParentRect.top
    };
    var scrollParentSpaceBelow = scrollableParentRect.height - relativeToScrollParentRect.bottom;
    var scrollParentSpaceAbove = relativeToScrollParentRect.top;
    var dropdownHeight = (_dropdown$current$get = dropdown.current.getBoundingClientRect().height) !== null && _dropdown$current$get !== void 0 ? _dropdown$current$get : 0;
    var windowSpaceBelow = window.innerHeight - toggleRect.bottom;
    setVerticalPosition(scrollParentSpaceBelow >= dropdownHeight && windowSpaceBelow >= dropdownHeight || windowSpaceBelow > scrollParentSpaceAbove ? "bottom" : "top");
  }, [positionNode]);

  // Update the position when the window fitment info changes.
  var onUpdateWindowFitment = useCallback(fitsWindow => {
    if (autoAdjust) {
      setAdjustedPosition(adjustForWindow(position, fitsWindow));
      updateVerticalPosition();
    }
    if (scrollOverflow) {
      setMaxHeight(fitsWindow.fromBottom.spaceBelow - 16);
    }
  }, [autoAdjust, position, scrollOverflow, setAdjustedPosition, updateVerticalPosition]);

  // Handle adjusting the horizontal position and scrolling of the dropdown so that it remains on screen.
  useWindowFitment(dropdown.current, positionNode, onUpdateWindowFitment, 0, isOpen && (autoAdjust || scrollOverflow));

  // Update the styles when the position changes.
  useEffect(() => {
    updatePositionStyle();
  }, [adjustedPosition, updatePositionStyle]);
  useEffect(() => {
    updateVerticalPosition();
  }, [updateVerticalPosition]);
  useEffect(() => {
    if (!dropdown.current) return;

    // align z-index: when we are in a modal context, we want the dropdown to be above the modal
    // apply the nearest parents z-index + 1
    var zIndex = getNearestParentsZIndex(positionNode);
    if (parseInt(zIndex) > 0) {
      var _dropdown$current$par;
      (_dropdown$current$par = dropdown.current.parentElement) === null || _dropdown$current$par === void 0 || _dropdown$current$par.style.setProperty("z-index", zIndex + 1);
    }
  }, [positionNode]);
  return (
    /*#__PURE__*/
    // Vanilla Framework uses .p-contextual-menu parent modifier classnames to determine the correct position of the .p-contextual-menu__dropdown dropdown (left, center, right).
    // Extra span wrapper is required as the dropdown is rendered in a portal.
    React.createElement("span", {
      className: contextualMenuClassName,
      style: positionStyle
    }, /*#__PURE__*/React.createElement("span", _extends({
      className: classNames("p-contextual-menu__dropdown", dropdownClassName),
      role: "menu",
      id: id,
      "aria-hidden": isOpen ? "false" : "true",
      "aria-label": Label.Dropdown,
      ref: dropdown,
      style: _objectSpread(_objectSpread(_objectSpread({}, constrainPanelWidth && positionStyle !== null && positionStyle !== void 0 && positionStyle.width ? {
        width: positionStyle.width,
        minWidth: 0,
        maxWidth: "none"
      } : {}), scrollOverflow ? {
        maxHeight,
        minHeight: "2rem",
        overflowX: "auto"
      } : {}), verticalPosition === "top" ? {
        bottom: "0"
      } : {})
    }, props), dropdownContent ? typeof dropdownContent === "function" ? dropdownContent(handleClose) : dropdownContent : links.map((item, i) => {
      if (Array.isArray(item)) {
        return /*#__PURE__*/React.createElement("span", {
          className: "p-contextual-menu__group",
          key: i
        }, item.map((link, j) => generateLink(link, j, handleClose)));
      } else if (typeof item === "string") {
        return /*#__PURE__*/React.createElement("div", {
          className: "p-contextual-menu__non-interactive",
          key: i
        }, item);
      }
      return generateLink(item, i, handleClose);
    })))
  );
};
export default ContextualMenuDropdown;
