// @canonical/react-components@4.2.0/dist/esm/components/PasswordToggle/PasswordToggle.js
var _excluded = ["caution", "className", "error", "help", "id", "label", "readOnly", "required", "success", "type", "wrapperClassName"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import classNames from "classnames";
import React, { useId, useState } from "react";
import Button from "../Button";
import Field from "../Field";
import VanillaLabel from "../Label";
import "./PasswordToggle.scss";
export var Label = /*#__PURE__*/function (Label) {
  Label["Hide"] = "Hide";
  Label["Show"] = "Show";
  return Label;
}({});

/**
 * The props for the Password Toggle component.
 */

/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Password Toggle](https://docs.vanillaframework.io/base/forms/#password-toggle).
 *
 * It can be used when an input needs to obscure its value, whilst giving the user a way to reveal it if needed.
 */
var PasswordToggle = /*#__PURE__*/React.forwardRef((_ref, ref) => {
  var {
      caution,
      className,
      error,
      help,
      id,
      label,
      readOnly,
      required,
      success,
      type,
      wrapperClassName
    } = _ref,
    inputProps = _objectWithoutProperties(_ref, _excluded);
  var [isPassword, setIsPassword] = useState(true);
  var validationId = useId();
  var helpId = useId();
  var hasError = !!error;
  var defaultPasswordToggleId = useId();
  var passwordToggleId = id || defaultPasswordToggleId;
  var togglePassword = () => {
    if (isPassword) {
      setIsPassword(false);
    } else {
      setIsPassword(true);
    }
  };
  return /*#__PURE__*/React.createElement(Field, {
    caution: caution,
    className: wrapperClassName,
    error: error,
    help: help,
    helpId: helpId,
    required: required,
    success: success,
    validationId: validationId
  }, /*#__PURE__*/React.createElement("div", {
    className: "password-toggle"
  }, /*#__PURE__*/React.createElement("input", _extends({
    "aria-describedby": [help ? helpId : null, success ? validationId : null].filter(Boolean).join(" "),
    "aria-errormessage": hasError ? validationId : null,
    "aria-invalid": hasError,
    className: classNames("p-form-validation__input", className),
    id: passwordToggleId,
    readOnly: readOnly,
    ref: ref,
    type: isPassword ? "password" : "text"
  }, inputProps)), /*#__PURE__*/React.createElement("div", {
    className: "p-form-password-toggle"
  }, /*#__PURE__*/React.createElement(VanillaLabel, {
    forId: passwordToggleId,
    required: required
  }, label), /*#__PURE__*/React.createElement(Button, {
    appearance: "base",
    type: type ? type : "button",
    className: "u-no-margin--bottom",
    hasIcon: true,
    "aria-controls": id,
    "aria-live": "polite",
    onClick: () => togglePassword()
  }, /*#__PURE__*/React.createElement("span", {
    className: "p-form-password-toggle__label"
  }, isPassword ? Label.Show : Label.Hide), /*#__PURE__*/React.createElement("i", {
    className: isPassword ? "p-icon--show" : "p-icon--hide"
  })))));
});
export default PasswordToggle;
