// @canonical/react-components@4.2.0/dist/esm/components/PasswordToggle/PasswordToggle.stories.js
import React from "react";
import PasswordToggle from "./PasswordToggle";
var meta = {
  component: PasswordToggle,
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  render: () => /*#__PURE__*/React.createElement(PasswordToggle, {
    id: "examplePassword1",
    label: "Password"
  }),
  name: "Default"
};
export var ReadOnly = {
  render: () => /*#__PURE__*/React.createElement(PasswordToggle, {
    id: "examplePassword2",
    label: "Password",
    readOnly: true,
    defaultValue: "password"
  }),
  name: "Read only"
};
export var Error = {
  render: () => /*#__PURE__*/React.createElement(PasswordToggle, {
    error: "This field is required.",
    id: "examplePassword3",
    label: "Password",
    defaultValue: "password"
  }),
  name: "Error"
};
export var Success = {
  render: () => /*#__PURE__*/React.createElement(PasswordToggle, {
    success: "Strong password",
    id: "examplePassword4",
    label: "Password",
    defaultValue: "password"
  }),
  name: "Success"
};
export var Caution = {
  render: () => /*#__PURE__*/React.createElement(PasswordToggle, {
    caution: "Weak password",
    id: "examplePassword5",
    label: "Password",
    defaultValue: "password"
  }),
  name: "Caution"
};
export var Help = {
  render: () => /*#__PURE__*/React.createElement(PasswordToggle, {
    help: "Enter a secure password",
    id: "examplePassword6",
    label: "Password"
  }),
  name: "Help"
};
export var Disabled = {
  render: () => /*#__PURE__*/React.createElement(PasswordToggle, {
    disabled: true,
    id: "examplePassword7",
    label: "Password",
    defaultValue: "password"
  }),
  name: "Disabled"
};
