// @canonical/react-components@4.2.0/dist/components/Accordion/Accordion.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _classnames = _interopRequireDefault(require("classnames"));
var _react = _interopRequireWildcard(require("react"));
var _AccordionSection = _interopRequireDefault(require("./AccordionSection"));
const _excluded = ["key"],
  _excluded2 = ["className", "expanded", "externallyControlled", "onExpandedChange", "sections", "titleElement"];
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
const generateSections = (sections, setExpanded, expanded, titleElement) => sections.map((_ref, i) => {
  let {
      key
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/_react.default.createElement(_AccordionSection.default, _extends({
    expanded: expanded,
    key: key || i,
    sectionKey: key === null || key === void 0 ? void 0 : key.toString(),
    setExpanded: setExpanded,
    titleElement: titleElement
  }, props));
});

/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Accordion](https://docs.vanillaframework.io/patterns/accordion/).
 *
 * The sidebar accordion, used in listing pages or as navigation, can hold multiple navigation items or to be used as a filter for content.
 */
const Accordion = _ref2 => {
  let {
      className,
      expanded,
      externallyControlled,
      onExpandedChange,
      sections,
      titleElement
    } = _ref2,
    asideProps = _objectWithoutProperties(_ref2, _excluded2);
  const [expandedSection, setExpandedSection] = (0, _react.useState)(expanded);
  const setExpanded = (id, title) => {
    setExpandedSection(id);
    onExpandedChange && onExpandedChange(id, title);
  };
  return /*#__PURE__*/_react.default.createElement("aside", _extends({
    className: (0, _classnames.default)(className, "p-accordion")
  }, asideProps, {
    role: "tablist",
    "aria-multiselectable": "true"
  }), /*#__PURE__*/_react.default.createElement("ul", {
    className: "p-accordion__list"
  }, generateSections(sections, setExpanded, externallyControlled ? expanded : expandedSection, titleElement)));
};
var _default = exports.default = Accordion;
