// @canonical/react-components@4.2.0/dist/esm/components/NotificationProvider/messageBuilder.js
import React from "react";
import { NotificationSeverity } from "../Notifications";
export var queue = notification => {
  return {
    state: {
      queuedNotification: notification
    }
  };
};
export var info = (message, title, actions) => {
  return {
    message,
    title,
    actions,
    type: NotificationSeverity.INFORMATION
  };
};
export var success = (message, title, actions) => {
  return {
    message,
    title,
    actions,
    type: NotificationSeverity.POSITIVE
  };
};
export var failure = (title, error, message, actions) => {
  return {
    actions,
    message: formatErrorMessage(message, error),
    title,
    type: NotificationSeverity.NEGATIVE
  };
};
export var caution = (message, title, actions) => {
  return {
    message,
    actions,
    title,
    type: NotificationSeverity.CAUTION
  };
};
export var formatErrorMessage = (message, error) => {
  if (error && error instanceof Error) {
    return /*#__PURE__*/React.createElement(React.Fragment, null, message, " ", error.message);
  } else {
    return message;
  }
};
