// @canonical/react-components@4.2.0/dist/esm/components/NotificationProvider/NotificationProvider.js
import React, { createContext, useContext, useEffect, useRef, useState } from "react";
import fastDeepEqual from "fast-deep-equal";
import { info as _info, failure as _failure, success as _success, queue } from "./messageBuilder";
import Notification from "../Notifications";
import { DefaultTitles } from "../Notifications/Notification/Notification";
var NotifyContext = /*#__PURE__*/createContext({
  notification: null,
  clear: () => undefined,
  failure: () => undefined,
  success: () => undefined,
  info: () => undefined,
  queue: () => undefined,
  setDeduplicated: () => undefined
});
export var NotificationProvider = _ref => {
  var {
    children,
    state,
    pathname
  } = _ref;
  var [notification, setNotification] = useState(null);
  var clear = () => notification !== null && setNotification(null);
  var setDeduplicated = value => {
    if (!fastDeepEqual(value, notification)) {
      setNotification(value);
    }
    return value;
  };
  useEffect(() => {
    if (state !== null && state !== void 0 && state.queuedNotification) {
      setDeduplicated(state.queuedNotification);
      window.history.replaceState({}, "");
    } else {
      clear();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [state, pathname]);
  var helper = {
    notification,
    clear,
    queue,
    failure: (title, error, message, actions) => setDeduplicated(_failure(title, error, message, actions)),
    info: (message, title) => setDeduplicated(_info(message, title)),
    success: (message, title) => setDeduplicated(_success(message, title)),
    setDeduplicated
  };
  return /*#__PURE__*/React.createElement(NotifyContext.Provider, {
    value: helper
  }, children);
};
export function useNotify() {
  return useContext(NotifyContext);
}
export var NotificationConsumer = () => {
  var notify = useNotify();
  var ref = useRef(null);
  useEffect(() => {
    var _ref$current;
    if ((_ref$current = ref.current) !== null && _ref$current !== void 0 && _ref$current.hasAttribute("scrollIntoView")) {
      ref.current.scrollIntoView({
        behavior: "smooth",
        block: "center",
        inline: "start"
      });
    }
  }, [notify.notification]);
  if (!notify.notification) {
    return null;
  }
  var {
    actions,
    title,
    type,
    message
  } = notify.notification;
  return /*#__PURE__*/React.createElement("div", {
    ref: ref
  }, /*#__PURE__*/React.createElement(Notification, {
    title: title !== null && title !== void 0 ? title : DefaultTitles[type],
    actions: actions,
    severity: type,
    onDismiss: notify.clear
  }, message));
};
