// @canonical/react-components@4.2.0/dist/esm/components/Slider/Slider.stories.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import React from "react";
import { useState } from "react";
import Slider from "./Slider";
var Template = args => {
  var [selectedValue, setSelectedValue] = useState(50);
  return /*#__PURE__*/React.createElement(Slider, _extends({}, args, {
    onChange: e => setSelectedValue(Number(e.target.value)),
    value: selectedValue
  }));
};
var meta = {
  component: Slider,
  render: Template,
  tags: ["autodocs"],
  argTypes: {
    caution: {
      control: {
        type: "text"
      }
    },
    error: {
      control: {
        type: "text"
      }
    },
    help: {
      control: {
        type: "text"
      }
    },
    label: {
      control: {
        type: "text"
      }
    }
  }
};
export default meta;
export var Default = {
  render: Template.bind({}),
  name: "Default",
  args: {
    disabled: false,
    label: "Volume",
    max: 100,
    min: 0,
    showInput: false
  }
};

/**
 * You can also optionally provide a numeric representation of the slider value.
 */
export var WithInput = {
  render: Template.bind({}),
  name: "With input",
  args: {
    disabled: false,
    inputDisabled: false,
    label: "Volume",
    max: 100,
    min: 0,
    showInput: true
  }
};
