// @canonical/react-components@4.2.0/dist/esm/components/Slider/Slider.js
var _excluded = ["caution", "disabled", "error", "help", "id", "inputDisabled", "label", "max", "min", "onChange", "required", "showInput"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import React, { useId } from "react";
import Field from "../Field";
export var FILLED_COLOR = "#0066CC";
export var EMPTY_COLOR = "#D9D9D9";
/**
 * This is the [React](https://reactjs.org/) component for the Vanilla [Slider](https://vanillaframework.io/docs/patterns/slider).
 *
 * Sliders should be used to let a user specify a numeric value between a defined
minimum and maximum value, however the precise value is not especially important.
 */
export var Slider = _ref => {
  var _navigator;
  var {
      caution,
      disabled = false,
      error,
      help,
      id,
      inputDisabled = false,
      label,
      max,
      min,
      onChange,
      required = false,
      showInput = false
    } = _ref,
    inputProps = _objectWithoutProperties(_ref, _excluded);
  var validationId = useId();
  var helpId = useId();
  var defaultSliderId = useId();
  var sliderId = id || defaultSliderId;
  var hasError = !!error;
  var style = {};
  if ((_navigator = navigator) !== null && _navigator !== void 0 && (_navigator = _navigator.userAgent) !== null && _navigator !== void 0 && _navigator.includes("AppleWebKit")) {
    var _inputProps$value;
    // Range inputs on Webkit browsers don't have a built-in "filled" portion,
    // so instead it is handled here as a background.
    var val = (_inputProps$value = inputProps.value) !== null && _inputProps$value !== void 0 ? _inputProps$value : inputProps.defaultValue;
    var filledPercentage = "".concat((Number(val) - min) / (max - min) * 100, "%");
    style = {
      background: "linear-gradient(\n        to right,\n        ".concat(FILLED_COLOR, " 0%,\n        ").concat(FILLED_COLOR, " ").concat(filledPercentage, ",\n        ").concat(EMPTY_COLOR, " ").concat(filledPercentage, ",\n        ").concat(EMPTY_COLOR, " 100%\n      )")
    };
  }
  return /*#__PURE__*/React.createElement(Field, {
    caution: caution,
    error: error,
    help: help,
    forId: sliderId,
    helpId: helpId,
    label: label,
    required: required,
    validationId: validationId
  }, /*#__PURE__*/React.createElement("div", {
    className: "p-slider__wrapper"
  }, /*#__PURE__*/React.createElement("input", _extends({
    "aria-describedby": help ? helpId : null,
    "aria-errormessage": hasError ? validationId : null,
    "aria-invalid": hasError,
    disabled: disabled,
    id: sliderId,
    max: max,
    min: min,
    onChange: onChange,
    required: required,
    style: style,
    type: "range"
  }, inputProps)), showInput && /*#__PURE__*/React.createElement("input", _extends({
    "aria-label": typeof label === "string" ? label : undefined,
    "aria-describedby": help ? helpId : null,
    "aria-errormessage": hasError ? validationId : null,
    "aria-invalid": hasError,
    className: "p-slider__input",
    disabled: disabled || inputDisabled,
    max: max,
    min: min,
    onChange: onChange,
    type: "number"
  }, inputProps))));
};
export default Slider;
