// @canonical/react-components@4.2.0/dist/external/useSSR.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.weAreServer = exports.useSSR = exports.Device = void 0;
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
/**
 * This is a reference implementation of the useSSR hook from use-ssr: https://github.com/iamthesiz/use-ssr/blob/master/useSSR.ts
 * The license for the content in this file is goverened by the original project's license: https://github.com/iamthesiz/use-ssr/blob/master/license.md
 */
let Device = exports.Device = /*#__PURE__*/function (Device) {
  Device["Browser"] = "browser";
  Device["Server"] = "server";
  return Device;
}({});
const {
  Browser,
  Server
} = Device;
const canUseDOM = !!(typeof window !== "undefined" && window.document && window.document.createElement);
const device = canUseDOM ? Browser : Server;
const SSRObject = {
  isBrowser: device === Browser,
  isServer: device === Server,
  device,
  canUseWorkers: typeof Worker !== "undefined",
  canUseEventListeners: device === Browser && !!window.addEventListener,
  canUseViewport: device === Browser && !!window.screen
};
const assign = function () {
  for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
    args[_key] = arguments[_key];
  }
  return args.reduce((acc, obj) => _objectSpread(_objectSpread({}, acc), obj), {});
};
const values = obj => Object.keys(obj).map(key => obj[key]);
const toArrayObject = () => assign((values(SSRObject), SSRObject));
let useSSRObject = toArrayObject();
const weAreServer = () => {
  SSRObject.isServer = true;
  useSSRObject = toArrayObject();
};
exports.weAreServer = weAreServer;
const useSSR = () => useSSRObject;
exports.useSSR = useSSR;
