// @canonical/react-components@4.2.0/dist/components/List/List.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.VerticalStepped = exports.TickedWithHorizontalDivider = exports.Ticked = exports.Stretch = exports.Split = exports.Middot = exports.Inline = exports.HorizontalStepped = exports.HorizontalDivider = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _List = _interopRequireDefault(require("./List"));
var _StatusLabel = _interopRequireDefault(require("../StatusLabel"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _List.default,
  tags: ["autodocs"],
  argTypes: {
    items: {
      control: {
        disable: true
      }
    },
    stepped: {
      control: {
        disable: true
      }
    }
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default",
  args: {
    items: ["Digital signage", "Robotics", "Gateways"]
  }
};
const Ticked = exports.Ticked = {
  render: () => /*#__PURE__*/_react.default.createElement(_List.default, {
    items: ["Hardware guidance and sizing", "Fixed-price deployment", "Reference architecture"],
    ticked: true
  }),
  name: "Ticked"
};
const HorizontalDivider = exports.HorizontalDivider = {
  render: () => /*#__PURE__*/_react.default.createElement(_List.default, {
    items: ["Single-command install and upgrade", "Postgres database for reliable state", "High availability for all services"],
    divided: true
  }),
  name: "Horizontal divider"
};
const TickedWithHorizontalDivider = exports.TickedWithHorizontalDivider = {
  render: () => /*#__PURE__*/_react.default.createElement(_List.default, {
    items: ["Balance compute load in the cloud", "Optimise for specific workloads", "Assess overcommit ratios"],
    divided: true,
    ticked: true
  }),
  name: "Ticked with horizontal divider"
};
const Inline = exports.Inline = {
  render: () => /*#__PURE__*/_react.default.createElement(_List.default, {
    items: ["Community", "Careers", "Press centre"],
    inline: true
  }),
  name: "Inline"
};
const Middot = exports.Middot = {
  render: () => /*#__PURE__*/_react.default.createElement(_List.default, {
    items: ["Legal information", "Data privacy", "Report a bug on this site"],
    middot: true
  }),
  name: "Middot"
};
const Stretch = exports.Stretch = {
  render: () => /*#__PURE__*/_react.default.createElement(_List.default, {
    items: [{
      content: /*#__PURE__*/_react.default.createElement("h3", null, "Documentation")
    }, {
      content: /*#__PURE__*/_react.default.createElement(_StatusLabel.default, {
        appearance: "positive"
      }, "New"),
      className: "u-vertically-center u-align--right"
    }],
    stretch: true
  }),
  name: "Stretch"
};
const VerticalStepped = exports.VerticalStepped = {
  render: () => /*#__PURE__*/_react.default.createElement(_List.default, {
    items: [{
      title: "Log in to JAAS",
      content: "Ensure you have an Ubuntu SSO account before contacting JAAS. Log in to JAAS now."
    }, {
      title: "Configure a model",
      content: 'Applications are contained within models and are installed via charms. Configure your model by pressing the "Start a new model" button.'
    }, {
      title: "Credentials and SSH keys",
      content: "After having selected a cloud, a form will appear for submitting your credentials to JAAS. The below resources are available if you need help with gathering credentials."
    }],
    stepped: true
  }),
  name: "Vertical stepped"
};
const HorizontalStepped = exports.HorizontalStepped = {
  render: () => /*#__PURE__*/_react.default.createElement(_List.default, {
    items: [{
      title: "Log in to JAAS",
      content: "Ensure you have an Ubuntu SSO account before contacting JAAS. Log in to JAAS now."
    }, {
      title: "Configure a model",
      content: 'Applications are contained within models and are installed via charms. Configure your model by pressing the "Start a new model" button.'
    }],
    stepped: true,
    detailed: true
  }),
  name: "Horizontal stepped"
};
const Split = exports.Split = {
  render: () => /*#__PURE__*/_react.default.createElement(_List.default, {
    items: ["Jointly shape the OpenStack architecture", "We help you plan your cloud hardware requirements", "We build OpenStack in your data center", "We operate the cloud to an SLA", "Transparent audit, logging, monitoring and management", "When your team is ready, we hand over the keys"],
    divided: true,
    split: true,
    ticked: true
  }),
  name: "Split"
};
