// @canonical/react-components@4.2.0/dist/esm/components/ApplicationLayout/AppAside/AppAside.stories.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
/* eslint-disable react-hooks/rules-of-hooks */
import React, { useState } from "react";
import Application from "../Application";
import AppMain from "../AppMain";
import Button from "../../Button";
import Col from "../../Col";
import Form from "../../Form";
import Icon from "../../Icon";
import Input from "../../Input";
import Panel from "../../Panel";
import Row from "../../Row";
import AppAside from "./AppAside";
var meta = {
  component: AppAside,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: false
    }
  }
};
export default meta;
/**
 * In most common cases an `AppAside` should contain a `<Panel>` to display the
 * content as intended in the application layout.
 *
 * `AppAside` should be a direct child of an `<Application>` or passed to the
 * application layout `<ApplicationLayout aside={<AppAside .../>}>`.
 */
export var Default = {
  render: args => {
    var [pinned, setPinned] = useState(false);
    var [width, setWidth] = useState(null);
    var [collapsed, setCollapsed] = useState(false);
    return /*#__PURE__*/React.createElement(Application, null, /*#__PURE__*/React.createElement(AppMain, null, /*#__PURE__*/React.createElement("p", null, "Scroll to the right to see the panel."), /*#__PURE__*/React.createElement(Button, {
      onClick: () => setCollapsed(false)
    }, "Open"), /*#__PURE__*/React.createElement(Button, {
      onClick: () => setWidth("narrow")
    }, "Narrow"), /*#__PURE__*/React.createElement(Button, {
      onClick: () => setWidth(null)
    }, "Default"), /*#__PURE__*/React.createElement(Button, {
      onClick: () => setWidth("wide")
    }, "Wide")), /*#__PURE__*/React.createElement(AppAside, _extends({}, args, {
      pinned: pinned,
      wide: width === "wide",
      narrow: width === "narrow",
      collapsed: collapsed
    }), /*#__PURE__*/React.createElement(Panel, {
      controls: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Button, {
        onClick: () => setPinned(!pinned),
        dense: true,
        className: "u-no-margin--bottom"
      }, "Pin"), /*#__PURE__*/React.createElement(Button, {
        appearance: "base",
        className: "u-no-margin--bottom",
        hasIcon: true,
        onClick: () => setCollapsed(!collapsed)
      }, /*#__PURE__*/React.createElement(Icon, {
        name: "close"
      }, "Close"))),
      title: "App aside"
    }, /*#__PURE__*/React.createElement(Form, {
      stacked: true
    }, /*#__PURE__*/React.createElement(Input, {
      label: "Full name",
      type: "text",
      name: "fullName",
      autoComplete: "name",
      stacked: true
    }), /*#__PURE__*/React.createElement(Input, {
      label: "Username",
      type: "text",
      name: "username-stacked",
      autoComplete: "username",
      "aria-describedby": "exampleHelpTextMessage",
      stacked: true,
      help: "30 characters or fewer."
    }), /*#__PURE__*/React.createElement(Input, {
      type: "text",
      label: "Email address",
      "aria-invalid": "true",
      name: "username-stackederror",
      autoComplete: "email",
      required: true,
      error: "This field is required.",
      stacked: true
    }), /*#__PURE__*/React.createElement(Input, {
      label: "Address line 1",
      type: "text",
      name: "address-optional-stacked",
      autoComplete: "address-line1",
      stacked: true
    }), /*#__PURE__*/React.createElement(Input, {
      label: "Address line 2",
      type: "text",
      name: "address-optional-stacked",
      autoComplete: "address-line3",
      stacked: true
    }), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 12
    }, /*#__PURE__*/React.createElement(Button, {
      appearance: "positive",
      className: "u-float-right",
      name: "add-details"
    }, "Add details")))))));
  }
};
