// @canonical/react-components@4.2.0/dist/components/SideNavigation/SideNavigationItem/SideNavigationItem.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Text = exports.Links = exports.Default = exports.CustomContent = void 0;
var _react = _interopRequireDefault(require("react"));
var _SideNavigation = _interopRequireDefault(require("../SideNavigation"));
var _SideNavigationItem = _interopRequireDefault(require("./SideNavigationItem"));
var _SideNavigationText = _interopRequireDefault(require("../SideNavigationText"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const meta = {
  component: _SideNavigationItem.default,
  render: args => /*#__PURE__*/_react.default.createElement(_SideNavigation.default, {
    dark: false,
    items: [{
      items: [/*#__PURE__*/_react.default.createElement(_SideNavigationItem.default, _extends({
        key: "item"
      }, args, {
        dark: false
      }))]
    }]
  }),
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  args: {
    icon: "drag",
    label: "Models",
    href: "/models"
  }
};

/**
 * The navigation item can appear as a link and accept the props from [`SideNavigationLink`](/docs/components-sidenavigation-sidenavigationlink--docs).
 */
const Links = exports.Links = {
  args: {
    icon: "drag",
    label: "Models",
    href: "/models"
  }
};

/**
 * By providing the `nonInteractive` prop the navigation item will appear as a
 * text entry and accept the props from
 * [`SideNavigationText`](/docs/components-sidenavigation-sidenavigationtext--docs).
 */
const Text = exports.Text = {
  args: {
    icon: "drag",
    label: "Models",
    nonInteractive: true
  }
};

/**
 * The navigation item can display custom content by providing the `children` prop.
 * In this case, any other attributes provided to the object will be given to
 * the list item.
 */
const CustomContent = exports.CustomContent = {
  args: {
    className: "custom-class",
    children: /*#__PURE__*/_react.default.createElement(_SideNavigationText.default, null, "This item has been given a custom class")
  }
};
