// @canonical/react-components@4.2.0/dist/esm/components/Accordion/Accordion.js
var _excluded = ["key"],
  _excluded2 = ["className", "expanded", "externallyControlled", "onExpandedChange", "sections", "titleElement"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import classNames from "classnames";
import React, { useState } from "react";
import AccordionSection from "./AccordionSection";
var generateSections = (sections, setExpanded, expanded, titleElement) => sections.map((_ref, i) => {
  var {
      key
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/React.createElement(AccordionSection, _extends({
    expanded: expanded,
    key: key || i,
    sectionKey: key === null || key === void 0 ? void 0 : key.toString(),
    setExpanded: setExpanded,
    titleElement: titleElement
  }, props));
});

/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Accordion](https://docs.vanillaframework.io/patterns/accordion/).
 *
 * The sidebar accordion, used in listing pages or as navigation, can hold multiple navigation items or to be used as a filter for content.
 */
var Accordion = _ref2 => {
  var {
      className,
      expanded,
      externallyControlled,
      onExpandedChange,
      sections,
      titleElement
    } = _ref2,
    asideProps = _objectWithoutProperties(_ref2, _excluded2);
  var [expandedSection, setExpandedSection] = useState(expanded);
  var setExpanded = (id, title) => {
    setExpandedSection(id);
    onExpandedChange && onExpandedChange(id, title);
  };
  return /*#__PURE__*/React.createElement("aside", _extends({
    className: classNames(className, "p-accordion")
  }, asideProps, {
    role: "tablist",
    "aria-multiselectable": "true"
  }), /*#__PURE__*/React.createElement("ul", {
    className: "p-accordion__list"
  }, generateSections(sections, setExpanded, externallyControlled ? expanded : expandedSection, titleElement)));
};
export default Accordion;
