// @canonical/react-components@4.2.0/dist/esm/components/Accordion/Accordion.stories.js
import { useState } from "react";
import React from "react";
import Accordion from "./Accordion";
var meta = {
  component: Accordion,
  tags: ["autodocs"],
  argTypes: {
    sections: {
      control: {
        disable: true
      }
    },
    externallyControlled: {
      control: {
        disable: true
      }
    }
  }
};
export default meta;
export var Default = {
  name: "Default",
  args: {
    sections: [{
      title: "Advanced topics",
      content: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("p", null, "Charm bundles"), /*#__PURE__*/React.createElement("p", null, "Machine authentication"), /*#__PURE__*/React.createElement("p", null, "Migrating models"), /*#__PURE__*/React.createElement("p", null, "Using storage"), /*#__PURE__*/React.createElement("p", null, "Working with actions"), /*#__PURE__*/React.createElement("p", null, "Working with resources"), /*#__PURE__*/React.createElement("p", null, "Cloud image metadata"), /*#__PURE__*/React.createElement("p", null, "Tools"))
    }, {
      title: "Networking",
      content: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("p", null, "Working offline"), /*#__PURE__*/React.createElement("p", null, "Fan container networking"), /*#__PURE__*/React.createElement("p", null, "Network spaces"))
    }, {
      title: "Miscellaneous",
      content: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("p", null, "Juju GUI"), /*#__PURE__*/React.createElement("p", null, "CentOS support"), /*#__PURE__*/React.createElement("p", null, "Collecting Juju metrics"))
    }]
  }
};

/**
 * The expanded accordion section can be controlled by external state.
 */
export var ExternalState = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [expandedSection, setExpandedSection] = useState();
    return /*#__PURE__*/React.createElement(Accordion, {
      expanded: expandedSection,
      externallyControlled: true,
      onExpandedChange: setExpandedSection,
      sections: [{
        title: "Advanced topics",
        key: "advanced-topics",
        content: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("p", null, "Charm bundles"), /*#__PURE__*/React.createElement("p", null, "Machine authentication"), /*#__PURE__*/React.createElement("p", null, "Migrating models"), /*#__PURE__*/React.createElement("p", null, "Using storage"), /*#__PURE__*/React.createElement("p", null, "Working with actions"), /*#__PURE__*/React.createElement("p", null, "Working with resources"), /*#__PURE__*/React.createElement("p", null, "Cloud image metadata"), /*#__PURE__*/React.createElement("p", null, "Tools"))
      }, {
        title: "Networking",
        key: "networking",
        content: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("p", null, "Working offline"), /*#__PURE__*/React.createElement("p", null, "Fan container networking"), /*#__PURE__*/React.createElement("p", null, "Network spaces"))
      }, {
        title: "Miscellaneous",
        key: "miscellaneous",
        content: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("p", null, "Juju GUI"), /*#__PURE__*/React.createElement("p", null, "CentOS support"), /*#__PURE__*/React.createElement("p", null, "Collecting Juju metrics"))
      }]
    });
  },
  name: "External state"
};

/**
 * `titleElement` prop can be used to define heading element for section titles.
 */
export var Headings = {
  name: "Headings",
  args: {
    sections: [{
      title: "Advanced topics",
      content: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("p", null, "Charm bundles"), /*#__PURE__*/React.createElement("p", null, "Machine authentication"), /*#__PURE__*/React.createElement("p", null, "Migrating models"), /*#__PURE__*/React.createElement("p", null, "Using storage"), /*#__PURE__*/React.createElement("p", null, "Working with actions"), /*#__PURE__*/React.createElement("p", null, "Working with resources"), /*#__PURE__*/React.createElement("p", null, "Cloud image metadata"), /*#__PURE__*/React.createElement("p", null, "Tools"))
    }, {
      title: "Networking",
      content: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("p", null, "Working offline"), /*#__PURE__*/React.createElement("p", null, "Fan container networking"), /*#__PURE__*/React.createElement("p", null, "Network spaces"))
    }, {
      title: "Miscellaneous",
      content: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("p", null, "Juju GUI"), /*#__PURE__*/React.createElement("p", null, "CentOS support"), /*#__PURE__*/React.createElement("p", null, "Collecting Juju metrics"))
    }],
    titleElement: "h3"
  }
};

/**
 * The `title` prop can be a `ReactNode`, allowing a higher degree of customisation of the section titles.
 */
export var CustomHeadings = {
  args: {
    sections: [{
      title: /*#__PURE__*/React.createElement(React.Fragment, null, "Advanced topics", " ", /*#__PURE__*/React.createElement("span", {
        className: "u-text--muted p-text--small"
      }, "optional")),
      content: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("p", null, "Charm bundles"), /*#__PURE__*/React.createElement("p", null, "Machine authentication"), /*#__PURE__*/React.createElement("p", null, "Migrating models"), /*#__PURE__*/React.createElement("p", null, "Using storage"), /*#__PURE__*/React.createElement("p", null, "Working with actions"), /*#__PURE__*/React.createElement("p", null, "Working with resources"), /*#__PURE__*/React.createElement("p", null, "Cloud image metadata"), /*#__PURE__*/React.createElement("p", null, "Tools"))
    }, {
      title: /*#__PURE__*/React.createElement(React.Fragment, null, "Networking", " ", /*#__PURE__*/React.createElement("span", {
        className: "u-text--muted p-text--small"
      }, "optional")),
      content: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("p", null, "Working offline"), /*#__PURE__*/React.createElement("p", null, "Fan container networking"), /*#__PURE__*/React.createElement("p", null, "Network spaces"))
    }, {
      title: /*#__PURE__*/React.createElement(React.Fragment, null, "Miscellaneous", " ", /*#__PURE__*/React.createElement("span", {
        className: "u-text--muted p-text--small"
      }, "optional")),
      content: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("p", null, "Juju GUI"), /*#__PURE__*/React.createElement("p", null, "CentOS support"), /*#__PURE__*/React.createElement("p", null, "Collecting Juju metrics"))
    }],
    titleElement: "h3"
  }
};
