// @canonical/react-components@4.2.0/dist/components/NotificationProvider/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "NotificationConsumer", {
  enumerable: true,
  get: function () {
    return _NotificationProvider.NotificationConsumer;
  }
});
Object.defineProperty(exports, "NotificationProvider", {
  enumerable: true,
  get: function () {
    return _NotificationProvider.NotificationProvider;
  }
});
Object.defineProperty(exports, "caution", {
  enumerable: true,
  get: function () {
    return _messageBuilder.caution;
  }
});
Object.defineProperty(exports, "failure", {
  enumerable: true,
  get: function () {
    return _messageBuilder.failure;
  }
});
Object.defineProperty(exports, "info", {
  enumerable: true,
  get: function () {
    return _messageBuilder.info;
  }
});
Object.defineProperty(exports, "queue", {
  enumerable: true,
  get: function () {
    return _messageBuilder.queue;
  }
});
Object.defineProperty(exports, "success", {
  enumerable: true,
  get: function () {
    return _messageBuilder.success;
  }
});
Object.defineProperty(exports, "useNotify", {
  enumerable: true,
  get: function () {
    return _NotificationProvider.useNotify;
  }
});
var _NotificationProvider = require("./NotificationProvider");
var _messageBuilder = require("./messageBuilder");
