// @canonical/react-components@4.2.0/dist/components/NotificationProvider/messageBuilder.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.success = exports.queue = exports.info = exports.formatErrorMessage = exports.failure = exports.caution = void 0;
var _react = _interopRequireDefault(require("react"));
var _Notifications = require("../Notifications");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const queue = notification => {
  return {
    state: {
      queuedNotification: notification
    }
  };
};
exports.queue = queue;
const info = (message, title, actions) => {
  return {
    message,
    title,
    actions,
    type: _Notifications.NotificationSeverity.INFORMATION
  };
};
exports.info = info;
const success = (message, title, actions) => {
  return {
    message,
    title,
    actions,
    type: _Notifications.NotificationSeverity.POSITIVE
  };
};
exports.success = success;
const failure = (title, error, message, actions) => {
  return {
    actions,
    message: formatErrorMessage(message, error),
    title,
    type: _Notifications.NotificationSeverity.NEGATIVE
  };
};
exports.failure = failure;
const caution = (message, title, actions) => {
  return {
    message,
    actions,
    title,
    type: _Notifications.NotificationSeverity.CAUTION
  };
};
exports.caution = caution;
const formatErrorMessage = (message, error) => {
  if (error && error instanceof Error) {
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, message, " ", error.message);
  } else {
    return message;
  }
};
exports.formatErrorMessage = formatErrorMessage;
