// @canonical/react-components@4.2.0/dist/components/NotificationProvider/NotificationProvider.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NotificationProvider = exports.NotificationConsumer = void 0;
exports.useNotify = useNotify;
var _react = _interopRequireWildcard(require("react"));
var _fastDeepEqual = _interopRequireDefault(require("fast-deep-equal"));
var _messageBuilder = require("./messageBuilder");
var _Notifications = _interopRequireDefault(require("../Notifications"));
var _Notification = require("../Notifications/Notification/Notification");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const NotifyContext = /*#__PURE__*/(0, _react.createContext)({
  notification: null,
  clear: () => undefined,
  failure: () => undefined,
  success: () => undefined,
  info: () => undefined,
  queue: () => undefined,
  setDeduplicated: () => undefined
});
const NotificationProvider = _ref => {
  let {
    children,
    state,
    pathname
  } = _ref;
  const [notification, setNotification] = (0, _react.useState)(null);
  const clear = () => notification !== null && setNotification(null);
  const setDeduplicated = value => {
    if (!(0, _fastDeepEqual.default)(value, notification)) {
      setNotification(value);
    }
    return value;
  };
  (0, _react.useEffect)(() => {
    if (state !== null && state !== void 0 && state.queuedNotification) {
      setDeduplicated(state.queuedNotification);
      window.history.replaceState({}, "");
    } else {
      clear();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [state, pathname]);
  const helper = {
    notification,
    clear,
    queue: _messageBuilder.queue,
    failure: (title, error, message, actions) => setDeduplicated((0, _messageBuilder.failure)(title, error, message, actions)),
    info: (message, title) => setDeduplicated((0, _messageBuilder.info)(message, title)),
    success: (message, title) => setDeduplicated((0, _messageBuilder.success)(message, title)),
    setDeduplicated
  };
  return /*#__PURE__*/_react.default.createElement(NotifyContext.Provider, {
    value: helper
  }, children);
};
exports.NotificationProvider = NotificationProvider;
function useNotify() {
  return (0, _react.useContext)(NotifyContext);
}
const NotificationConsumer = () => {
  const notify = useNotify();
  const ref = (0, _react.useRef)(null);
  (0, _react.useEffect)(() => {
    var _ref$current;
    if ((_ref$current = ref.current) !== null && _ref$current !== void 0 && _ref$current.hasAttribute("scrollIntoView")) {
      ref.current.scrollIntoView({
        behavior: "smooth",
        block: "center",
        inline: "start"
      });
    }
  }, [notify.notification]);
  if (!notify.notification) {
    return null;
  }
  const {
    actions,
    title,
    type,
    message
  } = notify.notification;
  return /*#__PURE__*/_react.default.createElement("div", {
    ref: ref
  }, /*#__PURE__*/_react.default.createElement(_Notifications.default, {
    title: title !== null && title !== void 0 ? title : _Notification.DefaultTitles[type],
    actions: actions,
    severity: type,
    onDismiss: notify.clear
  }, message));
};
exports.NotificationConsumer = NotificationConsumer;
