// @canonical/react-components@4.2.0/dist/components/ColumnSelector/ColumnSelectorStoriesHelper.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getRows = exports.USER_HIDEABLE_COLUMNS = exports.TYPE = exports.STATUS = exports.SNAPSHOTS = exports.PROJECT = exports.NAME = exports.MEMORY = exports.IPV6 = exports.IPV4 = exports.FILESYSTEM = exports.DESCRIPTION = exports.COLUMN_HEADERS = exports.CLUSTER_MEMBER = void 0;
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const STATUS = exports.STATUS = "Status";
const NAME = exports.NAME = "Name";
const TYPE = exports.TYPE = "Type";
const CLUSTER_MEMBER = exports.CLUSTER_MEMBER = "Cluster member";
const DESCRIPTION = exports.DESCRIPTION = "Description";
const MEMORY = exports.MEMORY = "Memory";
const FILESYSTEM = exports.FILESYSTEM = "Root filesystem";
const IPV4 = exports.IPV4 = "IPv4";
const IPV6 = exports.IPV6 = "IPv6";
const SNAPSHOTS = exports.SNAPSHOTS = "Snapshots";
const PROJECT = exports.PROJECT = "Project";
const COLUMN_HEADERS = exports.COLUMN_HEADERS = [NAME, TYPE, PROJECT, CLUSTER_MEMBER, MEMORY, FILESYSTEM, DESCRIPTION, IPV4, IPV6, SNAPSHOTS, STATUS];

// All columns except name and status
const USER_HIDEABLE_COLUMNS = exports.USER_HIDEABLE_COLUMNS = [TYPE, PROJECT, CLUSTER_MEMBER, MEMORY, FILESYSTEM, DESCRIPTION, IPV4, IPV6, SNAPSHOTS];
const INSTANCES = [{
  name: "darling-bunny",
  type: "VM",
  project: "default",
  clusterMember: "-",
  memory: "1.4 GiB of 1.9 GiB",
  fileSystem: "3.3 GiB of 9.5 GiB",
  description: "Meet darling-bunny, a speedy little instance hopping along with your latest project. It's got the ears for listening to your commands and the nose for sniffing out the best performance. Handle with care, and remember to feed it plenty of CPU.",
  ipv4: ["10.138.32.6"],
  ipv6: ["fd42:6a38:859a:587d:216:3eff:fe43:167e", "fe80::216:3eff:fe43:167e", "fe80::1050:62ff:feb2:964c"],
  status: "Running"
}, {
  name: "sassy-salamander",
  type: "VM",
  project: "default",
  clusterMember: "-",
  memory: "1.4 GiB of 1.9 GiB",
  fileSystem: "3.3 GiB of 9.5 GiB",
  description: "sassy-salamander is a resilient and feisty instance, perfectly at home in any environment—from the scorching heat of a web server to the damp coolness of a database. Don't let its size fool you; it's quick, adaptable, and not afraid to tackle complex tasks.",
  ipv4: ["10.138.32.6"],
  ipv6: ["fd42:6a38:859a:587d:216:3eff:fe43:167e", "fe80::216:3eff:fe43:167e", "fe80::1050:62ff:feb2:964c"],
  status: "Running"
}, {
  name: "fun-feline",
  type: "Container",
  project: "default",
  clusterMember: "-",
  memory: "0.7 GiB of 1.9 GiB",
  fileSystem: "1.1 GiB of 9.5 GiB",
  description: "fun-feline is a purr-fectly agile and independent instance. It’s got nine lives for all your testing needs and a mischievous spirit for tackling the toughest tasks. Just be sure to give it enough RAM—it loves to stretch out.",
  ipv4: ["10.138.32.6"],
  ipv6: ["fd42:6a38:859a:587d:216:3eff:fe43:167e", "fe80::216:3eff:fe43:167e", "fe80::1050:62ff:feb2:964c"],
  snapshots: ["1", "2"],
  status: "Stopped"
}];
const getRows = () => {
  const instanceRows = INSTANCES.map(instance => {
    var _instance$snapshots$l, _instance$snapshots;
    return {
      key: instance.name,
      name: instance.name,
      columns: [{
        content: instance.name,
        className: "u-truncate",
        title: "Instance ".concat(instance.name),
        role: "rowheader",
        "aria-label": NAME,
        // needed
        onClick: () => console.log("Click on instance", instance.name)
      }, {
        content: /*#__PURE__*/_react.default.createElement("span", null, instance.type),
        className: "clickable-cell",
        role: "cell",
        "aria-label": TYPE,
        // needed
        onClick: () => console.log("Click on instance", instance.type)
      }, {
        content: /*#__PURE__*/_react.default.createElement("a", null, instance.project),
        role: "cell",
        "aria-label": PROJECT // needed
      }, {
        content: /*#__PURE__*/_react.default.createElement("a", null, instance.clusterMember),
        role: "cell",
        "aria-label": CLUSTER_MEMBER // needed
      }, {
        content: instance.memory,
        className: "clickable-cell",
        role: "cell",
        "aria-label": MEMORY,
        // needed
        onClick: () => console.log("Click on instance", instance.memory)
      }, {
        content: instance.fileSystem,
        className: "clickable-cell",
        role: "cell",
        "aria-label": FILESYSTEM,
        // needed
        onClick: () => console.log("Click on instance", instance.fileSystem)
      }, {
        content: /*#__PURE__*/_react.default.createElement("div", {
          className: "u-truncate",
          title: instance.description
        }, instance.description),
        className: "clickable-cell",
        role: "cell",
        "aria-label": DESCRIPTION,
        // needed
        onClick: () => console.log("Click on instance", instance.description)
      }, {
        key: "ipv4-".concat(instance.ipv4.length),
        content: instance.ipv4.join(", "),
        className: "u-align--right clickable-cell",
        role: "cell",
        "aria-label": IPV4,
        // needed
        onClick: () => console.log("Click on instance", instance.ipv4)
      }, {
        key: "ipv6-".concat(instance.ipv6.length),
        content: instance.ipv6.join(", "),
        className: "clickable-cell",
        role: "cell",
        "aria-label": IPV6,
        // needed
        onClick: () => console.log("Click on instance", instance.ipv6)
      }, {
        content: (_instance$snapshots$l = (_instance$snapshots = instance.snapshots) === null || _instance$snapshots === void 0 ? void 0 : _instance$snapshots.length) !== null && _instance$snapshots$l !== void 0 ? _instance$snapshots$l : "0",
        className: "u-align--right clickable-cell",
        role: "cell",
        "aria-label": SNAPSHOTS,
        // needed
        onClick: () => {
          var _instance$snapshots2;
          return console.log("Click on instance", (_instance$snapshots2 = instance.snapshots) === null || _instance$snapshots2 === void 0 ? void 0 : _instance$snapshots2.length);
        }
      }, {
        content: instance.status,
        role: "cell",
        className: "clickable-cell",
        "aria-label": STATUS,
        // needed
        onClick: () => console.log("Click on instance", instance.status)
      }]
    };
  });
  return instanceRows;
};
exports.getRows = getRows;
