// @canonical/react-components@4.2.0/dist/components/ColumnSelector/ColumnSelector.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));
var _Tooltip = _interopRequireDefault(require("../Tooltip"));
var _ContextualMenu = _interopRequireDefault(require("../ContextualMenu"));
var _Icon = _interopRequireDefault(require("../Icon"));
var _CheckboxInput = _interopRequireDefault(require("../CheckboxInput"));
require("./ColumnSelector.scss");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
/**
This is a [React](https://reactjs.org/) component that extends from the Vanilla [Select](https://vanillaframework.io/docs/base/forms#select) element.
The aim of this component is to provide a dropdown menu to control the visibility of columns within a table.
This component allows users to customize their view, hiding or showing columns as needed, while also handling columns that are automatically hidden on smaller screens.
*/
const ColumnSelector = _ref => {
  let {
    className,
    columns,
    userHidden,
    sizeHidden,
    setUserHidden
  } = _ref;
  const selectedCount = columns.length - userHidden.length;
  const toggleHiddenColumn = column => {
    if (userHidden.includes(column)) {
      setUserHidden(userHidden.filter(c => c !== column));
    } else {
      setUserHidden([...userHidden, column]);
    }
  };
  const wrapTooltip = (element, column) => {
    if (!sizeHidden.includes(column)) return element;
    return /*#__PURE__*/_react.default.createElement(_Tooltip.default, {
      message: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, "Screen is too narrow to fit the column.", /*#__PURE__*/_react.default.createElement("br", null), "Disable columns above or use a bigger screen."),
      position: "left"
    }, element);
  };
  return /*#__PURE__*/_react.default.createElement(_ContextualMenu.default, {
    className: (0, _classnames.default)(className, "column-selector-toggle"),
    dropdownProps: {
      "aria-label": "columns menu"
    },
    position: "right",
    toggleClassName: "has-icon",
    toggleProps: {
      "aria-label": "Columns selection toggle"
    },
    toggleLabel: /*#__PURE__*/_react.default.createElement(_Icon.default, {
      name: "settings"
    }),
    toggleAppearance: "base",
    title: "Columns"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "column-selector-column-list"
  }, /*#__PURE__*/_react.default.createElement(_CheckboxInput.default, {
    checked: userHidden.length === 0,
    indeterminate: selectedCount > 0 && selectedCount < columns.length,
    label: "".concat(selectedCount, " out of ").concat(columns.length, " columns selected"),
    onChange: () => {
      if (userHidden.length > 0) {
        setUserHidden([]);
      } else {
        setUserHidden(columns);
      }
    }
  }), /*#__PURE__*/_react.default.createElement("hr", null), columns.map(column => /*#__PURE__*/_react.default.createElement("div", {
    key: column
  }, wrapTooltip(/*#__PURE__*/_react.default.createElement(_CheckboxInput.default, {
    "aria-label": column,
    labelClassName: (0, _classnames.default)({
      "size-hidden": sizeHidden.includes(column)
    }),
    checked: !userHidden.includes(column),
    label: column,
    onChange: () => {
      toggleHiddenColumn(column);
    },
    disabled: sizeHidden.includes(column)
  }), column)))));
};
var _default = exports.default = ColumnSelector;
