// @canonical/react-components@4.2.0/dist/components/ColumnSelector/ColumnSelector.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _ColumnSelector = _interopRequireDefault(require("./ColumnSelector"));
var _MainTable = _interopRequireDefault(require("../MainTable"));
var _ColumnSelectorStoriesHelper = require("./ColumnSelectorStoriesHelper");
var _columnSelectorHelper = require("./columnSelectorHelper");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const meta = {
  component: _ColumnSelector.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [userHidden, setUserHidden] = (0, _react.useState)([_ColumnSelectorStoriesHelper.FILESYSTEM, _ColumnSelectorStoriesHelper.SNAPSHOTS, _ColumnSelectorStoriesHelper.CLUSTER_MEMBER]);
    const sizeHidden = [_ColumnSelectorStoriesHelper.DESCRIPTION];
    const hiddenColumns = userHidden.concat(sizeHidden);
    const ROWS = (0, _ColumnSelectorStoriesHelper.getRows)();
    const setUserHiddenAndSaveInLocalStorage = columns => {
      setUserHidden(columns);
    };
    const displayedRows = (0, _columnSelectorHelper.visibleRowColumns)(ROWS, hiddenColumns);
    const displayedHeaders = (0, _columnSelectorHelper.visibleHeaderColumns)(_ColumnSelectorStoriesHelper.COLUMN_HEADERS.map(t => {
      return {
        content: t,
        sortKey: t
      };
    }), hiddenColumns);
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_ColumnSelector.default, {
      columns: _ColumnSelectorStoriesHelper.USER_HIDEABLE_COLUMNS,
      userHidden: userHidden,
      sizeHidden: sizeHidden,
      setUserHidden: setUserHiddenAndSaveInLocalStorage
    }), /*#__PURE__*/_react.default.createElement(_MainTable.default, {
      rows: displayedRows,
      headers: displayedHeaders
    }));
  },
  name: "Default"
};
