// @canonical/react-components@4.2.0/dist/esm/components/SidePanel/SidePanel.js
import React, { useState } from "react";
import classnames from "classnames";
import { createPortal } from "react-dom";
import Content from "./common/Content";
import Footer from "./common/Footer";
import Header from "./common/Header";
import HeaderControls from "./common/HeaderControls";
import HeaderTitle from "./common/HeaderTitle";
import Sticky from "./common/Sticky";
import AppAside from "../ApplicationLayout/AppAside";
import Spinner from "../Spinner";
import "./SidePanel.scss";

/**
 * The props for the SidePanelComponent component.
 */

/**
This is a [React](https://reactjs.org/) component for the Vanilla application layout [Side Panel](https://vanillaframework.io/docs/layouts/application#panels).

The sidepanel component should be used to show additional information relating to the main content on the page. It is using a composable structure with building blocks:

* **SidePanel.Header:** Defines the header of the side panel, which can include a title and controls.

* **SidePanel.HeaderTitle:** To show the title of the side panel as heading.

* **SidePanel.HeaderControls:** To show controls in the header, such as buttons or icons for actions like closing the panel.

* **SidePanel.Sticky:** Can be wrapped around the header or footer to make them sticky when scrolling. The scrollbar will use the full area of the side panel, not just the content area. To limit the scrollbar to the content area, use the `ScrollableContainer` component instead of this one.

* **SidePanel.Content:** To show the main content of the side panel.

* **SidePanel.Footer:** To show additional information or actions at the bottom of the side panel.

## Example

Usage of the composable structure:

```jsx
<SidePanel className="u-no-padding--top u-no-padding--bottom">
  <SidePanel.Sticky>
    <SidePanel.Header>
      <SidePanel.HeaderTitle>Edit panel</SidePanel.HeaderTitle>
    </SidePanel.Header>
  </SidePanel.Sticky>
  <SidePanel.Content className="u-no-padding">
    <p>Here be dragons!</p>
  </SidePanel.Content>
  <SidePanel.Sticky position="bottom">
    <SidePanel.Footer className="u-align--right">
      <Button appearance="base">Cancel</Button>
      <ActionButton appearance="positive">Save changes</ActionButton>
    </SidePanel.Footer>
  </SidePanel.Sticky>
</SidePanel>
```
*/
var SidePanelComponent = _ref => {
  var {
    children,
    className,
    hasError = false,
    loading = false,
    overlay,
    pinned,
    width = "",
    parentId = "l-application",
    isOpen = true,
    isAnimated
  } = _ref;
  var [hiding, setHiding] = useState(true);
  var [previousIsOpen, setPreviousIsOpen] = useState(false);
  var container = document.getElementById(parentId) || document.body;
  if (isOpen !== previousIsOpen) {
    setPreviousIsOpen(isOpen);
    setHiding(true);
  }

  // Pinned side panels don't get animation in Vanilla
  var isAnimatedFinal = isAnimated && !pinned;
  if (!isOpen && !(isAnimatedFinal && hiding)) {
    return null;
  }
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/createPortal(/*#__PURE__*/React.createElement(AppAside, {
    className: classnames("side-panel", className, {
      "is-overlay": overlay,
      "slide-in": isAnimatedFinal
    }),
    collapsed: !isOpen,
    "aria-label": "Side panel",
    pinned: pinned,
    narrow: width === "narrow",
    wide: width === "wide",
    onTransitionEnd: () => setHiding(false)
  }, loading ? /*#__PURE__*/React.createElement("div", {
    className: "loading"
  }, /*#__PURE__*/React.createElement(Spinner, null)) : hasError ? /*#__PURE__*/React.createElement("div", {
    className: "error"
  }, "Loading failed") : children), container));
};
var SidePanel = SidePanelComponent;
SidePanel.Header = Header;
SidePanel.HeaderTitle = HeaderTitle;
SidePanel.HeaderControls = HeaderControls;
SidePanel.Sticky = Sticky;
SidePanel.Content = Content;
SidePanel.Footer = Footer;
export default SidePanel;
