// @canonical/react-components@4.2.0/dist/esm/components/SidePanel/SidePanel.stories.js
import React, { useEffect, useId, useRef, useState } from "react";
import SidePanel from "./SidePanel";
import Button from "../Button";
import Application from "../ApplicationLayout/Application";
import AppMain from "../ApplicationLayout/AppMain";
import { ActionButton } from "../../index";
import Icon from "../Icon";
var meta = {
  component: SidePanel,
  tags: ["autodocs"],
  decorators: [Story => {
    var ref = useRef(null);
    useEffect(() => {
      var el = ref.current;
      if (el) {
        var _el$parentElement, _wrapper$parentElemen;
        var wrapper = el === null || el === void 0 || (_el$parentElement = el.parentElement) === null || _el$parentElement === void 0 || (_el$parentElement = _el$parentElement.parentElement) === null || _el$parentElement === void 0 ? void 0 : _el$parentElement.parentElement;
        var parent = wrapper === null || wrapper === void 0 || (_wrapper$parentElemen = wrapper.parentElement) === null || _wrapper$parentElemen === void 0 || (_wrapper$parentElemen = _wrapper$parentElemen.parentElement) === null || _wrapper$parentElemen === void 0 ? void 0 : _wrapper$parentElemen.parentElement;
        if (wrapper) {
          wrapper.style.setProperty("border", "0", "important");
        }
        if (parent) {
          parent.style.margin = "0";
          parent.style.padding = "0";
        }
      }
    }, []);
    document.body.style.padding = "0";
    return /*#__PURE__*/React.createElement("div", {
      ref: ref
    }, /*#__PURE__*/React.createElement(Story, null));
  }]
};
export default meta;
var StoryExample = args => {
  var [isOpen, setIsOpen] = useState(false);

  // Multiple stories get rendered on the docs page, ensure the ids are not colliding
  var id = useId();
  var parentId = args.parentId + id;
  var openPanel = () => {
    setIsOpen(true);
  };
  var closePanel = () => {
    setIsOpen(false);
  };

  // Delay opening the panel, so the l-application container is ready to receive it
  useEffect(() => void setTimeout(openPanel, 100), []);
  return /*#__PURE__*/React.createElement(Application, {
    id: parentId
  }, /*#__PURE__*/React.createElement(AppMain, null, /*#__PURE__*/React.createElement(Button, {
    onClick: openPanel,
    style: {
      margin: "1rem"
    }
  }, "Open side panel"), /*#__PURE__*/React.createElement(Button, {
    onClick: closePanel,
    style: {
      margin: "1rem"
    }
  }, "Close side panel")), /*#__PURE__*/React.createElement(SidePanel, {
    overlay: args.overlay,
    loading: args.loading,
    hasError: args.hasError,
    parentId: parentId,
    pinned: args.pinned,
    width: args.width,
    className: "u-no-padding--top u-no-padding--bottom",
    isOpen: isOpen,
    isAnimated: args.isAnimated
  }, /*#__PURE__*/React.createElement(SidePanel.Sticky, null, /*#__PURE__*/React.createElement(SidePanel.Header, null, /*#__PURE__*/React.createElement(SidePanel.HeaderTitle, null, "Edit panel"), /*#__PURE__*/React.createElement(SidePanel.HeaderControls, null, /*#__PURE__*/React.createElement(Button, {
    appearance: "base",
    className: "u-no-margin--bottom",
    hasIcon: true,
    onClick: closePanel,
    "aria-label": "Close"
  }, /*#__PURE__*/React.createElement(Icon, {
    name: "close"
  }))))), /*#__PURE__*/React.createElement(SidePanel.Content, {
    className: "u-no-padding"
  }, /*#__PURE__*/React.createElement("p", null, "Here be dragons!"), /*#__PURE__*/React.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras varius mi eu pretium vulputate. Nunc commodo sit amet nibh quis rhoncus. Aliquam rhoncus porttitor semper. Aenean faucibus consectetur neque in sodales. Sed cursus mauris id ex sollicitudin sodales. Quisque molestie rutrum odio, ornare pharetra ligula. Pellentesque ornare tristique feugiat. In a augue neque. Aenean eget arcu quis lacus tempus posuere in sit amet dui. Suspendisse faucibus sapien nisl, nec laoreet sem convallis nec."), /*#__PURE__*/React.createElement("p", null, "Vestibulum sed placerat lorem. Nam luctus ex id nisi luctus, id vestibulum sem bibendum. Vivamus turpis sem, pellentesque fermentum malesuada eu, faucibus porta libero. Duis eget venenatis odio. Etiam sed volutpat magna, non tempus erat. Nunc id tortor ac quam consectetur dapibus ac ut tellus. Pellentesque ut tellus venenatis elit vehicula condimentum eget quis lorem. Ut non consectetur est, a fringilla ipsum. Nunc vitae ligula ipsum. Etiam suscipit, libero ut lacinia viverra, nunc urna consequat ex, vel eleifend eros mauris vitae ipsum. Pellentesque sed dictum augue. Ut sit amet ullamcorper mauris. Nunc congue orci mollis purus sodales facilisis ac ut arcu. Maecenas feugiat sapien ac massa mollis sodales. Donec vitae turpis eu nisi laoreet pulvinar quis at nisl. Integer volutpat, metus eget elementum dictum, lacus sapien viverra felis, consequat fermentum nisl mi ac dui."), /*#__PURE__*/React.createElement("p", null, "Nullam nulla turpis, dignissim vel dapibus ut, volutpat ac dui. Donec vel elementum lacus. Mauris maximus nec felis at faucibus. Nunc faucibus gravida velit, id blandit lectus tincidunt ac. Vestibulum orci diam, elementum in congue eu, placerat id risus. Sed tempor tempus tellus, vitae iaculis turpis fringilla nec. Phasellus imperdiet facilisis velit, sit amet lobortis odio dignissim ut."), /*#__PURE__*/React.createElement("p", null, "Nam placerat urna vitae ligula hendrerit, ac tincidunt lorem maximus. Mauris eu odio nisi. Nulla facilisi. Sed egestas elit sed velit rutrum, sit amet bibendum metus hendrerit. Pellentesque luctus placerat tellus, eu bibendum justo. Cras eget leo ac ex volutpat gravida. Duis vitae mollis ante. Duis a congue nunc. Aenean aliquet, sapien quis tincidunt tincidunt, odio eros consectetur lacus, vel finibus mauris tortor id velit. Donec tincidunt vitae purus eu interdum. Pellentesque scelerisque dui viverra ex ullamcorper volutpat. Vestibulum lacinia vitae arcu volutpat porta. Etiam et cursus nulla, id aliquet felis. Nam ultricies, urna id mattis pretium, velit erat viverra elit, eu maximus diam eros id nisi."), /*#__PURE__*/React.createElement("p", null, "Nullam eget nisl lectus. Pellentesque eu mauris ut tortor malesuada sagittis. Cras dictum cursus est non ultricies. Duis mollis non neque at commodo. Nunc feugiat justo et consequat aliquam. Ut consectetur libero eu erat feugiat finibus. Duis varius convallis quam eu sagittis. Maecenas ac est arcu. Suspendisse at enim eget nibh ultricies dictum. Etiam aliquet tellus vel felis malesuada laoreet.")), /*#__PURE__*/React.createElement(SidePanel.Sticky, {
    position: "bottom"
  }, /*#__PURE__*/React.createElement(SidePanel.Footer, {
    className: "u-align--right"
  }, /*#__PURE__*/React.createElement(Button, {
    appearance: "base",
    onClick: closePanel
  }, "Cancel"), /*#__PURE__*/React.createElement(ActionButton, {
    appearance: "positive"
  }, "Save changes")))));
};
export var Default = {
  args: {
    className: "",
    hasError: false,
    parentId: "l-application-default",
    pinned: false,
    loading: false,
    overlay: true,
    width: ""
  },
  render: StoryExample
};
export var Pinned = {
  args: {
    className: "",
    hasError: false,
    parentId: "l-application-pinned",
    pinned: true,
    loading: false,
    overlay: false,
    width: ""
  },
  render: StoryExample,
  parameters: {
    docs: {
      disable: true
    }
  }
};
export var Loading = {
  args: {
    className: "",
    hasError: false,
    parentId: "l-application-loading",
    pinned: false,
    loading: true,
    overlay: true,
    width: ""
  },
  render: StoryExample
};
export var Error = {
  args: {
    className: "",
    hasError: true,
    parentId: "l-application-error",
    pinned: false,
    loading: false,
    overlay: true,
    width: ""
  },
  render: StoryExample
};
export var Narrow = {
  args: {
    className: "",
    hasError: false,
    parentId: "l-application-narrow",
    pinned: false,
    loading: false,
    overlay: true,
    width: "narrow"
  },
  render: StoryExample
};
export var Wide = {
  args: {
    className: "",
    hasError: false,
    parentId: "l-application-wide",
    pinned: false,
    loading: false,
    overlay: true,
    width: "wide"
  },
  render: StoryExample
};
export var Animated = {
  args: {
    className: "",
    hasError: false,
    parentId: "l-application-default",
    pinned: false,
    loading: false,
    overlay: false,
    width: "",
    isAnimated: true
  },
  render: StoryExample
};
