// @canonical/react-components@4.2.0/dist/esm/components/CustomSelect/CustomSelect.js
var _excluded = ["value", "options", "onChange", "onSearch", "id", "name", "disabled", "success", "error", "help", "wrapperClassName", "toggleClassName", "dropdownClassName", "defaultToggleLabel", "searchable", "searchPlaceholder", "takeFocus", "header", "selectRef", "initialPosition"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import classNames from "classnames";
import React, { useEffect, useId, useImperativeHandle, useState } from "react";
import Field from "../Field";
import ContextualMenu from "../ContextualMenu";
import { useListener } from "../../hooks";
import CustomSelectDropdown, { getOptionText } from "./CustomSelectDropdown";
import "./CustomSelect.scss";
/**
 * This is a [React](https://reactjs.org/) component that extends from the Vanilla [Select](https://vanillaframework.io/docs/base/forms#select) element.
 *
 * The aim of this component is to provide a select component with customisable options and a dropdown menu, whilst maintaining accessibility and usability.
 */
var CustomSelect = _ref => {
  var {
      value,
      options,
      onChange,
      onSearch,
      id,
      name,
      disabled,
      success,
      error,
      help,
      wrapperClassName,
      toggleClassName,
      dropdownClassName,
      defaultToggleLabel = "Select an option",
      searchable = "auto",
      searchPlaceholder = "Search",
      takeFocus,
      header,
      selectRef,
      initialPosition = "left"
    } = _ref,
    fieldProps = _objectWithoutProperties(_ref, _excluded);
  var [isOpen, setIsOpen] = useState(false);
  var validationId = useId();
  var defaultSelectId = useId();
  var selectId = id || defaultSelectId;
  var helpId = useId();
  var hasError = !!error;

  // Close the dropdown when the browser tab is hidden
  var onBrowserTabHidden = () => {
    if (document.visibilityState === "hidden") {
      setIsOpen(false);
    }
  };
  useListener(window, onBrowserTabHidden, "visibilitychange");

  // Close the dropdown when the browser window loses focus
  useListener(window, () => setIsOpen(false), "blur");
  useImperativeHandle(selectRef, () => ({
    open: () => {
      var _document$getElementB;
      setIsOpen(true);
      (_document$getElementB = document.getElementById(selectId)) === null || _document$getElementB === void 0 || _document$getElementB.focus();
    },
    focus: () => {
      var _document$getElementB2;
      return (_document$getElementB2 = document.getElementById(selectId)) === null || _document$getElementB2 === void 0 ? void 0 : _document$getElementB2.focus();
    },
    close: setIsOpen.bind(null, false),
    isOpen: isOpen
  }), [isOpen, selectId]);
  useEffect(() => {
    if (takeFocus) {
      var toggleButton = document.getElementById(selectId);
      toggleButton === null || toggleButton === void 0 || toggleButton.focus();
    }
  }, [takeFocus, selectId]);
  var selectedOption = options.find(option => option.value === value);
  var toggleLabel = /*#__PURE__*/React.createElement("span", {
    className: "toggle-label u-truncate"
  }, selectedOption ? selectedOption.selectedLabel || getOptionText(selectedOption) : defaultToggleLabel);
  var handleSelect = value => {
    var _document$getElementB3;
    (_document$getElementB3 = document.getElementById(selectId)) === null || _document$getElementB3 === void 0 || _document$getElementB3.focus();
    setIsOpen(false);
    onChange(value);
  };
  return /*#__PURE__*/React.createElement(Field, _extends({}, fieldProps, {
    className: classNames("p-custom-select", wrapperClassName),
    error: error,
    forId: selectId,
    help: help,
    helpId: helpId,
    isSelect: true,
    success: success,
    validationId: validationId
  }), /*#__PURE__*/React.createElement(ContextualMenu, {
    "aria-describedby": [help ? helpId : null, success ? validationId : null].filter(Boolean).join(" "),
    "aria-errormessage": hasError ? validationId : undefined,
    "aria-invalid": hasError,
    focusFirstItemOnOpen: false,
    toggleClassName: classNames("p-custom-select__toggle", "p-form-validation__input", toggleClassName, {
      active: isOpen
    }),
    toggleLabel: toggleLabel,
    visible: isOpen,
    onToggleMenu: open => {
      // Handle syncing the state when toggling the menu from within the
      // contextual menu component e.g. when clicking outside.
      if (open !== isOpen) {
        setIsOpen(open);
      }
    },
    toggleProps: {
      id: selectId,
      disabled: disabled,
      // tabIndex is set to -1 when disabled to prevent keyboard navigation to the select toggle
      tabIndex: disabled ? -1 : 0
    },
    className: "p-custom-select__wrapper",
    dropdownClassName: dropdownClassName,
    style: {
      width: "100%"
    },
    autoAdjust: true,
    position: initialPosition
  }, close => /*#__PURE__*/React.createElement(CustomSelectDropdown, {
    searchable: searchable,
    searchPlaceholder: searchPlaceholder,
    onSearch: onSearch,
    name: name || "",
    options: options || [],
    onSelect: handleSelect,
    onClose: () => {
      var _document$getElementB4;
      // When pressing ESC to close the dropdown, we keep focus on the toggle button
      close();
      (_document$getElementB4 = document.getElementById(selectId)) === null || _document$getElementB4 === void 0 || _document$getElementB4.focus();
    },
    header: header,
    toggleId: selectId
  })));
};
export default CustomSelect;
