// @canonical/react-components@4.2.0/dist/esm/components/CustomSelect/CustomSelect.stories.js
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectDestructuringEmpty(t) { if (null == t) throw new TypeError("Cannot destructure " + t); }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import CustomSelect from "./CustomSelect";
import React, { useState } from "react";
var generateStandardOptions = num => Array(num).fill(null).map((_, i) => ({
  value: "option-".concat(i + 1),
  label: "Option ".concat(i + 1),
  text: "Option ".concat(i + 1),
  disabled: false
}));
var generateCustomOptions = () => {
  return [{
    value: "smile",
    label: /*#__PURE__*/React.createElement("div", null, "\uD83D\uDE00"),
    text: "Smile",
    disabled: false
  }, {
    value: "grin",
    label: /*#__PURE__*/React.createElement("div", null, "\uD83D\uDE01"),
    text: "Grin",
    disabled: false
  }, {
    value: "cry",
    label: /*#__PURE__*/React.createElement("div", null, "\uD83D\uDE2D"),
    text: "Cry",
    disabled: false
  }, {
    value: "angry",
    label: /*#__PURE__*/React.createElement("div", null, "\uD83D\uDE21"),
    text: "Angry",
    disabled: false
  }, {
    value: "sad",
    label: /*#__PURE__*/React.createElement("div", null, "\uD83D\uDE22"),
    text: "Sad",
    disabled: false
  }];
};
var generateCustomOptionsWithSelectedLabel = () => {
  var options = [{
    type: "ovn",
    name: "ovntest",
    config: {
      "security.acls": "foo,bar"
    }
  }, {
    type: "bridge",
    name: "lxdbr0",
    config: {}
  }, {
    type: "bridge",
    name: "microbr0",
    config: {}
  }, {
    type: "macvlan",
    name: "macvlantest",
    config: {}
  }].map(network => {
    var _network$config$secur;
    return {
      label: /*#__PURE__*/React.createElement("div", {
        className: "label",
        style: {
          display: "flex",
          gap: "5px"
        }
      }, /*#__PURE__*/React.createElement("span", {
        title: network.name,
        className: "network-option u-truncate",
        style: {
          width: "12rem"
        }
      }, network.name), /*#__PURE__*/React.createElement("span", {
        title: network.type,
        className: "network-option u-truncate",
        style: {
          width: "8rem"
        }
      }, network.type), /*#__PURE__*/React.createElement("span", {
        title: "network ACLs",
        className: "network-option u-truncate u-align--right",
        style: {
          paddingRight: "8px",
          width: "4rem"
        }
      }, ((_network$config$secur = network.config["security.acls"]) === null || _network$config$secur === void 0 ? void 0 : _network$config$secur.length) || "-")),
      value: network.name,
      text: "".concat(network.name, " - ").concat(network.type),
      disabled: false,
      selectedLabel: /*#__PURE__*/React.createElement("span", null, network.name, "\xA0", /*#__PURE__*/React.createElement("span", {
        className: "u-text--muted"
      }, "(", network.type, ")"))
    };
  });
  return options;
};
var getHeader = () => {
  return /*#__PURE__*/React.createElement("div", {
    className: "header",
    style: {
      backgroundColor: "$colors--theme--background-alt",
      display: "flex",
      gap: "$sph--small",
      padding: "$sph--x-small $sph--small",
      position: "sticky",
      top: 0
    }
  }, /*#__PURE__*/React.createElement("span", {
    className: "network-option u-no-margin--bottom",
    style: {
      color: "$colors--theme--text-default",
      width: "12rem"
    }
  }, "Name"), /*#__PURE__*/React.createElement("span", {
    className: "network-option u-no-margin--bottom",
    style: {
      color: "$colors--theme--text-default",
      width: "8rem"
    }
  }, "Type"), /*#__PURE__*/React.createElement("span", {
    className: "network-option u-no-margin--bottom",
    style: {
      color: "$colors--theme--text-default",
      width: "4rem"
    }
  }, "ACLs"));
};
var Template = _ref => {
  var props = _extends({}, (_objectDestructuringEmpty(_ref), _ref));
  var [selected, setSelected] = useState(props.value || "");
  return /*#__PURE__*/React.createElement(CustomSelect, _extends({}, props, {
    value: selected,
    onChange: value => setSelected(value)
  }));
};
var meta = {
  component: CustomSelect,
  render: Template,
  tags: ["autodocs"],
  args: {
    name: "customSelect",
    label: "Custom Select",
    defaultToggleLabel: "Select an option",
    searchable: "auto",
    searchPlaceholder: "Search",
    initialPosition: "left"
  },
  argTypes: {
    searchable: {
      options: ["auto", "always", "async", "never"],
      control: {
        type: "select"
      }
    },
    initialPosition: {
      options: ["left", "right"],
      control: {
        type: "select"
      }
    }
  }
};
export default meta;
/**
 * If `label` is of `string` type. You do not have to do anything extra to render it.
 */
export var StandardOptions = {
  args: {
    options: generateStandardOptions(10)
  }
};

/**
 * If `label` is of `ReactNode` type. You can render custom content.
 * In this case, the `text` property for each option is required and is used for display in the toggle, search and sort functionalities.
 */
export var CustomOptions = {
  args: {
    options: generateCustomOptions()
  }
};

/**
 * If `label` is of `ReactNode` type. You can render custom content.
 * In this case, the `selectedLabel` for each option is provided and will be displayed in the toggle instead of `text`
 * The `text` property for each option is still required and is used for search and sort functionalities.
 */
export var CustomOptionsAndSelectedLabel = {
  args: {
    options: generateCustomOptionsWithSelectedLabel(),
    header: getHeader(),
    dropdownClassName: "network-select-dropdown"
  }
};

/**
 * For each option, if `disabled` is set to `true`, the option will be disabled.
 */
export var DisabledOptions = {
  args: {
    options: generateStandardOptions(5).map((option, i) => _objectSpread(_objectSpread({}, option), {}, {
      disabled: i % 2 === 0
    }))
  }
};

/**
 * Search is enabled by default when there are 5 or more options.
 */
export var AutoSearchable = {
  args: {
    options: generateStandardOptions(5),
    searchable: "auto"
  }
};

/**
 * Search can be enabled manually by setting `searchable` to `always`.
 */
export var ManualSearchable = {
  args: {
    options: generateStandardOptions(4),
    searchable: "always"
  }
};

/**
 * Search can be enabled manually by setting `searchable` to `async`.
 * This will always show the search input regardless of the number of options.
 */
export var AsyncSearchable = {
  args: {
    options: generateStandardOptions(0),
    searchable: "async"
  }
};
