// @canonical/react-components@4.2.0/dist/esm/components/Spinner/Spinner.js
import _pt from "prop-types";
var _excluded = ["className", "text", "isLight", "ariaLive", "role", "isMainComponent"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import React from "react";
import classNames from "classnames";
import CustomLayout from "../CustomLayout";
/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Spin](https://docs.vanillaframework.io/settings/animation-settings/#spin) animation.
 */
var Spinner = _ref => {
  var {
      className,
      text,
      isLight = false,
      ariaLive = "polite",
      role = "alert",
      isMainComponent = false
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  var loader = /*#__PURE__*/React.createElement("span", _extends({}, props, {
    className: classNames(className, "p-text--default"),
    role: role,
    "aria-live": ariaLive
  }), /*#__PURE__*/React.createElement("i", {
    className: classNames("p-icon--spinner", "u-animation--spin", {
      "is-light": isLight
    })
  }, text ? "" : "Loading"), text && /*#__PURE__*/React.createElement(React.Fragment, null, "\u2002", /*#__PURE__*/React.createElement("span", null, text)));
  return isMainComponent ? /*#__PURE__*/React.createElement(CustomLayout, null, loader) : loader;
};
Spinner.propTypes = {
  isLight: _pt.bool,
  text: _pt.string,
  role: _pt.string,
  ariaLive: _pt.oneOf(["assertive", "off", "polite"]),
  isMainComponent: _pt.bool
};
export default Spinner;
