// @canonical/react-components@4.2.0/dist/components/SidePanel/SidePanel.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Wide = exports.Pinned = exports.Narrow = exports.Loading = exports.Error = exports.Default = exports.Animated = void 0;
var _react = _interopRequireWildcard(require("react"));
var _SidePanel = _interopRequireDefault(require("./SidePanel"));
var _Button = _interopRequireDefault(require("../Button"));
var _Application = _interopRequireDefault(require("../ApplicationLayout/Application"));
var _AppMain = _interopRequireDefault(require("../ApplicationLayout/AppMain"));
var _index = require("../../index");
var _Icon = _interopRequireDefault(require("../Icon"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const meta = {
  component: _SidePanel.default,
  tags: ["autodocs"],
  decorators: [Story => {
    const ref = (0, _react.useRef)(null);
    (0, _react.useEffect)(() => {
      const el = ref.current;
      if (el) {
        var _el$parentElement, _wrapper$parentElemen;
        const wrapper = el === null || el === void 0 || (_el$parentElement = el.parentElement) === null || _el$parentElement === void 0 || (_el$parentElement = _el$parentElement.parentElement) === null || _el$parentElement === void 0 ? void 0 : _el$parentElement.parentElement;
        const parent = wrapper === null || wrapper === void 0 || (_wrapper$parentElemen = wrapper.parentElement) === null || _wrapper$parentElemen === void 0 || (_wrapper$parentElemen = _wrapper$parentElemen.parentElement) === null || _wrapper$parentElemen === void 0 ? void 0 : _wrapper$parentElemen.parentElement;
        if (wrapper) {
          wrapper.style.setProperty("border", "0", "important");
        }
        if (parent) {
          parent.style.margin = "0";
          parent.style.padding = "0";
        }
      }
    }, []);
    document.body.style.padding = "0";
    return /*#__PURE__*/_react.default.createElement("div", {
      ref: ref
    }, /*#__PURE__*/_react.default.createElement(Story, null));
  }]
};
var _default = exports.default = meta;
const StoryExample = args => {
  const [isOpen, setIsOpen] = (0, _react.useState)(false);

  // Multiple stories get rendered on the docs page, ensure the ids are not colliding
  const id = (0, _react.useId)();
  const parentId = args.parentId + id;
  const openPanel = () => {
    setIsOpen(true);
  };
  const closePanel = () => {
    setIsOpen(false);
  };

  // Delay opening the panel, so the l-application container is ready to receive it
  (0, _react.useEffect)(() => void setTimeout(openPanel, 100), []);
  return /*#__PURE__*/_react.default.createElement(_Application.default, {
    id: parentId
  }, /*#__PURE__*/_react.default.createElement(_AppMain.default, null, /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: openPanel,
    style: {
      margin: "1rem"
    }
  }, "Open side panel"), /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: closePanel,
    style: {
      margin: "1rem"
    }
  }, "Close side panel")), /*#__PURE__*/_react.default.createElement(_SidePanel.default, {
    overlay: args.overlay,
    loading: args.loading,
    hasError: args.hasError,
    parentId: parentId,
    pinned: args.pinned,
    width: args.width,
    className: "u-no-padding--top u-no-padding--bottom",
    isOpen: isOpen,
    isAnimated: args.isAnimated
  }, /*#__PURE__*/_react.default.createElement(_SidePanel.default.Sticky, null, /*#__PURE__*/_react.default.createElement(_SidePanel.default.Header, null, /*#__PURE__*/_react.default.createElement(_SidePanel.default.HeaderTitle, null, "Edit panel"), /*#__PURE__*/_react.default.createElement(_SidePanel.default.HeaderControls, null, /*#__PURE__*/_react.default.createElement(_Button.default, {
    appearance: "base",
    className: "u-no-margin--bottom",
    hasIcon: true,
    onClick: closePanel,
    "aria-label": "Close"
  }, /*#__PURE__*/_react.default.createElement(_Icon.default, {
    name: "close"
  }))))), /*#__PURE__*/_react.default.createElement(_SidePanel.default.Content, {
    className: "u-no-padding"
  }, /*#__PURE__*/_react.default.createElement("p", null, "Here be dragons!"), /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras varius mi eu pretium vulputate. Nunc commodo sit amet nibh quis rhoncus. Aliquam rhoncus porttitor semper. Aenean faucibus consectetur neque in sodales. Sed cursus mauris id ex sollicitudin sodales. Quisque molestie rutrum odio, ornare pharetra ligula. Pellentesque ornare tristique feugiat. In a augue neque. Aenean eget arcu quis lacus tempus posuere in sit amet dui. Suspendisse faucibus sapien nisl, nec laoreet sem convallis nec."), /*#__PURE__*/_react.default.createElement("p", null, "Vestibulum sed placerat lorem. Nam luctus ex id nisi luctus, id vestibulum sem bibendum. Vivamus turpis sem, pellentesque fermentum malesuada eu, faucibus porta libero. Duis eget venenatis odio. Etiam sed volutpat magna, non tempus erat. Nunc id tortor ac quam consectetur dapibus ac ut tellus. Pellentesque ut tellus venenatis elit vehicula condimentum eget quis lorem. Ut non consectetur est, a fringilla ipsum. Nunc vitae ligula ipsum. Etiam suscipit, libero ut lacinia viverra, nunc urna consequat ex, vel eleifend eros mauris vitae ipsum. Pellentesque sed dictum augue. Ut sit amet ullamcorper mauris. Nunc congue orci mollis purus sodales facilisis ac ut arcu. Maecenas feugiat sapien ac massa mollis sodales. Donec vitae turpis eu nisi laoreet pulvinar quis at nisl. Integer volutpat, metus eget elementum dictum, lacus sapien viverra felis, consequat fermentum nisl mi ac dui."), /*#__PURE__*/_react.default.createElement("p", null, "Nullam nulla turpis, dignissim vel dapibus ut, volutpat ac dui. Donec vel elementum lacus. Mauris maximus nec felis at faucibus. Nunc faucibus gravida velit, id blandit lectus tincidunt ac. Vestibulum orci diam, elementum in congue eu, placerat id risus. Sed tempor tempus tellus, vitae iaculis turpis fringilla nec. Phasellus imperdiet facilisis velit, sit amet lobortis odio dignissim ut."), /*#__PURE__*/_react.default.createElement("p", null, "Nam placerat urna vitae ligula hendrerit, ac tincidunt lorem maximus. Mauris eu odio nisi. Nulla facilisi. Sed egestas elit sed velit rutrum, sit amet bibendum metus hendrerit. Pellentesque luctus placerat tellus, eu bibendum justo. Cras eget leo ac ex volutpat gravida. Duis vitae mollis ante. Duis a congue nunc. Aenean aliquet, sapien quis tincidunt tincidunt, odio eros consectetur lacus, vel finibus mauris tortor id velit. Donec tincidunt vitae purus eu interdum. Pellentesque scelerisque dui viverra ex ullamcorper volutpat. Vestibulum lacinia vitae arcu volutpat porta. Etiam et cursus nulla, id aliquet felis. Nam ultricies, urna id mattis pretium, velit erat viverra elit, eu maximus diam eros id nisi."), /*#__PURE__*/_react.default.createElement("p", null, "Nullam eget nisl lectus. Pellentesque eu mauris ut tortor malesuada sagittis. Cras dictum cursus est non ultricies. Duis mollis non neque at commodo. Nunc feugiat justo et consequat aliquam. Ut consectetur libero eu erat feugiat finibus. Duis varius convallis quam eu sagittis. Maecenas ac est arcu. Suspendisse at enim eget nibh ultricies dictum. Etiam aliquet tellus vel felis malesuada laoreet.")), /*#__PURE__*/_react.default.createElement(_SidePanel.default.Sticky, {
    position: "bottom"
  }, /*#__PURE__*/_react.default.createElement(_SidePanel.default.Footer, {
    className: "u-align--right"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    appearance: "base",
    onClick: closePanel
  }, "Cancel"), /*#__PURE__*/_react.default.createElement(_index.ActionButton, {
    appearance: "positive"
  }, "Save changes")))));
};
const Default = exports.Default = {
  args: {
    className: "",
    hasError: false,
    parentId: "l-application-default",
    pinned: false,
    loading: false,
    overlay: true,
    width: ""
  },
  render: StoryExample
};
const Pinned = exports.Pinned = {
  args: {
    className: "",
    hasError: false,
    parentId: "l-application-pinned",
    pinned: true,
    loading: false,
    overlay: false,
    width: ""
  },
  render: StoryExample,
  parameters: {
    docs: {
      disable: true
    }
  }
};
const Loading = exports.Loading = {
  args: {
    className: "",
    hasError: false,
    parentId: "l-application-loading",
    pinned: false,
    loading: true,
    overlay: true,
    width: ""
  },
  render: StoryExample
};
const Error = exports.Error = {
  args: {
    className: "",
    hasError: true,
    parentId: "l-application-error",
    pinned: false,
    loading: false,
    overlay: true,
    width: ""
  },
  render: StoryExample
};
const Narrow = exports.Narrow = {
  args: {
    className: "",
    hasError: false,
    parentId: "l-application-narrow",
    pinned: false,
    loading: false,
    overlay: true,
    width: "narrow"
  },
  render: StoryExample
};
const Wide = exports.Wide = {
  args: {
    className: "",
    hasError: false,
    parentId: "l-application-wide",
    pinned: false,
    loading: false,
    overlay: true,
    width: "wide"
  },
  render: StoryExample
};
const Animated = exports.Animated = {
  args: {
    className: "",
    hasError: false,
    parentId: "l-application-default",
    pinned: false,
    loading: false,
    overlay: false,
    width: "",
    isAnimated: true
  },
  render: StoryExample
};
