// @canonical/react-components@4.2.0/dist/components/SidePanel/SidePanel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));
var _reactDom = require("react-dom");
var _Content = _interopRequireDefault(require("./common/Content"));
var _Footer = _interopRequireDefault(require("./common/Footer"));
var _Header = _interopRequireDefault(require("./common/Header"));
var _HeaderControls = _interopRequireDefault(require("./common/HeaderControls"));
var _HeaderTitle = _interopRequireDefault(require("./common/HeaderTitle"));
var _Sticky = _interopRequireDefault(require("./common/Sticky"));
var _AppAside = _interopRequireDefault(require("../ApplicationLayout/AppAside"));
var _Spinner = _interopRequireDefault(require("../Spinner"));
require("./SidePanel.scss");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
/**
 * The props for the SidePanelComponent component.
 */

/**
This is a [React](https://reactjs.org/) component for the Vanilla application layout [Side Panel](https://vanillaframework.io/docs/layouts/application#panels).

The sidepanel component should be used to show additional information relating to the main content on the page. It is using a composable structure with building blocks:

* **SidePanel.Header:** Defines the header of the side panel, which can include a title and controls.

* **SidePanel.HeaderTitle:** To show the title of the side panel as heading.

* **SidePanel.HeaderControls:** To show controls in the header, such as buttons or icons for actions like closing the panel.

* **SidePanel.Sticky:** Can be wrapped around the header or footer to make them sticky when scrolling. The scrollbar will use the full area of the side panel, not just the content area. To limit the scrollbar to the content area, use the `ScrollableContainer` component instead of this one.

* **SidePanel.Content:** To show the main content of the side panel.

* **SidePanel.Footer:** To show additional information or actions at the bottom of the side panel.

## Example

Usage of the composable structure:

```jsx
<SidePanel className="u-no-padding--top u-no-padding--bottom">
  <SidePanel.Sticky>
    <SidePanel.Header>
      <SidePanel.HeaderTitle>Edit panel</SidePanel.HeaderTitle>
    </SidePanel.Header>
  </SidePanel.Sticky>
  <SidePanel.Content className="u-no-padding">
    <p>Here be dragons!</p>
  </SidePanel.Content>
  <SidePanel.Sticky position="bottom">
    <SidePanel.Footer className="u-align--right">
      <Button appearance="base">Cancel</Button>
      <ActionButton appearance="positive">Save changes</ActionButton>
    </SidePanel.Footer>
  </SidePanel.Sticky>
</SidePanel>
```
*/
const SidePanelComponent = _ref => {
  let {
    children,
    className,
    hasError = false,
    loading = false,
    overlay,
    pinned,
    width = "",
    parentId = "l-application",
    isOpen = true,
    isAnimated
  } = _ref;
  const [hiding, setHiding] = (0, _react.useState)(true);
  const [previousIsOpen, setPreviousIsOpen] = (0, _react.useState)(false);
  const container = document.getElementById(parentId) || document.body;
  if (isOpen !== previousIsOpen) {
    setPreviousIsOpen(isOpen);
    setHiding(true);
  }

  // Pinned side panels don't get animation in Vanilla
  const isAnimatedFinal = isAnimated && !pinned;
  if (!isOpen && !(isAnimatedFinal && hiding)) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/(0, _reactDom.createPortal)(/*#__PURE__*/_react.default.createElement(_AppAside.default, {
    className: (0, _classnames.default)("side-panel", className, {
      "is-overlay": overlay,
      "slide-in": isAnimatedFinal
    }),
    collapsed: !isOpen,
    "aria-label": "Side panel",
    pinned: pinned,
    narrow: width === "narrow",
    wide: width === "wide",
    onTransitionEnd: () => setHiding(false)
  }, loading ? /*#__PURE__*/_react.default.createElement("div", {
    className: "loading"
  }, /*#__PURE__*/_react.default.createElement(_Spinner.default, null)) : hasError ? /*#__PURE__*/_react.default.createElement("div", {
    className: "error"
  }, "Loading failed") : children), container));
};
const SidePanel = SidePanelComponent;
SidePanel.Header = _Header.default;
SidePanel.HeaderTitle = _HeaderTitle.default;
SidePanel.HeaderControls = _HeaderControls.default;
SidePanel.Sticky = _Sticky.default;
SidePanel.Content = _Content.default;
SidePanel.Footer = _Footer.default;
var _default = exports.default = SidePanel;
