// @canonical/react-components@4.2.0/dist/esm/components/ConfirmationModal/ConfirmationModal.js
var _excluded = ["cancelButtonLabel", "cancelButtonProps", "children", "confirmButtonAppearance", "confirmButtonLabel", "confirmExtra", "onConfirm", "confirmButtonLoading", "confirmButtonDisabled", "confirmButtonProps", "renderInPortal", "portalRenderer"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import React from "react";
import Button from "../Button";
import Modal from "../Modal";
import ActionButton from "../ActionButton";
import { usePortal } from "../../external";
var InternalPortalRenderer = _ref => {
  var {
    children
  } = _ref;
  var {
    Portal
  } = usePortal();
  return /*#__PURE__*/React.createElement(Portal, null, children);
};

/**
 * `ConfirmationModal` is a specialised version of the [Modal](?path=/docs/modal--default-story) component to prompt a confirmation from the user before executing an action.
 */
export var ConfirmationModal = _ref2 => {
  var _cancelButtonProps$ty;
  var {
      cancelButtonLabel = "Cancel",
      cancelButtonProps,
      children,
      confirmButtonAppearance = "negative",
      confirmButtonLabel,
      confirmExtra,
      onConfirm,
      confirmButtonLoading,
      confirmButtonDisabled,
      confirmButtonProps,
      renderInPortal = false,
      portalRenderer: PortalRenderer
    } = _ref2,
    props = _objectWithoutProperties(_ref2, _excluded);
  var handleClick = action => event => {
    if (!props.shouldPropagateClickEvent) {
      event.stopPropagation();
    }
    if (action) {
      action(event);
    }
  };
  var ModalElement = /*#__PURE__*/React.createElement(Modal, _extends({
    buttonRow: /*#__PURE__*/React.createElement(React.Fragment, null, confirmExtra, /*#__PURE__*/React.createElement(Button, _extends({}, cancelButtonProps, {
      type: (_cancelButtonProps$ty = cancelButtonProps === null || cancelButtonProps === void 0 ? void 0 : cancelButtonProps.type) !== null && _cancelButtonProps$ty !== void 0 ? _cancelButtonProps$ty : "button",
      className: "u-no-margin--bottom",
      onClick: handleClick(props.close)
    }), cancelButtonLabel), /*#__PURE__*/React.createElement(ActionButton, _extends({}, confirmButtonProps, {
      appearance: confirmButtonAppearance,
      className: "u-no-margin--bottom",
      onClick: handleClick(onConfirm),
      loading: confirmButtonLoading,
      disabled: confirmButtonDisabled
    }), confirmButtonLabel))
  }, props), children);
  if (PortalRenderer) {
    return /*#__PURE__*/React.createElement(PortalRenderer, null, ModalElement);
  } else if (renderInPortal) {
    return /*#__PURE__*/React.createElement(InternalPortalRenderer, null, ModalElement);
  } else {
    return ModalElement;
  }
};
export default ConfirmationModal;
