// @canonical/react-components@4.2.0/dist/esm/components/ConfirmationModal/ConfirmationModal.stories.js
import { useState } from "react";
import React from "react";
import ConfirmationModal from "./ConfirmationModal";
import Input from "../Input";
var doNothing = () => {};
var meta = {
  component: ConfirmationModal,
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [modalOpen, setModalOpen] = useState(false);
    var closeHandler = () => setModalOpen(false);
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("button", {
      onClick: () => setModalOpen(true)
    }, "Open confirmation modal"), modalOpen ? /*#__PURE__*/React.createElement(ConfirmationModal, {
      title: "Confirm delete",
      confirmButtonLabel: "Delete",
      onConfirm: doNothing,
      close: closeHandler
    }, /*#__PURE__*/React.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/React.createElement("br", null), "You cannot undo this action.")) : null);
  },
  name: "Default"
};

/**
 * The title is optional, and the content is fully customisable.
 */
export var NoTitle = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [modalOpen, setModalOpen] = useState(false);
    var closeHandler = () => setModalOpen(false);
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("button", {
      onClick: () => setModalOpen(true)
    }, "Run"), modalOpen ? /*#__PURE__*/React.createElement(ConfirmationModal, {
      close: closeHandler,
      confirmButtonAppearance: "positive",
      confirmButtonLabel: "Confirm",
      onConfirm: doNothing
    }, /*#__PURE__*/React.createElement("h4", {
      style: {
        margin: "1rem 0"
      }
    }, "Run list-backups?"), /*#__PURE__*/React.createElement("div", {
      style: {
        width: "480px",
        marginBottom: "1rem"
      }
    }, /*#__PURE__*/React.createElement("div", {
      className: "u-text--muted"
    }, "UNIT COUNT"), /*#__PURE__*/React.createElement("div", null, "1")), /*#__PURE__*/React.createElement("div", {
      style: {
        marginBottom: "1.5rem"
      }
    }, /*#__PURE__*/React.createElement("div", {
      className: "u-text--muted"
    }, "UNIT NAME"), /*#__PURE__*/React.createElement("div", null, "mysql/0"))) : null);
  },
  name: "No title"
};

/**
 * The appearance of the confirm button can be any of the `Button`'s appearance values.
 */
export var Positive = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [modalOpen, setModalOpen] = useState(false);
    var closeHandler = () => setModalOpen(false);
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("button", {
      onClick: () => setModalOpen(true)
    }, "Apply changes"), modalOpen ? /*#__PURE__*/React.createElement(ConfirmationModal, {
      close: closeHandler,
      confirmButtonAppearance: "positive",
      confirmButtonLabel: "Yes, apply changes",
      onConfirm: doNothing
    }, /*#__PURE__*/React.createElement("h4", {
      style: {
        margin: "1rem 0"
      }
    }, "Are you sure you wish to apply these changes?"), /*#__PURE__*/React.createElement("p", null, "You have edited the following values to the mysql configuration:"), /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("h5", {
      className: "u-no-margin--bottom"
    }, "cluster-name"), /*#__PURE__*/React.createElement("pre", {
      className: "u-no-padding",
      style: {
        backgroundColor: "transparent"
      }
    }, "test")), /*#__PURE__*/React.createElement("div", {
      className: "p-text--small",
      style: {
        width: "480px"
      }
    }, "You can revert back to the applications default settings by clicking the \u201CReset all values\u201D button; or reset each edited field by clicking \u201CUse default\u201D.")) : null);
  },
  name: "Positive"
};

/**
 * Extra elements can be added to the button row.
 */
export var Extra = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    var [modalOpen, setModalOpen] = useState(false);
    var closeHandler = () => setModalOpen(false);
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("button", {
      onClick: () => setModalOpen(true)
    }, "Stop"), modalOpen ? /*#__PURE__*/React.createElement(ConfirmationModal, {
      confirmExtra: /*#__PURE__*/React.createElement("span", {
        className: "u-float-left"
      }, /*#__PURE__*/React.createElement(Input, {
        type: "checkbox",
        id: "extraCheckbox",
        label: "Force stop",
        tabIndex: -1
      })),
      title: "Confirm stop",
      confirmButtonLabel: "Stop",
      onConfirm: doNothing,
      close: closeHandler
    }, /*#__PURE__*/React.createElement("p", null, "This will stop instance", /*#__PURE__*/React.createElement("b", null, "whimsical-mouflon"), ".")) : null);
  },
  name: "Extra"
};
