// @canonical/react-components@4.2.0/dist/components/TablePagination/TablePagination.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.RenderBelow = exports.RenderAbove = exports.Default = exports.CustomPageLimit = exports.CustomDisplayTitle = exports.ControlsWithUnknownEntries = exports.ControlsWithPartiallyKnownEntries = exports.ControlsOnly = void 0;
var _react = _interopRequireDefault(require("react"));
var _TablePagination = _interopRequireDefault(require("./TablePagination"));
var _MainTable = _interopRequireDefault(require("../MainTable"));
var _TablePaginationControls = _interopRequireDefault(require("./TablePaginationControls"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _TablePagination.default,
  tags: ["autodocs"],
  argTypes: {
    totalItems: {
      if: {
        arg: "externallyControlled",
        truthy: true
      }
    },
    currentPage: {
      if: {
        arg: "externallyControlled",
        truthy: true
      }
    },
    pageSize: {
      if: {
        arg: "externallyControlled",
        truthy: true
      }
    },
    onPageChange: {
      if: {
        arg: "externallyControlled",
        truthy: true
      }
    },
    onPageSizeChange: {
      if: {
        arg: "externallyControlled",
        truthy: true
      }
    }
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default",
  args: {
    data: [{
      id: "row-1"
    }, {
      id: "row-2"
    }, {
      id: "row-3"
    }, {
      id: "row-4"
    }, {
      id: "row-5"
    }]
  }
};
const CustomPageLimit = exports.CustomPageLimit = {
  name: "CustomPageLimit",
  args: {
    data: [{
      id: "row-1"
    }, {
      id: "row-2"
    }, {
      id: "row-3"
    }, {
      id: "row-4"
    }, {
      id: "row-5"
    }],
    pageLimits: [1, 2, 3]
  }
};
const CustomDisplayTitle = exports.CustomDisplayTitle = {
  name: "CustomDisplayTitle",
  args: {
    data: [{
      id: "row-1"
    }, {
      id: "row-2"
    }, {
      id: "row-3"
    }, {
      id: "row-4"
    }, {
      id: "row-5"
    }],
    description: /*#__PURE__*/_react.default.createElement("b", null, "Hello there")
  }
};
const RenderAbove = exports.RenderAbove = {
  render: () => {
    const data = [{
      columns: [{
        content: "Ready",
        role: "rowheader"
      }, {
        content: 1,
        className: "u-align--right"
      }, {
        content: "1 GiB",
        className: "u-align--right"
      }, {
        content: 2,
        className: "u-align--right"
      }, {
        content: 42,
        className: "u-align--right"
      }],
      sortData: {
        status: "ready",
        cores: 2,
        ram: 1,
        disks: 2
      }
    }, {
      columns: [{
        content: "Idle",
        role: "rowheader"
      }, {
        content: 1,
        className: "u-align--right"
      }, {
        content: "1 GiB",
        className: "u-align--right"
      }, {
        content: 2,
        className: "u-align--right"
      }, {
        content: 23,
        className: "u-align--right"
      }],
      sortData: {
        status: "idle",
        cores: 1,
        ram: 1,
        disks: 2
      }
    }, {
      columns: [{
        content: "Waiting",
        role: "rowheader"
      }, {
        content: 8,
        className: "u-align--right"
      }, {
        content: "3.9 GiB",
        className: "u-align--right"
      }, {
        content: 3,
        className: "u-align--right"
      }, {
        content: 0,
        className: "u-align--right"
      }],
      sortData: {
        status: "waiting",
        cores: 8,
        ram: 3.9,
        disks: 3
      }
    }];
    const headers = [{
      content: "Status",
      sortKey: "status"
    }, {
      content: "Cores",
      sortKey: "cores",
      className: "u-align--right"
    }, {
      content: "RAM",
      sortKey: "ram",
      className: "u-align--right"
    }, {
      content: "Disks",
      sortKey: "disks",
      className: "u-align--right"
    }, {
      content: "Networks",
      className: "u-align--right"
    }];
    return /*#__PURE__*/_react.default.createElement(_TablePagination.default, {
      data: data,
      pageLimits: [1, 2, 3]
    }, /*#__PURE__*/_react.default.createElement(_MainTable.default, {
      headers: headers,
      rows: data,
      sortable: true
    }));
  },
  name: "RenderAbove"
};
const RenderBelow = exports.RenderBelow = {
  render: () => {
    const data = [{
      columns: [{
        content: "Ready",
        role: "rowheader"
      }, {
        content: 1,
        className: "u-align--right"
      }, {
        content: "1 GiB",
        className: "u-align--right"
      }, {
        content: 2,
        className: "u-align--right"
      }, {
        content: 42,
        className: "u-align--right"
      }],
      sortData: {
        status: "ready",
        cores: 2,
        ram: 1,
        disks: 2
      }
    }, {
      columns: [{
        content: "Idle",
        role: "rowheader"
      }, {
        content: 1,
        className: "u-align--right"
      }, {
        content: "1 GiB",
        className: "u-align--right"
      }, {
        content: 2,
        className: "u-align--right"
      }, {
        content: 23,
        className: "u-align--right"
      }],
      sortData: {
        status: "idle",
        cores: 1,
        ram: 1,
        disks: 2
      }
    }, {
      columns: [{
        content: "Waiting",
        role: "rowheader"
      }, {
        content: 8,
        className: "u-align--right"
      }, {
        content: "3.9 GiB",
        className: "u-align--right"
      }, {
        content: 3,
        className: "u-align--right"
      }, {
        content: 0,
        className: "u-align--right"
      }],
      sortData: {
        status: "waiting",
        cores: 8,
        ram: 3.9,
        disks: 3
      }
    }];
    const headers = [{
      content: "Status",
      sortKey: "status"
    }, {
      content: "Cores",
      sortKey: "cores",
      className: "u-align--right"
    }, {
      content: "RAM",
      sortKey: "ram",
      className: "u-align--right"
    }, {
      content: "Disks",
      sortKey: "disks",
      className: "u-align--right"
    }, {
      content: "Networks",
      className: "u-align--right"
    }];
    return /*#__PURE__*/_react.default.createElement(_TablePagination.default, {
      data: data,
      pageLimits: [1, 2, 3],
      position: "below"
    }, /*#__PURE__*/_react.default.createElement(_MainTable.default, {
      headers: headers,
      rows: data,
      sortable: true
    }));
  },
  name: "RenderBelow"
};

/** The table pagination controls can be used without wrapping MainTable by
 * using the `TablePaginationControls` component.
 */
const ControlsOnly = exports.ControlsOnly = {
  render: () => {
    return /*#__PURE__*/_react.default.createElement(_TablePaginationControls.default, {
      currentPage: 1,
      itemName: "row",
      nextButtonProps: {
        disabled: false
      },
      onInputPageChange: console.log,
      onNextPage: console.log,
      onPageSizeChange: console.log,
      onPreviousPage: console.log,
      pageLimits: [10, 25, 50],
      pageSize: 20,
      previousButtonProps: {
        disabled: false
      },
      showPageInput: true,
      totalItems: 100,
      visibleCount: 10
    });
  }
};

/** The `TablePaginationControls` component can be used when the total
 *  number of entries is unknown.
 */
const ControlsWithUnknownEntries = exports.ControlsWithUnknownEntries = {
  render: () => {
    return /*#__PURE__*/_react.default.createElement(_TablePaginationControls.default, {
      currentPage: 1,
      itemName: "row",
      nextButtonProps: {
        disabled: false
      },
      onInputPageChange: console.log,
      onNextPage: console.log,
      onPageSizeChange: console.log,
      onPreviousPage: console.log,
      pageLimits: [10, 25, 50],
      pageSize: 20,
      previousButtonProps: {
        disabled: false
      },
      showPageInput: true,
      visibleCount: 10
    });
  }
};

/** The `TablePaginationControls` component can be used when it is known
 *  that there are more entries than the amount displayed on the current page.
 */
const ControlsWithPartiallyKnownEntries = exports.ControlsWithPartiallyKnownEntries = {
  render: () => {
    return /*#__PURE__*/_react.default.createElement(_TablePaginationControls.default, {
      currentPage: 3,
      itemName: "row",
      nextButtonProps: {
        disabled: false
      },
      onInputPageChange: console.log,
      onNextPage: console.log,
      onPageSizeChange: console.log,
      onPreviousPage: console.log,
      pageLimits: [10, 25, 50],
      pageSize: 20,
      previousButtonProps: {
        disabled: false
      },
      showPageInput: true,
      visibleCount: 10
    });
  }
};
