// @canonical/react-components@4.2.0/dist/components/ContextualMenu/ContextualMenu.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Toggle = exports.OverflowingContainer = exports.InsideModal = exports.Default = exports.ChildFunction = exports.ChildElement = void 0;
var _react = _interopRequireDefault(require("react"));
var _Button = _interopRequireDefault(require("../Button"));
var _ContextualMenu = _interopRequireDefault(require("./ContextualMenu"));
var _Modal = _interopRequireDefault(require("../Modal"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const ScrollTemplate = args => /*#__PURE__*/_react.default.createElement("div", {
  style: {
    maxWidth: "30rem",
    maxHeight: "10rem",
    overflow: "auto",
    padding: "1rem"
  }
}, /*#__PURE__*/_react.default.createElement(_ContextualMenu.default, args), Array(3).fill(/*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Consequuntur cum dicta beatae nostrum eligendi similique earum, dolorem fuga quis, sequi voluptates architecto ipsa dolorum eaque rem expedita inventore voluptas odit aspernatur alias molestias facere.")), /*#__PURE__*/_react.default.createElement(_ContextualMenu.default, args));
const Template = args => /*#__PURE__*/_react.default.createElement("div", {
  className: "u-align--center"
}, /*#__PURE__*/_react.default.createElement(_ContextualMenu.default, args));
const meta = {
  component: _ContextualMenu.default,
  render: Template,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: {
        disable: true
      }
    },
    links: {
      control: {
        disable: true
      }
    }
  }
};
var _default = exports.default = meta;
/**
 * The contextual menu will provide a visual wrapper to any provided children. Visibility can be toggled via the `visible` prop.
 */
const Default = exports.Default = {
  name: "Default",
  args: {
    children: /*#__PURE__*/_react.default.createElement("span", {
      style: {
        padding: "1rem"
      }
    }, "This is a menu."),
    closeOnOutsideClick: false,
    constrainPanelWidth: false,
    position: "left",
    visible: true
  }
};
const Toggle = exports.Toggle = {
  name: "Toggle",
  args: {
    links: [{
      children: "Link 1",
      onClick: () => {}
    }, {
      children: "Link 2",
      onClick: () => {}
    }],
    hasToggleIcon: true,
    position: "right",
    toggleLabel: "Click me!"
  }
};
const OverflowingContainer = exports.OverflowingContainer = {
  render: ScrollTemplate.bind({}),
  name: "Overflowing container",
  args: {
    links: [{
      children: "Link 1",
      onClick: () => {}
    }, {
      children: "Long Link 2",
      onClick: () => {}
    }],
    hasToggleIcon: true,
    position: "right",
    toggleLabel: "Click me!"
  }
};
const ChildFunction = exports.ChildFunction = {
  name: "Child function",
  args: {
    children: close => /*#__PURE__*/_react.default.createElement(_Button.default, {
      onClick: close
    }, "child function"),
    hasToggleIcon: true,
    position: "right",
    toggleLabel: "Click me!"
  }
};
const ChildElement = exports.ChildElement = {
  name: "Child element",
  args: {
    children: /*#__PURE__*/_react.default.createElement(_Button.default, null, "child element"),
    hasToggleIcon: true,
    position: "right",
    toggleLabel: "Click me!"
  }
};
const InsideModal = exports.InsideModal = {
  name: "Inside modal",
  args: {
    links: Array.from({
      length: 5
    }, (_, index) => ({
      children: "Link ".concat(index + 1),
      onClick: () => {}
    })),
    hasToggleIcon: true,
    position: "right",
    toggleLabel: "Click me!"
  },
  render: args => /*#__PURE__*/_react.default.createElement(_Modal.default, {
    title: "Contextual Menu inside Modal"
  }, /*#__PURE__*/_react.default.createElement(Template, args))
};
