// @canonical/react-components@4.2.0/dist/components/ContextualMenu/ContextualMenu.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Label = void 0;
var _classnames = _interopRequireDefault(require("classnames"));
var _external = require("../../external");
var _hooks = require("../../hooks");
var _react = _interopRequireWildcard(require("react"));
var _Button = _interopRequireDefault(require("../Button"));
var _ContextualMenuDropdown = _interopRequireDefault(require("./ContextualMenuDropdown"));
const _excluded = ["autoAdjust", "children", "className", "closeOnEsc", "closeOnOutsideClick", "constrainPanelWidth", "dropdownClassName", "dropdownProps", "hasToggleIcon", "links", "onToggleMenu", "position", "positionNode", "scrollOverflow", "toggle", "toggleAppearance", "toggleClassName", "toggleDisabled", "toggleLabel", "toggleLabelFirst", "toggleProps", "visible", "focusFirstItemOnOpen"];
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
const focusableElementSelectors = 'a[href]:not([tabindex="-1"]), button:not([disabled]):not([aria-disabled="true"]), textarea:not([disabled]):not([aria-disabled="true"]):not([tabindex="-1"]), input:not([disabled]):not([aria-disabled="true"]):not([tabindex="-1"]), select:not([disabled]):not([aria-disabled="true"]):not([tabindex="-1"]), area[href]:not([tabindex="-1"]), iframe:not([tabindex="-1"]), [tabindex]:not([tabindex="-1"]), [contentEditable=true]:not([tabindex="-1"])';
let Label = exports.Label = /*#__PURE__*/function (Label) {
  Label["Toggle"] = "Toggle menu";
  return Label;
}({});
/**
 * The props for the ContextualMenu component.
 * @template L - The type of the link props.
 */
/**
 * Get the node to use for positioning the menu.
 * @param wrapper - The component's wrapping element.
 * @param positionNode - A positioning node, if supplied.
 * @return A node or null.
 */
const getPositionNode = (wrapper, positionNode) => {
  if (positionNode) {
    return positionNode;
  } else if (wrapper) {
    // We want to position the menu in relation to the toggle, if it exists.
    const toggle = wrapper.querySelector(".p-contextual-menu__toggle");
    return toggle || wrapper;
  }
  return null;
};

/**
 * Whether the positioning node is visible.
 * @param positionNode - The node that is used to position the menu.
 * @return Whether the positioning node is visible.
 */
const getPositionNodeVisible = positionNode => {
  return !positionNode || positionNode.offsetParent !== null;
};

/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Contextual menu](https://docs.vanillaframework.io/patterns//contextual-menu/).
 *
 * A contextual menu can be used in conjunction with any page element to provide a contextual menu.
 */
const ContextualMenu = _ref => {
  let {
      autoAdjust = true,
      children,
      className,
      closeOnEsc = true,
      closeOnOutsideClick = true,
      constrainPanelWidth,
      dropdownClassName,
      dropdownProps,
      hasToggleIcon,
      links,
      onToggleMenu,
      position = "right",
      positionNode,
      scrollOverflow,
      toggle,
      toggleAppearance,
      toggleClassName,
      toggleDisabled,
      toggleLabel,
      toggleLabelFirst = true,
      toggleProps,
      visible = false,
      focusFirstItemOnOpen = true
    } = _ref,
    wrapperProps = _objectWithoutProperties(_ref, _excluded);
  const dropdownId = (0, _react.useId)();
  const wrapper = (0, _react.useRef)(null);
  const [positionCoords, setPositionCoords] = (0, _react.useState)();
  const [adjustedPosition, setAdjustedPosition] = (0, _react.useState)(position);
  const focusAnimationFrameId = (0, _react.useRef)(null);
  (0, _react.useEffect)(() => {
    setAdjustedPosition(position);
  }, [position, autoAdjust]);

  // Update the coordinates of the position node.
  const updatePositionCoords = (0, _react.useCallback)(() => {
    const parent = getPositionNode(wrapper.current, positionNode);
    if (!parent) {
      return;
    }
    setPositionCoords(parent.getBoundingClientRect());
  }, [wrapper, positionNode]);

  /**
   * Gets the dropdopwn element (`ContextualMenuDropdown`).
   * @returns The dropdown element or null if it does not exist.
   */
  const getDropdown = (0, _react.useCallback)(() => {
    if (typeof document === "undefined") return null;
    /**
     * This is Using `document` instead of refs because `dropdownProps` may include a ref,
     * while `dropdownId` is unique and controlled by us.
     */
    return document.getElementById(dropdownId);
  }, [dropdownId]);

  /**
   * Gets all focusable items in the dropdown element.
   * @returns Array of focusable items in the dropdown element.
   */
  const getFocusableDropdownItems = (0, _react.useCallback)(() => {
    var _getDropdown;
    return Array.from(((_getDropdown = getDropdown()) === null || _getDropdown === void 0 ? void 0 : _getDropdown.querySelectorAll(focusableElementSelectors)) || []);
  }, [getDropdown]);

  /**
   * Focuses the first focusable item in the dropdown element.
   * This is useful for keyboard users (who expect focus to move into the menu when it opens).
   */
  const focusFirstDropdownItem = (0, _react.useCallback)(() => {
    var _focusableElements$;
    const focusableElements = getFocusableDropdownItems();
    (_focusableElements$ = focusableElements[0]) === null || _focusableElements$ === void 0 || _focusableElements$.focus();
  }, [getFocusableDropdownItems]);

  /**
   * Cleans up any pending dropdown focus animation frames.
   */
  const cleanupDropdownFocus = () => {
    if (focusAnimationFrameId.current) {
      cancelAnimationFrame(focusAnimationFrameId.current);
      focusAnimationFrameId.current = null;
    }
  };
  const returnFocusToTrigger = () => {
    var _wrapper$current;
    // Button element does not accept refs.
    const trigger = (_wrapper$current = wrapper.current) === null || _wrapper$current === void 0 ? void 0 : _wrapper$current.querySelector(".p-contextual-menu__toggle");
    // return focus to the trigger button when the menu closes
    if (trigger) {
      trigger.focus();
    }
  };
  (0, _react.useEffect)(() => {
    return () => cleanupDropdownFocus();
  }, []);
  const {
    openPortal,
    closePortal,
    isOpen,
    Portal
  } = (0, _external.usePortal)({
    closeOnEsc,
    closeOnOutsideClick,
    isOpen: visible,
    onOpen: event => {
      // Call the toggle callback, if supplied.
      onToggleMenu === null || onToggleMenu === void 0 || onToggleMenu(true);
      // When the menu opens then update the coordinates of the parent.
      updatePositionCoords();
      if (focusFirstItemOnOpen && // Don't focus the item unless it was opened by a keyboard event
      // This type silliness is because `detail` isn't on the type for `event.nativeEvent` passed from `usePortal`,
      // as we are using `CustomEvent<HTMLElement>` which does not have `detail` defined.
      event !== null && event !== void 0 && event.nativeEvent && "detail" in event.nativeEvent && event.nativeEvent.detail === 0) {
        cleanupDropdownFocus();
        // We need to wait a frame for any pending focus events to complete.
        focusAnimationFrameId.current = requestAnimationFrame(() => focusFirstDropdownItem());
      }
    },
    onClose: () => {
      // Call the toggle callback, if supplied.
      onToggleMenu === null || onToggleMenu === void 0 || onToggleMenu(false);
      cleanupDropdownFocus();
      returnFocusToTrigger();
    },
    programmaticallyOpen: true
  });

  /**
   * Trap focus within the dropdown
   */
  (0, _react.useEffect)(() => {
    const handleKeyDown = e => {
      if (e.key !== "Tab" || !isOpen) return;
      const items = getFocusableDropdownItems();
      if (items.length === 0) return;
      const active = document.activeElement;
      const first = items[0];
      const last = items[items.length - 1];
      if (!e.shiftKey && active === last) {
        // Tab on the last item: wrap back to the first focusable item
        e.preventDefault();
        first.focus();
      } else if (e.shiftKey && active === first) {
        // Shift+Tab on the first item: wrap back to the last focusable item
        e.preventDefault();
        last.focus();
      }
    };
    const dropdown = getDropdown();
    if (!dropdown) return undefined;
    dropdown.addEventListener("keydown", handleKeyDown);
    return () => {
      dropdown.removeEventListener("keydown", handleKeyDown);
    };
  }, [getDropdown, getFocusableDropdownItems, isOpen]);
  const previousVisible = (0, _hooks.usePrevious)(visible);
  const labelNode = toggleLabel && typeof toggleLabel === "string" ? /*#__PURE__*/_react.default.createElement("span", null, toggleLabel) : /*#__PURE__*/_react.default.isValidElement(toggleLabel) ? toggleLabel : null;
  const contextualMenuClassName = (0, _classnames.default)(className, "p-contextual-menu", {
    ["p-contextual-menu--".concat(adjustedPosition)]: adjustedPosition !== "right"
  });

  // Update the coordinates of the wrapper once it mounts to the dom. This uses
  // The callback ref pattern:
  // https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node
  const wrapperRef = (0, _react.useCallback)(node => {
    wrapper.current = node;
    if (node !== null) {
      updatePositionCoords();
    }
  }, [updatePositionCoords]);

  // Handle controlling updates to the menu visibility from outside
  // the component.
  (0, _react.useEffect)(() => {
    if (visible !== previousVisible) {
      if (visible && !isOpen) {
        openPortal();
      } else if (!visible && isOpen) {
        closePortal();
      }
    }
  }, [closePortal, openPortal, visible, isOpen, previousVisible]);
  const onResize = (0, _react.useCallback)(() => {
    const parent = getPositionNode(wrapper.current, positionNode);
    if (parent && !getPositionNodeVisible(parent)) {
      // Hide the menu if the item has become hidden. This might happen in
      // a responsive table when columns become hidden as the page
      // becomes smaller.
      closePortal();
    } else {
      // Only update if the coordinates have changed.
      // The check fixes a bug with chrome, where an input receiving focus and
      // opening the keyboard causes a resize and the keyboard closes right after
      // opening.
      const coords = parent.getBoundingClientRect();
      if (JSON.stringify(coords) !== JSON.stringify(positionCoords)) {
        // Update the coordinates so that the menu stays relative to the
        // toggle button.
        updatePositionCoords();
      }
    }
  }, [closePortal, positionNode, positionCoords, updatePositionCoords]);
  const onScroll = (0, _react.useCallback)(e => {
    const parent = getPositionNode(wrapper.current, positionNode);
    // update position if the scroll event is triggered by the parent of the menu
    if (parent && e.target.contains(parent)) {
      // Update the coordinates so that the menu stays relative to the
      // toggle button.
      updatePositionCoords();
    }
  }, [positionNode, updatePositionCoords]);
  (0, _hooks.useListener)(window, onResize, "resize", true, isOpen);
  (0, _hooks.useListener)(window, onScroll, "scroll", false, isOpen, true);
  let toggleNode = null;
  if (toggle) {
    toggleNode = toggle;
  } else if (hasToggleIcon || toggleLabel) {
    toggleNode = /*#__PURE__*/_react.default.createElement(_Button.default, _extends({
      appearance: toggleAppearance,
      "aria-controls": dropdownId,
      "aria-expanded": isOpen ? "true" : "false",
      "aria-label": toggleLabel ? null : Label.Toggle,
      "aria-pressed": isOpen ? "true" : "false",
      "aria-haspopup": "true",
      className: (0, _classnames.default)("p-contextual-menu__toggle", toggleClassName),
      disabled: toggleDisabled,
      hasIcon: hasToggleIcon,
      onClick: evt => {
        if (!isOpen) {
          openPortal(evt);
        } else {
          closePortal(evt);
        }
      },
      type: "button"
    }, toggleProps), toggleLabelFirst ? labelNode : null, hasToggleIcon ? /*#__PURE__*/_react.default.createElement("i", {
      className: (0, _classnames.default)("p-icon--chevron-down p-contextual-menu__indicator", {
        "is-light": ["negative", "positive"].includes(toggleAppearance)
      })
    }) : null, toggleLabelFirst ? null : labelNode);
  }
  return /*#__PURE__*/_react.default.createElement("span", _extends({
    className: contextualMenuClassName,
    ref: wrapperRef
  }, wrapperProps), toggleNode, isOpen && /*#__PURE__*/_react.default.createElement(Portal, null, /*#__PURE__*/_react.default.createElement(_ContextualMenuDropdown.default, _extends({
    adjustedPosition: adjustedPosition,
    autoAdjust: autoAdjust,
    handleClose: closePortal,
    constrainPanelWidth: constrainPanelWidth,
    dropdownClassName: dropdownClassName,
    dropdownContent: children,
    id: dropdownId,
    isOpen: isOpen,
    links: links,
    position: position,
    positionCoords: positionCoords,
    contextualMenuClassName: contextualMenuClassName,
    positionNode: getPositionNode(wrapper.current),
    scrollOverflow: scrollOverflow,
    setAdjustedPosition: setAdjustedPosition
  }, dropdownProps))));
};
var _default = exports.default = ContextualMenu;
