// @canonical/react-components@4.2.0/dist/esm/components/Col/Col.stories.js
import React from "react";
import Col from "./Col";
import Row from "../Row";
var Template = args => {
  return /*#__PURE__*/React.createElement("div", {
    className: "grid-demo"
  }, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, args)));
};
var meta = {
  component: Col,
  render: Template,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: {
        type: "text"
      }
    },
    size: {
      control: {
        type: "range",
        min: 1,
        max: 12,
        step: 1
      }
    }
  }
};
export default meta;
export var Default = {
  name: "Default",
  args: {
    children: "Hi, I'm Col.",
    size: 12
  }
};
export var Grid = {
  render: () => {
    return /*#__PURE__*/React.createElement("div", {
      className: "grid-demo"
    }, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 12
    }, /*#__PURE__*/React.createElement("span", null, ".col-12"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 11
    }, /*#__PURE__*/React.createElement("span", null, ".col-11")), /*#__PURE__*/React.createElement(Col, {
      size: 1
    }, /*#__PURE__*/React.createElement("span", null, ".col-1"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 10
    }, /*#__PURE__*/React.createElement("span", null, ".col-10")), /*#__PURE__*/React.createElement(Col, {
      size: 2
    }, /*#__PURE__*/React.createElement("span", null, ".col-2"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 9
    }, /*#__PURE__*/React.createElement("span", null, ".col-9")), /*#__PURE__*/React.createElement(Col, {
      size: 3
    }, /*#__PURE__*/React.createElement("span", null, ".col-3"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 8
    }, /*#__PURE__*/React.createElement("span", null, ".col-8")), /*#__PURE__*/React.createElement(Col, {
      size: 4
    }, /*#__PURE__*/React.createElement("span", null, ".col-4"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 7
    }, /*#__PURE__*/React.createElement("span", null, ".col-7")), /*#__PURE__*/React.createElement(Col, {
      size: 5
    }, /*#__PURE__*/React.createElement("span", null, ".col-5"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 6
    }, /*#__PURE__*/React.createElement("span", null, ".col-6")), /*#__PURE__*/React.createElement(Col, {
      size: 6
    }, /*#__PURE__*/React.createElement("span", null, ".col-6"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 5
    }, /*#__PURE__*/React.createElement("span", null, ".col-5")), /*#__PURE__*/React.createElement(Col, {
      size: 7
    }, /*#__PURE__*/React.createElement("span", null, ".col-7"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 4
    }, /*#__PURE__*/React.createElement("span", null, ".col-4")), /*#__PURE__*/React.createElement(Col, {
      size: 8
    }, /*#__PURE__*/React.createElement("span", null, ".col-8"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 3
    }, /*#__PURE__*/React.createElement("span", null, ".col-3")), /*#__PURE__*/React.createElement(Col, {
      size: 9
    }, /*#__PURE__*/React.createElement("span", null, ".col-9"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 2
    }, /*#__PURE__*/React.createElement("span", null, ".col-2")), /*#__PURE__*/React.createElement(Col, {
      size: 10
    }, /*#__PURE__*/React.createElement("span", null, ".col-10"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 1
    }, /*#__PURE__*/React.createElement("span", null, ".col-1")), /*#__PURE__*/React.createElement(Col, {
      size: 11
    }, /*#__PURE__*/React.createElement("span", null, ".col-11"))));
  },
  name: "Grid"
};
export var NestedColumns = {
  render: () => {
    return /*#__PURE__*/React.createElement("div", {
      className: "grid-demo"
    }, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      small: 4,
      medium: 6,
      size: 12
    }, "col-small-4 .col-medium-6 .col-12", /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      small: 3,
      medium: 3,
      size: 9
    }, "col-small-3 .col-medium-3 .col-9", /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      small: 1,
      medium: 1,
      size: 2
    }, "col-small-1 col-medium-1 col-2"), /*#__PURE__*/React.createElement(Col, {
      small: 1,
      medium: 1,
      size: 3
    }, "col-small-1 col-medium-1 col-2"), /*#__PURE__*/React.createElement(Col, {
      small: 1,
      medium: 1,
      size: 3
    }, "col-small-1 col-medium-1 col-2"))), /*#__PURE__*/React.createElement(Col, {
      small: 1,
      medium: 3,
      size: 3
    }, "col-small-2 col-medium-3 col-3")))));
  },
  name: "Nested columns"
};
export var EmptyColumns = {
  render: () => {
    return /*#__PURE__*/React.createElement("div", {
      className: "grid-demo"
    }, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 8
    }, ".col-8"), /*#__PURE__*/React.createElement(Col, {
      size: 4
    }, ".col-4")), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 7
    }, ".col-7"), /*#__PURE__*/React.createElement(Col, {
      size: 4
    }, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 3,
      emptyLarge: 2
    }, "col-3 col-start-large-2 inside col-4")))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 7
    }, ".col-7"), /*#__PURE__*/React.createElement(Col, {
      size: 4
    }, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
      size: 3
    }, "col-3 inside col-4")))));
  },
  name: "Empty columns"
};
