// @canonical/react-components@4.2.0/dist/components/StatusLabel/StatusLabel.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Positive = exports.Negative = exports.Information = exports.Default = exports.Caution = void 0;
var _react = _interopRequireDefault(require("react"));
var _StatusLabel = _interopRequireWildcard(require("./StatusLabel"));
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _StatusLabel.default,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: {
        type: "text"
      }
    },
    appearance: {
      control: {
        options: _StatusLabel.StatusLabelAppearance,
        type: "select"
      }
    }
  },
  args: {
    children: "Label",
    appearance: _StatusLabel.StatusLabelAppearance.POSITIVE
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: () => /*#__PURE__*/_react.default.createElement(_StatusLabel.default, null, "Default"),
  name: "Default"
};
const Positive = exports.Positive = {
  render: () => /*#__PURE__*/_react.default.createElement(_StatusLabel.default, {
    appearance: "positive"
  }, "Positive"),
  name: "Positive"
};
const Caution = exports.Caution = {
  render: () => /*#__PURE__*/_react.default.createElement(_StatusLabel.default, {
    appearance: "caution"
  }, "Caution"),
  name: "Caution"
};
const Negative = exports.Negative = {
  render: () => /*#__PURE__*/_react.default.createElement(_StatusLabel.default, {
    appearance: "negative"
  }, "Negative"),
  name: "Negative"
};
const Information = exports.Information = {
  render: () => /*#__PURE__*/_react.default.createElement(_StatusLabel.default, {
    appearance: "information"
  }, "Information"),
  name: "Information"
};
