// @canonical/react-components@4.2.0/dist/components/Col/Col.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.NestedColumns = exports.Grid = exports.EmptyColumns = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _Col = _interopRequireDefault(require("./Col"));
var _Row = _interopRequireDefault(require("../Row"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const Template = args => {
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "grid-demo"
  }, /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, args)));
};
const meta = {
  component: _Col.default,
  render: Template,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: {
        type: "text"
      }
    },
    size: {
      control: {
        type: "range",
        min: 1,
        max: 12,
        step: 1
      }
    }
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default",
  args: {
    children: "Hi, I'm Col.",
    size: 12
  }
};
const Grid = exports.Grid = {
  render: () => {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: "grid-demo"
    }, /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 12
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-12"))), /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 11
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-11")), /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 1
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-1"))), /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 10
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-10")), /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 2
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-2"))), /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 9
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-9")), /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 3
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-3"))), /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 8
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-8")), /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 4
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-4"))), /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 7
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-7")), /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 5
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-5"))), /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 6
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-6")), /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 6
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-6"))), /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 5
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-5")), /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 7
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-7"))), /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 4
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-4")), /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 8
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-8"))), /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 3
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-3")), /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 9
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-9"))), /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 2
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-2")), /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 10
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-10"))), /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 1
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-1")), /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 11
    }, /*#__PURE__*/_react.default.createElement("span", null, ".col-11"))));
  },
  name: "Grid"
};
const NestedColumns = exports.NestedColumns = {
  render: () => {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: "grid-demo"
    }, /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      small: 4,
      medium: 6,
      size: 12
    }, "col-small-4 .col-medium-6 .col-12", /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      small: 3,
      medium: 3,
      size: 9
    }, "col-small-3 .col-medium-3 .col-9", /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      small: 1,
      medium: 1,
      size: 2
    }, "col-small-1 col-medium-1 col-2"), /*#__PURE__*/_react.default.createElement(_Col.default, {
      small: 1,
      medium: 1,
      size: 3
    }, "col-small-1 col-medium-1 col-2"), /*#__PURE__*/_react.default.createElement(_Col.default, {
      small: 1,
      medium: 1,
      size: 3
    }, "col-small-1 col-medium-1 col-2"))), /*#__PURE__*/_react.default.createElement(_Col.default, {
      small: 1,
      medium: 3,
      size: 3
    }, "col-small-2 col-medium-3 col-3")))));
  },
  name: "Nested columns"
};
const EmptyColumns = exports.EmptyColumns = {
  render: () => {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: "grid-demo"
    }, /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 8
    }, ".col-8"), /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 4
    }, ".col-4")), /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 7
    }, ".col-7"), /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 4
    }, /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 3,
      emptyLarge: 2
    }, "col-3 col-start-large-2 inside col-4")))), /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 7
    }, ".col-7"), /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 4
    }, /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 3
    }, "col-3 inside col-4")))));
  },
  name: "Empty columns"
};
