// @canonical/react-components@4.2.0/dist/components/Stepper/Stepper.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.VerticalSelected = exports.VerticalOptional = exports.HorizontalStepper = exports.HorizontalSelected = exports.HorizontalOptional = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _Stepper = _interopRequireDefault(require("./Stepper"));
var _Step = _interopRequireDefault(require("./Step"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _Stepper.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default",
  args: {
    variant: "vertical",
    steps: [/*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Default Step 1",
      title: "Step 1",
      index: 1,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      handleClick: () => {}
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Default Step 2",
      title: "Step 2",
      index: 2,
      enabled: true,
      hasProgressLine: true,
      iconName: "spinner",
      iconClassName: "u-animation--spin",
      handleClick: () => {}
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Default Step 3",
      title: "Step 3",
      index: 3,
      enabled: false,
      hasProgressLine: false,
      iconName: "number",
      handleClick: () => {}
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Default Step 4",
      title: "Step 4",
      index: 4,
      enabled: false,
      hasProgressLine: false,
      iconName: "number",
      handleClick: () => {}
    })]
  }
};
const HorizontalStepper = exports.HorizontalStepper = {
  name: "Horizontal Stepper",
  render: () => /*#__PURE__*/_react.default.createElement(_Stepper.default, {
    variant: "horizontal",
    steps: [/*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Horizontal Step 1",
      title: "Step 1",
      index: 1,
      enabled: false,
      hasProgressLine: true,
      iconName: "success",
      handleClick: () => {}
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Horizontal Step 2",
      title: "Step 2",
      index: 2,
      enabled: false,
      hasProgressLine: true,
      iconName: "error",
      handleClick: () => {}
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Horizontal Step 3",
      title: "Step 3",
      index: 3,
      enabled: true,
      hasProgressLine: true,
      iconName: "spinner",
      iconClassName: "u-animation--spin",
      handleClick: () => {}
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Horizontal Step 4",
      title: "Step 4",
      index: 4,
      enabled: false,
      hasProgressLine: false,
      iconName: "number",
      handleClick: () => {}
    })]
  })
};
const VerticalOptional = exports.VerticalOptional = {
  name: "Vertical variant: Optional labels and links",
  render: () => /*#__PURE__*/_react.default.createElement(_Stepper.default, {
    steps: [/*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Vertical optional Step 1",
      title: "Step 1",
      index: 1,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      label: "Optional label",
      handleClick: () => {}
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Vertical optional Step 2",
      title: "Step 2",
      index: 2,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      handleClick: () => {},
      label: "Optional label"
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Vertical optional Step 3",
      title: "Step 3",
      index: 3,
      enabled: true,
      hasProgressLine: true,
      iconName: "number",
      label: "Optional label",
      linkProps: {
        children: "Optional link"
      },
      handleClick: () => {}
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Vertical optional Step 4",
      title: "Step 4",
      index: 4,
      enabled: false,
      hasProgressLine: false,
      linkProps: {
        children: "Optional link"
      },
      iconName: "number",
      handleClick: () => {}
    })]
  })
};
const HorizontalOptional = exports.HorizontalOptional = {
  name: "Horizontal variant: Optional labels and links",
  render: () => /*#__PURE__*/_react.default.createElement(_Stepper.default, {
    variant: "horizontal",
    steps: [/*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Horizontal optional Step 1",
      title: "Step 1",
      index: 1,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      label: "Optional label",
      handleClick: () => {}
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Horizontal optional Step 2",
      title: "Step 2",
      index: 2,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      handleClick: () => {},
      label: "Optional label"
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Horizontal optional Step 3",
      title: "Step 3",
      index: 3,
      enabled: true,
      hasProgressLine: true,
      iconName: "number",
      label: "Optional label",
      linkProps: {
        children: "Optional link"
      },
      handleClick: () => {}
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Horizontal optional Step 4",
      title: "Step 4",
      index: 4,
      enabled: false,
      hasProgressLine: false,
      iconName: "number",
      linkProps: {
        children: "Optional link"
      },
      handleClick: () => {}
    })]
  })
};
const VerticalSelected = exports.VerticalSelected = {
  name: "Vertical variant: Selected step",
  render: () => /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: "50%"
    }
  }, /*#__PURE__*/_react.default.createElement(_Stepper.default, {
    steps: [/*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Vertical Selected Step 1",
      title: "Step 1",
      index: 1,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      label: "Optional label",
      handleClick: () => {}
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Vertical Selected Step 2",
      title: "Step 2",
      index: 2,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      handleClick: () => {},
      label: "Optional label",
      selected: true
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Vertical Selected Step 3",
      title: "Step 3",
      index: 3,
      enabled: true,
      hasProgressLine: true,
      iconName: "number",
      label: "Optional label",
      linkProps: {
        children: "Optional link"
      },
      handleClick: () => {}
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Vertical Selected Step 4",
      title: "Step 4",
      index: 4,
      enabled: false,
      hasProgressLine: false,
      linkProps: {
        children: "Optional link"
      },
      iconName: "number",
      label: "Optional label",
      handleClick: () => {}
    })]
  }))
};
const HorizontalSelected = exports.HorizontalSelected = {
  name: "Horizontal variant: Selected step",
  render: () => /*#__PURE__*/_react.default.createElement(_Stepper.default, {
    variant: "horizontal",
    steps: [/*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Horizontal Selected Step 1",
      title: "Step 1",
      index: 1,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      label: "Optional label",
      handleClick: () => {}
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Horizontal Selected Step 2",
      title: "Step 2",
      index: 2,
      enabled: true,
      hasProgressLine: true,
      iconName: "success",
      selected: true,
      handleClick: () => {},
      label: "Optional label"
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Horizontal Selected Step 3",
      title: "Step 3",
      index: 3,
      enabled: true,
      hasProgressLine: true,
      iconName: "number",
      label: "Optional label",
      linkProps: {
        children: "Optional link"
      },
      handleClick: () => {}
    }), /*#__PURE__*/_react.default.createElement(_Step.default, {
      key: "Horizontal Selected Step 4",
      title: "Step 4",
      index: 4,
      enabled: false,
      hasProgressLine: false,
      iconName: "number",
      label: "Optional label",
      linkProps: {
        children: "Optional link"
      },
      handleClick: () => {}
    })]
  })
};
