// @canonical/react-components@4.2.0/dist/esm/components/RadioInput/RadioInput.stories.js
import React from "react";
import RadioInput from "./RadioInput";
var meta = {
  component: RadioInput,
  tags: ["autodocs"],
  argTypes: {
    label: {
      control: {
        type: "text"
      }
    },
    disabled: {
      control: {
        type: "boolean"
      }
    },
    required: {
      control: {
        type: "boolean"
      }
    },
    inline: {
      control: {
        type: "boolean"
      }
    },
    indeterminate: {
      control: {
        type: "boolean"
      }
    }
  },
  args: {
    label: "RadioInput"
  }
};
export default meta;
export var Default = {
  render: () => /*#__PURE__*/React.createElement(RadioInput, {
    label: "RadioInput"
  }),
  name: "Default"
};
export var Children = {
  render: () => /*#__PURE__*/React.createElement(RadioInput, {
    label: /*#__PURE__*/React.createElement(React.Fragment, null, "I agree to", /*#__PURE__*/React.createElement("a", {
      href: "http://ubuntu.com/legal"
    }, "Terms and Conditions")),
    name: "RadioInput"
  }),
  name: "Children"
};
export var Disabled = {
  render: () => /*#__PURE__*/React.createElement(RadioInput, {
    label: "Disabled",
    name: "RadioInput",
    disabled: true
  }),
  name: "Disabled"
};
export var Required = {
  render: () => /*#__PURE__*/React.createElement(RadioInput, {
    label: "Required",
    name: "RadioInput",
    required: true
  }),
  name: "Required"
};
export var Inline = {
  render: () => /*#__PURE__*/React.createElement(RadioInput, {
    label: "Inline",
    name: "RadioInput",
    inline: true
  }),
  name: "Inline"
};
