// @canonical/react-components@4.2.0/dist/esm/components/ApplicationLayout/AppMain/AppMain.stories.js
import Application from "../Application";
import Button from "../../Button";
import React from "react";
import Panel from "../../Panel";
import AppMain from "./AppMain";
var meta = {
  component: AppMain,
  tags: ["autodocs"]
};
export default meta;
/**
 * In most common cases an `AppMain` should contain a `<Panel>` to display the
 * content as intended in the application layout.
 *
 * `AppMain` should be a direct child of an `<Application>` or when using `ApplicationLayout`
 * it will automatically wrap the component's children.
 */
export var Default = {
  args: {
    children: "AppMain"
  }
};
export var Content = {
  render: args => {
    return /*#__PURE__*/React.createElement(Application, null, /*#__PURE__*/React.createElement(AppMain, args, /*#__PURE__*/React.createElement(Panel, {
      controls: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Button, {
        appearance: "positive",
        onClick: () => {},
        className: "u-no-margin--bottom"
      }, "Add"), /*#__PURE__*/React.createElement(Button, {
        appearance: "negative",
        onClick: () => {},
        className: "u-no-margin--bottom"
      }, "Delete")),
      title: "App main"
    }, /*#__PURE__*/React.createElement("p", null, "App main content."), /*#__PURE__*/React.createElement("p", null, "Scroll to the right to see the controls."))));
  }
};
