// @canonical/react-components@4.2.0/dist/esm/components/StatusLabel/StatusLabel.stories.js
import React from "react";
import StatusLabel, { StatusLabelAppearance } from "./StatusLabel";
var meta = {
  component: StatusLabel,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: {
        type: "text"
      }
    },
    appearance: {
      control: {
        options: StatusLabelAppearance,
        type: "select"
      }
    }
  },
  args: {
    children: "Label",
    appearance: StatusLabelAppearance.POSITIVE
  }
};
export default meta;
export var Default = {
  render: () => /*#__PURE__*/React.createElement(StatusLabel, null, "Default"),
  name: "Default"
};
export var Positive = {
  render: () => /*#__PURE__*/React.createElement(StatusLabel, {
    appearance: "positive"
  }, "Positive"),
  name: "Positive"
};
export var Caution = {
  render: () => /*#__PURE__*/React.createElement(StatusLabel, {
    appearance: "caution"
  }, "Caution"),
  name: "Caution"
};
export var Negative = {
  render: () => /*#__PURE__*/React.createElement(StatusLabel, {
    appearance: "negative"
  }, "Negative"),
  name: "Negative"
};
export var Information = {
  render: () => /*#__PURE__*/React.createElement(StatusLabel, {
    appearance: "information"
  }, "Information"),
  name: "Information"
};
