// @canonical/react-components@4.2.0/dist/esm/components/StatusLabel/StatusLabel.js
var _excluded = ["appearance", "children", "className"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import React from "react";
import classNames from "classnames";
export var StatusLabelAppearance = {
  CAUTION: "caution",
  DEFAULT: "",
  INFORMATION: "information",
  NEGATIVE: "negative",
  POSITIVE: "positive"
};

/**
 * The type of the StatusLabel props.
 */

/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Label](https://vanillaframework.io/docs/patterns/labels).
 * Labels are static elements which you can apply to signify status, tags or any other information you find useful.
 * @deprecated StatusLabel is deprecated. Use Read-only Chip instead.
 */
var StatusLabel = _ref => {
  var {
      appearance,
      children,
      className
    } = _ref,
    labelProps = _objectWithoutProperties(_ref, _excluded);
  var classes = classNames(appearance ? "p-status-label--".concat(appearance) : "p-status-label", className);
  return /*#__PURE__*/React.createElement("div", _extends({}, labelProps, {
    className: classes
  }), children);
};
export default StatusLabel;
