// @canonical/react-components@4.2.0/dist/components/ApplicationLayout/AppStatus/AppStatus.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _Application = _interopRequireDefault(require("../Application"));
var _Button = _interopRequireDefault(require("../../Button"));
var _Panel = _interopRequireDefault(require("../../Panel"));
var _AppStatus = _interopRequireDefault(require("./AppStatus"));
var _AppMain = _interopRequireDefault(require("../AppMain"));
var _Strip = _interopRequireDefault(require("../../Strip"));
var _Spinner = _interopRequireDefault(require("../../Spinner"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _AppStatus.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
/**
 * `AppStatus` can contain a `<Panel>` to display the content as intended in the
 * application layout.
 *
 * `AppStatus` should be a direct child of an `<Application>` or when using `ApplicationLayout`
 * it will automatically wrap the content passed to the `status` prop.
 */
const Default = exports.Default = {
  render: args => {
    return /*#__PURE__*/_react.default.createElement(_Application.default, null, /*#__PURE__*/_react.default.createElement(_AppMain.default, null, /*#__PURE__*/_react.default.createElement(_Strip.default, null, /*#__PURE__*/_react.default.createElement("p", null, "Scroll to the bottom to see the status bar."))), /*#__PURE__*/_react.default.createElement(_AppStatus.default, args, /*#__PURE__*/_react.default.createElement(_Panel.default, {
      wrapContent: false
    }, /*#__PURE__*/_react.default.createElement("p", {
      className: "u-no-margin--bottom"
    }, /*#__PURE__*/_react.default.createElement(_Spinner.default, null), " Update in progress...", " ", /*#__PURE__*/_react.default.createElement(_Button.default, {
      appearance: "base",
      onClick: () => {},
      dense: true,
      className: "u-no-margin"
    }, "Retry")))));
  }
};
