// @canonical/react-components@4.2.0/dist/components/Spinner/Spinner.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));
var _CustomLayout = _interopRequireDefault(require("../CustomLayout"));
const _excluded = ["className", "text", "isLight", "ariaLive", "role", "isMainComponent"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Spin](https://docs.vanillaframework.io/settings/animation-settings/#spin) animation.
 */
const Spinner = _ref => {
  let {
      className,
      text,
      isLight = false,
      ariaLive = "polite",
      role = "alert",
      isMainComponent = false
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  const loader = /*#__PURE__*/_react.default.createElement("span", _extends({}, props, {
    className: (0, _classnames.default)(className, "p-text--default"),
    role: role,
    "aria-live": ariaLive
  }), /*#__PURE__*/_react.default.createElement("i", {
    className: (0, _classnames.default)("p-icon--spinner", "u-animation--spin", {
      "is-light": isLight
    })
  }, text ? "" : "Loading"), text && /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, "\u2002", /*#__PURE__*/_react.default.createElement("span", null, text)));
  return isMainComponent ? /*#__PURE__*/_react.default.createElement(_CustomLayout.default, null, loader) : loader;
};
Spinner.propTypes = {
  isLight: _propTypes.default.bool,
  text: _propTypes.default.string,
  role: _propTypes.default.string,
  ariaLive: _propTypes.default.oneOf(["assertive", "off", "polite"]),
  isMainComponent: _propTypes.default.bool
};
var _default = exports.default = Spinner;
