// @canonical/react-components@4.2.0/dist/esm/components/Strip/Strip.stories.js
import React from "react";
import Col from "../Col";
import Row from "../Row";
import Strip from "./Strip";
var meta = {
  component: Strip,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: {
        disable: true
      }
    }
  }
};
export default meta;
export var LightStrip = {
  name: "Light strip",
  args: {
    type: "light",
    children: /*#__PURE__*/React.createElement("p", null, "This is a light row")
  }
};
export var DarkStrip = {
  render: () => /*#__PURE__*/React.createElement(Strip, {
    type: "dark"
  }, /*#__PURE__*/React.createElement("p", null, "This is a dark row")),
  name: "Dark strip"
};
export var AccentStrip = {
  render: () => /*#__PURE__*/React.createElement(Strip, {
    type: "accent",
    element: "section",
    includeCol: false
  }, /*#__PURE__*/React.createElement(Col, {
    size: 8
  }, /*#__PURE__*/React.createElement("h1", null, "Still running Ubuntu 14.04 LTS?"), /*#__PURE__*/React.createElement("p", null, "Learn how to maintain ongoing security compliance for your Ubuntu 14.04 LTS systems.")), /*#__PURE__*/React.createElement(Col, {
    size: 4
  }, /*#__PURE__*/React.createElement("img", {
    src: "https://assets.ubuntu.com/v1/2217d1c8-Security.svg",
    alt: "Placeholder"
  }))),
  name: "Accent strip"
};
export var ImageStrip = {
  render: () => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Strip, {
    background: "https://assets.ubuntu.com/sites/ubuntu/latest/u/img/backgrounds/image-background-paper.png",
    includeCol: false,
    element: "section",
    light: true,
    rowClassName: "u-vertically-center",
    type: "image"
  }, /*#__PURE__*/React.createElement(Col, {
    size: 8
  }, /*#__PURE__*/React.createElement("h1", null, "Get started with big software, fast"), /*#__PURE__*/React.createElement("p", null, "conjure-up lets you summon up a big-software stack as a \u201Cspell\u201D \u2014 a model of the stack, combined with extra know-how to get you from an installed stack to a fully usable one. Start using your big software instead of learning how to deploy it.")), /*#__PURE__*/React.createElement(Col, {
    size: 4,
    className: "u-hide--small u-align--center"
  }, /*#__PURE__*/React.createElement("img", {
    src: "https://assets.ubuntu.com/v1/1abb8716-conjure-up-illustration.svg",
    alt: "Placeholder"
  }))), /*#__PURE__*/React.createElement(Strip, {
    background: "https://assets.ubuntu.com/v1/9b68976e-Aubergine_suru_background_2.png",
    dark: true,
    includeCol: false,
    element: "section",
    type: "image"
  }, /*#__PURE__*/React.createElement(Row, {
    className: "u-vertically-center"
  }, /*#__PURE__*/React.createElement(Col, {
    size: 8
  }, /*#__PURE__*/React.createElement("h1", null, "We are Canonical"), /*#__PURE__*/React.createElement("p", null, "It is our mission to make open source software available to people everywhere. We believe the best way to fuel innovation is to give the innovators the technology they need.")), /*#__PURE__*/React.createElement(Col, {
    size: 4,
    className: "u-hide--small u-align--center"
  }, /*#__PURE__*/React.createElement("img", {
    src: "https://assets.ubuntu.com/v1/9c74eb2d-logo-canonical-white.svg",
    alt: "Placeholder"
  }))))),
  name: "Image strip",
  parameters: {
    percy: {
      waitForTimeout: 1000 // Wait for 1 second before taking a snapshot to allow the image to load
    }
  }
};
export var BorderedStrip = {
  render: () => /*#__PURE__*/React.createElement(Strip, {
    bordered: true,
    colSize: 8,
    element: "section"
  }, /*#__PURE__*/React.createElement("h2", null, "The node lifecycle"), /*#__PURE__*/React.createElement("p", null, "Each machine (\u201Cnode\u201D) managed by MAAS goes through a lifecycle \u2014 from its enlistment or onboarding to MAAS, through commissioning when we inventory and can setup firmware or other hardware-specific elements, then allocation to a user and deployment, and finally they are released back to the pool or retired altogether.")),
  name: "Bordered strip"
};
export var DeepStrip = {
  render: () => /*#__PURE__*/React.createElement(Strip, {
    deep: true,
    includeCol: false,
    element: "section",
    type: "light",
    rowClassName: "u-vertically-center"
  }, /*#__PURE__*/React.createElement(Col, {
    size: 8
  }, /*#__PURE__*/React.createElement("h2", null, "The fastest way to go from development to production in IoT"), /*#__PURE__*/React.createElement("p", null, "Learn about how Ubuntu Core and snaps can help you build your connected devices.")), /*#__PURE__*/React.createElement(Col, {
    size: 4,
    className: "u-hide--small u-align--center"
  }, /*#__PURE__*/React.createElement("img", {
    src: "https://assets.ubuntu.com/v1/808a4e5b-iot.png?h=300",
    alt: "Placeholder"
  }))),
  name: "Deep strip"
};
export var ShallowStrip = {
  render: () => /*#__PURE__*/React.createElement(Strip, {
    shallow: true,
    includeCol: false,
    element: "section",
    type: "light",
    rowClassName: "u-vertically-center"
  }, /*#__PURE__*/React.createElement(Col, {
    size: 8
  }, /*#__PURE__*/React.createElement("h2", null, "The fastest way to go from development to production in IoT"), /*#__PURE__*/React.createElement("p", null, "Learn about how Ubuntu Core and snaps can help you build your connected devices.")), /*#__PURE__*/React.createElement(Col, {
    size: 4,
    className: "u-hide--small u-align--center"
  }, /*#__PURE__*/React.createElement("img", {
    src: "https://assets.ubuntu.com/v1/808a4e5b-iot.png?h=300",
    alt: "Placeholder"
  }))),
  name: "Shallow strip"
};
