// @canonical/react-components@4.2.0/dist/esm/components/Panel/Panel.stories.js
import React from "react";
import Panel from "./Panel";
import Button from "../Button";
import Icon from "../Icon";
var meta = {
  component: Panel,
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  args: {
    children: "Panel content",
    title: "Panel"
  }
};
export var Header = {
  args: {
    controls: /*#__PURE__*/React.createElement(Button, {
      appearance: "positive"
    }, /*#__PURE__*/React.createElement(Icon, {
      name: "plus",
      light: true
    }), " Create"),
    title: "Panel title",
    titleComponent: "h1"
  }
};

/**
 * The logo may be provided as attributes to use the standard logo. If this is
 * not sufficient the a `ReactNode` can be passed to the `logo` prop instead.
 */
export var Logo = {
  args: {
    logo: {
      icon: "https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg",
      name: "https://assets.ubuntu.com/v1/a85f7947-juju_black-text-only.svg",
      nameAlt: "Juju"
    }
  }
};

/**
 * If the default header does not meet your needs then a `ReactNode` can be
 * passed to the `header` prop to replace the header.
 */
export var CustomHeader = {
  args: {
    header: /*#__PURE__*/React.createElement("div", {
      className: "p-panel__header"
    }, "This header replaces the entire header area")
  }
};
