// @canonical/react-components@4.2.0/dist/components/ApplicationLayout/ApplicationLayout.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.NoNavigation = exports.Navigation = exports.ExternalStates = exports.Default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _Form = _interopRequireDefault(require("../Form"));
var _Input = _interopRequireDefault(require("../Input"));
var _Button = _interopRequireDefault(require("../Button"));
var _Row = _interopRequireDefault(require("../Row"));
var _Col = _interopRequireDefault(require("../Col"));
var _ApplicationLayout = _interopRequireDefault(require("./ApplicationLayout"));
var _AppAside = _interopRequireDefault(require("./AppAside"));
var _Panel = _interopRequireDefault(require("../Panel"));
var _Icon = _interopRequireDefault(require("../Icon"));
var _SideNavigation = _interopRequireDefault(require("../SideNavigation"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
/* eslint-disable react-hooks/rules-of-hooks */

const meta = {
  component: _ApplicationLayout.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: () => {
    const [showAside, setShowAside] = (0, _react.useState)(true);
    const [asidePinned, setAsidePinned] = (0, _react.useState)(false);
    return /*#__PURE__*/_react.default.createElement(_ApplicationLayout.default, {
      logo: {
        icon: "https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg",
        name: "https://assets.ubuntu.com/v1/2e04d794-logo-jaas.svg",
        nameAlt: "JAAS",
        href: "/"
      },
      navItems: [{
        items: [{
          icon: "drag",
          label: "Models",
          href: "/models"
        }, {
          icon: "menu",
          label: "Controllers",
          href: "/controllers"
        }, {
          icon: "user",
          label: "Permissions",
          href: "/users"
        }]
      }],
      aside: showAside ? /*#__PURE__*/_react.default.createElement(_AppAside.default, {
        pinned: asidePinned
      }, /*#__PURE__*/_react.default.createElement(_Panel.default, {
        title: "Aside panel",
        controls: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_Button.default, {
          onClick: () => setAsidePinned(!asidePinned),
          dense: true,
          className: "u-no-margin"
        }, "Pin aside"), /*#__PURE__*/_react.default.createElement(_Button.default, {
          appearance: "base",
          className: "u-no-margin--bottom",
          hasIcon: true,
          onClick: () => {
            setShowAside(false);
            setAsidePinned(false);
          }
        }, /*#__PURE__*/_react.default.createElement(_Icon.default, {
          name: "close"
        }, "Close")))
      }, /*#__PURE__*/_react.default.createElement(_Form.default, {
        stacked: true
      }, /*#__PURE__*/_react.default.createElement(_Input.default, {
        label: "Full name",
        type: "text",
        name: "fullName",
        autoComplete: "name",
        stacked: true
      }), /*#__PURE__*/_react.default.createElement(_Input.default, {
        label: "Username",
        type: "text",
        name: "username-stacked",
        autoComplete: "username",
        "aria-describedby": "exampleHelpTextMessage",
        stacked: true,
        help: "30 characters or fewer."
      }), /*#__PURE__*/_react.default.createElement(_Input.default, {
        type: "text",
        label: "Email address",
        "aria-invalid": "true",
        name: "username-stackederror",
        autoComplete: "email",
        required: true,
        error: "This field is required.",
        stacked: true
      }), /*#__PURE__*/_react.default.createElement(_Input.default, {
        label: "Address line 1",
        type: "text",
        name: "address-optional-stacked",
        autoComplete: "address-line1",
        stacked: true
      }), /*#__PURE__*/_react.default.createElement(_Input.default, {
        label: "Address line 2",
        type: "text",
        name: "address-optional-stacked",
        autoComplete: "address-line3",
        stacked: true
      }), /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
        size: 12
      }, /*#__PURE__*/_react.default.createElement(_Button.default, {
        appearance: "positive",
        className: "u-float-right",
        name: "add-details"
      }, "Add details")))))) : null,
      status: /*#__PURE__*/_react.default.createElement(_Button.default, {
        onClick: () => setShowAside(!showAside),
        dense: true,
        appearance: "base",
        className: "u-no-margin"
      }, "Toggle aside")
    }, /*#__PURE__*/_react.default.createElement(_Panel.default, {
      title: "Application Layout"
    }, /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 12
    }, "Content"))));
  }
};

/**
 * The pinned and collapsed states can be handled externally if you need access
 * to these outside the application layout.
 */
const ExternalStates = exports.ExternalStates = {
  render: () => {
    const [menuPinned, setMenuPinned] = (0, _react.useState)(false);
    const [menuCollapsed, setMenuCollapsed] = (0, _react.useState)(true);
    return /*#__PURE__*/_react.default.createElement(_ApplicationLayout.default, {
      menuPinned: menuPinned,
      onPinMenu: setMenuPinned,
      menuCollapsed: menuCollapsed,
      onCollapseMenu: setMenuCollapsed,
      logo: {
        icon: "https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg",
        name: "https://assets.ubuntu.com/v1/2e04d794-logo-jaas.svg",
        nameAlt: "JAAS",
        href: "/"
      },
      navItems: [{
        items: [{
          icon: "drag",
          label: "Models",
          href: "/models"
        }, {
          icon: "menu",
          label: "Controllers",
          href: "/controllers"
        }, {
          icon: "user",
          label: "Permissions",
          href: "/users"
        }]
      }]
    }, /*#__PURE__*/_react.default.createElement(_Panel.default, {
      title: "Application Layout"
    }, /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 12
    }, "Content"))));
  }
};

/**
 * Menu items can either be provided as an array given to `navItems` or when
 * more control is needed then by proving `sideNavigation`.
 *
 * For more details about how to provide navigation items see the
 * [`SideNavigation`](/docs/components-sidenavigation--docs) docs.
 */
const Navigation = exports.Navigation = {
  render: () => {
    const [menuPinned, setMenuPinned] = (0, _react.useState)(false);
    const [menuCollapsed, setMenuCollapsed] = (0, _react.useState)(true);
    return /*#__PURE__*/_react.default.createElement(_ApplicationLayout.default, {
      menuPinned: menuPinned,
      onPinMenu: setMenuPinned,
      menuCollapsed: menuCollapsed,
      onCollapseMenu: setMenuCollapsed,
      logo: {
        icon: "https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg",
        name: "https://assets.ubuntu.com/v1/2e04d794-logo-jaas.svg",
        nameAlt: "JAAS",
        href: "/"
      },
      sideNavigation: /*#__PURE__*/_react.default.createElement(_SideNavigation.default, {
        items: [{
          items: [{
            icon: "drag",
            label: "Models",
            href: "/models"
          }, {
            icon: "menu",
            label: "Controllers",
            href: "/controllers"
          }, {
            icon: "user",
            label: "Permissions",
            href: "/users"
          }]
        }]
      })
    }, /*#__PURE__*/_react.default.createElement(_Panel.default, {
      title: "Application Layout"
    }, /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 12
    }, "Content"))));
  }
};

/**
 * App navigation can also be completely opted out of, for some
 * particular scenarios (e.g. login page, error pages).
 */
const NoNavigation = exports.NoNavigation = {
  render: () => {
    return /*#__PURE__*/_react.default.createElement(_ApplicationLayout.default, null, /*#__PURE__*/_react.default.createElement(_Panel.default, {
      title: "Application Layout with no navigation"
    }, /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 12
    }, "Content"))));
  }
};
