// @canonical/react-components@4.2.0/dist/esm/components/Chip/Chip.stories.js
import React from "react";
import Chip from "./Chip";
var meta = {
  component: Chip,
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  name: "Default",
  args: {
    value: "Default"
  }
};
export var LeadValue = {
  render: () => /*#__PURE__*/React.createElement(Chip, {
    lead: "Owner",
    value: "Bob"
  }),
  name: "Lead-value"
};
export var Appearance = {
  render: () => /*#__PURE__*/React.createElement(Chip, {
    lead: "Type",
    value: "Positive",
    appearance: "positive"
  }),
  name: "Appearance"
};
export var Dismissible = {
  render: () => /*#__PURE__*/React.createElement(Chip, {
    lead: "Owner",
    value: "Bob",
    onDismiss: () => {
      console.log("onDismiss called");
    }
  }),
  name: "Dismissible"
};
export var Dense = {
  render: () => /*#__PURE__*/React.createElement(Chip, {
    lead: "Owner",
    value: "Bob",
    isDense: true
  }),
  name: "Dense"
};
export var Inline = {
  render: () => /*#__PURE__*/React.createElement("p", null, "This is text with an inline", " ", /*#__PURE__*/React.createElement(Chip, {
    value: "chip",
    isDense: true,
    isInline: true
  })),
  name: "Inline"
};
export var ReadOnly = {
  render: () => /*#__PURE__*/React.createElement(Chip, {
    lead: "Owner",
    value: "Bob",
    isReadOnly: true
  }),
  name: "Read-only"
};
export var WithIcon = {
  render: () => /*#__PURE__*/React.createElement(Chip, {
    value: "Users",
    iconName: "user"
  }),
  name: "With Icon"
};
export var WithBadge = {
  render: () => /*#__PURE__*/React.createElement(Chip, {
    lead: "Owner",
    value: "Bob",
    badge: /*#__PURE__*/React.createElement("span", {
      className: "p-badge",
      "aria-label": "More than 2.5 billion items"
    }, "2.5B")
  }),
  name: "With Badge"
};
