// @canonical/react-components@4.2.0/dist/components/SearchAndFilter/SearchAndFilter.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithExistingSearchData = exports.WithDataSet = exports.Default = void 0;
var _SearchAndFilter = _interopRequireDefault(require("./SearchAndFilter"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _SearchAndFilter.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default",
  args: {
    returnSearchData: () => {},
    filterPanelData: [{
      id: 0,
      heading: "Cloud",
      chips: [{
        lead: "Cloud",
        value: "Google"
      }, {
        lead: "Cloud",
        value: "AWS"
      }, {
        lead: "Cloud",
        value: "Azure"
      }]
    }, {
      id: 1,
      heading: "Region",
      chips: [{
        lead: "Region",
        value: "us-east1"
      }, {
        lead: "Region",
        value: "us-north2"
      }, {
        lead: "Region",
        value: "us-south3"
      }, {
        lead: "Region",
        value: "us-north4"
      }, {
        lead: "Region",
        value: "us-east5"
      }, {
        lead: "Region",
        value: "us-south6"
      }, {
        lead: "Region",
        value: "us-east7"
      }, {
        lead: "Region",
        value: "us-east8"
      }, {
        lead: "Region",
        value: "us-east9"
      }, {
        lead: "Region",
        value: "us-east10"
      }]
    }, {
      id: 2,
      heading: "Owner",
      chips: [{
        lead: "Owner",
        value: "foo"
      }, {
        lead: "Owner",
        value: "bar"
      }, {
        lead: "Owner",
        value: "baz"
      }]
    }]
  }
};
const WithDataSet = exports.WithDataSet = {
  name: "With data set",
  args: {
    filterPanelData: [{
      id: 0,
      heading: "Cloud",
      chips: [{
        lead: "Cloud",
        value: "Google"
      }, {
        lead: "Cloud",
        value: "AWS"
      }, {
        lead: "Cloud",
        value: "Azure"
      }]
    }, {
      id: 1,
      heading: "Region",
      chips: [{
        lead: "Region",
        value: "us-east1"
      }, {
        lead: "Region",
        value: "us-north2"
      }, {
        lead: "Region",
        value: "us-south3"
      }, {
        lead: "Region",
        value: "us-north4"
      }, {
        lead: "Region",
        value: "us-east5"
      }, {
        lead: "Region",
        value: "us-south6"
      }, {
        lead: "Region",
        value: "us-east7"
      }, {
        lead: "Region",
        value: "us-east8"
      }, {
        lead: "Region",
        value: "us-east9"
      }, {
        lead: "Region",
        value: "us-east10"
      }]
    }, {
      id: 2,
      heading: "Owner",
      chips: [{
        lead: "Owner",
        value: "foo"
      }, {
        lead: "Owner",
        value: "bar"
      }, {
        lead: "Owner",
        value: "baz"
      }]
    }],
    returnSearchData: () => {}
  }
};
const WithExistingSearchData = exports.WithExistingSearchData = {
  name: "With existing search data",
  args: {
    existingSearchData: [{
      lead: "Cloud",
      value: "Google"
    }],
    filterPanelData: [{
      id: 0,
      heading: "Cloud",
      chips: [{
        lead: "Cloud",
        value: "Google"
      }, {
        lead: "Cloud",
        value: "AWS"
      }, {
        lead: "Cloud",
        value: "Azure"
      }]
    }, {
      id: 1,
      heading: "Region",
      chips: [{
        lead: "Region",
        value: "us-east1"
      }, {
        lead: "Region",
        value: "us-north2"
      }, {
        lead: "Region",
        value: "us-south3"
      }, {
        lead: "Region",
        value: "us-north4"
      }, {
        lead: "Region",
        value: "us-east5"
      }, {
        lead: "Region",
        value: "us-south6"
      }, {
        lead: "Region",
        value: "us-east7"
      }, {
        lead: "Region",
        value: "us-east8"
      }, {
        lead: "Region",
        value: "us-east9"
      }, {
        lead: "Region",
        value: "us-east10"
      }]
    }, {
      id: 2,
      heading: "Owner",
      chips: [{
        lead: "Owner",
        value: "foo"
      }, {
        lead: "Owner",
        value: "bar"
      }, {
        lead: "Owner",
        value: "baz"
      }]
    }],
    returnSearchData: () => {}
  }
};
