// @canonical/react-components@4.2.0/dist/esm/components/EventQueue/EventQueue.stories.js
import React, { useState } from "react";
import { createEventQueue } from "./EventQueue";
import Button from "../Button/Button";
var {
  EventQueueProvider,
  useEventQueue
} = createEventQueue();
var meta = {
  title: "Components/EventQueue",
  component: EventQueueProvider,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component: "\nThis provides an event queue system for managing callbacks associated with\nasynchronous operations (e.g., API calls) in an application.\n\nIt allows components to register success, failure,\nand optional finish handlers for a given operation ID, and later retrieve or remove them.\n\nThis is useful for handling side effects when dealing\nwith multiple operations that need to be tracked and updated independently.\n\nThe `createEventQueue` function should be used to create a single provider and context that are shared throughout your application.\nThe returned `EventQueueProvider` and `useEventQueue` hook should be exported\nfrom a shared module and reused across the app to ensure a single\ncontext instance is used.\n\n**Usage pattern:**\n\n```tsx\n// eventQueue.ts\nexport const { EventQueueProvider, useEventQueue } = createEventQueue<EventType>();\n\n// App.tsx\nimport { EventQueueProvider } from \"./eventQueue\";\n\n<EventQueueProvider>\n  <App />\n</EventQueueProvider>\n\n// In any other component\nimport { useEventQueue } from \"./eventQueue\";\n\nconst eventQueue = useEventQueue();\neventQueue.set(operationId, onSuccess, onFailure);\n```"
      }
    }
  },
  decorators: [Story => /*#__PURE__*/React.createElement(EventQueueProvider, null, /*#__PURE__*/React.createElement(Story, null))]
};
export default meta;
export var Default = {
  render: () => {
    var EventQueueDemo = () => {
      var queue = useEventQueue();
      var operationId = "storybook-op-id";
      var [logs, setLogs] = useState([]);
      var [isRegistered, setIsRegistered] = useState(false);
      var appendLog = msg => setLogs(prev => [...prev, msg]);

      // This simulates the code that is executed when the asynchronous operation starts
      var register = () => {
        queue.set(operationId, event => appendLog("Success: ".concat(event.message)), msg => appendLog("Failure: ".concat(msg)), () => appendLog("Finished"));
        appendLog("Registered callbacks");
        setIsRegistered(true);
      };

      // This simulates the code that is executed when the asynchronous operation succeeds
      var triggerSuccess = () => {
        var event = queue.get(operationId);
        if (event) {
          var _event$onFinish;
          event.onSuccess({
            message: "This was successful!"
          });
          (_event$onFinish = event.onFinish) === null || _event$onFinish === void 0 || _event$onFinish.call(event);
          queue.remove(operationId);
          setIsRegistered(false);
        }
      };

      // This simulates the code that is executed when the asynchronous operation fails
      var triggerFailure = () => {
        var event = queue.get(operationId);
        if (event) {
          var _event$onFinish2;
          event.onFailure("Something failed!");
          (_event$onFinish2 = event.onFinish) === null || _event$onFinish2 === void 0 || _event$onFinish2.call(event);
          queue.remove(operationId);
          setIsRegistered(false);
        }
      };
      return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("div", {
        style: {
          marginBottom: "0.5rem",
          display: "flex"
        }
      }, /*#__PURE__*/React.createElement(Button, {
        onClick: register,
        disabled: isRegistered
      }, "Register Callbacks"), /*#__PURE__*/React.createElement(Button, {
        onClick: triggerSuccess,
        disabled: !isRegistered
      }, "Trigger Success"), /*#__PURE__*/React.createElement(Button, {
        onClick: triggerFailure,
        disabled: !isRegistered
      }, "Trigger Failure")), /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("strong", null, "Logs:"), /*#__PURE__*/React.createElement("ul", null, logs.map((log, index) => /*#__PURE__*/React.createElement("li", {
        key: index
      }, log)))));
    };
    return /*#__PURE__*/React.createElement(EventQueueDemo, null);
  }
};
