// @canonical/react-components@4.2.0/dist/esm/components/ApplicationLayout/AppNavigation/AppNavigation.stories.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
/* eslint-disable react-hooks/rules-of-hooks */
import { useState } from "react";
import classNames from "classnames";
import React from "react";
import Application from "../Application";
import AppMain from "../AppMain";
import Button from "../../Button";
import Icon from "../../Icon";
import Panel from "../../Panel";
import SideNavigation from "../../SideNavigation";
import Strip from "../../Strip";
import AppNavigation from "./AppNavigation";
var meta = {
  component: AppNavigation,
  tags: ["autodocs"]
};
export default meta;
/**
 * In most common cases an `AppNavigation` should contain a `<Panel>` and `<SideNavigation>`.
 *
 * `AppNavigation` should be a direct child of an `<Application>` or when using `ApplicationLayout`
 * it will wrap `navItems` and `sideNavigation`.
 */
export var Default = {
  render: args => {
    var [pinned, setPinned] = useState(false);
    var [collapsed, setCollapsed] = useState(true);
    return /*#__PURE__*/React.createElement(Application, null, /*#__PURE__*/React.createElement(AppNavigation, _extends({}, args, {
      pinned: pinned,
      collapsed: collapsed
    }), /*#__PURE__*/React.createElement(Panel, {
      dark: true,
      controls: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Button, {
        hasIcon: true,
        appearance: "base",
        className: classNames("u-no-margin u-hide--medium", {
          "is-dark": true
        }),
        onClick: evt => {
          setCollapsed(true);
          // The menu stays open while its content has focus, so the
          // close button must blur to actually close the menu.
          evt.currentTarget.blur();
        }
      }, /*#__PURE__*/React.createElement(Icon, {
        name: "close",
        className: classNames({
          "is-light": true
        })
      }, "Close menu")), /*#__PURE__*/React.createElement(Button, {
        hasIcon: true,
        appearance: "base",
        className: classNames("u-no-margin u-hide--small", {
          "is-dark": true
        }),
        onClick: () => {
          setPinned(!pinned);
        }
      }, /*#__PURE__*/React.createElement(Icon, {
        name: pinned ? "close" : "pin",
        className: classNames({
          "is-light": true
        })
      }, pinned ? "Unpin menu" : "Pin menu"))),
      controlsClassName: "u-hide--large",
      stickyHeader: true,
      logo: {
        icon: "https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg",
        name: "https://assets.ubuntu.com/v1/2e04d794-logo-jaas.svg",
        nameAlt: "JAAS",
        href: "/"
      }
    }, /*#__PURE__*/React.createElement(SideNavigation, {
      dark: true,
      items: [{
        items: [{
          icon: "drag",
          label: "Models",
          href: "/models"
        }, {
          icon: "menu",
          label: "Controllers",
          href: "/controllers"
        }, {
          icon: "user",
          label: "Permissions",
          href: "/permissions"
        }]
      }]
    }))), /*#__PURE__*/React.createElement(AppMain, null, /*#__PURE__*/React.createElement(Strip, null, /*#__PURE__*/React.createElement("p", null, "These behaviours are only available at smaller screen sizes:"), /*#__PURE__*/React.createElement(Button, {
      onClick: () => setPinned(!pinned)
    }, "Pin"), /*#__PURE__*/React.createElement(Button, {
      onClick: () => setCollapsed(!collapsed)
    }, "Toggle"))));
  }
};
