// @canonical/react-components@4.2.0/dist/esm/components/Tabs/Tabs.stories.js
import React from "react";
import Row from "../Row";
import Tabs from "./Tabs";
var meta = {
  component: Tabs,
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  name: "Default",
  args: {
    links: [{
      active: true,
      label: "Summary"
    }, {
      active: false,
      label: "Network"
    }, {
      active: false,
      label: "Storage"
    }, {
      active: false,
      label: "Settings"
    }]
  }
};

/**
 * To horizontally align the Tabs with other content, they can be contained within a Row component to
provide correct gutters.
 */
export var HorizontallyAligned = {
  render: () => /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Tabs, {
    links: [{
      active: true,
      label: "Summary"
    }, {
      active: false,
      label: "Network"
    }, {
      active: false,
      label: "Storage"
    }, {
      active: false,
      label: "Settings"
    }]
  }), /*#__PURE__*/React.createElement("p", null, "There should be gutters and the text should be horizontally aligned")),
  name: "Horizontally aligned"
};
