// @canonical/react-components@4.2.0/dist/esm/components/Pagination/Pagination.stories.js
import React from "react";
import Pagination from "./Pagination";
var meta = {
  component: Pagination,
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  name: "Default",
  args: {
    itemsPerPage: 20,
    totalItems: 100,
    paginate: () => {},
    currentPage: 3
  }
};
export var TruncatedNumberedPagination = {
  render: () => /*#__PURE__*/React.createElement(Pagination, {
    itemsPerPage: 10,
    totalItems: 1000,
    paginate: () => {},
    currentPage: 34
  }),
  name: "Truncated (Numbered Pagination)"
};
export var DisabledControlsNumberedPagination = {
  render: () => /*#__PURE__*/React.createElement(Pagination, {
    itemsPerPage: 20,
    totalItems: 100,
    paginate: () => {},
    currentPage: 1
  }),
  name: "Disabled controls (Numbered Pagination)"
};
export var CenteredNumberedPagination = {
  render: () => /*#__PURE__*/React.createElement(Pagination, {
    itemsPerPage: 10,
    totalItems: 50,
    paginate: () => {},
    currentPage: 1,
    centered: true
  }),
  name: "Centered (Numbered Pagination)"
};
export var DisabledPaginationItemsNumberedPagination = {
  render: () => /*#__PURE__*/React.createElement(Pagination, {
    itemsPerPage: 10,
    totalItems: 50,
    paginate: () => {},
    currentPage: 1,
    hideNumbers: true
  }),
  name: "Disabled pagination items (Numbered Pagination)"
};
export var DefaultButtonsOnlyPagination = {
  render: () => /*#__PURE__*/React.createElement(Pagination, {
    onForward: () => {},
    onBack: () => {}
  }),
  name: "Default (Buttons-only Pagination)"
};
export var DefaultButtonLabelsButtonsOnlyPagination = {
  render: () => /*#__PURE__*/React.createElement(Pagination, {
    onForward: () => {},
    onBack: () => {},
    showLabels: true
  }),
  name: "Default button labels (Buttons-only Pagination)"
};
export var CustomButtonLabelsButtonsOnlyPagination = {
  render: () => /*#__PURE__*/React.createElement(Pagination, {
    onForward: () => {},
    onBack: () => {},
    showLabels: true,
    forwardLabel: "Custom forward label",
    backLabel: "Custom back label"
  }),
  name: "Custom button labels (Buttons-only Pagination)"
};
export var CenteredButtonsOnlyPagination = {
  render: () => /*#__PURE__*/React.createElement(Pagination, {
    onForward: () => {},
    onBack: () => {},
    centered: true
  }),
  name: "Centered (Buttons-only Pagination)"
};
